{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "initial_id",
   "metadata": {
    "collapsed": true,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:16.637157Z",
     "start_time": "2023-12-13T16:52:16.617990600Z"
    }
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "from biomart import BiomartServer\n",
    "import io"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "outputs": [
    {
     "data": {
      "text/plain": "['biomart',\n 'depmap',\n 'missing_gene_names.csv',\n 'pathway_commons',\n 'reactome',\n 'reactome_rows_to_drop.csv',\n 'string']"
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#os.getcwd()\n",
    "\n",
    "# Linux workstation\n",
    "#data_path = '/home/db600/phd/data/'\n",
    "\n",
    "# Laptop\n",
    "data_path = 'C:\\\\Users\\\\dan\\\\Documents\\\\phd\\\\data\\\\'\n",
    "\n",
    "os.listdir(data_path)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:18.228980300Z",
     "start_time": "2023-12-13T16:52:18.206280900Z"
    }
   },
   "id": "e734db8459feaf32"
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "outputs": [
    {
     "data": {
      "text/plain": "['PathwayCommons12.reactome.hgnc.gmt',\n 'PathwayCommons12.reactome.hgnc.gmt.gz',\n 'PathwayCommons12.reactome.hgnc.sif',\n 'PathwayCommons12.reactome.hgnc.sif.gz',\n 'PathwayCommons12.reactome.hgnc.txt',\n 'PathwayCommons12.reactome.hgnc.txt.gz',\n 'PathwayCommons12.reactome.uniprot.gmt',\n 'PathwayCommons12.reactome.uniprot.gmt.gz']"
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir(data_path + '/pathway_commons/')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:20.239557400Z",
     "start_time": "2023-12-13T16:52:20.236611500Z"
    }
   },
   "id": "fd659b1ed25390e0"
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "outputs": [
    {
     "data": {
      "text/plain": "Index(['PARTICIPANT_A', 'INTERACTION_TYPE', 'PARTICIPANT_B'], dtype='object')"
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_sif = pd.read_csv(data_path + \"pathway_commons/PathwayCommons12.reactome.hgnc.sif\", sep=\"\\t\")\n",
    "reactome_sif.head()\n",
    "reactome_sif.columns = ['PARTICIPANT_A', 'INTERACTION_TYPE', 'PARTICIPANT_B']\n",
    "reactome_sif.columns"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:21.939195600Z",
     "start_time": "2023-12-13T16:52:21.815381300Z"
    }
   },
   "id": "911a8569b59173b9"
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['in-complex-with' 'controls-state-change-of' 'controls-production-of'\n",
      " 'controls-transport-of' 'controls-transport-of-chemical'\n",
      " 'catalysis-precedes' 'controls-phosphorylation-of'\n",
      " 'controls-expression-of' 'chemical-affects' 'used-to-produce'\n",
      " 'consumption-controlled-by' 'reacts-with']\n"
     ]
    }
   ],
   "source": [
    "print(reactome_sif['INTERACTION_TYPE'].unique())"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:24.831277100Z",
     "start_time": "2023-12-13T16:52:24.783746200Z"
    }
   },
   "id": "7527938f700d6de5"
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "outputs": [
    {
     "data": {
      "text/plain": "  PARTICIPANT_A INTERACTION_TYPE PARTICIPANT_B\n0          A1CF  in-complex-with       APOBEC2\n1          A1CF  in-complex-with      APOBEC3A\n2          A1CF  in-complex-with      APOBEC3B\n3          A1CF  in-complex-with      APOBEC3C\n4          A1CF  in-complex-with      APOBEC3H",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>PARTICIPANT_A</th>\n      <th>INTERACTION_TYPE</th>\n      <th>PARTICIPANT_B</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC2</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3A</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3B</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3C</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3H</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_sif.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:26.055153100Z",
     "start_time": "2023-12-13T16:52:26.043805100Z"
    }
   },
   "id": "35cf88030f92149c"
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "outputs": [
    {
     "data": {
      "text/plain": "353635"
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get row count for number of edges in the dataset\n",
    "len(reactome_sif)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:27.650020400Z",
     "start_time": "2023-12-13T16:52:27.621010600Z"
    }
   },
   "id": "eaa487c299371ea7"
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "353635\n",
      "353635\n",
      "707270\n"
     ]
    },
    {
     "data": {
      "text/plain": "<Figure size 640x480 with 1 Axes>",
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjoAAAGdCAYAAAAbudkLAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAsNElEQVR4nO3de1TVdb7/8dcWBMEQU0cUBcSVZogyiVSWlnSx0HRG1+nYTF4q7eS4K41udpwzTXbB6kROx61l09Kazkx2MVcXy8F00rIZEcU0TnYZEi8YacX2MorC5/dHi/1riyBsNnw3n/18rMVa873w2e/3huQ1n/35fr8uY4wRAACAhdo5XQAAAEBLIegAAABrEXQAAIC1CDoAAMBaBB0AAGAtgg4AALAWQQcAAFiLoAMAAKwV6XQBTqupqdG+ffsUFxcnl8vldDkAAKARjDE6dOiQEhMT1a5d/fM2YR909u3bp6SkJKfLAAAAAdi9e7d69+5d7/GwDzpxcXGSfnyjOnXq5HA1AACgMbxer5KSknx/x+sTtkHH4/HI4/GourpaktSpUyeCDgAAbcyZlp24wv2hnl6vV/Hx8aqsrCToAADQRjT27zdXXQEAAGsRdAAAgLXCdo0OAACBMsbo5MmTvnWeCL6IiAhFRkY2+9YvBB0AAJqgqqpK5eXlOnr0qNOlWC82NlY9e/ZUVFRUwGMQdAAAaKSamhqVlpYqIiJCiYmJioqK4mazLcAYo6qqKn377bcqLS1Vv379GrwpYEPCNuicenk5AABnUlVVpZqaGiUlJSk2NtbpcqwWExOj9u3ba9euXaqqqlKHDh0CGidsFyO73W6VlJSosLDQ6VIAAG1MoLMLaJpgvM/8pAAAgLUIOgAAoI6//e1vcrlc+uGHH5wupVnCdo0OAADB0mfOO636el/PH9Oqr9eWMaMDAACsRdABACAMGGP0+OOPq2/fvoqJiVFGRoZee+013/FVq1apf//+iomJUXZ2tr7++us6Yzz33HO+K87Gjx+v/Px8de7c2e+ct956S5mZmerQoYP69u2rBx98UCdPnvQd//3vf6/k5GRFR0crMTFRd9xxR0u1LImPrlpU7VQmU4wAAKf99re/1YoVK7R48WL169dP69ev16RJk/Szn/1Mffv21YQJEzRjxgz95je/0ebNm3XXXXf5ff9HH32kGTNm6LHHHtO4ceO0Zs0a/dd//ZffOatXr9akSZP09NNPa8SIEfrqq6/0H//xH5KkBx54QK+99pqeeuopvfzyyxo4cKD279+vbdu2tWjfBB0AACx35MgR5efna+3atRo2bJgkqW/fvvrwww/17LPPqk+fPurbt6+eeuopuVwunXvuudq+fbsee+wx3xj/8z//o5ycHN19992SpP79+2vjxo16++23fec88sgjmjNnjqZOnep7jYceekj33nuvHnjgAZWVlalHjx668sor1b59eyUnJ+uCCy5o0d7D9qMrj8ejtLQ0ZWVlOV0KAAAtqqSkRMeOHdNVV12ls846y/f14osv6quvvtL//d//6aKLLvK7y3NtIKq1c+fOOqHk1O2ioiLNmzfP7zVuueUW3yMzrrvuOv3rX/9S3759dcstt+iNN97w+1irJYTtjI7b7Zbb7ZbX61V8fLzT5QAA0GJqamokSe+884569erldyw6Olq33377GccwxtR53IUxps7rPPjgg5owYUKd7+/QoYOSkpK0c+dOFRQUaM2aNZo5c6aeeOIJffDBB2rfvn1T22qUsA06AACEi7S0NEVHR6usrEyXXXbZaY+vXLnSb9/f//53v+0BAwZo06ZNfvs2b97stz1kyBDt3LlT55xzTr21xMTEaNy4cRo3bpzcbrcGDBig7du3a8iQIU3sqnEIOgAAWC4uLk5333237rzzTtXU1Gj48OHyer3auHGjzjrrLM2YMUNPPvmkcnNzdeutt6qoqEjLli3zG+P222/XpZdeqvz8fI0dO1Zr167Vu+++6zfL87vf/U7XXnutkpKSdN1116ldu3b65JNPtH37dj388MNatmyZqqurdeGFFyo2NlZ/+tOfFBMTo5SUlBbrPWzX6AAAEE4eeugh/e53v1NeXp7OO+88XX311XrrrbeUmpqq5ORkvf7663rrrbeUkZGhZ555Ro8++qjf919yySV65plnlJ+fr4yMDL333nu68847/R62efXVV+vtt99WQUGBsrKydNFFFyk/P98XZDp37qznnntOl1xyiQYPHqz3339fb731lrp27dpifbvMqR+whZnaNTqVlZXq1KlTUMfm8nIAsMuxY8dUWlqq1NTUgJ+mbZNbbrlFn332mTZs2NAi4zf0fjf27zcfXQEAgEb57//+b1111VXq2LGj3n33Xb3wwgtatGiR02U1iKADAAAaZdOmTXr88cd16NAh9e3bV08//bSmT5/udFkNIugAAIBGeeWVV5wuoclYjAwAAKxF0AEAANYi6AAA0ERhfsFyqwnG+xy2QYdnXQEAmqr2MQVHjx51uJLwUPs+N+fxEGG7GJlnXQEAmioiIkKdO3dWRUWFJCk2NrbO85/QfMYYHT16VBUVFercubMiIiICHitsgw4AAIHo0aOHJPnCDlpO586dfe93oAg6AAA0gcvlUs+ePdW9e3edOHHC6XKs1b59+2bN5NQi6AAAEICIiIig/CFGywrbxcgAAMB+BB0AAGAtgg4AALAWQQcAAFiLoAMAAKxF0AEAANYi6AAAAGsRdAAAgLUIOgAAwFoEHQAAYC2CDgAAsFbYBh2Px6O0tDRlZWU5XQoAAGghYRt03G63SkpKVFhY6HQpAACghYRt0AEAAPYj6AAAAGsRdAAAgLUIOgAAwFoEHQAAYK1IpwsIB33mvOP731/PH+NgJQAAhBdmdAAAgLUIOgAAwFoEHQAAYC2CDgAAsBZBBwAAWIugAwAArEXQAQAA1iLoAAAAaxF0AACAtQg6AADAWgQdAABgLYIOAACwFkEHAABYi6DTyvrMecfvaeYAAKDltPmgc+jQIWVlZennP/+5Bg0apOeee87pkgAAQIiIdLqA5oqNjdUHH3yg2NhYHT16VOnp6ZowYYK6du3qdGkAAMBhbX5GJyIiQrGxsZKkY8eOqbq6WsYYh6sCAAChwPGgs379eo0dO1aJiYlyuVxauXJlnXMWLVqk1NRUdejQQZmZmdqwYYPf8R9++EEZGRnq3bu37r33XnXr1q2VqgcAAKHM8aBz5MgRZWRkaOHChac9vnz5cs2ePVtz587V1q1bNWLECOXk5KisrMx3TufOnbVt2zaVlpbqz3/+s7755pvWKh8AAIQwx4NOTk6OHn74YU2YMOG0x/Pz8zVt2jRNnz5d5513nhYsWKCkpCQtXry4zrkJCQkaPHiw1q9fX+/rHT9+XF6v1+8LAADYyfGg05CqqioVFRVp1KhRfvtHjRqljRs3SpK++eYbX1jxer1av369zj333HrHzMvLU3x8vO8rKSmp5RoAAACOCumgc+DAAVVXVyshIcFvf0JCgvbv3y9J2rNnjy699FJlZGRo+PDhuu222zR48OB6x7z//vtVWVnp+9q9e3eL9gAAAJzTJi4vd7lcftvGGN++zMxMFRcXN3qs6OhoRUdHB7M8AAAQokJ6Rqdbt26KiIjwzd7UqqioqDPLAwAAcKqQDjpRUVHKzMxUQUGB3/6CggJdfPHFzRrb4/EoLS1NWVlZzRoHAACELsc/ujp8+LC+/PJL33ZpaamKi4vVpUsXJScnKzc3V5MnT9bQoUM1bNgwLVmyRGVlZZoxY0azXtftdsvtdsvr9So+Pr65bQAAgBDkeNDZvHmzsrOzfdu5ubmSpKlTp2rZsmWaOHGiDh48qHnz5qm8vFzp6elatWqVUlJSnCoZAAC0ES4T5s9LqJ3RqaysVKdOnYI6dkNPKf96/pigvhYAAOGksX+/Q3qNDgAAQHOEbdBhMTIAAPYL26DjdrtVUlKiwsJCp0sBAAAtJGyDDgAAsB9BBwAAWIugAwAArBW2QYfFyAAA2I/76Dh0H51a3E8HAICm4z46AAAg7BF0AACAtQg6AADAWgQdAABgrbANOlx1BQCA/cI26PAICAAA7Be2QQcAANiPoAMAAKxF0AEAANYi6Disz5x3GnUHZQAA0HQEHQAAYC2CDgAAsFbYBh3uowMAgP3CNuhwHx0AAOwXtkEHAADYj6ADAACsRdABAADWIugAAABrEXQAAIC1CDoAAMBaBJ0QxGMhAAAIjkinC3CKx+ORx+NRdXW106VIEsEGAIAWELYzOtwwEAAA+4Vt0AEAAPYj6AAAAGsRdAAAgLUIOgAAwFoEHQAAYC2CDgAAsBZBBwAAWIugAwAArEXQAQAA1grboOPxeJSWlqasrCynSwEAAC0kbIMOj4AAAMB+YRt0AACA/Qg6AADAWgQdAABgLYJOCOsz5x31mfOO02UAANBmEXQAAIC1CDoAAMBaBB0AAGCtSKcLQOP9dL3O1/PHOFgJAABtAzM6AADAWgQdAABgLYIOAACwFkEHAABYi6ADAACsRdABAADW4vLyNuB0j4Go3cdl5gAA1C9sZ3Q8Ho/S0tKUlZXldCkAAKCFhG3QcbvdKikpUWFhodOlAACAFhK2QQcAANiPoAMAAKxF0AEAANYi6AAAAGsRdAAAgLUIOm1cnznvnPY+OwAAgKADAAAsRtCxBDM7AADURdABAADWIugAAABr8VBPBOynH5XxcFEAQCgKaEantLQ02HUAAAAEXUBB55xzzlF2drZeeuklHTt2LNg1AQAABEVAQWfbtm06//zzddddd6lHjx669dZbtWnTpmDXBgAA0CwBBZ309HTl5+dr7969Wrp0qfbv36/hw4dr4MCBys/P17fffhvsOgEAAJqsWVddRUZGavz48XrllVf02GOP6auvvtLdd9+t3r17a8qUKSovLw9WnQAAAE3WrKCzefNmzZw5Uz179lR+fr7uvvtuffXVV1q7dq327t2rX/ziF8GqEy2MGw4CAGwU0OXl+fn5Wrp0qXbu3KnRo0frxRdf1OjRo9Wu3Y+5KTU1Vc8++6wGDBgQ1GIBAACaIqCgs3jxYt1888266aab1KNHj9Oek5ycrOeff75ZxaF5amdouMcNACBcBRR0vvjiizOeExUVpalTpwYyPAAAQFAEtEZn6dKlevXVV+vsf/XVV/XCCy80u6im2L17t0aOHKm0tDQNHjz4tHUBAIDwFFDQmT9/vrp161Znf/fu3fXoo482u6imiIyM1IIFC1RSUqI1a9bozjvv1JEjR1q1BgAAEJoC+uhq165dSk1NrbM/JSVFZWVlzS6qKXr27KmePXtK+jFodenSRd999506duzYqnUAAIDQE9CMTvfu3fXJJ5/U2b9t2zZ17dq1SWOtX79eY8eOVWJiolwul1auXFnnnEWLFik1NVUdOnRQZmamNmzYcNqxNm/erJqaGiUlJTWpBgAAYKeAgs7111+vO+64Q+vWrVN1dbWqq6u1du1azZo1S9dff32Txjpy5IgyMjK0cOHC0x5fvny5Zs+erblz52rr1q0aMWKEcnJy6swcHTx4UFOmTNGSJUsCaQkAAFgooI+uHn74Ye3atUtXXHGFIiN/HKKmpkZTpkxp8hqdnJwc5eTk1Hs8Pz9f06ZN0/Tp0yVJCxYs0OrVq7V48WLl5eVJko4fP67x48fr/vvv18UXX9zg6x0/flzHjx/3bXu93ibVCwAA2o6Agk5UVJSWL1+uhx56SNu2bVNMTIwGDRqklJSUoBZXVVWloqIizZkzx2//qFGjtHHjRkmSMUY33nijLr/8ck2ePPmMY+bl5enBBx8Map02+uldkrkPDwCgrQoo6NTq37+/+vfvH6xa6jhw4ICqq6uVkJDgtz8hIUH79++XJH300Udavny5Bg8e7Fvf86c//UmDBg067Zj333+/cnNzfdter5c1PQAAWCqgoFNdXa1ly5bp/fffV0VFhWpqavyOr127NijF1XK5XH7bxhjfvuHDh9d5/YZER0crOjo6qPUBAIDQFFDQmTVrlpYtW6YxY8YoPT29ThAJlm7duikiIsI3e1OroqKiziwP6tfQoyB4kCcAwGYBBZ2XX35Zr7zyikaPHh3sevxERUUpMzNTBQUFGj9+vG9/QUFBs5+M7vF45PF4VF1d3dwyAQBAiAp4MfI555wTlAIOHz6sL7/80rddWlqq4uJidenSRcnJycrNzdXkyZM1dOhQDRs2TEuWLFFZWZlmzJjRrNd1u91yu93yer2Kj49vbhthgYeEAgDamoCCzl133aU//OEPWrhwYbM/ttq8ebOys7N927ULhadOnaply5Zp4sSJOnjwoObNm6fy8nKlp6dr1apVQb/CCwAA2CegoPPhhx9q3bp1evfddzVw4EC1b9/e7/iKFSsaPdbIkSNljGnwnJkzZ2rmzJmBlAoAAMJYQEGnc+fOfmtmEF5YwAwAaCsCCjpLly4Ndh2tztbFyIQQAAD+v4CedSVJJ0+e1Jo1a/Tss8/q0KFDkqR9+/bp8OHDQSuuJbndbpWUlKiwsNDpUgAAQAsJaEZn165duuaaa1RWVqbjx4/rqquuUlxcnB5//HEdO3ZMzzzzTLDrhIOYJQIAtFUBzejMmjVLQ4cO1ffff6+YmBjf/vHjx+v9998PWnEAAADNEfBVVx999JGioqL89qekpGjv3r1BKQxtC/fYAQCEooBmdGpqak67iHfPnj2Ki4trdlGtwePxKC0tTVlZWU6XAgAAWkhAQeeqq67SggULfNsul0uHDx/WAw880OKPhQgWFiMDAGC/gD66euqpp5Sdna20tDQdO3ZMv/71r/XFF1+oW7du+stf/hLsGgEAAAISUNBJTExUcXGx/vKXv2jLli2qqanRtGnTdMMNN/gtTgYAAHBSQEFHkmJiYnTzzTfr5ptvDmY9AAAAQRNQ0HnxxRcbPD5lypSAigEAAAimgILOrFmz/LZPnDiho0ePKioqSrGxsW0i6Nj6CAgAAPD/BXTV1ffff+/3dfjwYe3cuVPDhw9vM4uRueoKAAD7Bfysq1P169dP8+fPrzPbAwAA4JSAFyOfTkREhPbt2xfMIdECeHYVACBcBBR03nzzTb9tY4zKy8u1cOFCXXLJJUEpDMFHwAEAhJuAgs4vf/lLv22Xy6Wf/exnuvzyy/Xkk08Goy4AAIBmCyjo1NTUBLsOAACAoAvaYmQAAIBQE9CMTm5ubqPPzc/PD+QlWhz30QEAwH4BBZ2tW7dqy5YtOnnypM4991xJ0ueff66IiAgNGTLEd57L5QpOlS3A7XbL7XbL6/UqPj7e6XKsVrsI+uv5YxyuBAAQbgIKOmPHjlVcXJxeeOEFnX322ZJ+vIngTTfdpBEjRuiuu+4KapEAAACBCGiNzpNPPqm8vDxfyJGks88+Ww8//DBXXYW5PnPe4TJ2AEDICCjoeL1effPNN3X2V1RU6NChQ80uCgAAIBgCCjrjx4/XTTfdpNdee0179uzRnj179Nprr2natGmaMGFCsGsEAAAISEBrdJ555hndfffdmjRpkk6cOPHjQJGRmjZtmp544omgFggAABCogIJObGysFi1apCeeeEJfffWVjDE655xz1LFjx2DXBwAAELBm3TCwvLxc5eXl6t+/vzp27ChjTLDqAgAAaLaAgs7Bgwd1xRVXqH///ho9erTKy8slSdOnT28zl5Z7PB6lpaUpKyvL6VIAAEALCSjo3HnnnWrfvr3KysoUGxvr2z9x4kS99957QSuuJbndbpWUlKiwsNDpUgAAQAsJaI3OX//6V61evVq9e/f229+vXz/t2rUrKIWhbeNeOgCAUBDQjM6RI0f8ZnJqHThwQNHR0c0uCgAAIBgCCjqXXnqpXnzxRd+2y+VSTU2NnnjiCWVnZwetOLQ+7mwMALBJQB9dPfHEExo5cqQ2b96sqqoq3Xvvvfr000/13Xff6aOPPgp2jQAAAAEJKOikpaXpk08+0eLFixUREaEjR45owoQJcrvd6tmzZ7BrhKV+OnPEk80BAC2hyUHnxIkTGjVqlJ599lk9+OCDLVETAABAUDR5jU779u21Y8cOuVyulqgHAAAgaAJajDxlyhQ9//zzwa4FYYIFzwCA1hLQGp2qqir98Y9/VEFBgYYOHVrnGVf5+flBKQ5oC2pDG+uMACD0NCno/POf/1SfPn20Y8cODRkyRJL0+eef+53TVj7S8ng88ng8qq6udroUAADQQpoUdPr166fy8nKtW7dO0o+PfHj66aeVkJDQIsW1JLfbLbfbLa/Xq/j4eKfLAQAALaBJa3ROfTr5u+++qyNHjgS1IAAAgGAJaI1OrVODDwDuDwQAoaRJMzoul6vOGpy2siYHAACEnybN6BhjdOONN/oe3Hns2DHNmDGjzlVXK1asCF6FsAaXlAMAWluTgs7UqVP9tidNmhTUYgAAAIKpSUFn6dKlLVUHAABA0AV0Z2QAAIC2gKCDNoNHRwAAmoqgAwAArEXQAQAA1iLoAAAAazXrzsgAGufUJ5xz92QAaB3M6AAAAGsRdBCSAr3CiiuzAAA/RdABAADWCts1Oh6PRx6PR9XV1U6XAgSMtT4A0LCwndFxu90qKSlRYWGh06UAAIAWErYzOggtp16V1BaxNggAQk/YzugAAAD7EXTQZnGFFQDgTAg6AADAWqzRQVgIp6uTbFjvBADBwowOAACwFkEHAABYi6ADK7FQGQAgEXQAAIDFWIyMkBLogzwBADgdZnQAAIC1CDoAAMBaBB0AfljIDcAmBB0AAGAtgg4AALAWQQcAAFiLoAMAAKxF0AEAANbihoEIaU5f/RNqTwIPtXoAINQxowMAAKxlRdAZP368zj77bP3bv/2b06UAAIAQYkXQueOOO/Tiiy86XQYAAAgxVgSd7OxsxcXFOV0GEBDuRAwALcfxoLN+/XqNHTtWiYmJcrlcWrlyZZ1zFi1apNTUVHXo0EGZmZnasGFD6xcKAADaHMeDzpEjR5SRkaGFCxee9vjy5cs1e/ZszZ07V1u3btWIESOUk5OjsrKyVq4UAAC0NY5fXp6Tk6OcnJx6j+fn52vatGmaPn26JGnBggVavXq1Fi9erLy8vCa/3vHjx3X8+HHfttfrbXrRAACgTXB8RqchVVVVKioq0qhRo/z2jxo1Shs3bgxozLy8PMXHx/u+kpKSglEqcFqBrL9hzQ4ABE9IB50DBw6ourpaCQkJfvsTEhK0f/9+3/bVV1+t6667TqtWrVLv3r1VWFhY75j333+/KisrfV+7d+9usfoBAICzHP/oqjFcLpfftjHGb9/q1asbPVZ0dLSio6ODVhsAAAhdIT2j061bN0VERPjN3khSRUVFnVkeAACAU4X0jE5UVJQyMzNVUFCg8ePH+/YXFBToF7/4RbPG9ng88ng8qq6ubm6ZaGXNXb9S3/e3xvOjAq2dNTsIVTx/DaHO8aBz+PBhffnll77t0tJSFRcXq0uXLkpOTlZubq4mT56soUOHatiwYVqyZInKyso0Y8aMZr2u2+2W2+2W1+tVfHx8c9sAAAAhyPGgs3nzZmVnZ/u2c3NzJUlTp07VsmXLNHHiRB08eFDz5s1TeXm50tPTtWrVKqWkpDhVMgAAaCMcDzojR46UMabBc2bOnKmZM2e2UkUAAMAWIb0YGQAAoDkcn9FxCouR0ZBQWPwbCjUAQFsXtjM6brdbJSUlDd5cEAAAtG1hG3QAAID9CDoAAMBaBB0AAGAtFiOzGBlN8NMFwtwJFgBCX9jO6LAYGQAA+4Vt0AEAAPYj6AAAAGsRdAAAgLUIOgAAwFpcdcVVVwiQk49oaI3Xrn0Nri4D0JaF7YwOV10BAGC/sA06AADAfgQdAABgLYIOAACwFkEHAABYi6ADAACsRdABAADWCtug4/F4lJaWpqysLKdLAdqcPnPecfQ+QgDQWGEbdLiPDgAA9gvboAMAAOxH0AEAANYi6AAAAGsRdAAAgLUIOgAAwFoEHQAAYC2CDgAAsFak0wU4xePxyOPxqLq62ulS0ILC6aZ2Tem19tyv549pqXIAICSE7YwONwwEAMB+YRt0AACA/Qg6AADAWgQdAABgLYIOAACwFkEHAABYi6ADAACsRdABAADWIugAAABrEXQAAIC1eAQEj4BAGAmVR2LU9wiKn9bH4ykABEPYzujwCAgAAOwXtkEHAADYj6ADAACsRdABAADWIugAAABrEXQAAIC1CDoAAMBaBB0AAGAtgg4AALAWQQcAAFiLoAMAAKxF0AEAANYi6AAAAGsRdAAAgLUIOgAAwFoEHQAAYK1IpwtwisfjkcfjUXV1tdOlIET0mfOOVa9fO97X88cEddyWeI3T9V7f2D89t77Xbcw5+FFr/J4ATgrbGR23262SkhIVFhY6XQoAAGghYRt0AACA/Qg6AADAWgQdAABgLYIOAACwFkEHAABYi6ADAACsRdABAADWIugAAABrEXQAAIC1CDoAAMBaBB0AAGAtgg4AALAWQQcAAFiLoAMAAKxF0AEAANYi6AAAAGsRdAAAgLUIOgAAwFoEHQAAYC2CDgAAsJYVQeftt9/Wueeeq379+umPf/yj0+UAAIAQEel0Ac118uRJ5ebmat26derUqZOGDBmiCRMmqEuXLk6XBgAAHNbmZ3Q2bdqkgQMHqlevXoqLi9Po0aO1evVqp8sCAAAhwPGgs379eo0dO1aJiYlyuVxauXJlnXMWLVqk1NRUdejQQZmZmdqwYYPv2L59+9SrVy/fdu/evbV3797WKB0AAIQ4x4POkSNHlJGRoYULF572+PLlyzV79mzNnTtXW7du1YgRI5STk6OysjJJkjGmzve4XK56X+/48ePyer1+XwAAwE6Or9HJyclRTk5Ovcfz8/M1bdo0TZ8+XZK0YMECrV69WosXL1ZeXp569erlN4OzZ88eXXjhhfWOl5eXpwcffDB4DQCW6zPnHUnS1/PHnPGcM53npJ/WeConam7M++qE071PTamxoe8/tedQ/r1pys/n1J5DrZfWcLqfZaj8jjs+o9OQqqoqFRUVadSoUX77R40apY0bN0qSLrjgAu3YsUN79+7VoUOHtGrVKl199dX1jnn//fersrLS97V79+4W7QEAADjH8Rmdhhw4cEDV1dVKSEjw25+QkKD9+/dLkiIjI/Xkk08qOztbNTU1uvfee9W1a9d6x4yOjlZ0dHSL1g0AAEJDSAedWqeuuTHG+O0bN26cxo0b19plAQCAEBfSH11169ZNERERvtmbWhUVFXVmeQAAAE4V0kEnKipKmZmZKigo8NtfUFCgiy++uFljezwepaWlKSsrq1njAACA0OX4R1eHDx/Wl19+6dsuLS1VcXGxunTpouTkZOXm5mry5MkaOnSohg0bpiVLlqisrEwzZsxo1uu63W653W55vV7Fx8c3tw0AABCCHA86mzdvVnZ2tm87NzdXkjR16lQtW7ZMEydO1MGDBzVv3jyVl5crPT1dq1atUkpKilMlAwCANsLxoDNy5MjT3vTvp2bOnKmZM2e2UkUAAMAWIb1GpyWxRgcAAPuFbdBxu90qKSlRYWGh06UAAIAWErZBBwAA2I+gAwAArEXQAQAA1iLoAAAAa4Vt0OGqKwAA7Of4fXScUntn5MrKSnXu3Flerzfor1Fz/GjQx4Qzan8/2uLP9Ke/26fWf7rf+/p6bGicM41Z32s05n09dbyfnlvfa53unObWHGynvgdOaczPoqEam/L9DZ3r9Ptwqqb8fBrz35XtGvpvrqXej9pxz3QvPpc50xmW27Nnj5KSkpwuAwAABGD37t3q3bt3vcfDPujU1NRo3759iouLk8vlCtq4Xq9XSUlJ2r17tzp16hS0cUMV/dov3HqmX/uFW8+29WuM0aFDh5SYmKh27epfiRO2H13VateuXYNJsLk6depkxS9UY9Gv/cKtZ/q1X7j1bFO/jXkod9guRgYAAPYj6AAAAGsRdFpIdHS0HnjgAUVHRztdSqugX/uFW8/0a79w6znc+q0V9ouRAQCAvZjRAQAA1iLoAAAAaxF0AACAtQg6AADAWgSdFrBo0SKlpqaqQ4cOyszM1IYNG5wuKSB5eXnKyspSXFycunfvrl/+8pfauXOn3znGGP3+979XYmKiYmJiNHLkSH366ad+5xw/fly33367unXrpo4dO2rcuHHas2dPa7YSkLy8PLlcLs2ePdu3z7Z+9+7dq0mTJqlr166KjY3Vz3/+cxUVFfmO29bvyZMn9dvf/lapqamKiYlR3759NW/ePNXU1PjOacs9r1+/XmPHjlViYqJcLpdWrlzpdzxYvX3//feaPHmy4uPjFR8fr8mTJ+uHH35o4e5Or6GeT5w4ofvuu0+DBg1Sx44dlZiYqClTpmjfvn1+Y7Slns/0M/6pW2+9VS6XSwsWLPDb35b6DQqDoHr55ZdN+/btzXPPPWdKSkrMrFmzTMeOHc2uXbucLq3Jrr76arN06VKzY8cOU1xcbMaMGWOSk5PN4cOHfefMnz/fxMXFmddff91s377dTJw40fTs2dN4vV7fOTNmzDC9evUyBQUFZsuWLSY7O9tkZGSYkydPOtFWo2zatMn06dPHDB482MyaNcu336Z+v/vuO5OSkmJuvPFG849//MOUlpaaNWvWmC+//NJ3jk39GmPMww8/bLp27WrefvttU1paal599VVz1llnmQULFvjOacs9r1q1ysydO9e8/vrrRpJ54403/I4Hq7drrrnGpKenm40bN5qNGzea9PR0c+2117ZWm34a6vmHH34wV155pVm+fLn57LPPzMcff2wuvPBCk5mZ6TdGW+r5TD/jWm+88YbJyMgwiYmJ5qmnnvI71pb6DQaCTpBdcMEFZsaMGX77BgwYYObMmeNQRcFTUVFhJJkPPvjAGGNMTU2N6dGjh5k/f77vnGPHjpn4+HjzzDPPGGN+/Iemffv25uWXX/ads3fvXtOuXTvz3nvvtW4DjXTo0CHTr18/U1BQYC677DJf0LGt3/vuu88MHz683uO29WuMMWPGjDE333yz374JEyaYSZMmGWPs6vnUP4LB6q2kpMRIMn//+99953z88cdGkvnss89auKuGNfSHv9amTZuMJN//+WzLPdfX7549e0yvXr3Mjh07TEpKil/Qacv9BoqProKoqqpKRUVFGjVqlN/+UaNGaePGjQ5VFTyVlZWSpC5dukiSSktLtX//fr9+o6Ojddlll/n6LSoq0okTJ/zOSUxMVHp6esi+J263W2PGjNGVV17pt9+2ft98800NHTpU1113nbp3767zzz9fzz33nO+4bf1K0vDhw/X+++/r888/lyRt27ZNH374oUaPHi3Jzp5rBau3jz/+WPHx8brwwgt951x00UWKj48P6f5rVVZWyuVyqXPnzpLs67mmpkaTJ0/WPffco4EDB9Y5blu/jRH2D/UMpgMHDqi6uloJCQl++xMSErR//36HqgoOY4xyc3M1fPhwpaenS5Kvp9P1u2vXLt85UVFROvvss+ucE4rvycsvv6wtW7aosLCwzjHb+v3nP/+pxYsXKzc3V//5n/+pTZs26Y477lB0dLSmTJliXb+SdN9996myslIDBgxQRESEqqur9cgjj+hXv/qVJPt+xj8VrN7279+v7t271xm/e/fuId2/JB07dkxz5szRr3/9a99DLW3r+bHHHlNkZKTuuOOO0x63rd/GIOi0AJfL5bdtjKmzr6257bbb9Mknn+jDDz+scyyQfkPxPdm9e7dmzZqlv/71r+rQoUO959nSb01NjYYOHapHH31UknT++efr008/1eLFizVlyhTfebb0K0nLly/XSy+9pD//+c8aOHCgiouLNXv2bCUmJmrq1Km+82zq+VTB6O1054d6/ydOnND111+vmpoaLVq06Iznt8Wei4qK9Ic//EFbtmxpcl1tsd/G4qOrIOrWrZsiIiLqJN6Kioo6/y+qLbn99tv15ptvat26derdu7dvf48ePSSpwX579Oihqqoqff/99/WeEyqKiopUUVGhzMxMRUZGKjIyUh988IGefvppRUZG+uq1pd+ePXsqLS3Nb995552nsrIySfb9fCXpnnvu0Zw5c3T99ddr0KBBmjx5su68807l5eVJsrPnWsHqrUePHvrmm2/qjP/tt9+GbP8nTpzQv//7v6u0tFQFBQW+2RzJrp43bNigiooKJScn+/4N27Vrl+666y716dNHkl39NhZBJ4iioqKUmZmpgoICv/0FBQW6+OKLHaoqcMYY3XbbbVqxYoXWrl2r1NRUv+Opqanq0aOHX79VVVX64IMPfP1mZmaqffv2fueUl5drx44dIfeeXHHFFdq+fbuKi4t9X0OHDtUNN9yg4uJi9e3b16p+L7nkkjq3C/j888+VkpIiyb6fryQdPXpU7dr5/7MXERHhu7zcxp5rBau3YcOGqbKyUps2bfKd849//EOVlZUh2X9tyPniiy+0Zs0ade3a1e+4TT1PnjxZn3zyid+/YYmJibrnnnu0evVqSXb122itvfrZdrWXlz///POmpKTEzJ4923Ts2NF8/fXXTpfWZL/5zW9MfHy8+dvf/mbKy8t9X0ePHvWdM3/+fBMfH29WrFhhtm/fbn71q1+d9nLV3r17mzVr1pgtW7aYyy+/PCQuxW2Mn151ZYxd/W7atMlERkaaRx55xHzxxRfmf//3f01sbKx56aWXfOfY1K8xxkydOtX06tXLd3n5ihUrTLdu3cy9997rO6ct93zo0CGzdetWs3XrViPJ5Ofnm61bt/quMApWb9dcc40ZPHiw+fjjj83HH39sBg0a5Nilxw31fOLECTNu3DjTu3dvU1xc7Pfv2PHjx31jtKWez/QzPtWpV10Z07b6DQaCTgvweDwmJSXFREVFmSFDhvgux25rJJ32a+nSpb5zampqzAMPPGB69OhhoqOjzaWXXmq2b9/uN86//vUvc9ttt5kuXbqYmJgYc+2115qysrJW7iYwpwYd2/p96623THp6uomOjjYDBgwwS5Ys8TtuW79er9fMmjXLJCcnmw4dOpi+ffuauXPn+v3Ra8s9r1u37rT/zU6dOtUYE7zeDh48aG644QYTFxdn4uLizA033GC+//77VurSX0M9l5aW1vvv2Lp163xjtKWez/QzPtXpgk5b6jcYXMYY0xozRwAAAK2NNToAAMBaBB0AAGAtgg4AALAWQQcAAFiLoAMAAKxF0AEAANYi6AAAAGsRdAAAgLUIOgAAwFoEHQAAYC2CDgAAsBZBBwAAWOv/Ad9E3cKYT1RaAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Get number of nodes and edges and plot histogram of edges showing number of nodes with X edges \n",
    "total_gene_a = reactome_sif['PARTICIPANT_A'].tolist()\n",
    "total_gene_b = reactome_sif['PARTICIPANT_B'].tolist()\n",
    "total_genes = total_gene_a + total_gene_b\n",
    "\n",
    "total_genes_df = pd.DataFrame(total_genes, columns=['gene'])\n",
    "\n",
    "total_genes_df['edges'] = total_genes_df['gene'].map(total_genes_df['gene'].value_counts())\n",
    "\n",
    "print(len(total_gene_a))\n",
    "print(len(total_gene_b))\n",
    "print(len(total_genes))\n",
    "\n",
    "total_genes_df.head()\n",
    "\n",
    "unique_genes_df = total_genes_df.drop_duplicates('gene') \n",
    "unique_genes_df.sort_values(by='edges', ascending=False)\n",
    "\n",
    "fig = unique_genes_df.plot(kind='hist', bins=200)\n",
    "\n",
    "plt.yscale('log')\n",
    "plt.show()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:31.021847300Z",
     "start_time": "2023-12-13T16:52:30.498905800Z"
    }
   },
   "id": "2fa02ebdb6baf5dd"
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10055\n",
      "10907\n",
      "12323\n"
     ]
    }
   ],
   "source": [
    "# See how many nodes there are\n",
    "unique_gene_a = reactome_sif['PARTICIPANT_A'].unique().tolist()\n",
    "unique_gene_b = reactome_sif['PARTICIPANT_B'].unique().tolist()\n",
    "unique_gene_all = set(unique_gene_a + unique_gene_b)\n",
    "\n",
    "print(len(unique_gene_a))\n",
    "print(len(unique_gene_b))\n",
    "print(len(unique_gene_all))\n",
    "\n",
    "#df = pd.DataFrame.from_dict(count, orient='index')\n",
    "#df.plot(kind='bar')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:33.334911200Z",
     "start_time": "2023-12-13T16:52:33.288177400Z"
    }
   },
   "id": "d7c55fd4c10e2704"
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "outputs": [
    {
     "data": {
      "text/plain": "329941"
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_no_duplicates = reactome_sif.drop_duplicates(subset=['PARTICIPANT_A','PARTICIPANT_B'])\n",
    "len(reactome_no_duplicates)\n",
    "\n",
    "#reactome_no_duplicates[(reactome_no_duplicates['PARTICIPANT_A']== 'A1CF') & (reactome_no_duplicates['PARTICIPANT_B']=='APOBEC3A')]"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:41.344011200Z",
     "start_time": "2023-12-13T16:52:41.280578500Z"
    }
   },
   "id": "16d8aede53e99ca8"
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "outputs": [
    {
     "data": {
      "text/plain": "  PARTICIPANT_A INTERACTION_TYPE PARTICIPANT_B\n0          A1CF  in-complex-with       APOBEC2\n1          A1CF  in-complex-with      APOBEC3A\n2          A1CF  in-complex-with      APOBEC3B\n3          A1CF  in-complex-with      APOBEC3C\n4          A1CF  in-complex-with      APOBEC3H",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>PARTICIPANT_A</th>\n      <th>INTERACTION_TYPE</th>\n      <th>PARTICIPANT_B</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC2</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3A</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3B</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3C</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3H</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_no_duplicates.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T16:52:54.135146Z",
     "start_time": "2023-12-13T16:52:54.124776700Z"
    }
   },
   "id": "93e9e2268c2aae64"
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[BiomartServer:'http://useast.ensembl.org/biomart/martservice'] Fetching datasets\n",
      "[BiomartServer:'http://useast.ensembl.org/biomart/martservice'] Fetching databases\n",
      "[BiomartDatabase:'Ensembl Genes 110'] Fetching datasets\n",
      "[BiomartDatabase:'Mouse strains 110'] Fetching datasets\n",
      "[BiomartDatabase:'Sequence'] Fetching datasets\n",
      "[BiomartDatabase:'Ontology'] Fetching datasets\n",
      "[BiomartDatabase:'Genomic features 110'] Fetching datasets\n",
      "[BiomartDatabase:'Ensembl Variation 110'] Fetching datasets\n",
      "[BiomartDatabase:'Ensembl Regulation 110'] Fetching datasets\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Fetching attributes\n",
      "{'3_utr_end': '3' UTR end' (default: False),\n",
      " '3_utr_start': '3' UTR start' (default: False),\n",
      " '3utr': '3' UTR' (default: False),\n",
      " '5_utr_end': '5' UTR end' (default: False),\n",
      " '5_utr_start': '5' UTR start' (default: False),\n",
      " '5utr': '5' UTR' (default: False),\n",
      " 'abrachyrhynchus_homolog_associated_gene_name': 'Pink-footed goose gene name' (default: False),\n",
      " 'abrachyrhynchus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_chrom_end': 'Pink-footed goose chromosome/scaffold end (bp)' (default: False),\n",
      " 'abrachyrhynchus_homolog_chrom_start': 'Pink-footed goose chromosome/scaffold start (bp)' (default: False),\n",
      " 'abrachyrhynchus_homolog_chromosome': 'Pink-footed goose chromosome/scaffold name' (default: False),\n",
      " 'abrachyrhynchus_homolog_ensembl_gene': 'Pink-footed goose gene stable ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_ensembl_peptide': 'Pink-footed goose protein or transcript stable ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_goc_score': 'Pink-footed goose Gene-order conservation score' (default: False),\n",
      " 'abrachyrhynchus_homolog_orthology_confidence': 'Pink-footed goose orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'abrachyrhynchus_homolog_orthology_type': 'Pink-footed goose homology type' (default: False),\n",
      " 'abrachyrhynchus_homolog_perc_id': '%id. target Pink-footed goose gene identical to query gene' (default: False),\n",
      " 'abrachyrhynchus_homolog_perc_id_r1': '%id. query gene identical to target Pink-footed goose gene' (default: False),\n",
      " 'abrachyrhynchus_homolog_subtype': 'Last common ancestor with Pink-footed goose' (default: False),\n",
      " 'abrachyrhynchus_homolog_wga_coverage': 'Pink-footed goose Whole-genome alignment coverage' (default: False),\n",
      " 'acalliptera_homolog_associated_gene_name': 'Eastern happy gene name' (default: False),\n",
      " 'acalliptera_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acalliptera_homolog_chrom_end': 'Eastern happy chromosome/scaffold end (bp)' (default: False),\n",
      " 'acalliptera_homolog_chrom_start': 'Eastern happy chromosome/scaffold start (bp)' (default: False),\n",
      " 'acalliptera_homolog_chromosome': 'Eastern happy chromosome/scaffold name' (default: False),\n",
      " 'acalliptera_homolog_ensembl_gene': 'Eastern happy gene stable ID' (default: False),\n",
      " 'acalliptera_homolog_ensembl_peptide': 'Eastern happy protein or transcript stable ID' (default: False),\n",
      " 'acalliptera_homolog_goc_score': 'Eastern happy Gene-order conservation score' (default: False),\n",
      " 'acalliptera_homolog_orthology_confidence': 'Eastern happy orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acalliptera_homolog_orthology_type': 'Eastern happy homology type' (default: False),\n",
      " 'acalliptera_homolog_perc_id': '%id. target Eastern happy gene identical to query gene' (default: False),\n",
      " 'acalliptera_homolog_perc_id_r1': '%id. query gene identical to target Eastern happy gene' (default: False),\n",
      " 'acalliptera_homolog_subtype': 'Last common ancestor with Eastern happy' (default: False),\n",
      " 'acalliptera_homolog_wga_coverage': 'Eastern happy Whole-genome alignment coverage' (default: False),\n",
      " 'acarolinensis_homolog_associated_gene_name': 'Green anole gene name' (default: False),\n",
      " 'acarolinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acarolinensis_homolog_chrom_end': 'Green anole chromosome/scaffold end (bp)' (default: False),\n",
      " 'acarolinensis_homolog_chrom_start': 'Green anole chromosome/scaffold start (bp)' (default: False),\n",
      " 'acarolinensis_homolog_chromosome': 'Green anole chromosome/scaffold name' (default: False),\n",
      " 'acarolinensis_homolog_ensembl_gene': 'Green anole gene stable ID' (default: False),\n",
      " 'acarolinensis_homolog_ensembl_peptide': 'Green anole protein or transcript stable ID' (default: False),\n",
      " 'acarolinensis_homolog_goc_score': 'Green anole Gene-order conservation score' (default: False),\n",
      " 'acarolinensis_homolog_orthology_confidence': 'Green anole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acarolinensis_homolog_orthology_type': 'Green anole homology type' (default: False),\n",
      " 'acarolinensis_homolog_perc_id': '%id. target Green anole gene identical to query gene' (default: False),\n",
      " 'acarolinensis_homolog_perc_id_r1': '%id. query gene identical to target Green anole gene' (default: False),\n",
      " 'acarolinensis_homolog_subtype': 'Last common ancestor with Green anole' (default: False),\n",
      " 'acchrysaetos_homolog_associated_gene_name': 'Golden eagle gene name' (default: False),\n",
      " 'acchrysaetos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acchrysaetos_homolog_chrom_end': 'Golden eagle chromosome/scaffold end (bp)' (default: False),\n",
      " 'acchrysaetos_homolog_chrom_start': 'Golden eagle chromosome/scaffold start (bp)' (default: False),\n",
      " 'acchrysaetos_homolog_chromosome': 'Golden eagle chromosome/scaffold name' (default: False),\n",
      " 'acchrysaetos_homolog_ensembl_gene': 'Golden eagle gene stable ID' (default: False),\n",
      " 'acchrysaetos_homolog_ensembl_peptide': 'Golden eagle protein or transcript stable ID' (default: False),\n",
      " 'acchrysaetos_homolog_goc_score': 'Golden eagle Gene-order conservation score' (default: False),\n",
      " 'acchrysaetos_homolog_orthology_confidence': 'Golden eagle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acchrysaetos_homolog_orthology_type': 'Golden eagle homology type' (default: False),\n",
      " 'acchrysaetos_homolog_perc_id': '%id. target Golden eagle gene identical to query gene' (default: False),\n",
      " 'acchrysaetos_homolog_perc_id_r1': '%id. query gene identical to target Golden eagle gene' (default: False),\n",
      " 'acchrysaetos_homolog_subtype': 'Last common ancestor with Golden eagle' (default: False),\n",
      " 'acchrysaetos_homolog_wga_coverage': 'Golden eagle Whole-genome alignment coverage' (default: False),\n",
      " 'acitrinellus_homolog_associated_gene_name': 'Midas cichlid gene name' (default: False),\n",
      " 'acitrinellus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acitrinellus_homolog_chrom_end': 'Midas cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'acitrinellus_homolog_chrom_start': 'Midas cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'acitrinellus_homolog_chromosome': 'Midas cichlid chromosome/scaffold name' (default: False),\n",
      " 'acitrinellus_homolog_ensembl_gene': 'Midas cichlid gene stable ID' (default: False),\n",
      " 'acitrinellus_homolog_ensembl_peptide': 'Midas cichlid protein or transcript stable ID' (default: False),\n",
      " 'acitrinellus_homolog_goc_score': 'Midas cichlid Gene-order conservation score' (default: False),\n",
      " 'acitrinellus_homolog_orthology_confidence': 'Midas cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acitrinellus_homolog_orthology_type': 'Midas cichlid homology type' (default: False),\n",
      " 'acitrinellus_homolog_perc_id': '%id. target Midas cichlid gene identical to query gene' (default: False),\n",
      " 'acitrinellus_homolog_perc_id_r1': '%id. query gene identical to target Midas cichlid gene' (default: False),\n",
      " 'acitrinellus_homolog_subtype': 'Last common ancestor with Midas cichlid' (default: False),\n",
      " 'acitrinellus_homolog_wga_coverage': 'Midas cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'affy_hc_g110': 'AFFY HC G110 probe' (default: False),\n",
      " 'affy_hg_focus': 'AFFY HG Focus probe' (default: False),\n",
      " 'affy_hg_u133_plus_2': 'AFFY HG U133 Plus 2 probe' (default: False),\n",
      " 'affy_hg_u133a_2': 'AFFY HG U133A 2 probe' (default: False),\n",
      " 'affy_hg_u133b': 'AFFY HG U133B probe' (default: False),\n",
      " 'affy_hg_u95a': 'AFFY HG U95A probe' (default: False),\n",
      " 'affy_hg_u95av2': 'AFFY HG U95Av2 probe' (default: False),\n",
      " 'affy_hg_u95b': 'AFFY HG U95B probe' (default: False),\n",
      " 'affy_hg_u95c': 'AFFY HG U95C probe' (default: False),\n",
      " 'affy_hg_u95d': 'AFFY HG U95D probe' (default: False),\n",
      " 'affy_hg_u95e': 'AFFY HG U95E probe' (default: False),\n",
      " 'affy_ht_hg_u133_plus_pm': 'AFFY HT HG U133 Plus PM probe' (default: False),\n",
      " 'affy_hta_2_0': 'AFFY HTA 2 0 probe' (default: False),\n",
      " 'affy_huex_1_0_st_v2': 'AFFY HuEx 1 0 st v2 probe' (default: False),\n",
      " 'affy_hugene_1_0_st_v1': 'AFFY HuGene 1 0 st v1 probe' (default: False),\n",
      " 'affy_hugene_2_0_st_v1': 'AFFY HuGene 2 0 st v1 probe' (default: False),\n",
      " 'affy_hugene_2_1_st_v1': 'AFFY HuGene 2 1 st v1 probe' (default: False),\n",
      " 'affy_hugenefl': 'AFFY HuGeneFL probe' (default: False),\n",
      " 'affy_primeview': 'AFFY PrimeView probe' (default: False),\n",
      " 'affy_u133_x3p': 'AFFY U133 X3P probe' (default: False),\n",
      " 'agilent_cgh_44b': 'AGILENT CGH 44b probe' (default: False),\n",
      " 'agilent_gpl19072': 'AGILENT GPL19072 probe' (default: False),\n",
      " 'agilent_gpl26966': 'AGILENT GPL26966 probe' (default: False),\n",
      " 'agilent_gpl6848': 'AGILENT GPL6848 probe' (default: False),\n",
      " 'agilent_sureprint_g3_ge_8x60k': 'AGILENT SurePrint G3 GE 8x60k probe' (default: False),\n",
      " 'agilent_sureprint_g3_ge_8x60k_v2': 'AGILENT SurePrint G3 GE 8x60k v2 probe' (default: False),\n",
      " 'agilent_wholegenome': 'AGILENT WholeGenome probe' (default: False),\n",
      " 'agilent_wholegenome_4x44k_v1': 'AGILENT WholeGenome 4x44k v1 probe' (default: False),\n",
      " 'agilent_wholegenome_4x44k_v2': 'AGILENT WholeGenome 4x44k v2 probe' (default: False),\n",
      " 'allele': 'Variant alleles' (default: False),\n",
      " 'allele_string_2076': 'Consequence specific allele' (default: False),\n",
      " 'alphafold': 'AFDB-ENSP mappings' (default: False),\n",
      " 'alphafold_end': 'AFDB-ENSP mappings end' (default: False),\n",
      " 'alphafold_start': 'AFDB-ENSP mappings start' (default: False),\n",
      " 'amelanoleuca_homolog_associated_gene_name': 'Giant panda gene name' (default: False),\n",
      " 'amelanoleuca_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'amelanoleuca_homolog_chrom_end': 'Giant panda chromosome/scaffold end (bp)' (default: False),\n",
      " 'amelanoleuca_homolog_chrom_start': 'Giant panda chromosome/scaffold start (bp)' (default: False),\n",
      " 'amelanoleuca_homolog_chromosome': 'Giant panda chromosome/scaffold name' (default: False),\n",
      " 'amelanoleuca_homolog_ensembl_gene': 'Giant panda gene stable ID' (default: False),\n",
      " 'amelanoleuca_homolog_ensembl_peptide': 'Giant panda protein or transcript stable ID' (default: False),\n",
      " 'amelanoleuca_homolog_goc_score': 'Giant panda Gene-order conservation score' (default: False),\n",
      " 'amelanoleuca_homolog_orthology_confidence': 'Giant panda orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'amelanoleuca_homolog_orthology_type': 'Giant panda homology type' (default: False),\n",
      " 'amelanoleuca_homolog_perc_id': '%id. target Giant panda gene identical to query gene' (default: False),\n",
      " 'amelanoleuca_homolog_perc_id_r1': '%id. query gene identical to target Giant panda gene' (default: False),\n",
      " 'amelanoleuca_homolog_subtype': 'Last common ancestor with Giant panda' (default: False),\n",
      " 'amexicanus_homolog_associated_gene_name': 'Mexican tetra gene name' (default: False),\n",
      " 'amexicanus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'amexicanus_homolog_chrom_end': 'Mexican tetra chromosome/scaffold end (bp)' (default: False),\n",
      " 'amexicanus_homolog_chrom_start': 'Mexican tetra chromosome/scaffold start (bp)' (default: False),\n",
      " 'amexicanus_homolog_chromosome': 'Mexican tetra chromosome/scaffold name' (default: False),\n",
      " 'amexicanus_homolog_ensembl_gene': 'Mexican tetra gene stable ID' (default: False),\n",
      " 'amexicanus_homolog_ensembl_peptide': 'Mexican tetra protein or transcript stable ID' (default: False),\n",
      " 'amexicanus_homolog_goc_score': 'Mexican tetra Gene-order conservation score' (default: False),\n",
      " 'amexicanus_homolog_orthology_confidence': 'Mexican tetra orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'amexicanus_homolog_orthology_type': 'Mexican tetra homology type' (default: False),\n",
      " 'amexicanus_homolog_perc_id': '%id. target Mexican tetra gene identical to query gene' (default: False),\n",
      " 'amexicanus_homolog_perc_id_r1': '%id. query gene identical to target Mexican tetra gene' (default: False),\n",
      " 'amexicanus_homolog_subtype': 'Last common ancestor with Mexican tetra' (default: False),\n",
      " 'amexicanus_homolog_wga_coverage': 'Mexican tetra Whole-genome alignment coverage' (default: False),\n",
      " 'anancymaae_homolog_associated_gene_name': 'Ma's night monkey gene name' (default: False),\n",
      " 'anancymaae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'anancymaae_homolog_chrom_end': 'Ma's night monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'anancymaae_homolog_chrom_start': 'Ma's night monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'anancymaae_homolog_chromosome': 'Ma's night monkey chromosome/scaffold name' (default: False),\n",
      " 'anancymaae_homolog_ensembl_gene': 'Ma's night monkey gene stable ID' (default: False),\n",
      " 'anancymaae_homolog_ensembl_peptide': 'Ma's night monkey protein or transcript stable ID' (default: False),\n",
      " 'anancymaae_homolog_goc_score': 'Ma's night monkey Gene-order conservation score' (default: False),\n",
      " 'anancymaae_homolog_orthology_confidence': 'Ma's night monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'anancymaae_homolog_orthology_type': 'Ma's night monkey homology type' (default: False),\n",
      " 'anancymaae_homolog_perc_id': '%id. target Ma's night monkey gene identical to query gene' (default: False),\n",
      " 'anancymaae_homolog_perc_id_r1': '%id. query gene identical to target Ma's night monkey gene' (default: False),\n",
      " 'anancymaae_homolog_subtype': 'Last common ancestor with Ma's night monkey' (default: False),\n",
      " 'anancymaae_homolog_wga_coverage': 'Ma's night monkey Whole-genome alignment coverage' (default: False),\n",
      " 'aocellaris_homolog_associated_gene_name': 'Clown anemonefish gene name' (default: False),\n",
      " 'aocellaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'aocellaris_homolog_chrom_end': 'Clown anemonefish chromosome/scaffold end (bp)' (default: False),\n",
      " 'aocellaris_homolog_chrom_start': 'Clown anemonefish chromosome/scaffold start (bp)' (default: False),\n",
      " 'aocellaris_homolog_chromosome': 'Clown anemonefish chromosome/scaffold name' (default: False),\n",
      " 'aocellaris_homolog_ensembl_gene': 'Clown anemonefish gene stable ID' (default: False),\n",
      " 'aocellaris_homolog_ensembl_peptide': 'Clown anemonefish protein or transcript stable ID' (default: False),\n",
      " 'aocellaris_homolog_goc_score': 'Clown anemonefish Gene-order conservation score' (default: False),\n",
      " 'aocellaris_homolog_orthology_confidence': 'Clown anemonefish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'aocellaris_homolog_orthology_type': 'Clown anemonefish homology type' (default: False),\n",
      " 'aocellaris_homolog_perc_id': '%id. target Clown anemonefish gene identical to query gene' (default: False),\n",
      " 'aocellaris_homolog_perc_id_r1': '%id. query gene identical to target Clown anemonefish gene' (default: False),\n",
      " 'aocellaris_homolog_subtype': 'Last common ancestor with Clown anemonefish' (default: False),\n",
      " 'aocellaris_homolog_wga_coverage': 'Clown anemonefish Whole-genome alignment coverage' (default: False),\n",
      " 'apercula_homolog_associated_gene_name': 'Orange clownfish gene name' (default: False),\n",
      " 'apercula_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'apercula_homolog_chrom_end': 'Orange clownfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'apercula_homolog_chrom_start': 'Orange clownfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'apercula_homolog_chromosome': 'Orange clownfish chromosome/scaffold name' (default: False),\n",
      " 'apercula_homolog_ensembl_gene': 'Orange clownfish gene stable ID' (default: False),\n",
      " 'apercula_homolog_ensembl_peptide': 'Orange clownfish protein or transcript stable ID' (default: False),\n",
      " 'apercula_homolog_goc_score': 'Orange clownfish Gene-order conservation score' (default: False),\n",
      " 'apercula_homolog_orthology_confidence': 'Orange clownfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'apercula_homolog_orthology_type': 'Orange clownfish homology type' (default: False),\n",
      " 'apercula_homolog_perc_id': '%id. target Orange clownfish gene identical to query gene' (default: False),\n",
      " 'apercula_homolog_perc_id_r1': '%id. query gene identical to target Orange clownfish gene' (default: False),\n",
      " 'apercula_homolog_subtype': 'Last common ancestor with Orange clownfish' (default: False),\n",
      " 'apercula_homolog_wga_coverage': 'Orange clownfish Whole-genome alignment coverage' (default: False),\n",
      " 'apolyacanthus_homolog_associated_gene_name': 'Spiny chromis gene name' (default: False),\n",
      " 'apolyacanthus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'apolyacanthus_homolog_chrom_end': 'Spiny chromis chromosome/scaffold end (bp)' (default: False),\n",
      " 'apolyacanthus_homolog_chrom_start': 'Spiny chromis chromosome/scaffold start (bp)' (default: False),\n",
      " 'apolyacanthus_homolog_chromosome': 'Spiny chromis chromosome/scaffold name' (default: False),\n",
      " 'apolyacanthus_homolog_ensembl_gene': 'Spiny chromis gene stable ID' (default: False),\n",
      " 'apolyacanthus_homolog_ensembl_peptide': 'Spiny chromis protein or transcript stable ID' (default: False),\n",
      " 'apolyacanthus_homolog_goc_score': 'Spiny chromis Gene-order conservation score' (default: False),\n",
      " 'apolyacanthus_homolog_orthology_confidence': 'Spiny chromis orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'apolyacanthus_homolog_orthology_type': 'Spiny chromis homology type' (default: False),\n",
      " 'apolyacanthus_homolog_perc_id': '%id. target Spiny chromis gene identical to query gene' (default: False),\n",
      " 'apolyacanthus_homolog_perc_id_r1': '%id. query gene identical to target Spiny chromis gene' (default: False),\n",
      " 'apolyacanthus_homolog_subtype': 'Last common ancestor with Spiny chromis' (default: False),\n",
      " 'apolyacanthus_homolog_wga_coverage': 'Spiny chromis Whole-genome alignment coverage' (default: False),\n",
      " 'applatyrhynchos_homolog_associated_gene_name': 'Duck gene name' (default: False),\n",
      " 'applatyrhynchos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'applatyrhynchos_homolog_chrom_end': 'Duck chromosome/scaffold end (bp)' (default: False),\n",
      " 'applatyrhynchos_homolog_chrom_start': 'Duck chromosome/scaffold start (bp)' (default: False),\n",
      " 'applatyrhynchos_homolog_chromosome': 'Duck chromosome/scaffold name' (default: False),\n",
      " 'applatyrhynchos_homolog_ensembl_gene': 'Duck gene stable ID' (default: False),\n",
      " 'applatyrhynchos_homolog_ensembl_peptide': 'Duck protein or transcript stable ID' (default: False),\n",
      " 'applatyrhynchos_homolog_goc_score': 'Duck Gene-order conservation score' (default: False),\n",
      " 'applatyrhynchos_homolog_orthology_confidence': 'Duck orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'applatyrhynchos_homolog_orthology_type': 'Duck homology type' (default: False),\n",
      " 'applatyrhynchos_homolog_perc_id': '%id. target Duck gene identical to query gene' (default: False),\n",
      " 'applatyrhynchos_homolog_perc_id_r1': '%id. query gene identical to target Duck gene' (default: False),\n",
      " 'applatyrhynchos_homolog_subtype': 'Last common ancestor with Duck' (default: False),\n",
      " 'applatyrhynchos_homolog_wga_coverage': 'Duck Whole-genome alignment coverage' (default: False),\n",
      " 'arrayexpress': 'Expression Atlas ID' (default: False),\n",
      " 'atestudineus_homolog_associated_gene_name': 'Climbing perch gene name' (default: False),\n",
      " 'atestudineus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'atestudineus_homolog_chrom_end': 'Climbing perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'atestudineus_homolog_chrom_start': 'Climbing perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'atestudineus_homolog_chromosome': 'Climbing perch chromosome/scaffold name' (default: False),\n",
      " 'atestudineus_homolog_ensembl_gene': 'Climbing perch gene stable ID' (default: False),\n",
      " 'atestudineus_homolog_ensembl_peptide': 'Climbing perch protein or transcript stable ID' (default: False),\n",
      " 'atestudineus_homolog_goc_score': 'Climbing perch Gene-order conservation score' (default: False),\n",
      " 'atestudineus_homolog_orthology_confidence': 'Climbing perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'atestudineus_homolog_orthology_type': 'Climbing perch homology type' (default: False),\n",
      " 'atestudineus_homolog_perc_id': '%id. target Climbing perch gene identical to query gene' (default: False),\n",
      " 'atestudineus_homolog_perc_id_r1': '%id. query gene identical to target Climbing perch gene' (default: False),\n",
      " 'atestudineus_homolog_subtype': 'Last common ancestor with Climbing perch' (default: False),\n",
      " 'band': 'Karyotype band' (default: False),\n",
      " 'bbbison_homolog_associated_gene_name': 'American bison gene name' (default: False),\n",
      " 'bbbison_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bbbison_homolog_chrom_end': 'American bison chromosome/scaffold end (bp)' (default: False),\n",
      " 'bbbison_homolog_chrom_start': 'American bison chromosome/scaffold start (bp)' (default: False),\n",
      " 'bbbison_homolog_chromosome': 'American bison chromosome/scaffold name' (default: False),\n",
      " 'bbbison_homolog_ensembl_gene': 'American bison gene stable ID' (default: False),\n",
      " 'bbbison_homolog_ensembl_peptide': 'American bison protein or transcript stable ID' (default: False),\n",
      " 'bbbison_homolog_goc_score': 'American bison Gene-order conservation score' (default: False),\n",
      " 'bbbison_homolog_orthology_confidence': 'American bison orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bbbison_homolog_orthology_type': 'American bison homology type' (default: False),\n",
      " 'bbbison_homolog_perc_id': '%id. target American bison gene identical to query gene' (default: False),\n",
      " 'bbbison_homolog_perc_id_r1': '%id. query gene identical to target American bison gene' (default: False),\n",
      " 'bbbison_homolog_subtype': 'Last common ancestor with American bison' (default: False),\n",
      " 'bbbison_homolog_wga_coverage': 'American bison Whole-genome alignment coverage' (default: False),\n",
      " 'bgrunniens_homolog_associated_gene_name': 'Domestic yak gene name' (default: False),\n",
      " 'bgrunniens_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bgrunniens_homolog_chrom_end': 'Domestic yak chromosome/scaffold end (bp)' (default: False),\n",
      " 'bgrunniens_homolog_chrom_start': 'Domestic yak chromosome/scaffold start (bp)' (default: False),\n",
      " 'bgrunniens_homolog_chromosome': 'Domestic yak chromosome/scaffold name' (default: False),\n",
      " 'bgrunniens_homolog_ensembl_gene': 'Domestic yak gene stable ID' (default: False),\n",
      " 'bgrunniens_homolog_ensembl_peptide': 'Domestic yak protein or transcript stable ID' (default: False),\n",
      " 'bgrunniens_homolog_goc_score': 'Domestic yak Gene-order conservation score' (default: False),\n",
      " 'bgrunniens_homolog_orthology_confidence': 'Domestic yak orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bgrunniens_homolog_orthology_type': 'Domestic yak homology type' (default: False),\n",
      " 'bgrunniens_homolog_perc_id': '%id. target Domestic yak gene identical to query gene' (default: False),\n",
      " 'bgrunniens_homolog_perc_id_r1': '%id. query gene identical to target Domestic yak gene' (default: False),\n",
      " 'bgrunniens_homolog_subtype': 'Last common ancestor with Domestic yak' (default: False),\n",
      " 'bgrunniens_homolog_wga_coverage': 'Domestic yak Whole-genome alignment coverage' (default: False),\n",
      " 'bihybrid_homolog_associated_gene_name': 'Hybrid - Bos Indicus gene name' (default: False),\n",
      " 'bihybrid_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bihybrid_homolog_chrom_end': 'Hybrid - Bos Indicus chromosome/scaffold end (bp)' (default: False),\n",
      " 'bihybrid_homolog_chrom_start': 'Hybrid - Bos Indicus chromosome/scaffold start (bp)' (default: False),\n",
      " 'bihybrid_homolog_chromosome': 'Hybrid - Bos Indicus chromosome/scaffold name' (default: False),\n",
      " 'bihybrid_homolog_ensembl_gene': 'Hybrid - Bos Indicus gene stable ID' (default: False),\n",
      " 'bihybrid_homolog_ensembl_peptide': 'Hybrid - Bos Indicus protein or transcript stable ID' (default: False),\n",
      " 'bihybrid_homolog_goc_score': 'Hybrid - Bos Indicus Gene-order conservation score' (default: False),\n",
      " 'bihybrid_homolog_orthology_confidence': 'Hybrid - Bos Indicus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bihybrid_homolog_orthology_type': 'Hybrid - Bos Indicus homology type' (default: False),\n",
      " 'bihybrid_homolog_perc_id': '%id. target Hybrid - Bos Indicus gene identical to query gene' (default: False),\n",
      " 'bihybrid_homolog_perc_id_r1': '%id. query gene identical to target Hybrid - Bos Indicus gene' (default: False),\n",
      " 'bihybrid_homolog_subtype': 'Last common ancestor with Hybrid - Bos Indicus' (default: False),\n",
      " 'bihybrid_homolog_wga_coverage': 'Hybrid - Bos Indicus Whole-genome alignment coverage' (default: False),\n",
      " 'biogrid': 'BioGRID Interaction data, The General Repository for Interaction Datasets ID' (default: False),\n",
      " 'bmusculus_homolog_associated_gene_name': 'Blue whale gene name' (default: False),\n",
      " 'bmusculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bmusculus_homolog_chrom_end': 'Blue whale chromosome/scaffold end (bp)' (default: False),\n",
      " 'bmusculus_homolog_chrom_start': 'Blue whale chromosome/scaffold start (bp)' (default: False),\n",
      " 'bmusculus_homolog_chromosome': 'Blue whale chromosome/scaffold name' (default: False),\n",
      " 'bmusculus_homolog_ensembl_gene': 'Blue whale gene stable ID' (default: False),\n",
      " 'bmusculus_homolog_ensembl_peptide': 'Blue whale protein or transcript stable ID' (default: False),\n",
      " 'bmusculus_homolog_goc_score': 'Blue whale Gene-order conservation score' (default: False),\n",
      " 'bmusculus_homolog_orthology_confidence': 'Blue whale orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bmusculus_homolog_orthology_type': 'Blue whale homology type' (default: False),\n",
      " 'bmusculus_homolog_perc_id': '%id. target Blue whale gene identical to query gene' (default: False),\n",
      " 'bmusculus_homolog_perc_id_r1': '%id. query gene identical to target Blue whale gene' (default: False),\n",
      " 'bmusculus_homolog_subtype': 'Last common ancestor with Blue whale' (default: False),\n",
      " 'bmusculus_homolog_wga_coverage': 'Blue whale Whole-genome alignment coverage' (default: False),\n",
      " 'bmutus_homolog_associated_gene_name': 'Wild yak gene name' (default: False),\n",
      " 'bmutus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bmutus_homolog_chrom_end': 'Wild yak chromosome/scaffold end (bp)' (default: False),\n",
      " 'bmutus_homolog_chrom_start': 'Wild yak chromosome/scaffold start (bp)' (default: False),\n",
      " 'bmutus_homolog_chromosome': 'Wild yak chromosome/scaffold name' (default: False),\n",
      " 'bmutus_homolog_ensembl_gene': 'Wild yak gene stable ID' (default: False),\n",
      " 'bmutus_homolog_ensembl_peptide': 'Wild yak protein or transcript stable ID' (default: False),\n",
      " 'bmutus_homolog_goc_score': 'Wild yak Gene-order conservation score' (default: False),\n",
      " 'bmutus_homolog_orthology_confidence': 'Wild yak orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bmutus_homolog_orthology_type': 'Wild yak homology type' (default: False),\n",
      " 'bmutus_homolog_perc_id': '%id. target Wild yak gene identical to query gene' (default: False),\n",
      " 'bmutus_homolog_perc_id_r1': '%id. query gene identical to target Wild yak gene' (default: False),\n",
      " 'bmutus_homolog_subtype': 'Last common ancestor with Wild yak' (default: False),\n",
      " 'bmutus_homolog_wga_coverage': 'Wild yak Whole-genome alignment coverage' (default: False),\n",
      " 'bsplendens_homolog_associated_gene_name': 'Siamese fighting fish gene name' (default: False),\n",
      " 'bsplendens_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bsplendens_homolog_chrom_end': 'Siamese fighting fish chromosome/scaffold end (bp)' (default: False),\n",
      " 'bsplendens_homolog_chrom_start': 'Siamese fighting fish chromosome/scaffold start (bp)' (default: False),\n",
      " 'bsplendens_homolog_chromosome': 'Siamese fighting fish chromosome/scaffold name' (default: False),\n",
      " 'bsplendens_homolog_ensembl_gene': 'Siamese fighting fish gene stable ID' (default: False),\n",
      " 'bsplendens_homolog_ensembl_peptide': 'Siamese fighting fish protein or transcript stable ID' (default: False),\n",
      " 'bsplendens_homolog_goc_score': 'Siamese fighting fish Gene-order conservation score' (default: False),\n",
      " 'bsplendens_homolog_orthology_confidence': 'Siamese fighting fish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bsplendens_homolog_orthology_type': 'Siamese fighting fish homology type' (default: False),\n",
      " 'bsplendens_homolog_perc_id': '%id. target Siamese fighting fish gene identical to query gene' (default: False),\n",
      " 'bsplendens_homolog_perc_id_r1': '%id. query gene identical to target Siamese fighting fish gene' (default: False),\n",
      " 'bsplendens_homolog_subtype': 'Last common ancestor with Siamese fighting fish' (default: False),\n",
      " 'bsplendens_homolog_wga_coverage': 'Siamese fighting fish Whole-genome alignment coverage' (default: False),\n",
      " 'btaurus_homolog_associated_gene_name': 'Cow gene name' (default: False),\n",
      " 'btaurus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'btaurus_homolog_chrom_end': 'Cow chromosome/scaffold end (bp)' (default: False),\n",
      " 'btaurus_homolog_chrom_start': 'Cow chromosome/scaffold start (bp)' (default: False),\n",
      " 'btaurus_homolog_chromosome': 'Cow chromosome/scaffold name' (default: False),\n",
      " 'btaurus_homolog_ensembl_gene': 'Cow gene stable ID' (default: False),\n",
      " 'btaurus_homolog_ensembl_peptide': 'Cow protein or transcript stable ID' (default: False),\n",
      " 'btaurus_homolog_goc_score': 'Cow Gene-order conservation score' (default: False),\n",
      " 'btaurus_homolog_orthology_confidence': 'Cow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'btaurus_homolog_orthology_type': 'Cow homology type' (default: False),\n",
      " 'btaurus_homolog_perc_id': '%id. target Cow gene identical to query gene' (default: False),\n",
      " 'btaurus_homolog_perc_id_r1': '%id. query gene identical to target Cow gene' (default: False),\n",
      " 'btaurus_homolog_subtype': 'Last common ancestor with Cow' (default: False),\n",
      " 'btaurus_homolog_wga_coverage': 'Cow Whole-genome alignment coverage' (default: False),\n",
      " 'cabingdonii_homolog_associated_gene_name': 'Abingdon island giant tortoise gene name' (default: False),\n",
      " 'cabingdonii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cabingdonii_homolog_chrom_end': 'Abingdon island giant tortoise chromosome/scaffold end (bp)' (default: False),\n",
      " 'cabingdonii_homolog_chrom_start': 'Abingdon island giant tortoise chromosome/scaffold start (bp)' (default: False),\n",
      " 'cabingdonii_homolog_chromosome': 'Abingdon island giant tortoise chromosome/scaffold name' (default: False),\n",
      " 'cabingdonii_homolog_ensembl_gene': 'Abingdon island giant tortoise gene stable ID' (default: False),\n",
      " 'cabingdonii_homolog_ensembl_peptide': 'Abingdon island giant tortoise protein or transcript stable ID' (default: False),\n",
      " 'cabingdonii_homolog_goc_score': 'Abingdon island giant tortoise Gene-order conservation score' (default: False),\n",
      " 'cabingdonii_homolog_orthology_confidence': 'Abingdon island giant tortoise orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cabingdonii_homolog_orthology_type': 'Abingdon island giant tortoise homology type' (default: False),\n",
      " 'cabingdonii_homolog_perc_id': '%id. target Abingdon island giant tortoise gene identical to query gene' (default: False),\n",
      " 'cabingdonii_homolog_perc_id_r1': '%id. query gene identical to target Abingdon island giant tortoise gene' (default: False),\n",
      " 'cabingdonii_homolog_subtype': 'Last common ancestor with Abingdon island giant tortoise' (default: False),\n",
      " 'cabingdonii_homolog_wga_coverage': 'Abingdon island giant tortoise Whole-genome alignment coverage' (default: False),\n",
      " 'catys_homolog_associated_gene_name': 'Sooty mangabey gene name' (default: False),\n",
      " 'catys_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'catys_homolog_chrom_end': 'Sooty mangabey chromosome/scaffold end (bp)' (default: False),\n",
      " 'catys_homolog_chrom_start': 'Sooty mangabey chromosome/scaffold start (bp)' (default: False),\n",
      " 'catys_homolog_chromosome': 'Sooty mangabey chromosome/scaffold name' (default: False),\n",
      " 'catys_homolog_ensembl_gene': 'Sooty mangabey gene stable ID' (default: False),\n",
      " 'catys_homolog_ensembl_peptide': 'Sooty mangabey protein or transcript stable ID' (default: False),\n",
      " 'catys_homolog_goc_score': 'Sooty mangabey Gene-order conservation score' (default: False),\n",
      " 'catys_homolog_orthology_confidence': 'Sooty mangabey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'catys_homolog_orthology_type': 'Sooty mangabey homology type' (default: False),\n",
      " 'catys_homolog_perc_id': '%id. target Sooty mangabey gene identical to query gene' (default: False),\n",
      " 'catys_homolog_perc_id_r1': '%id. query gene identical to target Sooty mangabey gene' (default: False),\n",
      " 'catys_homolog_subtype': 'Last common ancestor with Sooty mangabey' (default: False),\n",
      " 'catys_homolog_wga_coverage': 'Sooty mangabey Whole-genome alignment coverage' (default: False),\n",
      " 'cauratus_homolog_associated_gene_name': 'Goldfish gene name' (default: False),\n",
      " 'cauratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cauratus_homolog_chrom_end': 'Goldfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'cauratus_homolog_chrom_start': 'Goldfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'cauratus_homolog_chromosome': 'Goldfish chromosome/scaffold name' (default: False),\n",
      " 'cauratus_homolog_ensembl_gene': 'Goldfish gene stable ID' (default: False),\n",
      " 'cauratus_homolog_ensembl_peptide': 'Goldfish protein or transcript stable ID' (default: False),\n",
      " 'cauratus_homolog_goc_score': 'Goldfish Gene-order conservation score' (default: False),\n",
      " 'cauratus_homolog_orthology_confidence': 'Goldfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cauratus_homolog_orthology_type': 'Goldfish homology type' (default: False),\n",
      " 'cauratus_homolog_perc_id': '%id. target Goldfish gene identical to query gene' (default: False),\n",
      " 'cauratus_homolog_perc_id_r1': '%id. query gene identical to target Goldfish gene' (default: False),\n",
      " 'cauratus_homolog_subtype': 'Last common ancestor with Goldfish' (default: False),\n",
      " 'cauratus_homolog_wga_coverage': 'Goldfish Whole-genome alignment coverage' (default: False),\n",
      " 'cccarpio_homolog_associated_gene_name': 'Common carp gene name' (default: False),\n",
      " 'cccarpio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cccarpio_homolog_chrom_end': 'Common carp chromosome/scaffold end (bp)' (default: False),\n",
      " 'cccarpio_homolog_chrom_start': 'Common carp chromosome/scaffold start (bp)' (default: False),\n",
      " 'cccarpio_homolog_chromosome': 'Common carp chromosome/scaffold name' (default: False),\n",
      " 'cccarpio_homolog_ensembl_gene': 'Common carp gene stable ID' (default: False),\n",
      " 'cccarpio_homolog_ensembl_peptide': 'Common carp protein or transcript stable ID' (default: False),\n",
      " 'cccarpio_homolog_goc_score': 'Common carp Gene-order conservation score' (default: False),\n",
      " 'cccarpio_homolog_orthology_confidence': 'Common carp orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cccarpio_homolog_orthology_type': 'Common carp homology type' (default: False),\n",
      " 'cccarpio_homolog_perc_id': '%id. target Common carp gene identical to query gene' (default: False),\n",
      " 'cccarpio_homolog_perc_id_r1': '%id. query gene identical to target Common carp gene' (default: False),\n",
      " 'cccarpio_homolog_subtype': 'Last common ancestor with Common carp' (default: False),\n",
      " 'ccds': 'CCDS ID' (default: False),\n",
      " 'cdd': 'CDD ID' (default: False),\n",
      " 'cdd_end': 'CDD end' (default: False),\n",
      " 'cdd_start': 'CDD start' (default: False),\n",
      " 'cdna': 'cDNA sequences' (default: False),\n",
      " 'cdna_coding_end': 'cDNA coding end' (default: False),\n",
      " 'cdna_coding_start': 'cDNA coding start' (default: False),\n",
      " 'cdromedarius_homolog_associated_gene_name': 'Arabian camel gene name' (default: False),\n",
      " 'cdromedarius_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cdromedarius_homolog_chrom_end': 'Arabian camel chromosome/scaffold end (bp)' (default: False),\n",
      " 'cdromedarius_homolog_chrom_start': 'Arabian camel chromosome/scaffold start (bp)' (default: False),\n",
      " 'cdromedarius_homolog_chromosome': 'Arabian camel chromosome/scaffold name' (default: False),\n",
      " 'cdromedarius_homolog_ensembl_gene': 'Arabian camel gene stable ID' (default: False),\n",
      " 'cdromedarius_homolog_ensembl_peptide': 'Arabian camel protein or transcript stable ID' (default: False),\n",
      " 'cdromedarius_homolog_goc_score': 'Arabian camel Gene-order conservation score' (default: False),\n",
      " 'cdromedarius_homolog_orthology_confidence': 'Arabian camel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cdromedarius_homolog_orthology_type': 'Arabian camel homology type' (default: False),\n",
      " 'cdromedarius_homolog_perc_id': '%id. target Arabian camel gene identical to query gene' (default: False),\n",
      " 'cdromedarius_homolog_perc_id_r1': '%id. query gene identical to target Arabian camel gene' (default: False),\n",
      " 'cdromedarius_homolog_subtype': 'Last common ancestor with Arabian camel' (default: False),\n",
      " 'cdromedarius_homolog_wga_coverage': 'Arabian camel Whole-genome alignment coverage' (default: False),\n",
      " 'cds_end': 'CDS end' (default: False),\n",
      " 'cds_end_2076': 'CDS end' (default: False),\n",
      " 'cds_length': 'CDS Length' (default: False),\n",
      " 'cds_start': 'CDS start' (default: False),\n",
      " 'cds_start_2076': 'CDS start' (default: False),\n",
      " 'celegans_homolog_associated_gene_name': 'Caenorhabditis elegans (PRJNA13758) gene name' (default: False),\n",
      " 'celegans_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'celegans_homolog_chrom_end': 'Caenorhabditis elegans (PRJNA13758) chromosome/scaffold end (bp)' (default: False),\n",
      " 'celegans_homolog_chrom_start': 'Caenorhabditis elegans (PRJNA13758) chromosome/scaffold start (bp)' (default: False),\n",
      " 'celegans_homolog_chromosome': 'Caenorhabditis elegans (PRJNA13758) chromosome/scaffold name' (default: False),\n",
      " 'celegans_homolog_ensembl_gene': 'Caenorhabditis elegans (PRJNA13758) gene stable ID' (default: False),\n",
      " 'celegans_homolog_ensembl_peptide': 'Caenorhabditis elegans (PRJNA13758) protein or transcript stable ID' (default: False),\n",
      " 'celegans_homolog_orthology_confidence': 'Caenorhabditis elegans (PRJNA13758) orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'celegans_homolog_orthology_type': 'Caenorhabditis elegans (PRJNA13758) homology type' (default: False),\n",
      " 'celegans_homolog_perc_id': '%id. target Caenorhabditis elegans (PRJNA13758) gene identical to query gene' (default: False),\n",
      " 'celegans_homolog_perc_id_r1': '%id. query gene identical to target Caenorhabditis elegans (PRJNA13758) gene' (default: False),\n",
      " 'celegans_homolog_subtype': 'Last common ancestor with Caenorhabditis elegans (PRJNA13758)' (default: False),\n",
      " 'cgchok1gshd_homolog_associated_gene_name': 'Chinese hamster CHOK1GS gene name' (default: False),\n",
      " 'cgchok1gshd_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgchok1gshd_homolog_chrom_end': 'Chinese hamster CHOK1GS chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgchok1gshd_homolog_chrom_start': 'Chinese hamster CHOK1GS chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgchok1gshd_homolog_chromosome': 'Chinese hamster CHOK1GS chromosome/scaffold name' (default: False),\n",
      " 'cgchok1gshd_homolog_ensembl_gene': 'Chinese hamster CHOK1GS gene stable ID' (default: False),\n",
      " 'cgchok1gshd_homolog_ensembl_peptide': 'Chinese hamster CHOK1GS protein or transcript stable ID' (default: False),\n",
      " 'cgchok1gshd_homolog_goc_score': 'Chinese hamster CHOK1GS Gene-order conservation score' (default: False),\n",
      " 'cgchok1gshd_homolog_orthology_confidence': 'Chinese hamster CHOK1GS orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgchok1gshd_homolog_orthology_type': 'Chinese hamster CHOK1GS homology type' (default: False),\n",
      " 'cgchok1gshd_homolog_perc_id': '%id. target Chinese hamster CHOK1GS gene identical to query gene' (default: False),\n",
      " 'cgchok1gshd_homolog_perc_id_r1': '%id. query gene identical to target Chinese hamster CHOK1GS gene' (default: False),\n",
      " 'cgchok1gshd_homolog_subtype': 'Last common ancestor with Chinese hamster CHOK1GS' (default: False),\n",
      " 'cgchok1gshd_homolog_wga_coverage': 'Chinese hamster CHOK1GS Whole-genome alignment coverage' (default: False),\n",
      " 'cgobio_homolog_associated_gene_name': 'Channel bull blenny gene name' (default: False),\n",
      " 'cgobio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgobio_homolog_chrom_end': 'Channel bull blenny chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgobio_homolog_chrom_start': 'Channel bull blenny chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgobio_homolog_chromosome': 'Channel bull blenny chromosome/scaffold name' (default: False),\n",
      " 'cgobio_homolog_ensembl_gene': 'Channel bull blenny gene stable ID' (default: False),\n",
      " 'cgobio_homolog_ensembl_peptide': 'Channel bull blenny protein or transcript stable ID' (default: False),\n",
      " 'cgobio_homolog_goc_score': 'Channel bull blenny Gene-order conservation score' (default: False),\n",
      " 'cgobio_homolog_orthology_confidence': 'Channel bull blenny orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgobio_homolog_orthology_type': 'Channel bull blenny homology type' (default: False),\n",
      " 'cgobio_homolog_perc_id': '%id. target Channel bull blenny gene identical to query gene' (default: False),\n",
      " 'cgobio_homolog_perc_id_r1': '%id. query gene identical to target Channel bull blenny gene' (default: False),\n",
      " 'cgobio_homolog_subtype': 'Last common ancestor with Channel bull blenny' (default: False),\n",
      " 'cgobio_homolog_wga_coverage': 'Channel bull blenny Whole-genome alignment coverage' (default: False),\n",
      " 'charengus_homolog_associated_gene_name': 'Atlantic herring gene name' (default: False),\n",
      " 'charengus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'charengus_homolog_chrom_end': 'Atlantic herring chromosome/scaffold end (bp)' (default: False),\n",
      " 'charengus_homolog_chrom_start': 'Atlantic herring chromosome/scaffold start (bp)' (default: False),\n",
      " 'charengus_homolog_chromosome': 'Atlantic herring chromosome/scaffold name' (default: False),\n",
      " 'charengus_homolog_ensembl_gene': 'Atlantic herring gene stable ID' (default: False),\n",
      " 'charengus_homolog_ensembl_peptide': 'Atlantic herring protein or transcript stable ID' (default: False),\n",
      " 'charengus_homolog_goc_score': 'Atlantic herring Gene-order conservation score' (default: False),\n",
      " 'charengus_homolog_orthology_confidence': 'Atlantic herring orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'charengus_homolog_orthology_type': 'Atlantic herring homology type' (default: False),\n",
      " 'charengus_homolog_perc_id': '%id. target Atlantic herring gene identical to query gene' (default: False),\n",
      " 'charengus_homolog_perc_id_r1': '%id. query gene identical to target Atlantic herring gene' (default: False),\n",
      " 'charengus_homolog_subtype': 'Last common ancestor with Atlantic herring' (default: False),\n",
      " 'charengus_homolog_wga_coverage': 'Atlantic herring Whole-genome alignment coverage' (default: False),\n",
      " 'chembl': 'ChEMBL ID' (default: False),\n",
      " 'chircus_homolog_associated_gene_name': 'Goat gene name' (default: False),\n",
      " 'chircus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'chircus_homolog_chrom_end': 'Goat chromosome/scaffold end (bp)' (default: False),\n",
      " 'chircus_homolog_chrom_start': 'Goat chromosome/scaffold start (bp)' (default: False),\n",
      " 'chircus_homolog_chromosome': 'Goat chromosome/scaffold name' (default: False),\n",
      " 'chircus_homolog_ensembl_gene': 'Goat gene stable ID' (default: False),\n",
      " 'chircus_homolog_ensembl_peptide': 'Goat protein or transcript stable ID' (default: False),\n",
      " 'chircus_homolog_goc_score': 'Goat Gene-order conservation score' (default: False),\n",
      " 'chircus_homolog_orthology_confidence': 'Goat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'chircus_homolog_orthology_type': 'Goat homology type' (default: False),\n",
      " 'chircus_homolog_perc_id': '%id. target Goat gene identical to query gene' (default: False),\n",
      " 'chircus_homolog_perc_id_r1': '%id. query gene identical to target Goat gene' (default: False),\n",
      " 'chircus_homolog_subtype': 'Last common ancestor with Goat' (default: False),\n",
      " 'chircus_homolog_wga_coverage': 'Goat Whole-genome alignment coverage' (default: False),\n",
      " 'choffmanni_homolog_associated_gene_name': 'Sloth gene name' (default: False),\n",
      " 'choffmanni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'choffmanni_homolog_chrom_end': 'Sloth chromosome/scaffold end (bp)' (default: False),\n",
      " 'choffmanni_homolog_chrom_start': 'Sloth chromosome/scaffold start (bp)' (default: False),\n",
      " 'choffmanni_homolog_chromosome': 'Sloth chromosome/scaffold name' (default: False),\n",
      " 'choffmanni_homolog_ensembl_gene': 'Sloth gene stable ID' (default: False),\n",
      " 'choffmanni_homolog_ensembl_peptide': 'Sloth protein or transcript stable ID' (default: False),\n",
      " 'choffmanni_homolog_goc_score': 'Sloth Gene-order conservation score' (default: False),\n",
      " 'choffmanni_homolog_orthology_confidence': 'Sloth orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'choffmanni_homolog_orthology_type': 'Sloth homology type' (default: False),\n",
      " 'choffmanni_homolog_perc_id': '%id. target Sloth gene identical to query gene' (default: False),\n",
      " 'choffmanni_homolog_perc_id_r1': '%id. query gene identical to target Sloth gene' (default: False),\n",
      " 'choffmanni_homolog_subtype': 'Last common ancestor with Sloth' (default: False),\n",
      " 'choffmanni_homolog_wga_coverage': 'Sloth Whole-genome alignment coverage' (default: False),\n",
      " 'chromosome_end': 'Chromosome/scaffold position end (bp)' (default: False),\n",
      " 'chromosome_name': 'Chromosome/scaffold name' (default: False),\n",
      " 'chromosome_start': 'chromosome/scaffold position start (bp)' (default: False),\n",
      " 'chyarkandensis_homolog_associated_gene_name': 'Yarkand deer gene name' (default: False),\n",
      " 'chyarkandensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'chyarkandensis_homolog_chrom_end': 'Yarkand deer chromosome/scaffold end (bp)' (default: False),\n",
      " 'chyarkandensis_homolog_chrom_start': 'Yarkand deer chromosome/scaffold start (bp)' (default: False),\n",
      " 'chyarkandensis_homolog_chromosome': 'Yarkand deer chromosome/scaffold name' (default: False),\n",
      " 'chyarkandensis_homolog_ensembl_gene': 'Yarkand deer gene stable ID' (default: False),\n",
      " 'chyarkandensis_homolog_ensembl_peptide': 'Yarkand deer protein or transcript stable ID' (default: False),\n",
      " 'chyarkandensis_homolog_goc_score': 'Yarkand deer Gene-order conservation score' (default: False),\n",
      " 'chyarkandensis_homolog_orthology_confidence': 'Yarkand deer orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'chyarkandensis_homolog_orthology_type': 'Yarkand deer homology type' (default: False),\n",
      " 'chyarkandensis_homolog_perc_id': '%id. target Yarkand deer gene identical to query gene' (default: False),\n",
      " 'chyarkandensis_homolog_perc_id_r1': '%id. query gene identical to target Yarkand deer gene' (default: False),\n",
      " 'chyarkandensis_homolog_subtype': 'Last common ancestor with Yarkand deer' (default: False),\n",
      " 'chyarkandensis_homolog_wga_coverage': 'Yarkand deer Whole-genome alignment coverage' (default: False),\n",
      " 'cimitator_homolog_associated_gene_name': 'Panamanian white-faced capuchin gene name' (default: False),\n",
      " 'cimitator_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cimitator_homolog_chrom_end': 'Panamanian white-faced capuchin chromosome/scaffold end (bp)' (default: False),\n",
      " 'cimitator_homolog_chrom_start': 'Panamanian white-faced capuchin chromosome/scaffold start (bp)' (default: False),\n",
      " 'cimitator_homolog_chromosome': 'Panamanian white-faced capuchin chromosome/scaffold name' (default: False),\n",
      " 'cimitator_homolog_ensembl_gene': 'Panamanian white-faced capuchin gene stable ID' (default: False),\n",
      " 'cimitator_homolog_ensembl_peptide': 'Panamanian white-faced capuchin protein or transcript stable ID' (default: False),\n",
      " 'cimitator_homolog_goc_score': 'Panamanian white-faced capuchin Gene-order conservation score' (default: False),\n",
      " 'cimitator_homolog_orthology_confidence': 'Panamanian white-faced capuchin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cimitator_homolog_orthology_type': 'Panamanian white-faced capuchin homology type' (default: False),\n",
      " 'cimitator_homolog_perc_id': '%id. target Panamanian white-faced capuchin gene identical to query gene' (default: False),\n",
      " 'cimitator_homolog_perc_id_r1': '%id. query gene identical to target Panamanian white-faced capuchin gene' (default: False),\n",
      " 'cimitator_homolog_subtype': 'Last common ancestor with Panamanian white-faced capuchin' (default: False),\n",
      " 'cimitator_homolog_wga_coverage': 'Panamanian white-faced capuchin Whole-genome alignment coverage' (default: False),\n",
      " 'cintestinalis_homolog_associated_gene_name': 'C.intestinalis gene name' (default: False),\n",
      " 'cintestinalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cintestinalis_homolog_chrom_end': 'C.intestinalis chromosome/scaffold end (bp)' (default: False),\n",
      " 'cintestinalis_homolog_chrom_start': 'C.intestinalis chromosome/scaffold start (bp)' (default: False),\n",
      " 'cintestinalis_homolog_chromosome': 'C.intestinalis chromosome/scaffold name' (default: False),\n",
      " 'cintestinalis_homolog_ensembl_gene': 'C.intestinalis gene stable ID' (default: False),\n",
      " 'cintestinalis_homolog_ensembl_peptide': 'C.intestinalis protein or transcript stable ID' (default: False),\n",
      " 'cintestinalis_homolog_orthology_confidence': 'C.intestinalis orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cintestinalis_homolog_orthology_type': 'C.intestinalis homology type' (default: False),\n",
      " 'cintestinalis_homolog_perc_id': '%id. target C.intestinalis gene identical to query gene' (default: False),\n",
      " 'cintestinalis_homolog_perc_id_r1': '%id. query gene identical to target C.intestinalis gene' (default: False),\n",
      " 'cintestinalis_homolog_subtype': 'Last common ancestor with C.intestinalis' (default: False),\n",
      " 'cintestinalis_homolog_wga_coverage': 'C.intestinalis Whole-genome alignment coverage' (default: False),\n",
      " 'cjacchus_homolog_associated_gene_name': 'White-tufted-ear marmoset gene name' (default: False),\n",
      " 'cjacchus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cjacchus_homolog_chrom_end': 'White-tufted-ear marmoset chromosome/scaffold end (bp)' (default: False),\n",
      " 'cjacchus_homolog_chrom_start': 'White-tufted-ear marmoset chromosome/scaffold start (bp)' (default: False),\n",
      " 'cjacchus_homolog_chromosome': 'White-tufted-ear marmoset chromosome/scaffold name' (default: False),\n",
      " 'cjacchus_homolog_ensembl_gene': 'White-tufted-ear marmoset gene stable ID' (default: False),\n",
      " 'cjacchus_homolog_ensembl_peptide': 'White-tufted-ear marmoset protein or transcript stable ID' (default: False),\n",
      " 'cjacchus_homolog_goc_score': 'White-tufted-ear marmoset Gene-order conservation score' (default: False),\n",
      " 'cjacchus_homolog_orthology_confidence': 'White-tufted-ear marmoset orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cjacchus_homolog_orthology_type': 'White-tufted-ear marmoset homology type' (default: False),\n",
      " 'cjacchus_homolog_perc_id': '%id. target White-tufted-ear marmoset gene identical to query gene' (default: False),\n",
      " 'cjacchus_homolog_perc_id_r1': '%id. query gene identical to target White-tufted-ear marmoset gene' (default: False),\n",
      " 'cjacchus_homolog_subtype': 'Last common ancestor with White-tufted-ear marmoset' (default: False),\n",
      " 'cjacchus_homolog_wga_coverage': 'White-tufted-ear marmoset Whole-genome alignment coverage' (default: False),\n",
      " 'cjaponica_homolog_associated_gene_name': 'Japanese quail gene name' (default: False),\n",
      " 'cjaponica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cjaponica_homolog_chrom_end': 'Japanese quail chromosome/scaffold end (bp)' (default: False),\n",
      " 'cjaponica_homolog_chrom_start': 'Japanese quail chromosome/scaffold start (bp)' (default: False),\n",
      " 'cjaponica_homolog_chromosome': 'Japanese quail chromosome/scaffold name' (default: False),\n",
      " 'cjaponica_homolog_ensembl_gene': 'Japanese quail gene stable ID' (default: False),\n",
      " 'cjaponica_homolog_ensembl_peptide': 'Japanese quail protein or transcript stable ID' (default: False),\n",
      " 'cjaponica_homolog_goc_score': 'Japanese quail Gene-order conservation score' (default: False),\n",
      " 'cjaponica_homolog_orthology_confidence': 'Japanese quail orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cjaponica_homolog_orthology_type': 'Japanese quail homology type' (default: False),\n",
      " 'cjaponica_homolog_perc_id': '%id. target Japanese quail gene identical to query gene' (default: False),\n",
      " 'cjaponica_homolog_perc_id_r1': '%id. query gene identical to target Japanese quail gene' (default: False),\n",
      " 'cjaponica_homolog_subtype': 'Last common ancestor with Japanese quail' (default: False),\n",
      " 'cjaponica_homolog_wga_coverage': 'Japanese quail Whole-genome alignment coverage' (default: False),\n",
      " 'clanigera_homolog_associated_gene_name': 'Long-tailed chinchilla gene name' (default: False),\n",
      " 'clanigera_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clanigera_homolog_chrom_end': 'Long-tailed chinchilla chromosome/scaffold end (bp)' (default: False),\n",
      " 'clanigera_homolog_chrom_start': 'Long-tailed chinchilla chromosome/scaffold start (bp)' (default: False),\n",
      " 'clanigera_homolog_chromosome': 'Long-tailed chinchilla chromosome/scaffold name' (default: False),\n",
      " 'clanigera_homolog_ensembl_gene': 'Long-tailed chinchilla gene stable ID' (default: False),\n",
      " 'clanigera_homolog_ensembl_peptide': 'Long-tailed chinchilla protein or transcript stable ID' (default: False),\n",
      " 'clanigera_homolog_goc_score': 'Long-tailed chinchilla Gene-order conservation score' (default: False),\n",
      " 'clanigera_homolog_orthology_confidence': 'Long-tailed chinchilla orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clanigera_homolog_orthology_type': 'Long-tailed chinchilla homology type' (default: False),\n",
      " 'clanigera_homolog_perc_id': '%id. target Long-tailed chinchilla gene identical to query gene' (default: False),\n",
      " 'clanigera_homolog_perc_id_r1': '%id. query gene identical to target Long-tailed chinchilla gene' (default: False),\n",
      " 'clanigera_homolog_subtype': 'Last common ancestor with Long-tailed chinchilla' (default: False),\n",
      " 'clanigera_homolog_wga_coverage': 'Long-tailed chinchilla Whole-genome alignment coverage' (default: False),\n",
      " 'cldingo_homolog_associated_gene_name': 'Dingo gene name' (default: False),\n",
      " 'cldingo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cldingo_homolog_chrom_end': 'Dingo chromosome/scaffold end (bp)' (default: False),\n",
      " 'cldingo_homolog_chrom_start': 'Dingo chromosome/scaffold start (bp)' (default: False),\n",
      " 'cldingo_homolog_chromosome': 'Dingo chromosome/scaffold name' (default: False),\n",
      " 'cldingo_homolog_ensembl_gene': 'Dingo gene stable ID' (default: False),\n",
      " 'cldingo_homolog_ensembl_peptide': 'Dingo protein or transcript stable ID' (default: False),\n",
      " 'cldingo_homolog_goc_score': 'Dingo Gene-order conservation score' (default: False),\n",
      " 'cldingo_homolog_orthology_confidence': 'Dingo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cldingo_homolog_orthology_type': 'Dingo homology type' (default: False),\n",
      " 'cldingo_homolog_perc_id': '%id. target Dingo gene identical to query gene' (default: False),\n",
      " 'cldingo_homolog_perc_id_r1': '%id. query gene identical to target Dingo gene' (default: False),\n",
      " 'cldingo_homolog_subtype': 'Last common ancestor with Dingo' (default: False),\n",
      " 'cldingo_homolog_wga_coverage': 'Dingo Whole-genome alignment coverage' (default: False),\n",
      " 'clfamiliaris_homolog_associated_gene_name': 'Dog gene name' (default: False),\n",
      " 'clfamiliaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clfamiliaris_homolog_chrom_end': 'Dog chromosome/scaffold end (bp)' (default: False),\n",
      " 'clfamiliaris_homolog_chrom_start': 'Dog chromosome/scaffold start (bp)' (default: False),\n",
      " 'clfamiliaris_homolog_chromosome': 'Dog chromosome/scaffold name' (default: False),\n",
      " 'clfamiliaris_homolog_ensembl_gene': 'Dog gene stable ID' (default: False),\n",
      " 'clfamiliaris_homolog_ensembl_peptide': 'Dog protein or transcript stable ID' (default: False),\n",
      " 'clfamiliaris_homolog_goc_score': 'Dog Gene-order conservation score' (default: False),\n",
      " 'clfamiliaris_homolog_orthology_confidence': 'Dog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clfamiliaris_homolog_orthology_type': 'Dog homology type' (default: False),\n",
      " 'clfamiliaris_homolog_perc_id': '%id. target Dog gene identical to query gene' (default: False),\n",
      " 'clfamiliaris_homolog_perc_id_r1': '%id. query gene identical to target Dog gene' (default: False),\n",
      " 'clfamiliaris_homolog_subtype': 'Last common ancestor with Dog' (default: False),\n",
      " 'clfamiliaris_homolog_wga_coverage': 'Dog Whole-genome alignment coverage' (default: False),\n",
      " 'clinical_significance': 'Clinical significance' (default: False),\n",
      " 'clumpus_homolog_associated_gene_name': 'Lumpfish gene name' (default: False),\n",
      " 'clumpus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clumpus_homolog_chrom_end': 'Lumpfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'clumpus_homolog_chrom_start': 'Lumpfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'clumpus_homolog_chromosome': 'Lumpfish chromosome/scaffold name' (default: False),\n",
      " 'clumpus_homolog_ensembl_gene': 'Lumpfish gene stable ID' (default: False),\n",
      " 'clumpus_homolog_ensembl_peptide': 'Lumpfish protein or transcript stable ID' (default: False),\n",
      " 'clumpus_homolog_goc_score': 'Lumpfish Gene-order conservation score' (default: False),\n",
      " 'clumpus_homolog_orthology_confidence': 'Lumpfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clumpus_homolog_orthology_type': 'Lumpfish homology type' (default: False),\n",
      " 'clumpus_homolog_perc_id': '%id. target Lumpfish gene identical to query gene' (default: False),\n",
      " 'clumpus_homolog_perc_id_r1': '%id. query gene identical to target Lumpfish gene' (default: False),\n",
      " 'clumpus_homolog_subtype': 'Last common ancestor with Lumpfish' (default: False),\n",
      " 'clumpus_homolog_wga_coverage': 'Lumpfish Whole-genome alignment coverage' (default: False),\n",
      " 'cmilii_homolog_associated_gene_name': 'Elephant shark gene name' (default: False),\n",
      " 'cmilii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cmilii_homolog_chrom_end': 'Elephant shark chromosome/scaffold end (bp)' (default: False),\n",
      " 'cmilii_homolog_chrom_start': 'Elephant shark chromosome/scaffold start (bp)' (default: False),\n",
      " 'cmilii_homolog_chromosome': 'Elephant shark chromosome/scaffold name' (default: False),\n",
      " 'cmilii_homolog_ensembl_gene': 'Elephant shark gene stable ID' (default: False),\n",
      " 'cmilii_homolog_ensembl_peptide': 'Elephant shark protein or transcript stable ID' (default: False),\n",
      " 'cmilii_homolog_orthology_confidence': 'Elephant shark orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cmilii_homolog_orthology_type': 'Elephant shark homology type' (default: False),\n",
      " 'cmilii_homolog_perc_id': '%id. target Elephant shark gene identical to query gene' (default: False),\n",
      " 'cmilii_homolog_perc_id_r1': '%id. query gene identical to target Elephant shark gene' (default: False),\n",
      " 'cmilii_homolog_subtype': 'Last common ancestor with Elephant shark' (default: False),\n",
      " 'cmilii_homolog_wga_coverage': 'Elephant shark Whole-genome alignment coverage' (default: False),\n",
      " 'codelink_codelink': 'CODELINK CODELINK probe' (default: False),\n",
      " 'coding': 'Coding sequence' (default: False),\n",
      " 'coding_gene_flank': 'Flank-coding region (Gene)' (default: False),\n",
      " 'coding_transcript_flank': 'Flank-coding region (Transcript)' (default: False),\n",
      " 'cpbellii_homolog_associated_gene_name': 'Painted turtle gene name' (default: False),\n",
      " 'cpbellii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cpbellii_homolog_chrom_end': 'Painted turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'cpbellii_homolog_chrom_start': 'Painted turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'cpbellii_homolog_chromosome': 'Painted turtle chromosome/scaffold name' (default: False),\n",
      " 'cpbellii_homolog_ensembl_gene': 'Painted turtle gene stable ID' (default: False),\n",
      " 'cpbellii_homolog_ensembl_peptide': 'Painted turtle protein or transcript stable ID' (default: False),\n",
      " 'cpbellii_homolog_goc_score': 'Painted turtle Gene-order conservation score' (default: False),\n",
      " 'cpbellii_homolog_orthology_confidence': 'Painted turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cpbellii_homolog_orthology_type': 'Painted turtle homology type' (default: False),\n",
      " 'cpbellii_homolog_perc_id': '%id. target Painted turtle gene identical to query gene' (default: False),\n",
      " 'cpbellii_homolog_perc_id_r1': '%id. query gene identical to target Painted turtle gene' (default: False),\n",
      " 'cpbellii_homolog_subtype': 'Last common ancestor with Painted turtle' (default: False),\n",
      " 'cpbellii_homolog_wga_coverage': 'Painted turtle Whole-genome alignment coverage' (default: False),\n",
      " 'cporcellus_homolog_associated_gene_name': 'Guinea Pig gene name' (default: False),\n",
      " 'cporcellus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cporcellus_homolog_chrom_end': 'Guinea Pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'cporcellus_homolog_chrom_start': 'Guinea Pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'cporcellus_homolog_chromosome': 'Guinea Pig chromosome/scaffold name' (default: False),\n",
      " 'cporcellus_homolog_ensembl_gene': 'Guinea Pig gene stable ID' (default: False),\n",
      " 'cporcellus_homolog_ensembl_peptide': 'Guinea Pig protein or transcript stable ID' (default: False),\n",
      " 'cporcellus_homolog_goc_score': 'Guinea Pig Gene-order conservation score' (default: False),\n",
      " 'cporcellus_homolog_orthology_confidence': 'Guinea Pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cporcellus_homolog_orthology_type': 'Guinea Pig homology type' (default: False),\n",
      " 'cporcellus_homolog_perc_id': '%id. target Guinea Pig gene identical to query gene' (default: False),\n",
      " 'cporcellus_homolog_perc_id_r1': '%id. query gene identical to target Guinea Pig gene' (default: False),\n",
      " 'cporcellus_homolog_subtype': 'Last common ancestor with Guinea Pig' (default: False),\n",
      " 'cporcellus_homolog_wga_coverage': 'Guinea Pig Whole-genome alignment coverage' (default: False),\n",
      " 'cporosus_homolog_associated_gene_name': 'Australian saltwater crocodile gene name' (default: False),\n",
      " 'cporosus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cporosus_homolog_chrom_end': 'Australian saltwater crocodile chromosome/scaffold end (bp)' (default: False),\n",
      " 'cporosus_homolog_chrom_start': 'Australian saltwater crocodile chromosome/scaffold start (bp)' (default: False),\n",
      " 'cporosus_homolog_chromosome': 'Australian saltwater crocodile chromosome/scaffold name' (default: False),\n",
      " 'cporosus_homolog_ensembl_gene': 'Australian saltwater crocodile gene stable ID' (default: False),\n",
      " 'cporosus_homolog_ensembl_peptide': 'Australian saltwater crocodile protein or transcript stable ID' (default: False),\n",
      " 'cporosus_homolog_goc_score': 'Australian saltwater crocodile Gene-order conservation score' (default: False),\n",
      " 'cporosus_homolog_orthology_confidence': 'Australian saltwater crocodile orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cporosus_homolog_orthology_type': 'Australian saltwater crocodile homology type' (default: False),\n",
      " 'cporosus_homolog_perc_id': '%id. target Australian saltwater crocodile gene identical to query gene' (default: False),\n",
      " 'cporosus_homolog_perc_id_r1': '%id. query gene identical to target Australian saltwater crocodile gene' (default: False),\n",
      " 'cporosus_homolog_subtype': 'Last common ancestor with Australian saltwater crocodile' (default: False),\n",
      " 'cporosus_homolog_wga_coverage': 'Australian saltwater crocodile Whole-genome alignment coverage' (default: False),\n",
      " 'csabaeus_homolog_associated_gene_name': 'Vervet-AGM gene name' (default: False),\n",
      " 'csabaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csabaeus_homolog_chrom_end': 'Vervet-AGM chromosome/scaffold end (bp)' (default: False),\n",
      " 'csabaeus_homolog_chrom_start': 'Vervet-AGM chromosome/scaffold start (bp)' (default: False),\n",
      " 'csabaeus_homolog_chromosome': 'Vervet-AGM chromosome/scaffold name' (default: False),\n",
      " 'csabaeus_homolog_ensembl_gene': 'Vervet-AGM gene stable ID' (default: False),\n",
      " 'csabaeus_homolog_ensembl_peptide': 'Vervet-AGM protein or transcript stable ID' (default: False),\n",
      " 'csabaeus_homolog_goc_score': 'Vervet-AGM Gene-order conservation score' (default: False),\n",
      " 'csabaeus_homolog_orthology_confidence': 'Vervet-AGM orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csabaeus_homolog_orthology_type': 'Vervet-AGM homology type' (default: False),\n",
      " 'csabaeus_homolog_perc_id': '%id. target Vervet-AGM gene identical to query gene' (default: False),\n",
      " 'csabaeus_homolog_perc_id_r1': '%id. query gene identical to target Vervet-AGM gene' (default: False),\n",
      " 'csabaeus_homolog_subtype': 'Last common ancestor with Vervet-AGM' (default: False),\n",
      " 'csabaeus_homolog_wga_coverage': 'Vervet-AGM Whole-genome alignment coverage' (default: False),\n",
      " 'csavignyi_homolog_associated_gene_name': 'C.savignyi gene name' (default: False),\n",
      " 'csavignyi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csavignyi_homolog_chrom_end': 'C.savignyi chromosome/scaffold end (bp)' (default: False),\n",
      " 'csavignyi_homolog_chrom_start': 'C.savignyi chromosome/scaffold start (bp)' (default: False),\n",
      " 'csavignyi_homolog_chromosome': 'C.savignyi chromosome/scaffold name' (default: False),\n",
      " 'csavignyi_homolog_ensembl_gene': 'C.savignyi gene stable ID' (default: False),\n",
      " 'csavignyi_homolog_ensembl_peptide': 'C.savignyi protein or transcript stable ID' (default: False),\n",
      " 'csavignyi_homolog_orthology_confidence': 'C.savignyi orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csavignyi_homolog_orthology_type': 'C.savignyi homology type' (default: False),\n",
      " 'csavignyi_homolog_perc_id': '%id. target C.savignyi gene identical to query gene' (default: False),\n",
      " 'csavignyi_homolog_perc_id_r1': '%id. query gene identical to target C.savignyi gene' (default: False),\n",
      " 'csavignyi_homolog_subtype': 'Last common ancestor with C.savignyi' (default: False),\n",
      " 'csavignyi_homolog_wga_coverage': 'C.savignyi Whole-genome alignment coverage' (default: False),\n",
      " 'csemilaevis_homolog_associated_gene_name': 'Tongue sole gene name' (default: False),\n",
      " 'csemilaevis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csemilaevis_homolog_chrom_end': 'Tongue sole chromosome/scaffold end (bp)' (default: False),\n",
      " 'csemilaevis_homolog_chrom_start': 'Tongue sole chromosome/scaffold start (bp)' (default: False),\n",
      " 'csemilaevis_homolog_chromosome': 'Tongue sole chromosome/scaffold name' (default: False),\n",
      " 'csemilaevis_homolog_ensembl_gene': 'Tongue sole gene stable ID' (default: False),\n",
      " 'csemilaevis_homolog_ensembl_peptide': 'Tongue sole protein or transcript stable ID' (default: False),\n",
      " 'csemilaevis_homolog_goc_score': 'Tongue sole Gene-order conservation score' (default: False),\n",
      " 'csemilaevis_homolog_orthology_confidence': 'Tongue sole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csemilaevis_homolog_orthology_type': 'Tongue sole homology type' (default: False),\n",
      " 'csemilaevis_homolog_perc_id': '%id. target Tongue sole gene identical to query gene' (default: False),\n",
      " 'csemilaevis_homolog_perc_id_r1': '%id. query gene identical to target Tongue sole gene' (default: False),\n",
      " 'csemilaevis_homolog_subtype': 'Last common ancestor with Tongue sole' (default: False),\n",
      " 'csemilaevis_homolog_wga_coverage': 'Tongue sole Whole-genome alignment coverage' (default: False),\n",
      " 'csyrichta_homolog_associated_gene_name': 'Tarsier gene name' (default: False),\n",
      " 'csyrichta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csyrichta_homolog_chrom_end': 'Tarsier chromosome/scaffold end (bp)' (default: False),\n",
      " 'csyrichta_homolog_chrom_start': 'Tarsier chromosome/scaffold start (bp)' (default: False),\n",
      " 'csyrichta_homolog_chromosome': 'Tarsier chromosome/scaffold name' (default: False),\n",
      " 'csyrichta_homolog_ensembl_gene': 'Tarsier gene stable ID' (default: False),\n",
      " 'csyrichta_homolog_ensembl_peptide': 'Tarsier protein or transcript stable ID' (default: False),\n",
      " 'csyrichta_homolog_goc_score': 'Tarsier Gene-order conservation score' (default: False),\n",
      " 'csyrichta_homolog_orthology_confidence': 'Tarsier orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csyrichta_homolog_orthology_type': 'Tarsier homology type' (default: False),\n",
      " 'csyrichta_homolog_perc_id': '%id. target Tarsier gene identical to query gene' (default: False),\n",
      " 'csyrichta_homolog_perc_id_r1': '%id. query gene identical to target Tarsier gene' (default: False),\n",
      " 'csyrichta_homolog_subtype': 'Last common ancestor with Tarsier' (default: False),\n",
      " 'csyrichta_homolog_wga_coverage': 'Tarsier Whole-genome alignment coverage' (default: False),\n",
      " 'cvariegatus_homolog_associated_gene_name': 'Sheepshead minnow gene name' (default: False),\n",
      " 'cvariegatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cvariegatus_homolog_chrom_end': 'Sheepshead minnow chromosome/scaffold end (bp)' (default: False),\n",
      " 'cvariegatus_homolog_chrom_start': 'Sheepshead minnow chromosome/scaffold start (bp)' (default: False),\n",
      " 'cvariegatus_homolog_chromosome': 'Sheepshead minnow chromosome/scaffold name' (default: False),\n",
      " 'cvariegatus_homolog_ensembl_gene': 'Sheepshead minnow gene stable ID' (default: False),\n",
      " 'cvariegatus_homolog_ensembl_peptide': 'Sheepshead minnow protein or transcript stable ID' (default: False),\n",
      " 'cvariegatus_homolog_goc_score': 'Sheepshead minnow Gene-order conservation score' (default: False),\n",
      " 'cvariegatus_homolog_orthology_confidence': 'Sheepshead minnow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cvariegatus_homolog_orthology_type': 'Sheepshead minnow homology type' (default: False),\n",
      " 'cvariegatus_homolog_perc_id': '%id. target Sheepshead minnow gene identical to query gene' (default: False),\n",
      " 'cvariegatus_homolog_perc_id_r1': '%id. query gene identical to target Sheepshead minnow gene' (default: False),\n",
      " 'cvariegatus_homolog_subtype': 'Last common ancestor with Sheepshead minnow' (default: False),\n",
      " 'cvariegatus_homolog_wga_coverage': 'Sheepshead minnow Whole-genome alignment coverage' (default: False),\n",
      " 'cwagneri_homolog_associated_gene_name': 'Chacoan peccary gene name' (default: False),\n",
      " 'cwagneri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cwagneri_homolog_chrom_end': 'Chacoan peccary chromosome/scaffold end (bp)' (default: False),\n",
      " 'cwagneri_homolog_chrom_start': 'Chacoan peccary chromosome/scaffold start (bp)' (default: False),\n",
      " 'cwagneri_homolog_chromosome': 'Chacoan peccary chromosome/scaffold name' (default: False),\n",
      " 'cwagneri_homolog_ensembl_gene': 'Chacoan peccary gene stable ID' (default: False),\n",
      " 'cwagneri_homolog_ensembl_peptide': 'Chacoan peccary protein or transcript stable ID' (default: False),\n",
      " 'cwagneri_homolog_goc_score': 'Chacoan peccary Gene-order conservation score' (default: False),\n",
      " 'cwagneri_homolog_orthology_confidence': 'Chacoan peccary orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cwagneri_homolog_orthology_type': 'Chacoan peccary homology type' (default: False),\n",
      " 'cwagneri_homolog_perc_id': '%id. target Chacoan peccary gene identical to query gene' (default: False),\n",
      " 'cwagneri_homolog_perc_id_r1': '%id. query gene identical to target Chacoan peccary gene' (default: False),\n",
      " 'cwagneri_homolog_subtype': 'Last common ancestor with Chacoan peccary' (default: False),\n",
      " 'cwagneri_homolog_wga_coverage': 'Chacoan peccary Whole-genome alignment coverage' (default: False),\n",
      " 'dbass3_id': 'DataBase of Aberrant 3' Splice Sites ID' (default: False),\n",
      " 'dbass3_name': 'DataBase of Aberrant 3' Splice Sites name' (default: False),\n",
      " 'dbass5_id': 'DataBase of Aberrant 5' Splice Sites ID' (default: False),\n",
      " 'dbass5_name': 'DataBase of Aberrant 5' Splice Sites name' (default: False),\n",
      " 'dclupeoides_homolog_associated_gene_name': 'Denticle herring gene name' (default: False),\n",
      " 'dclupeoides_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dclupeoides_homolog_chrom_end': 'Denticle herring chromosome/scaffold end (bp)' (default: False),\n",
      " 'dclupeoides_homolog_chrom_start': 'Denticle herring chromosome/scaffold start (bp)' (default: False),\n",
      " 'dclupeoides_homolog_chromosome': 'Denticle herring chromosome/scaffold name' (default: False),\n",
      " 'dclupeoides_homolog_ensembl_gene': 'Denticle herring gene stable ID' (default: False),\n",
      " 'dclupeoides_homolog_ensembl_peptide': 'Denticle herring protein or transcript stable ID' (default: False),\n",
      " 'dclupeoides_homolog_goc_score': 'Denticle herring Gene-order conservation score' (default: False),\n",
      " 'dclupeoides_homolog_orthology_confidence': 'Denticle herring orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dclupeoides_homolog_orthology_type': 'Denticle herring homology type' (default: False),\n",
      " 'dclupeoides_homolog_perc_id': '%id. target Denticle herring gene identical to query gene' (default: False),\n",
      " 'dclupeoides_homolog_perc_id_r1': '%id. query gene identical to target Denticle herring gene' (default: False),\n",
      " 'dclupeoides_homolog_subtype': 'Last common ancestor with Denticle herring' (default: False),\n",
      " 'dclupeoides_homolog_wga_coverage': 'Denticle herring Whole-genome alignment coverage' (default: False),\n",
      " 'definition_1006': 'GO term definition' (default: False),\n",
      " 'description': 'Gene description' (default: False),\n",
      " 'distance_to_transcript_2076': 'Distance to transcript' (default: False),\n",
      " 'dlabrax_homolog_associated_gene_name': 'European seabass gene name' (default: False),\n",
      " 'dlabrax_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dlabrax_homolog_chrom_end': 'European seabass chromosome/scaffold end (bp)' (default: False),\n",
      " 'dlabrax_homolog_chrom_start': 'European seabass chromosome/scaffold start (bp)' (default: False),\n",
      " 'dlabrax_homolog_chromosome': 'European seabass chromosome/scaffold name' (default: False),\n",
      " 'dlabrax_homolog_ensembl_gene': 'European seabass gene stable ID' (default: False),\n",
      " 'dlabrax_homolog_ensembl_peptide': 'European seabass protein or transcript stable ID' (default: False),\n",
      " 'dlabrax_homolog_goc_score': 'European seabass Gene-order conservation score' (default: False),\n",
      " 'dlabrax_homolog_orthology_confidence': 'European seabass orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dlabrax_homolog_orthology_type': 'European seabass homology type' (default: False),\n",
      " 'dlabrax_homolog_perc_id': '%id. target European seabass gene identical to query gene' (default: False),\n",
      " 'dlabrax_homolog_perc_id_r1': '%id. query gene identical to target European seabass gene' (default: False),\n",
      " 'dlabrax_homolog_subtype': 'Last common ancestor with European seabass' (default: False),\n",
      " 'dleucas_homolog_associated_gene_name': 'Beluga whale gene name' (default: False),\n",
      " 'dleucas_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dleucas_homolog_chrom_end': 'Beluga whale chromosome/scaffold end (bp)' (default: False),\n",
      " 'dleucas_homolog_chrom_start': 'Beluga whale chromosome/scaffold start (bp)' (default: False),\n",
      " 'dleucas_homolog_chromosome': 'Beluga whale chromosome/scaffold name' (default: False),\n",
      " 'dleucas_homolog_ensembl_gene': 'Beluga whale gene stable ID' (default: False),\n",
      " 'dleucas_homolog_ensembl_peptide': 'Beluga whale protein or transcript stable ID' (default: False),\n",
      " 'dleucas_homolog_goc_score': 'Beluga whale Gene-order conservation score' (default: False),\n",
      " 'dleucas_homolog_orthology_confidence': 'Beluga whale orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dleucas_homolog_orthology_type': 'Beluga whale homology type' (default: False),\n",
      " 'dleucas_homolog_perc_id': '%id. target Beluga whale gene identical to query gene' (default: False),\n",
      " 'dleucas_homolog_perc_id_r1': '%id. query gene identical to target Beluga whale gene' (default: False),\n",
      " 'dleucas_homolog_subtype': 'Last common ancestor with Beluga whale' (default: False),\n",
      " 'dleucas_homolog_wga_coverage': 'Beluga whale Whole-genome alignment coverage' (default: False),\n",
      " 'dmelanogaster_homolog_associated_gene_name': 'Drosophila melanogaster (Fruit fly) gene name' (default: False),\n",
      " 'dmelanogaster_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dmelanogaster_homolog_chrom_end': 'Drosophila melanogaster (Fruit fly) chromosome/scaffold end (bp)' (default: False),\n",
      " 'dmelanogaster_homolog_chrom_start': 'Drosophila melanogaster (Fruit fly) chromosome/scaffold start (bp)' (default: False),\n",
      " 'dmelanogaster_homolog_chromosome': 'Drosophila melanogaster (Fruit fly) chromosome/scaffold name' (default: False),\n",
      " 'dmelanogaster_homolog_ensembl_gene': 'Drosophila melanogaster (Fruit fly) gene stable ID' (default: False),\n",
      " 'dmelanogaster_homolog_ensembl_peptide': 'Drosophila melanogaster (Fruit fly) protein or transcript stable ID' (default: False),\n",
      " 'dmelanogaster_homolog_orthology_confidence': 'Drosophila melanogaster (Fruit fly) orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dmelanogaster_homolog_orthology_type': 'Drosophila melanogaster (Fruit fly) homology type' (default: False),\n",
      " 'dmelanogaster_homolog_perc_id': '%id. target Drosophila melanogaster (Fruit fly) gene identical to query gene' (default: False),\n",
      " 'dmelanogaster_homolog_perc_id_r1': '%id. query gene identical to target Drosophila melanogaster (Fruit fly) gene' (default: False),\n",
      " 'dmelanogaster_homolog_subtype': 'Last common ancestor with Drosophila melanogaster (Fruit fly)' (default: False),\n",
      " 'dnovemcinctus_homolog_associated_gene_name': 'Armadillo gene name' (default: False),\n",
      " 'dnovemcinctus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dnovemcinctus_homolog_chrom_end': 'Armadillo chromosome/scaffold end (bp)' (default: False),\n",
      " 'dnovemcinctus_homolog_chrom_start': 'Armadillo chromosome/scaffold start (bp)' (default: False),\n",
      " 'dnovemcinctus_homolog_chromosome': 'Armadillo chromosome/scaffold name' (default: False),\n",
      " 'dnovemcinctus_homolog_ensembl_gene': 'Armadillo gene stable ID' (default: False),\n",
      " 'dnovemcinctus_homolog_ensembl_peptide': 'Armadillo protein or transcript stable ID' (default: False),\n",
      " 'dnovemcinctus_homolog_goc_score': 'Armadillo Gene-order conservation score' (default: False),\n",
      " 'dnovemcinctus_homolog_orthology_confidence': 'Armadillo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dnovemcinctus_homolog_orthology_type': 'Armadillo homology type' (default: False),\n",
      " 'dnovemcinctus_homolog_perc_id': '%id. target Armadillo gene identical to query gene' (default: False),\n",
      " 'dnovemcinctus_homolog_perc_id_r1': '%id. query gene identical to target Armadillo gene' (default: False),\n",
      " 'dnovemcinctus_homolog_subtype': 'Last common ancestor with Armadillo' (default: False),\n",
      " 'dnovemcinctus_homolog_wga_coverage': 'Armadillo Whole-genome alignment coverage' (default: False),\n",
      " 'dordii_homolog_associated_gene_name': 'Kangaroo rat gene name' (default: False),\n",
      " 'dordii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dordii_homolog_chrom_end': 'Kangaroo rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'dordii_homolog_chrom_start': 'Kangaroo rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'dordii_homolog_chromosome': 'Kangaroo rat chromosome/scaffold name' (default: False),\n",
      " 'dordii_homolog_ensembl_gene': 'Kangaroo rat gene stable ID' (default: False),\n",
      " 'dordii_homolog_ensembl_peptide': 'Kangaroo rat protein or transcript stable ID' (default: False),\n",
      " 'dordii_homolog_goc_score': 'Kangaroo rat Gene-order conservation score' (default: False),\n",
      " 'dordii_homolog_orthology_confidence': 'Kangaroo rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dordii_homolog_orthology_type': 'Kangaroo rat homology type' (default: False),\n",
      " 'dordii_homolog_perc_id': '%id. target Kangaroo rat gene identical to query gene' (default: False),\n",
      " 'dordii_homolog_perc_id_r1': '%id. query gene identical to target Kangaroo rat gene' (default: False),\n",
      " 'dordii_homolog_subtype': 'Last common ancestor with Kangaroo rat' (default: False),\n",
      " 'dordii_homolog_wga_coverage': 'Kangaroo rat Whole-genome alignment coverage' (default: False),\n",
      " 'downstream_flank': 'downstream_flank' (default: False),\n",
      " 'drerio_homolog_associated_gene_name': 'Zebrafish gene name' (default: False),\n",
      " 'drerio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'drerio_homolog_chrom_end': 'Zebrafish chromosome/scaffold end (bp)' (default: False),\n",
      " 'drerio_homolog_chrom_start': 'Zebrafish chromosome/scaffold start (bp)' (default: False),\n",
      " 'drerio_homolog_chromosome': 'Zebrafish chromosome/scaffold name' (default: False),\n",
      " 'drerio_homolog_ensembl_gene': 'Zebrafish gene stable ID' (default: False),\n",
      " 'drerio_homolog_ensembl_peptide': 'Zebrafish protein or transcript stable ID' (default: False),\n",
      " 'drerio_homolog_goc_score': 'Zebrafish Gene-order conservation score' (default: False),\n",
      " 'drerio_homolog_orthology_confidence': 'Zebrafish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'drerio_homolog_orthology_type': 'Zebrafish homology type' (default: False),\n",
      " 'drerio_homolog_perc_id': '%id. target Zebrafish gene identical to query gene' (default: False),\n",
      " 'drerio_homolog_perc_id_r1': '%id. query gene identical to target Zebrafish gene' (default: False),\n",
      " 'drerio_homolog_subtype': 'Last common ancestor with Zebrafish' (default: False),\n",
      " 'drerio_homolog_wga_coverage': 'Zebrafish Whole-genome alignment coverage' (default: False),\n",
      " 'easinus_homolog_associated_gene_name': 'Donkey gene name' (default: False),\n",
      " 'easinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'easinus_homolog_chrom_end': 'Donkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'easinus_homolog_chrom_start': 'Donkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'easinus_homolog_chromosome': 'Donkey chromosome/scaffold name' (default: False),\n",
      " 'easinus_homolog_ensembl_gene': 'Donkey gene stable ID' (default: False),\n",
      " 'easinus_homolog_ensembl_peptide': 'Donkey protein or transcript stable ID' (default: False),\n",
      " 'easinus_homolog_goc_score': 'Donkey Gene-order conservation score' (default: False),\n",
      " 'easinus_homolog_orthology_confidence': 'Donkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'easinus_homolog_orthology_type': 'Donkey homology type' (default: False),\n",
      " 'easinus_homolog_perc_id': '%id. target Donkey gene identical to query gene' (default: False),\n",
      " 'easinus_homolog_perc_id_r1': '%id. query gene identical to target Donkey gene' (default: False),\n",
      " 'easinus_homolog_subtype': 'Last common ancestor with Donkey' (default: False),\n",
      " 'eburgeri_homolog_associated_gene_name': 'Hagfish gene name' (default: False),\n",
      " 'eburgeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eburgeri_homolog_chrom_end': 'Hagfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'eburgeri_homolog_chrom_start': 'Hagfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'eburgeri_homolog_chromosome': 'Hagfish chromosome/scaffold name' (default: False),\n",
      " 'eburgeri_homolog_ensembl_gene': 'Hagfish gene stable ID' (default: False),\n",
      " 'eburgeri_homolog_ensembl_peptide': 'Hagfish protein or transcript stable ID' (default: False),\n",
      " 'eburgeri_homolog_orthology_confidence': 'Hagfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eburgeri_homolog_orthology_type': 'Hagfish homology type' (default: False),\n",
      " 'eburgeri_homolog_perc_id': '%id. target Hagfish gene identical to query gene' (default: False),\n",
      " 'eburgeri_homolog_perc_id_r1': '%id. query gene identical to target Hagfish gene' (default: False),\n",
      " 'eburgeri_homolog_subtype': 'Last common ancestor with Hagfish' (default: False),\n",
      " 'eburgeri_homolog_wga_coverage': 'Hagfish Whole-genome alignment coverage' (default: False),\n",
      " 'ecaballus_homolog_associated_gene_name': 'Horse gene name' (default: False),\n",
      " 'ecaballus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ecaballus_homolog_chrom_end': 'Horse chromosome/scaffold end (bp)' (default: False),\n",
      " 'ecaballus_homolog_chrom_start': 'Horse chromosome/scaffold start (bp)' (default: False),\n",
      " 'ecaballus_homolog_chromosome': 'Horse chromosome/scaffold name' (default: False),\n",
      " 'ecaballus_homolog_ensembl_gene': 'Horse gene stable ID' (default: False),\n",
      " 'ecaballus_homolog_ensembl_peptide': 'Horse protein or transcript stable ID' (default: False),\n",
      " 'ecaballus_homolog_goc_score': 'Horse Gene-order conservation score' (default: False),\n",
      " 'ecaballus_homolog_orthology_confidence': 'Horse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ecaballus_homolog_orthology_type': 'Horse homology type' (default: False),\n",
      " 'ecaballus_homolog_perc_id': '%id. target Horse gene identical to query gene' (default: False),\n",
      " 'ecaballus_homolog_perc_id_r1': '%id. query gene identical to target Horse gene' (default: False),\n",
      " 'ecaballus_homolog_subtype': 'Last common ancestor with Horse' (default: False),\n",
      " 'ecaballus_homolog_wga_coverage': 'Horse Whole-genome alignment coverage' (default: False),\n",
      " 'ecalabaricus_homolog_associated_gene_name': 'Reedfish gene name' (default: False),\n",
      " 'ecalabaricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ecalabaricus_homolog_chrom_end': 'Reedfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ecalabaricus_homolog_chrom_start': 'Reedfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ecalabaricus_homolog_chromosome': 'Reedfish chromosome/scaffold name' (default: False),\n",
      " 'ecalabaricus_homolog_ensembl_gene': 'Reedfish gene stable ID' (default: False),\n",
      " 'ecalabaricus_homolog_ensembl_peptide': 'Reedfish protein or transcript stable ID' (default: False),\n",
      " 'ecalabaricus_homolog_goc_score': 'Reedfish Gene-order conservation score' (default: False),\n",
      " 'ecalabaricus_homolog_orthology_confidence': 'Reedfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ecalabaricus_homolog_orthology_type': 'Reedfish homology type' (default: False),\n",
      " 'ecalabaricus_homolog_perc_id': '%id. target Reedfish gene identical to query gene' (default: False),\n",
      " 'ecalabaricus_homolog_perc_id_r1': '%id. query gene identical to target Reedfish gene' (default: False),\n",
      " 'ecalabaricus_homolog_subtype': 'Last common ancestor with Reedfish' (default: False),\n",
      " 'ecalabaricus_homolog_wga_coverage': 'Reedfish Whole-genome alignment coverage' (default: False),\n",
      " 'eelectricus_homolog_associated_gene_name': 'Electric eel gene name' (default: False),\n",
      " 'eelectricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eelectricus_homolog_chrom_end': 'Electric eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'eelectricus_homolog_chrom_start': 'Electric eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'eelectricus_homolog_chromosome': 'Electric eel chromosome/scaffold name' (default: False),\n",
      " 'eelectricus_homolog_ensembl_gene': 'Electric eel gene stable ID' (default: False),\n",
      " 'eelectricus_homolog_ensembl_peptide': 'Electric eel protein or transcript stable ID' (default: False),\n",
      " 'eelectricus_homolog_goc_score': 'Electric eel Gene-order conservation score' (default: False),\n",
      " 'eelectricus_homolog_orthology_confidence': 'Electric eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eelectricus_homolog_orthology_type': 'Electric eel homology type' (default: False),\n",
      " 'eelectricus_homolog_perc_id': '%id. target Electric eel gene identical to query gene' (default: False),\n",
      " 'eelectricus_homolog_perc_id_r1': '%id. query gene identical to target Electric eel gene' (default: False),\n",
      " 'eelectricus_homolog_subtype': 'Last common ancestor with Electric eel' (default: False),\n",
      " 'eelectricus_homolog_wga_coverage': 'Electric eel Whole-genome alignment coverage' (default: False),\n",
      " 'eeuropaeus_homolog_associated_gene_name': 'Hedgehog gene name' (default: False),\n",
      " 'eeuropaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eeuropaeus_homolog_chrom_end': 'Hedgehog chromosome/scaffold end (bp)' (default: False),\n",
      " 'eeuropaeus_homolog_chrom_start': 'Hedgehog chromosome/scaffold start (bp)' (default: False),\n",
      " 'eeuropaeus_homolog_chromosome': 'Hedgehog chromosome/scaffold name' (default: False),\n",
      " 'eeuropaeus_homolog_ensembl_gene': 'Hedgehog gene stable ID' (default: False),\n",
      " 'eeuropaeus_homolog_ensembl_peptide': 'Hedgehog protein or transcript stable ID' (default: False),\n",
      " 'eeuropaeus_homolog_goc_score': 'Hedgehog Gene-order conservation score' (default: False),\n",
      " 'eeuropaeus_homolog_orthology_confidence': 'Hedgehog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eeuropaeus_homolog_orthology_type': 'Hedgehog homology type' (default: False),\n",
      " 'eeuropaeus_homolog_perc_id': '%id. target Hedgehog gene identical to query gene' (default: False),\n",
      " 'eeuropaeus_homolog_perc_id_r1': '%id. query gene identical to target Hedgehog gene' (default: False),\n",
      " 'eeuropaeus_homolog_subtype': 'Last common ancestor with Hedgehog' (default: False),\n",
      " 'eeuropaeus_homolog_wga_coverage': 'Hedgehog Whole-genome alignment coverage' (default: False),\n",
      " 'elucius_homolog_associated_gene_name': 'Northern pike gene name' (default: False),\n",
      " 'elucius_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'elucius_homolog_chrom_end': 'Northern pike chromosome/scaffold end (bp)' (default: False),\n",
      " 'elucius_homolog_chrom_start': 'Northern pike chromosome/scaffold start (bp)' (default: False),\n",
      " 'elucius_homolog_chromosome': 'Northern pike chromosome/scaffold name' (default: False),\n",
      " 'elucius_homolog_ensembl_gene': 'Northern pike gene stable ID' (default: False),\n",
      " 'elucius_homolog_ensembl_peptide': 'Northern pike protein or transcript stable ID' (default: False),\n",
      " 'elucius_homolog_goc_score': 'Northern pike Gene-order conservation score' (default: False),\n",
      " 'elucius_homolog_orthology_confidence': 'Northern pike orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'elucius_homolog_orthology_type': 'Northern pike homology type' (default: False),\n",
      " 'elucius_homolog_perc_id': '%id. target Northern pike gene identical to query gene' (default: False),\n",
      " 'elucius_homolog_perc_id_r1': '%id. query gene identical to target Northern pike gene' (default: False),\n",
      " 'elucius_homolog_subtype': 'Last common ancestor with Northern pike' (default: False),\n",
      " 'elucius_homolog_wga_coverage': 'Northern pike Whole-genome alignment coverage' (default: False),\n",
      " 'embl': 'European Nucleotide Archive ID' (default: False),\n",
      " 'end_phase': 'End phase' (default: False),\n",
      " 'end_position': 'Gene end (bp)' (default: False),\n",
      " 'ens_lrg_gene': 'LRG display in Ensembl gene ID' (default: False),\n",
      " 'ens_lrg_transcript': 'LRG display in Ensembl transcript ID' (default: False),\n",
      " 'ensembl_exon_id': 'Exon stable ID' (default: False),\n",
      " 'ensembl_gene_id': 'Gene stable ID' (default: False),\n",
      " 'ensembl_gene_id_version': 'Gene stable ID version' (default: False),\n",
      " 'ensembl_peptide_id': 'Protein stable ID' (default: False),\n",
      " 'ensembl_peptide_id_version': 'Protein stable ID version' (default: False),\n",
      " 'ensembl_transcript_id': 'Transcript stable ID' (default: False),\n",
      " 'ensembl_transcript_id_version': 'Transcript stable ID version' (default: False),\n",
      " 'entrezgene_accession': 'NCBI gene (formerly Entrezgene) accession' (default: False),\n",
      " 'entrezgene_description': 'NCBI gene (formerly Entrezgene) description' (default: False),\n",
      " 'entrezgene_id': 'NCBI gene (formerly Entrezgene) ID' (default: False),\n",
      " 'entrezgene_trans_name': 'EntrezGene transcript name ID' (default: False),\n",
      " 'etelfairi_homolog_associated_gene_name': 'Lesser hedgehog tenrec gene name' (default: False),\n",
      " 'etelfairi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'etelfairi_homolog_chrom_end': 'Lesser hedgehog tenrec chromosome/scaffold end (bp)' (default: False),\n",
      " 'etelfairi_homolog_chrom_start': 'Lesser hedgehog tenrec chromosome/scaffold start (bp)' (default: False),\n",
      " 'etelfairi_homolog_chromosome': 'Lesser hedgehog tenrec chromosome/scaffold name' (default: False),\n",
      " 'etelfairi_homolog_ensembl_gene': 'Lesser hedgehog tenrec gene stable ID' (default: False),\n",
      " 'etelfairi_homolog_ensembl_peptide': 'Lesser hedgehog tenrec protein or transcript stable ID' (default: False),\n",
      " 'etelfairi_homolog_goc_score': 'Lesser hedgehog tenrec Gene-order conservation score' (default: False),\n",
      " 'etelfairi_homolog_orthology_confidence': 'Lesser hedgehog tenrec orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'etelfairi_homolog_orthology_type': 'Lesser hedgehog tenrec homology type' (default: False),\n",
      " 'etelfairi_homolog_perc_id': '%id. target Lesser hedgehog tenrec gene identical to query gene' (default: False),\n",
      " 'etelfairi_homolog_perc_id_r1': '%id. query gene identical to target Lesser hedgehog tenrec gene' (default: False),\n",
      " 'etelfairi_homolog_subtype': 'Last common ancestor with Lesser hedgehog tenrec' (default: False),\n",
      " 'etelfairi_homolog_wga_coverage': 'Lesser hedgehog tenrec Whole-genome alignment coverage' (default: False),\n",
      " 'exon_chrom_end': 'Exon region end (bp)' (default: False),\n",
      " 'exon_chrom_start': 'Exon region start (bp)' (default: False),\n",
      " 'external_gene_name': 'Gene name' (default: False),\n",
      " 'external_gene_source': 'Source of gene name' (default: False),\n",
      " 'external_synonym': 'Gene Synonym' (default: False),\n",
      " 'external_transcript_name': 'Transcript name' (default: False),\n",
      " 'external_transcript_source_name': 'Source of transcript name' (default: False),\n",
      " 'falbicollis_homolog_associated_gene_name': 'Collared flycatcher gene name' (default: False),\n",
      " 'falbicollis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'falbicollis_homolog_chrom_end': 'Collared flycatcher chromosome/scaffold end (bp)' (default: False),\n",
      " 'falbicollis_homolog_chrom_start': 'Collared flycatcher chromosome/scaffold start (bp)' (default: False),\n",
      " 'falbicollis_homolog_chromosome': 'Collared flycatcher chromosome/scaffold name' (default: False),\n",
      " 'falbicollis_homolog_ensembl_gene': 'Collared flycatcher gene stable ID' (default: False),\n",
      " 'falbicollis_homolog_ensembl_peptide': 'Collared flycatcher protein or transcript stable ID' (default: False),\n",
      " 'falbicollis_homolog_goc_score': 'Collared flycatcher Gene-order conservation score' (default: False),\n",
      " 'falbicollis_homolog_orthology_confidence': 'Collared flycatcher orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'falbicollis_homolog_orthology_type': 'Collared flycatcher homology type' (default: False),\n",
      " 'falbicollis_homolog_perc_id': '%id. target Collared flycatcher gene identical to query gene' (default: False),\n",
      " 'falbicollis_homolog_perc_id_r1': '%id. query gene identical to target Collared flycatcher gene' (default: False),\n",
      " 'falbicollis_homolog_subtype': 'Last common ancestor with Collared flycatcher' (default: False),\n",
      " 'fcatus_homolog_associated_gene_name': 'Cat gene name' (default: False),\n",
      " 'fcatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fcatus_homolog_chrom_end': 'Cat chromosome/scaffold end (bp)' (default: False),\n",
      " 'fcatus_homolog_chrom_start': 'Cat chromosome/scaffold start (bp)' (default: False),\n",
      " 'fcatus_homolog_chromosome': 'Cat chromosome/scaffold name' (default: False),\n",
      " 'fcatus_homolog_ensembl_gene': 'Cat gene stable ID' (default: False),\n",
      " 'fcatus_homolog_ensembl_peptide': 'Cat protein or transcript stable ID' (default: False),\n",
      " 'fcatus_homolog_goc_score': 'Cat Gene-order conservation score' (default: False),\n",
      " 'fcatus_homolog_orthology_confidence': 'Cat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fcatus_homolog_orthology_type': 'Cat homology type' (default: False),\n",
      " 'fcatus_homolog_perc_id': '%id. target Cat gene identical to query gene' (default: False),\n",
      " 'fcatus_homolog_perc_id_r1': '%id. query gene identical to target Cat gene' (default: False),\n",
      " 'fcatus_homolog_subtype': 'Last common ancestor with Cat' (default: False),\n",
      " 'fcatus_homolog_wga_coverage': 'Cat Whole-genome alignment coverage' (default: False),\n",
      " 'fheteroclitus_homolog_associated_gene_name': 'Mummichog gene name' (default: False),\n",
      " 'fheteroclitus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fheteroclitus_homolog_chrom_end': 'Mummichog chromosome/scaffold end (bp)' (default: False),\n",
      " 'fheteroclitus_homolog_chrom_start': 'Mummichog chromosome/scaffold start (bp)' (default: False),\n",
      " 'fheteroclitus_homolog_chromosome': 'Mummichog chromosome/scaffold name' (default: False),\n",
      " 'fheteroclitus_homolog_ensembl_gene': 'Mummichog gene stable ID' (default: False),\n",
      " 'fheteroclitus_homolog_ensembl_peptide': 'Mummichog protein or transcript stable ID' (default: False),\n",
      " 'fheteroclitus_homolog_goc_score': 'Mummichog Gene-order conservation score' (default: False),\n",
      " 'fheteroclitus_homolog_orthology_confidence': 'Mummichog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fheteroclitus_homolog_orthology_type': 'Mummichog homology type' (default: False),\n",
      " 'fheteroclitus_homolog_perc_id': '%id. target Mummichog gene identical to query gene' (default: False),\n",
      " 'fheteroclitus_homolog_perc_id_r1': '%id. query gene identical to target Mummichog gene' (default: False),\n",
      " 'fheteroclitus_homolog_subtype': 'Last common ancestor with Mummichog' (default: False),\n",
      " 'fheteroclitus_homolog_wga_coverage': 'Mummichog Whole-genome alignment coverage' (default: False),\n",
      " 'gaculeatus_homolog_associated_gene_name': 'Stickleback gene name' (default: False),\n",
      " 'gaculeatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gaculeatus_homolog_chrom_end': 'Stickleback chromosome/scaffold end (bp)' (default: False),\n",
      " 'gaculeatus_homolog_chrom_start': 'Stickleback chromosome/scaffold start (bp)' (default: False),\n",
      " 'gaculeatus_homolog_chromosome': 'Stickleback chromosome/scaffold name' (default: False),\n",
      " 'gaculeatus_homolog_ensembl_gene': 'Stickleback gene stable ID' (default: False),\n",
      " 'gaculeatus_homolog_ensembl_peptide': 'Stickleback protein or transcript stable ID' (default: False),\n",
      " 'gaculeatus_homolog_goc_score': 'Stickleback Gene-order conservation score' (default: False),\n",
      " 'gaculeatus_homolog_orthology_confidence': 'Stickleback orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gaculeatus_homolog_orthology_type': 'Stickleback homology type' (default: False),\n",
      " 'gaculeatus_homolog_perc_id': '%id. target Stickleback gene identical to query gene' (default: False),\n",
      " 'gaculeatus_homolog_perc_id_r1': '%id. query gene identical to target Stickleback gene' (default: False),\n",
      " 'gaculeatus_homolog_subtype': 'Last common ancestor with Stickleback' (default: False),\n",
      " 'gaculeatus_homolog_wga_coverage': 'Stickleback Whole-genome alignment coverage' (default: False),\n",
      " 'gene3d': 'Gene3D ID' (default: False),\n",
      " 'gene3d_end': 'Gene3D end' (default: False),\n",
      " 'gene3d_start': 'Gene3D start' (default: False),\n",
      " 'gene_biotype': 'Gene type' (default: False),\n",
      " 'gene_exon': 'Exon sequences' (default: False),\n",
      " 'gene_exon_intron': 'Unspliced (Gene)' (default: False),\n",
      " 'gene_flank': 'Flank (Gene)' (default: False),\n",
      " 'genecards': 'GeneCards ID' (default: False),\n",
      " 'genomic_coding_end': 'Genomic coding end' (default: False),\n",
      " 'genomic_coding_start': 'Genomic coding start' (default: False),\n",
      " 'germ_line_variation_source': 'Variant source' (default: False),\n",
      " 'gevgoodei_homolog_associated_gene_name': 'Goodes thornscrub tortoise gene name' (default: False),\n",
      " 'gevgoodei_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gevgoodei_homolog_chrom_end': 'Goodes thornscrub tortoise chromosome/scaffold end (bp)' (default: False),\n",
      " 'gevgoodei_homolog_chrom_start': 'Goodes thornscrub tortoise chromosome/scaffold start (bp)' (default: False),\n",
      " 'gevgoodei_homolog_chromosome': 'Goodes thornscrub tortoise chromosome/scaffold name' (default: False),\n",
      " 'gevgoodei_homolog_ensembl_gene': 'Goodes thornscrub tortoise gene stable ID' (default: False),\n",
      " 'gevgoodei_homolog_ensembl_peptide': 'Goodes thornscrub tortoise protein or transcript stable ID' (default: False),\n",
      " 'gevgoodei_homolog_goc_score': 'Goodes thornscrub tortoise Gene-order conservation score' (default: False),\n",
      " 'gevgoodei_homolog_orthology_confidence': 'Goodes thornscrub tortoise orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gevgoodei_homolog_orthology_type': 'Goodes thornscrub tortoise homology type' (default: False),\n",
      " 'gevgoodei_homolog_perc_id': '%id. target Goodes thornscrub tortoise gene identical to query gene' (default: False),\n",
      " 'gevgoodei_homolog_perc_id_r1': '%id. query gene identical to target Goodes thornscrub tortoise gene' (default: False),\n",
      " 'gevgoodei_homolog_subtype': 'Last common ancestor with Goodes thornscrub tortoise' (default: False),\n",
      " 'gevgoodei_homolog_wga_coverage': 'Goodes thornscrub tortoise Whole-genome alignment coverage' (default: False),\n",
      " 'gfortis_homolog_associated_gene_name': 'Medium ground-finch gene name' (default: False),\n",
      " 'gfortis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gfortis_homolog_chrom_end': 'Medium ground-finch chromosome/scaffold end (bp)' (default: False),\n",
      " 'gfortis_homolog_chrom_start': 'Medium ground-finch chromosome/scaffold start (bp)' (default: False),\n",
      " 'gfortis_homolog_chromosome': 'Medium ground-finch chromosome/scaffold name' (default: False),\n",
      " 'gfortis_homolog_ensembl_gene': 'Medium ground-finch gene stable ID' (default: False),\n",
      " 'gfortis_homolog_ensembl_peptide': 'Medium ground-finch protein or transcript stable ID' (default: False),\n",
      " 'gfortis_homolog_goc_score': 'Medium ground-finch Gene-order conservation score' (default: False),\n",
      " 'gfortis_homolog_orthology_confidence': 'Medium ground-finch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gfortis_homolog_orthology_type': 'Medium ground-finch homology type' (default: False),\n",
      " 'gfortis_homolog_perc_id': '%id. target Medium ground-finch gene identical to query gene' (default: False),\n",
      " 'gfortis_homolog_perc_id_r1': '%id. query gene identical to target Medium ground-finch gene' (default: False),\n",
      " 'gfortis_homolog_subtype': 'Last common ancestor with Medium ground-finch' (default: False),\n",
      " 'gfortis_homolog_wga_coverage': 'Medium ground-finch Whole-genome alignment coverage' (default: False),\n",
      " 'ggallus_homolog_associated_gene_name': 'Chicken gene name' (default: False),\n",
      " 'ggallus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ggallus_homolog_chrom_end': 'Chicken chromosome/scaffold end (bp)' (default: False),\n",
      " 'ggallus_homolog_chrom_start': 'Chicken chromosome/scaffold start (bp)' (default: False),\n",
      " 'ggallus_homolog_chromosome': 'Chicken chromosome/scaffold name' (default: False),\n",
      " 'ggallus_homolog_ensembl_gene': 'Chicken gene stable ID' (default: False),\n",
      " 'ggallus_homolog_ensembl_peptide': 'Chicken protein or transcript stable ID' (default: False),\n",
      " 'ggallus_homolog_goc_score': 'Chicken Gene-order conservation score' (default: False),\n",
      " 'ggallus_homolog_orthology_confidence': 'Chicken orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ggallus_homolog_orthology_type': 'Chicken homology type' (default: False),\n",
      " 'ggallus_homolog_perc_id': '%id. target Chicken gene identical to query gene' (default: False),\n",
      " 'ggallus_homolog_perc_id_r1': '%id. query gene identical to target Chicken gene' (default: False),\n",
      " 'ggallus_homolog_subtype': 'Last common ancestor with Chicken' (default: False),\n",
      " 'ggallus_homolog_wga_coverage': 'Chicken Whole-genome alignment coverage' (default: False),\n",
      " 'ggorilla_homolog_associated_gene_name': 'Gorilla gene name' (default: False),\n",
      " 'ggorilla_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ggorilla_homolog_chrom_end': 'Gorilla chromosome/scaffold end (bp)' (default: False),\n",
      " 'ggorilla_homolog_chrom_start': 'Gorilla chromosome/scaffold start (bp)' (default: False),\n",
      " 'ggorilla_homolog_chromosome': 'Gorilla chromosome/scaffold name' (default: False),\n",
      " 'ggorilla_homolog_ensembl_gene': 'Gorilla gene stable ID' (default: False),\n",
      " 'ggorilla_homolog_ensembl_peptide': 'Gorilla protein or transcript stable ID' (default: False),\n",
      " 'ggorilla_homolog_goc_score': 'Gorilla Gene-order conservation score' (default: False),\n",
      " 'ggorilla_homolog_orthology_confidence': 'Gorilla orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ggorilla_homolog_orthology_type': 'Gorilla homology type' (default: False),\n",
      " 'ggorilla_homolog_perc_id': '%id. target Gorilla gene identical to query gene' (default: False),\n",
      " 'ggorilla_homolog_perc_id_r1': '%id. query gene identical to target Gorilla gene' (default: False),\n",
      " 'ggorilla_homolog_subtype': 'Last common ancestor with Gorilla' (default: False),\n",
      " 'ggorilla_homolog_wga_coverage': 'Gorilla Whole-genome alignment coverage' (default: False),\n",
      " 'gmorhua_homolog_associated_gene_name': 'Atlantic cod gene name' (default: False),\n",
      " 'gmorhua_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gmorhua_homolog_chrom_end': 'Atlantic cod chromosome/scaffold end (bp)' (default: False),\n",
      " 'gmorhua_homolog_chrom_start': 'Atlantic cod chromosome/scaffold start (bp)' (default: False),\n",
      " 'gmorhua_homolog_chromosome': 'Atlantic cod chromosome/scaffold name' (default: False),\n",
      " 'gmorhua_homolog_ensembl_gene': 'Atlantic cod gene stable ID' (default: False),\n",
      " 'gmorhua_homolog_ensembl_peptide': 'Atlantic cod protein or transcript stable ID' (default: False),\n",
      " 'gmorhua_homolog_goc_score': 'Atlantic cod Gene-order conservation score' (default: False),\n",
      " 'gmorhua_homolog_orthology_confidence': 'Atlantic cod orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gmorhua_homolog_orthology_type': 'Atlantic cod homology type' (default: False),\n",
      " 'gmorhua_homolog_perc_id': '%id. target Atlantic cod gene identical to query gene' (default: False),\n",
      " 'gmorhua_homolog_perc_id_r1': '%id. query gene identical to target Atlantic cod gene' (default: False),\n",
      " 'gmorhua_homolog_subtype': 'Last common ancestor with Atlantic cod' (default: False),\n",
      " 'go_id': 'GO term accession' (default: False),\n",
      " 'go_linkage_type': 'GO term evidence code' (default: False),\n",
      " 'goslim_goa_accession': 'GOSlim GOA Accession(s)' (default: False),\n",
      " 'goslim_goa_description': 'GOSlim GOA Description' (default: False),\n",
      " 'hamap': 'HAMAP ID' (default: False),\n",
      " 'hamap_end': 'HAMAP end' (default: False),\n",
      " 'hamap_start': 'HAMAP start' (default: False),\n",
      " 'hburtoni_homolog_associated_gene_name': 'Burton's mouthbrooder gene name' (default: False),\n",
      " 'hburtoni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hburtoni_homolog_chrom_end': 'Burton's mouthbrooder chromosome/scaffold end (bp)' (default: False),\n",
      " 'hburtoni_homolog_chrom_start': 'Burton's mouthbrooder chromosome/scaffold start (bp)' (default: False),\n",
      " 'hburtoni_homolog_chromosome': 'Burton's mouthbrooder chromosome/scaffold name' (default: False),\n",
      " 'hburtoni_homolog_ensembl_gene': 'Burton's mouthbrooder gene stable ID' (default: False),\n",
      " 'hburtoni_homolog_ensembl_peptide': 'Burton's mouthbrooder protein or transcript stable ID' (default: False),\n",
      " 'hburtoni_homolog_goc_score': 'Burton's mouthbrooder Gene-order conservation score' (default: False),\n",
      " 'hburtoni_homolog_orthology_confidence': 'Burton's mouthbrooder orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hburtoni_homolog_orthology_type': 'Burton's mouthbrooder homology type' (default: False),\n",
      " 'hburtoni_homolog_perc_id': '%id. target Burton's mouthbrooder gene identical to query gene' (default: False),\n",
      " 'hburtoni_homolog_perc_id_r1': '%id. query gene identical to target Burton's mouthbrooder gene' (default: False),\n",
      " 'hburtoni_homolog_subtype': 'Last common ancestor with Burton's mouthbrooder' (default: False),\n",
      " 'hburtoni_homolog_wga_coverage': 'Burton's mouthbrooder Whole-genome alignment coverage' (default: False),\n",
      " 'hcomes_homolog_associated_gene_name': 'Tiger tail seahorse gene name' (default: False),\n",
      " 'hcomes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hcomes_homolog_chrom_end': 'Tiger tail seahorse chromosome/scaffold end (bp)' (default: False),\n",
      " 'hcomes_homolog_chrom_start': 'Tiger tail seahorse chromosome/scaffold start (bp)' (default: False),\n",
      " 'hcomes_homolog_chromosome': 'Tiger tail seahorse chromosome/scaffold name' (default: False),\n",
      " 'hcomes_homolog_ensembl_gene': 'Tiger tail seahorse gene stable ID' (default: False),\n",
      " 'hcomes_homolog_ensembl_peptide': 'Tiger tail seahorse protein or transcript stable ID' (default: False),\n",
      " 'hcomes_homolog_goc_score': 'Tiger tail seahorse Gene-order conservation score' (default: False),\n",
      " 'hcomes_homolog_orthology_confidence': 'Tiger tail seahorse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hcomes_homolog_orthology_type': 'Tiger tail seahorse homology type' (default: False),\n",
      " 'hcomes_homolog_perc_id': '%id. target Tiger tail seahorse gene identical to query gene' (default: False),\n",
      " 'hcomes_homolog_perc_id_r1': '%id. query gene identical to target Tiger tail seahorse gene' (default: False),\n",
      " 'hcomes_homolog_subtype': 'Last common ancestor with Tiger tail seahorse' (default: False),\n",
      " 'hcomes_homolog_wga_coverage': 'Tiger tail seahorse Whole-genome alignment coverage' (default: False),\n",
      " 'hgfemale_homolog_associated_gene_name': 'Naked mole-rat female gene name' (default: False),\n",
      " 'hgfemale_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hgfemale_homolog_chrom_end': 'Naked mole-rat female chromosome/scaffold end (bp)' (default: False),\n",
      " 'hgfemale_homolog_chrom_start': 'Naked mole-rat female chromosome/scaffold start (bp)' (default: False),\n",
      " 'hgfemale_homolog_chromosome': 'Naked mole-rat female chromosome/scaffold name' (default: False),\n",
      " 'hgfemale_homolog_ensembl_gene': 'Naked mole-rat female gene stable ID' (default: False),\n",
      " 'hgfemale_homolog_ensembl_peptide': 'Naked mole-rat female protein or transcript stable ID' (default: False),\n",
      " 'hgfemale_homolog_goc_score': 'Naked mole-rat female Gene-order conservation score' (default: False),\n",
      " 'hgfemale_homolog_orthology_confidence': 'Naked mole-rat female orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hgfemale_homolog_orthology_type': 'Naked mole-rat female homology type' (default: False),\n",
      " 'hgfemale_homolog_perc_id': '%id. target Naked mole-rat female gene identical to query gene' (default: False),\n",
      " 'hgfemale_homolog_perc_id_r1': '%id. query gene identical to target Naked mole-rat female gene' (default: False),\n",
      " 'hgfemale_homolog_subtype': 'Last common ancestor with Naked mole-rat female' (default: False),\n",
      " 'hgnc_id': 'HGNC ID' (default: False),\n",
      " 'hgnc_symbol': 'HGNC symbol' (default: False),\n",
      " 'hgnc_trans_name': 'Transcript name ID' (default: False),\n",
      " 'hhucho_homolog_associated_gene_name': 'Huchen gene name' (default: False),\n",
      " 'hhucho_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hhucho_homolog_chrom_end': 'Huchen chromosome/scaffold end (bp)' (default: False),\n",
      " 'hhucho_homolog_chrom_start': 'Huchen chromosome/scaffold start (bp)' (default: False),\n",
      " 'hhucho_homolog_chromosome': 'Huchen chromosome/scaffold name' (default: False),\n",
      " 'hhucho_homolog_ensembl_gene': 'Huchen gene stable ID' (default: False),\n",
      " 'hhucho_homolog_ensembl_peptide': 'Huchen protein or transcript stable ID' (default: False),\n",
      " 'hhucho_homolog_goc_score': 'Huchen Gene-order conservation score' (default: False),\n",
      " 'hhucho_homolog_orthology_confidence': 'Huchen orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hhucho_homolog_orthology_type': 'Huchen homology type' (default: False),\n",
      " 'hhucho_homolog_perc_id': '%id. target Huchen gene identical to query gene' (default: False),\n",
      " 'hhucho_homolog_perc_id_r1': '%id. query gene identical to target Huchen gene' (default: False),\n",
      " 'hhucho_homolog_subtype': 'Last common ancestor with Huchen' (default: False),\n",
      " 'hhucho_homolog_wga_coverage': 'Huchen Whole-genome alignment coverage' (default: False),\n",
      " 'hmmpanther': 'PANTHER ID' (default: False),\n",
      " 'hmmpanther_end': 'PANTHER end' (default: False),\n",
      " 'hmmpanther_start': 'PANTHER start' (default: False),\n",
      " 'hpa_accession': 'Human Protein Atlas accession' (default: False),\n",
      " 'hpa_id': 'Human Protein Atlas ID' (default: False),\n",
      " 'hsapiens_paralog_associated_gene_name': 'Human paralogue associated gene name' (default: False),\n",
      " 'hsapiens_paralog_canonical_transcript_protein': 'Paralogue query protein or transcript ID' (default: False),\n",
      " 'hsapiens_paralog_chrom_end': 'Human paralogue chromosome/scaffold end (bp)' (default: False),\n",
      " 'hsapiens_paralog_chrom_start': 'Human paralogue chromosome/scaffold start (bp)' (default: False),\n",
      " 'hsapiens_paralog_chromosome': 'Human paralogue chromosome/scaffold name' (default: False),\n",
      " 'hsapiens_paralog_ensembl_gene': 'Human paralogue gene stable ID' (default: False),\n",
      " 'hsapiens_paralog_ensembl_peptide': 'Human paralogue protein or transcript ID' (default: False),\n",
      " 'hsapiens_paralog_orthology_type': 'Human paralogue homology type' (default: False),\n",
      " 'hsapiens_paralog_perc_id': 'Paralogue %id. target Human gene identical to query gene' (default: False),\n",
      " 'hsapiens_paralog_perc_id_r1': 'Paralogue %id. query gene identical to target Human gene' (default: False),\n",
      " 'hsapiens_paralog_subtype': 'Paralogue last common ancestor with Human' (default: False),\n",
      " 'illumina_humanref_8_v3': 'ILLUMINA HumanRef 8 V3 probe' (default: False),\n",
      " 'illumina_humanwg_6_v3': 'ILLUMINA HumanWG 6 V3 probe' (default: False),\n",
      " 'interpro': 'Interpro ID' (default: False),\n",
      " 'interpro_description': 'Interpro Description' (default: False),\n",
      " 'interpro_end': 'Interpro end' (default: False),\n",
      " 'interpro_short_description': 'Interpro Short Description' (default: False),\n",
      " 'interpro_start': 'Interpro start' (default: False),\n",
      " 'ipunctatus_homolog_associated_gene_name': 'Channel catfish gene name' (default: False),\n",
      " 'ipunctatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ipunctatus_homolog_chrom_end': 'Channel catfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ipunctatus_homolog_chrom_start': 'Channel catfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ipunctatus_homolog_chromosome': 'Channel catfish chromosome/scaffold name' (default: False),\n",
      " 'ipunctatus_homolog_ensembl_gene': 'Channel catfish gene stable ID' (default: False),\n",
      " 'ipunctatus_homolog_ensembl_peptide': 'Channel catfish protein or transcript stable ID' (default: False),\n",
      " 'ipunctatus_homolog_goc_score': 'Channel catfish Gene-order conservation score' (default: False),\n",
      " 'ipunctatus_homolog_orthology_confidence': 'Channel catfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ipunctatus_homolog_orthology_type': 'Channel catfish homology type' (default: False),\n",
      " 'ipunctatus_homolog_perc_id': '%id. target Channel catfish gene identical to query gene' (default: False),\n",
      " 'ipunctatus_homolog_perc_id_r1': '%id. query gene identical to target Channel catfish gene' (default: False),\n",
      " 'ipunctatus_homolog_subtype': 'Last common ancestor with Channel catfish' (default: False),\n",
      " 'ipunctatus_homolog_wga_coverage': 'Channel catfish Whole-genome alignment coverage' (default: False),\n",
      " 'is_constitutive': 'Constitutive exon' (default: False),\n",
      " 'itridecemlineatus_homolog_associated_gene_name': 'Squirrel gene name' (default: False),\n",
      " 'itridecemlineatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'itridecemlineatus_homolog_chrom_end': 'Squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'itridecemlineatus_homolog_chrom_start': 'Squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'itridecemlineatus_homolog_chromosome': 'Squirrel chromosome/scaffold name' (default: False),\n",
      " 'itridecemlineatus_homolog_ensembl_gene': 'Squirrel gene stable ID' (default: False),\n",
      " 'itridecemlineatus_homolog_ensembl_peptide': 'Squirrel protein or transcript stable ID' (default: False),\n",
      " 'itridecemlineatus_homolog_goc_score': 'Squirrel Gene-order conservation score' (default: False),\n",
      " 'itridecemlineatus_homolog_orthology_confidence': 'Squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'itridecemlineatus_homolog_orthology_type': 'Squirrel homology type' (default: False),\n",
      " 'itridecemlineatus_homolog_perc_id': '%id. target Squirrel gene identical to query gene' (default: False),\n",
      " 'itridecemlineatus_homolog_perc_id_r1': '%id. query gene identical to target Squirrel gene' (default: False),\n",
      " 'itridecemlineatus_homolog_subtype': 'Last common ancestor with Squirrel' (default: False),\n",
      " 'itridecemlineatus_homolog_wga_coverage': 'Squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'jjaculus_homolog_associated_gene_name': 'Lesser Egyptian jerboa gene name' (default: False),\n",
      " 'jjaculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'jjaculus_homolog_chrom_end': 'Lesser Egyptian jerboa chromosome/scaffold end (bp)' (default: False),\n",
      " 'jjaculus_homolog_chrom_start': 'Lesser Egyptian jerboa chromosome/scaffold start (bp)' (default: False),\n",
      " 'jjaculus_homolog_chromosome': 'Lesser Egyptian jerboa chromosome/scaffold name' (default: False),\n",
      " 'jjaculus_homolog_ensembl_gene': 'Lesser Egyptian jerboa gene stable ID' (default: False),\n",
      " 'jjaculus_homolog_ensembl_peptide': 'Lesser Egyptian jerboa protein or transcript stable ID' (default: False),\n",
      " 'jjaculus_homolog_goc_score': 'Lesser Egyptian jerboa Gene-order conservation score' (default: False),\n",
      " 'jjaculus_homolog_orthology_confidence': 'Lesser Egyptian jerboa orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'jjaculus_homolog_orthology_type': 'Lesser Egyptian jerboa homology type' (default: False),\n",
      " 'jjaculus_homolog_perc_id': '%id. target Lesser Egyptian jerboa gene identical to query gene' (default: False),\n",
      " 'jjaculus_homolog_perc_id_r1': '%id. query gene identical to target Lesser Egyptian jerboa gene' (default: False),\n",
      " 'jjaculus_homolog_subtype': 'Last common ancestor with Lesser Egyptian jerboa' (default: False),\n",
      " 'jjaculus_homolog_wga_coverage': 'Lesser Egyptian jerboa Whole-genome alignment coverage' (default: False),\n",
      " 'kmarmoratus_homolog_associated_gene_name': 'Mangrove rivulus gene name' (default: False),\n",
      " 'kmarmoratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'kmarmoratus_homolog_chrom_end': 'Mangrove rivulus chromosome/scaffold end (bp)' (default: False),\n",
      " 'kmarmoratus_homolog_chrom_start': 'Mangrove rivulus chromosome/scaffold start (bp)' (default: False),\n",
      " 'kmarmoratus_homolog_chromosome': 'Mangrove rivulus chromosome/scaffold name' (default: False),\n",
      " 'kmarmoratus_homolog_ensembl_gene': 'Mangrove rivulus gene stable ID' (default: False),\n",
      " 'kmarmoratus_homolog_ensembl_peptide': 'Mangrove rivulus protein or transcript stable ID' (default: False),\n",
      " 'kmarmoratus_homolog_goc_score': 'Mangrove rivulus Gene-order conservation score' (default: False),\n",
      " 'kmarmoratus_homolog_orthology_confidence': 'Mangrove rivulus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'kmarmoratus_homolog_orthology_type': 'Mangrove rivulus homology type' (default: False),\n",
      " 'kmarmoratus_homolog_perc_id': '%id. target Mangrove rivulus gene identical to query gene' (default: False),\n",
      " 'kmarmoratus_homolog_perc_id_r1': '%id. query gene identical to target Mangrove rivulus gene' (default: False),\n",
      " 'kmarmoratus_homolog_subtype': 'Last common ancestor with Mangrove rivulus' (default: False),\n",
      " 'kmarmoratus_homolog_wga_coverage': 'Mangrove rivulus Whole-genome alignment coverage' (default: False),\n",
      " 'lafricana_homolog_associated_gene_name': 'Elephant gene name' (default: False),\n",
      " 'lafricana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lafricana_homolog_chrom_end': 'Elephant chromosome/scaffold end (bp)' (default: False),\n",
      " 'lafricana_homolog_chrom_start': 'Elephant chromosome/scaffold start (bp)' (default: False),\n",
      " 'lafricana_homolog_chromosome': 'Elephant chromosome/scaffold name' (default: False),\n",
      " 'lafricana_homolog_ensembl_gene': 'Elephant gene stable ID' (default: False),\n",
      " 'lafricana_homolog_ensembl_peptide': 'Elephant protein or transcript stable ID' (default: False),\n",
      " 'lafricana_homolog_goc_score': 'Elephant Gene-order conservation score' (default: False),\n",
      " 'lafricana_homolog_orthology_confidence': 'Elephant orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lafricana_homolog_orthology_type': 'Elephant homology type' (default: False),\n",
      " 'lafricana_homolog_perc_id': '%id. target Elephant gene identical to query gene' (default: False),\n",
      " 'lafricana_homolog_perc_id_r1': '%id. query gene identical to target Elephant gene' (default: False),\n",
      " 'lafricana_homolog_subtype': 'Last common ancestor with Elephant' (default: False),\n",
      " 'lafricana_homolog_wga_coverage': 'Elephant Whole-genome alignment coverage' (default: False),\n",
      " 'lbergylta_homolog_associated_gene_name': 'Ballan wrasse gene name' (default: False),\n",
      " 'lbergylta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lbergylta_homolog_chrom_end': 'Ballan wrasse chromosome/scaffold end (bp)' (default: False),\n",
      " 'lbergylta_homolog_chrom_start': 'Ballan wrasse chromosome/scaffold start (bp)' (default: False),\n",
      " 'lbergylta_homolog_chromosome': 'Ballan wrasse chromosome/scaffold name' (default: False),\n",
      " 'lbergylta_homolog_ensembl_gene': 'Ballan wrasse gene stable ID' (default: False),\n",
      " 'lbergylta_homolog_ensembl_peptide': 'Ballan wrasse protein or transcript stable ID' (default: False),\n",
      " 'lbergylta_homolog_goc_score': 'Ballan wrasse Gene-order conservation score' (default: False),\n",
      " 'lbergylta_homolog_orthology_confidence': 'Ballan wrasse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lbergylta_homolog_orthology_type': 'Ballan wrasse homology type' (default: False),\n",
      " 'lbergylta_homolog_perc_id': '%id. target Ballan wrasse gene identical to query gene' (default: False),\n",
      " 'lbergylta_homolog_perc_id_r1': '%id. query gene identical to target Ballan wrasse gene' (default: False),\n",
      " 'lbergylta_homolog_subtype': 'Last common ancestor with Ballan wrasse' (default: False),\n",
      " 'lbergylta_homolog_wga_coverage': 'Ballan wrasse Whole-genome alignment coverage' (default: False),\n",
      " 'lcalcarifer_homolog_associated_gene_name': 'Barramundi perch gene name' (default: False),\n",
      " 'lcalcarifer_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcalcarifer_homolog_chrom_end': 'Barramundi perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcalcarifer_homolog_chrom_start': 'Barramundi perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcalcarifer_homolog_chromosome': 'Barramundi perch chromosome/scaffold name' (default: False),\n",
      " 'lcalcarifer_homolog_ensembl_gene': 'Barramundi perch gene stable ID' (default: False),\n",
      " 'lcalcarifer_homolog_ensembl_peptide': 'Barramundi perch protein or transcript stable ID' (default: False),\n",
      " 'lcalcarifer_homolog_goc_score': 'Barramundi perch Gene-order conservation score' (default: False),\n",
      " 'lcalcarifer_homolog_orthology_confidence': 'Barramundi perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcalcarifer_homolog_orthology_type': 'Barramundi perch homology type' (default: False),\n",
      " 'lcalcarifer_homolog_perc_id': '%id. target Barramundi perch gene identical to query gene' (default: False),\n",
      " 'lcalcarifer_homolog_perc_id_r1': '%id. query gene identical to target Barramundi perch gene' (default: False),\n",
      " 'lcalcarifer_homolog_subtype': 'Last common ancestor with Barramundi perch' (default: False),\n",
      " 'lcalcarifer_homolog_wga_coverage': 'Barramundi perch Whole-genome alignment coverage' (default: False),\n",
      " 'lchalumnae_homolog_associated_gene_name': 'Coelacanth gene name' (default: False),\n",
      " 'lchalumnae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lchalumnae_homolog_chrom_end': 'Coelacanth chromosome/scaffold end (bp)' (default: False),\n",
      " 'lchalumnae_homolog_chrom_start': 'Coelacanth chromosome/scaffold start (bp)' (default: False),\n",
      " 'lchalumnae_homolog_chromosome': 'Coelacanth chromosome/scaffold name' (default: False),\n",
      " 'lchalumnae_homolog_ensembl_gene': 'Coelacanth gene stable ID' (default: False),\n",
      " 'lchalumnae_homolog_ensembl_peptide': 'Coelacanth protein or transcript stable ID' (default: False),\n",
      " 'lchalumnae_homolog_goc_score': 'Coelacanth Gene-order conservation score' (default: False),\n",
      " 'lchalumnae_homolog_orthology_confidence': 'Coelacanth orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lchalumnae_homolog_orthology_type': 'Coelacanth homology type' (default: False),\n",
      " 'lchalumnae_homolog_perc_id': '%id. target Coelacanth gene identical to query gene' (default: False),\n",
      " 'lchalumnae_homolog_perc_id_r1': '%id. query gene identical to target Coelacanth gene' (default: False),\n",
      " 'lchalumnae_homolog_subtype': 'Last common ancestor with Coelacanth' (default: False),\n",
      " 'lchalumnae_homolog_wga_coverage': 'Coelacanth Whole-genome alignment coverage' (default: False),\n",
      " 'lcrocea_homolog_associated_gene_name': 'Large yellow croaker gene name' (default: False),\n",
      " 'lcrocea_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcrocea_homolog_chrom_end': 'Large yellow croaker chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcrocea_homolog_chrom_start': 'Large yellow croaker chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcrocea_homolog_chromosome': 'Large yellow croaker chromosome/scaffold name' (default: False),\n",
      " 'lcrocea_homolog_ensembl_gene': 'Large yellow croaker gene stable ID' (default: False),\n",
      " 'lcrocea_homolog_ensembl_peptide': 'Large yellow croaker protein or transcript stable ID' (default: False),\n",
      " 'lcrocea_homolog_goc_score': 'Large yellow croaker Gene-order conservation score' (default: False),\n",
      " 'lcrocea_homolog_orthology_confidence': 'Large yellow croaker orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcrocea_homolog_orthology_type': 'Large yellow croaker homology type' (default: False),\n",
      " 'lcrocea_homolog_perc_id': '%id. target Large yellow croaker gene identical to query gene' (default: False),\n",
      " 'lcrocea_homolog_perc_id_r1': '%id. query gene identical to target Large yellow croaker gene' (default: False),\n",
      " 'lcrocea_homolog_subtype': 'Last common ancestor with Large yellow croaker' (default: False),\n",
      " 'lcrocea_homolog_wga_coverage': 'Large yellow croaker Whole-genome alignment coverage' (default: False),\n",
      " 'llaticaudata_homolog_associated_gene_name': 'Blue-ringed sea krait gene name' (default: False),\n",
      " 'llaticaudata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'llaticaudata_homolog_chrom_end': 'Blue-ringed sea krait chromosome/scaffold end (bp)' (default: False),\n",
      " 'llaticaudata_homolog_chrom_start': 'Blue-ringed sea krait chromosome/scaffold start (bp)' (default: False),\n",
      " 'llaticaudata_homolog_chromosome': 'Blue-ringed sea krait chromosome/scaffold name' (default: False),\n",
      " 'llaticaudata_homolog_ensembl_gene': 'Blue-ringed sea krait gene stable ID' (default: False),\n",
      " 'llaticaudata_homolog_ensembl_peptide': 'Blue-ringed sea krait protein or transcript stable ID' (default: False),\n",
      " 'llaticaudata_homolog_goc_score': 'Blue-ringed sea krait Gene-order conservation score' (default: False),\n",
      " 'llaticaudata_homolog_orthology_confidence': 'Blue-ringed sea krait orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'llaticaudata_homolog_orthology_type': 'Blue-ringed sea krait homology type' (default: False),\n",
      " 'llaticaudata_homolog_perc_id': '%id. target Blue-ringed sea krait gene identical to query gene' (default: False),\n",
      " 'llaticaudata_homolog_perc_id_r1': '%id. query gene identical to target Blue-ringed sea krait gene' (default: False),\n",
      " 'llaticaudata_homolog_subtype': 'Last common ancestor with Blue-ringed sea krait' (default: False),\n",
      " 'llaticaudata_homolog_wga_coverage': 'Blue-ringed sea krait Whole-genome alignment coverage' (default: False),\n",
      " 'lleishanense_homolog_associated_gene_name': 'Leishan spiny toad gene name' (default: False),\n",
      " 'lleishanense_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lleishanense_homolog_chrom_end': 'Leishan spiny toad chromosome/scaffold end (bp)' (default: False),\n",
      " 'lleishanense_homolog_chrom_start': 'Leishan spiny toad chromosome/scaffold start (bp)' (default: False),\n",
      " 'lleishanense_homolog_chromosome': 'Leishan spiny toad chromosome/scaffold name' (default: False),\n",
      " 'lleishanense_homolog_ensembl_gene': 'Leishan spiny toad gene stable ID' (default: False),\n",
      " 'lleishanense_homolog_ensembl_peptide': 'Leishan spiny toad protein or transcript stable ID' (default: False),\n",
      " 'lleishanense_homolog_goc_score': 'Leishan spiny toad Gene-order conservation score' (default: False),\n",
      " 'lleishanense_homolog_orthology_confidence': 'Leishan spiny toad orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lleishanense_homolog_orthology_type': 'Leishan spiny toad homology type' (default: False),\n",
      " 'lleishanense_homolog_perc_id': '%id. target Leishan spiny toad gene identical to query gene' (default: False),\n",
      " 'lleishanense_homolog_perc_id_r1': '%id. query gene identical to target Leishan spiny toad gene' (default: False),\n",
      " 'lleishanense_homolog_subtype': 'Last common ancestor with Leishan spiny toad' (default: False),\n",
      " 'lleishanense_homolog_wga_coverage': 'Leishan spiny toad Whole-genome alignment coverage' (default: False),\n",
      " 'loculatus_homolog_associated_gene_name': 'Spotted gar gene name' (default: False),\n",
      " 'loculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'loculatus_homolog_chrom_end': 'Spotted gar chromosome/scaffold end (bp)' (default: False),\n",
      " 'loculatus_homolog_chrom_start': 'Spotted gar chromosome/scaffold start (bp)' (default: False),\n",
      " 'loculatus_homolog_chromosome': 'Spotted gar chromosome/scaffold name' (default: False),\n",
      " 'loculatus_homolog_ensembl_gene': 'Spotted gar gene stable ID' (default: False),\n",
      " 'loculatus_homolog_ensembl_peptide': 'Spotted gar protein or transcript stable ID' (default: False),\n",
      " 'loculatus_homolog_goc_score': 'Spotted gar Gene-order conservation score' (default: False),\n",
      " 'loculatus_homolog_orthology_confidence': 'Spotted gar orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'loculatus_homolog_orthology_type': 'Spotted gar homology type' (default: False),\n",
      " 'loculatus_homolog_perc_id': '%id. target Spotted gar gene identical to query gene' (default: False),\n",
      " 'loculatus_homolog_perc_id_r1': '%id. query gene identical to target Spotted gar gene' (default: False),\n",
      " 'loculatus_homolog_subtype': 'Last common ancestor with Spotted gar' (default: False),\n",
      " 'loculatus_homolog_wga_coverage': 'Spotted gar Whole-genome alignment coverage' (default: False),\n",
      " 'mapweight': 'Mapweight' (default: False),\n",
      " 'marmatus_homolog_associated_gene_name': 'Zig-zag eel gene name' (default: False),\n",
      " 'marmatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'marmatus_homolog_chrom_end': 'Zig-zag eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'marmatus_homolog_chrom_start': 'Zig-zag eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'marmatus_homolog_chromosome': 'Zig-zag eel chromosome/scaffold name' (default: False),\n",
      " 'marmatus_homolog_ensembl_gene': 'Zig-zag eel gene stable ID' (default: False),\n",
      " 'marmatus_homolog_ensembl_peptide': 'Zig-zag eel protein or transcript stable ID' (default: False),\n",
      " 'marmatus_homolog_goc_score': 'Zig-zag eel Gene-order conservation score' (default: False),\n",
      " 'marmatus_homolog_orthology_confidence': 'Zig-zag eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'marmatus_homolog_orthology_type': 'Zig-zag eel homology type' (default: False),\n",
      " 'marmatus_homolog_perc_id': '%id. target Zig-zag eel gene identical to query gene' (default: False),\n",
      " 'marmatus_homolog_perc_id_r1': '%id. query gene identical to target Zig-zag eel gene' (default: False),\n",
      " 'marmatus_homolog_subtype': 'Last common ancestor with Zig-zag eel' (default: False),\n",
      " 'mart_transcript_variation_som__dm_allele_string_2076': 'Consequence specific allele' (default: False),\n",
      " 'mart_transcript_variation_som__dm_distance_to_transcript_2076': 'Distance to transcript' (default: False),\n",
      " 'mauratus_homolog_associated_gene_name': 'Golden Hamster gene name' (default: False),\n",
      " 'mauratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mauratus_homolog_chrom_end': 'Golden Hamster chromosome/scaffold end (bp)' (default: False),\n",
      " 'mauratus_homolog_chrom_start': 'Golden Hamster chromosome/scaffold start (bp)' (default: False),\n",
      " 'mauratus_homolog_chromosome': 'Golden Hamster chromosome/scaffold name' (default: False),\n",
      " 'mauratus_homolog_ensembl_gene': 'Golden Hamster gene stable ID' (default: False),\n",
      " 'mauratus_homolog_ensembl_peptide': 'Golden Hamster protein or transcript stable ID' (default: False),\n",
      " 'mauratus_homolog_goc_score': 'Golden Hamster Gene-order conservation score' (default: False),\n",
      " 'mauratus_homolog_orthology_confidence': 'Golden Hamster orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mauratus_homolog_orthology_type': 'Golden Hamster homology type' (default: False),\n",
      " 'mauratus_homolog_perc_id': '%id. target Golden Hamster gene identical to query gene' (default: False),\n",
      " 'mauratus_homolog_perc_id_r1': '%id. query gene identical to target Golden Hamster gene' (default: False),\n",
      " 'mauratus_homolog_subtype': 'Last common ancestor with Golden Hamster' (default: False),\n",
      " 'mauratus_homolog_wga_coverage': 'Golden Hamster Whole-genome alignment coverage' (default: False),\n",
      " 'mcaroli_homolog_associated_gene_name': 'Ryukyu mouse gene name' (default: False),\n",
      " 'mcaroli_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mcaroli_homolog_chrom_end': 'Ryukyu mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mcaroli_homolog_chrom_start': 'Ryukyu mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mcaroli_homolog_chromosome': 'Ryukyu mouse chromosome/scaffold name' (default: False),\n",
      " 'mcaroli_homolog_ensembl_gene': 'Ryukyu mouse gene stable ID' (default: False),\n",
      " 'mcaroli_homolog_ensembl_peptide': 'Ryukyu mouse protein or transcript stable ID' (default: False),\n",
      " 'mcaroli_homolog_goc_score': 'Ryukyu mouse Gene-order conservation score' (default: False),\n",
      " 'mcaroli_homolog_orthology_confidence': 'Ryukyu mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mcaroli_homolog_orthology_type': 'Ryukyu mouse homology type' (default: False),\n",
      " 'mcaroli_homolog_perc_id': '%id. target Ryukyu mouse gene identical to query gene' (default: False),\n",
      " 'mcaroli_homolog_perc_id_r1': '%id. query gene identical to target Ryukyu mouse gene' (default: False),\n",
      " 'mcaroli_homolog_subtype': 'Last common ancestor with Ryukyu mouse' (default: False),\n",
      " 'mcaroli_homolog_wga_coverage': 'Ryukyu mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mdomestica_homolog_associated_gene_name': 'Opossum gene name' (default: False),\n",
      " 'mdomestica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mdomestica_homolog_chrom_end': 'Opossum chromosome/scaffold end (bp)' (default: False),\n",
      " 'mdomestica_homolog_chrom_start': 'Opossum chromosome/scaffold start (bp)' (default: False),\n",
      " 'mdomestica_homolog_chromosome': 'Opossum chromosome/scaffold name' (default: False),\n",
      " 'mdomestica_homolog_ensembl_gene': 'Opossum gene stable ID' (default: False),\n",
      " 'mdomestica_homolog_ensembl_peptide': 'Opossum protein or transcript stable ID' (default: False),\n",
      " 'mdomestica_homolog_goc_score': 'Opossum Gene-order conservation score' (default: False),\n",
      " 'mdomestica_homolog_orthology_confidence': 'Opossum orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mdomestica_homolog_orthology_type': 'Opossum homology type' (default: False),\n",
      " 'mdomestica_homolog_perc_id': '%id. target Opossum gene identical to query gene' (default: False),\n",
      " 'mdomestica_homolog_perc_id_r1': '%id. query gene identical to target Opossum gene' (default: False),\n",
      " 'mdomestica_homolog_subtype': 'Last common ancestor with Opossum' (default: False),\n",
      " 'mdomestica_homolog_wga_coverage': 'Opossum Whole-genome alignment coverage' (default: False),\n",
      " 'merops': 'MEROPS - the Peptidase Database ID' (default: False),\n",
      " 'mfascicularis_homolog_associated_gene_name': 'Crab-eating macaque gene name' (default: False),\n",
      " 'mfascicularis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mfascicularis_homolog_chrom_end': 'Crab-eating macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mfascicularis_homolog_chrom_start': 'Crab-eating macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mfascicularis_homolog_chromosome': 'Crab-eating macaque chromosome/scaffold name' (default: False),\n",
      " 'mfascicularis_homolog_ensembl_gene': 'Crab-eating macaque gene stable ID' (default: False),\n",
      " 'mfascicularis_homolog_ensembl_peptide': 'Crab-eating macaque protein or transcript stable ID' (default: False),\n",
      " 'mfascicularis_homolog_goc_score': 'Crab-eating macaque Gene-order conservation score' (default: False),\n",
      " 'mfascicularis_homolog_orthology_confidence': 'Crab-eating macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mfascicularis_homolog_orthology_type': 'Crab-eating macaque homology type' (default: False),\n",
      " 'mfascicularis_homolog_perc_id': '%id. target Crab-eating macaque gene identical to query gene' (default: False),\n",
      " 'mfascicularis_homolog_perc_id_r1': '%id. query gene identical to target Crab-eating macaque gene' (default: False),\n",
      " 'mfascicularis_homolog_subtype': 'Last common ancestor with Crab-eating macaque' (default: False),\n",
      " 'mfascicularis_homolog_wga_coverage': 'Crab-eating macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mgallopavo_homolog_associated_gene_name': 'Turkey gene name' (default: False),\n",
      " 'mgallopavo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mgallopavo_homolog_chrom_end': 'Turkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'mgallopavo_homolog_chrom_start': 'Turkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'mgallopavo_homolog_chromosome': 'Turkey chromosome/scaffold name' (default: False),\n",
      " 'mgallopavo_homolog_ensembl_gene': 'Turkey gene stable ID' (default: False),\n",
      " 'mgallopavo_homolog_ensembl_peptide': 'Turkey protein or transcript stable ID' (default: False),\n",
      " 'mgallopavo_homolog_goc_score': 'Turkey Gene-order conservation score' (default: False),\n",
      " 'mgallopavo_homolog_orthology_confidence': 'Turkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mgallopavo_homolog_orthology_type': 'Turkey homology type' (default: False),\n",
      " 'mgallopavo_homolog_perc_id': '%id. target Turkey gene identical to query gene' (default: False),\n",
      " 'mgallopavo_homolog_perc_id_r1': '%id. query gene identical to target Turkey gene' (default: False),\n",
      " 'mgallopavo_homolog_subtype': 'Last common ancestor with Turkey' (default: False),\n",
      " 'mim_gene_accession': 'MIM gene accession' (default: False),\n",
      " 'mim_gene_description': 'MIM gene description' (default: False),\n",
      " 'mim_morbid_accession': 'MIM morbid accession' (default: False),\n",
      " 'mim_morbid_description': 'MIM morbid description' (default: False),\n",
      " 'minor_allele': 'Minor allele' (default: False),\n",
      " 'minor_allele_count': 'Minor allele count' (default: False),\n",
      " 'minor_allele_freq': 'Minor allele frequency' (default: False),\n",
      " 'mirbase_accession': 'miRBase accession' (default: False),\n",
      " 'mirbase_id': 'miRBase ID' (default: False),\n",
      " 'mirbase_trans_name': 'miRBase transcript name ID' (default: False),\n",
      " 'mleucophaeus_homolog_associated_gene_name': 'Drill gene name' (default: False),\n",
      " 'mleucophaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mleucophaeus_homolog_chrom_end': 'Drill chromosome/scaffold end (bp)' (default: False),\n",
      " 'mleucophaeus_homolog_chrom_start': 'Drill chromosome/scaffold start (bp)' (default: False),\n",
      " 'mleucophaeus_homolog_chromosome': 'Drill chromosome/scaffold name' (default: False),\n",
      " 'mleucophaeus_homolog_ensembl_gene': 'Drill gene stable ID' (default: False),\n",
      " 'mleucophaeus_homolog_ensembl_peptide': 'Drill protein or transcript stable ID' (default: False),\n",
      " 'mleucophaeus_homolog_goc_score': 'Drill Gene-order conservation score' (default: False),\n",
      " 'mleucophaeus_homolog_orthology_confidence': 'Drill orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mleucophaeus_homolog_orthology_type': 'Drill homology type' (default: False),\n",
      " 'mleucophaeus_homolog_perc_id': '%id. target Drill gene identical to query gene' (default: False),\n",
      " 'mleucophaeus_homolog_perc_id_r1': '%id. query gene identical to target Drill gene' (default: False),\n",
      " 'mleucophaeus_homolog_subtype': 'Last common ancestor with Drill' (default: False),\n",
      " 'mleucophaeus_homolog_wga_coverage': 'Drill Whole-genome alignment coverage' (default: False),\n",
      " 'mlucifugus_homolog_associated_gene_name': 'Microbat gene name' (default: False),\n",
      " 'mlucifugus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mlucifugus_homolog_chrom_end': 'Microbat chromosome/scaffold end (bp)' (default: False),\n",
      " 'mlucifugus_homolog_chrom_start': 'Microbat chromosome/scaffold start (bp)' (default: False),\n",
      " 'mlucifugus_homolog_chromosome': 'Microbat chromosome/scaffold name' (default: False),\n",
      " 'mlucifugus_homolog_ensembl_gene': 'Microbat gene stable ID' (default: False),\n",
      " 'mlucifugus_homolog_ensembl_peptide': 'Microbat protein or transcript stable ID' (default: False),\n",
      " 'mlucifugus_homolog_goc_score': 'Microbat Gene-order conservation score' (default: False),\n",
      " 'mlucifugus_homolog_orthology_confidence': 'Microbat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mlucifugus_homolog_orthology_type': 'Microbat homology type' (default: False),\n",
      " 'mlucifugus_homolog_perc_id': '%id. target Microbat gene identical to query gene' (default: False),\n",
      " 'mlucifugus_homolog_perc_id_r1': '%id. query gene identical to target Microbat gene' (default: False),\n",
      " 'mlucifugus_homolog_subtype': 'Last common ancestor with Microbat' (default: False),\n",
      " 'mlucifugus_homolog_wga_coverage': 'Microbat Whole-genome alignment coverage' (default: False),\n",
      " 'mmmarmota_homolog_associated_gene_name': 'Alpine marmot gene name' (default: False),\n",
      " 'mmmarmota_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmmarmota_homolog_chrom_end': 'Alpine marmot chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmmarmota_homolog_chrom_start': 'Alpine marmot chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmmarmota_homolog_chromosome': 'Alpine marmot chromosome/scaffold name' (default: False),\n",
      " 'mmmarmota_homolog_ensembl_gene': 'Alpine marmot gene stable ID' (default: False),\n",
      " 'mmmarmota_homolog_ensembl_peptide': 'Alpine marmot protein or transcript stable ID' (default: False),\n",
      " 'mmmarmota_homolog_goc_score': 'Alpine marmot Gene-order conservation score' (default: False),\n",
      " 'mmmarmota_homolog_orthology_confidence': 'Alpine marmot orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmmarmota_homolog_orthology_type': 'Alpine marmot homology type' (default: False),\n",
      " 'mmmarmota_homolog_perc_id': '%id. target Alpine marmot gene identical to query gene' (default: False),\n",
      " 'mmmarmota_homolog_perc_id_r1': '%id. query gene identical to target Alpine marmot gene' (default: False),\n",
      " 'mmmarmota_homolog_subtype': 'Last common ancestor with Alpine marmot' (default: False),\n",
      " 'mmmarmota_homolog_wga_coverage': 'Alpine marmot Whole-genome alignment coverage' (default: False),\n",
      " 'mmonoceros_homolog_associated_gene_name': 'Narwhal gene name' (default: False),\n",
      " 'mmonoceros_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmonoceros_homolog_chrom_end': 'Narwhal chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmonoceros_homolog_chrom_start': 'Narwhal chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmonoceros_homolog_chromosome': 'Narwhal chromosome/scaffold name' (default: False),\n",
      " 'mmonoceros_homolog_ensembl_gene': 'Narwhal gene stable ID' (default: False),\n",
      " 'mmonoceros_homolog_ensembl_peptide': 'Narwhal protein or transcript stable ID' (default: False),\n",
      " 'mmonoceros_homolog_goc_score': 'Narwhal Gene-order conservation score' (default: False),\n",
      " 'mmonoceros_homolog_orthology_confidence': 'Narwhal orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmonoceros_homolog_orthology_type': 'Narwhal homology type' (default: False),\n",
      " 'mmonoceros_homolog_perc_id': '%id. target Narwhal gene identical to query gene' (default: False),\n",
      " 'mmonoceros_homolog_perc_id_r1': '%id. query gene identical to target Narwhal gene' (default: False),\n",
      " 'mmonoceros_homolog_subtype': 'Last common ancestor with Narwhal' (default: False),\n",
      " 'mmonoceros_homolog_wga_coverage': 'Narwhal Whole-genome alignment coverage' (default: False),\n",
      " 'mmoschiferus_homolog_associated_gene_name': 'Siberian musk deer gene name' (default: False),\n",
      " 'mmoschiferus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmoschiferus_homolog_chrom_end': 'Siberian musk deer chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmoschiferus_homolog_chrom_start': 'Siberian musk deer chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmoschiferus_homolog_chromosome': 'Siberian musk deer chromosome/scaffold name' (default: False),\n",
      " 'mmoschiferus_homolog_ensembl_gene': 'Siberian musk deer gene stable ID' (default: False),\n",
      " 'mmoschiferus_homolog_ensembl_peptide': 'Siberian musk deer protein or transcript stable ID' (default: False),\n",
      " 'mmoschiferus_homolog_goc_score': 'Siberian musk deer Gene-order conservation score' (default: False),\n",
      " 'mmoschiferus_homolog_orthology_confidence': 'Siberian musk deer orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmoschiferus_homolog_orthology_type': 'Siberian musk deer homology type' (default: False),\n",
      " 'mmoschiferus_homolog_perc_id': '%id. target Siberian musk deer gene identical to query gene' (default: False),\n",
      " 'mmoschiferus_homolog_perc_id_r1': '%id. query gene identical to target Siberian musk deer gene' (default: False),\n",
      " 'mmoschiferus_homolog_subtype': 'Last common ancestor with Siberian musk deer' (default: False),\n",
      " 'mmoschiferus_homolog_wga_coverage': 'Siberian musk deer Whole-genome alignment coverage' (default: False),\n",
      " 'mmulatta_homolog_associated_gene_name': 'Macaque gene name' (default: False),\n",
      " 'mmulatta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmulatta_homolog_chrom_end': 'Macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmulatta_homolog_chrom_start': 'Macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmulatta_homolog_chromosome': 'Macaque chromosome/scaffold name' (default: False),\n",
      " 'mmulatta_homolog_ensembl_gene': 'Macaque gene stable ID' (default: False),\n",
      " 'mmulatta_homolog_ensembl_peptide': 'Macaque protein or transcript stable ID' (default: False),\n",
      " 'mmulatta_homolog_goc_score': 'Macaque Gene-order conservation score' (default: False),\n",
      " 'mmulatta_homolog_orthology_confidence': 'Macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmulatta_homolog_orthology_type': 'Macaque homology type' (default: False),\n",
      " 'mmulatta_homolog_perc_id': '%id. target Macaque gene identical to query gene' (default: False),\n",
      " 'mmulatta_homolog_perc_id_r1': '%id. query gene identical to target Macaque gene' (default: False),\n",
      " 'mmulatta_homolog_subtype': 'Last common ancestor with Macaque' (default: False),\n",
      " 'mmulatta_homolog_wga_coverage': 'Macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mmurdjan_homolog_associated_gene_name': 'Pinecone soldierfish gene name' (default: False),\n",
      " 'mmurdjan_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmurdjan_homolog_chrom_end': 'Pinecone soldierfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmurdjan_homolog_chrom_start': 'Pinecone soldierfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmurdjan_homolog_chromosome': 'Pinecone soldierfish chromosome/scaffold name' (default: False),\n",
      " 'mmurdjan_homolog_ensembl_gene': 'Pinecone soldierfish gene stable ID' (default: False),\n",
      " 'mmurdjan_homolog_ensembl_peptide': 'Pinecone soldierfish protein or transcript stable ID' (default: False),\n",
      " 'mmurdjan_homolog_goc_score': 'Pinecone soldierfish Gene-order conservation score' (default: False),\n",
      " 'mmurdjan_homolog_orthology_confidence': 'Pinecone soldierfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmurdjan_homolog_orthology_type': 'Pinecone soldierfish homology type' (default: False),\n",
      " 'mmurdjan_homolog_perc_id': '%id. target Pinecone soldierfish gene identical to query gene' (default: False),\n",
      " 'mmurdjan_homolog_perc_id_r1': '%id. query gene identical to target Pinecone soldierfish gene' (default: False),\n",
      " 'mmurdjan_homolog_subtype': 'Last common ancestor with Pinecone soldierfish' (default: False),\n",
      " 'mmurdjan_homolog_wga_coverage': 'Pinecone soldierfish Whole-genome alignment coverage' (default: False),\n",
      " 'mmurinus_homolog_associated_gene_name': 'Mouse Lemur gene name' (default: False),\n",
      " 'mmurinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmurinus_homolog_chrom_end': 'Mouse Lemur chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmurinus_homolog_chrom_start': 'Mouse Lemur chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmurinus_homolog_chromosome': 'Mouse Lemur chromosome/scaffold name' (default: False),\n",
      " 'mmurinus_homolog_ensembl_gene': 'Mouse Lemur gene stable ID' (default: False),\n",
      " 'mmurinus_homolog_ensembl_peptide': 'Mouse Lemur protein or transcript stable ID' (default: False),\n",
      " 'mmurinus_homolog_goc_score': 'Mouse Lemur Gene-order conservation score' (default: False),\n",
      " 'mmurinus_homolog_orthology_confidence': 'Mouse Lemur orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmurinus_homolog_orthology_type': 'Mouse Lemur homology type' (default: False),\n",
      " 'mmurinus_homolog_perc_id': '%id. target Mouse Lemur gene identical to query gene' (default: False),\n",
      " 'mmurinus_homolog_perc_id_r1': '%id. query gene identical to target Mouse Lemur gene' (default: False),\n",
      " 'mmurinus_homolog_subtype': 'Last common ancestor with Mouse Lemur' (default: False),\n",
      " 'mmurinus_homolog_wga_coverage': 'Mouse Lemur Whole-genome alignment coverage' (default: False),\n",
      " 'mmusculus_homolog_associated_gene_name': 'Mouse gene name' (default: False),\n",
      " 'mmusculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmusculus_homolog_chrom_end': 'Mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmusculus_homolog_chrom_start': 'Mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmusculus_homolog_chromosome': 'Mouse chromosome/scaffold name' (default: False),\n",
      " 'mmusculus_homolog_ensembl_gene': 'Mouse gene stable ID' (default: False),\n",
      " 'mmusculus_homolog_ensembl_peptide': 'Mouse protein or transcript stable ID' (default: False),\n",
      " 'mmusculus_homolog_goc_score': 'Mouse Gene-order conservation score' (default: False),\n",
      " 'mmusculus_homolog_orthology_confidence': 'Mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmusculus_homolog_orthology_type': 'Mouse homology type' (default: False),\n",
      " 'mmusculus_homolog_perc_id': '%id. target Mouse gene identical to query gene' (default: False),\n",
      " 'mmusculus_homolog_perc_id_r1': '%id. query gene identical to target Mouse gene' (default: False),\n",
      " 'mmusculus_homolog_subtype': 'Last common ancestor with Mouse' (default: False),\n",
      " 'mmusculus_homolog_wga_coverage': 'Mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mnemestrina_homolog_associated_gene_name': 'Pig-tailed macaque gene name' (default: False),\n",
      " 'mnemestrina_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mnemestrina_homolog_chrom_end': 'Pig-tailed macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mnemestrina_homolog_chrom_start': 'Pig-tailed macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mnemestrina_homolog_chromosome': 'Pig-tailed macaque chromosome/scaffold name' (default: False),\n",
      " 'mnemestrina_homolog_ensembl_gene': 'Pig-tailed macaque gene stable ID' (default: False),\n",
      " 'mnemestrina_homolog_ensembl_peptide': 'Pig-tailed macaque protein or transcript stable ID' (default: False),\n",
      " 'mnemestrina_homolog_goc_score': 'Pig-tailed macaque Gene-order conservation score' (default: False),\n",
      " 'mnemestrina_homolog_orthology_confidence': 'Pig-tailed macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mnemestrina_homolog_orthology_type': 'Pig-tailed macaque homology type' (default: False),\n",
      " 'mnemestrina_homolog_perc_id': '%id. target Pig-tailed macaque gene identical to query gene' (default: False),\n",
      " 'mnemestrina_homolog_perc_id_r1': '%id. query gene identical to target Pig-tailed macaque gene' (default: False),\n",
      " 'mnemestrina_homolog_subtype': 'Last common ancestor with Pig-tailed macaque' (default: False),\n",
      " 'mnemestrina_homolog_wga_coverage': 'Pig-tailed macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mobidblite': 'MobiDB lite' (default: False),\n",
      " 'mobidblite_end': 'MobiDB lite end' (default: False),\n",
      " 'mobidblite_start': 'MobiDB lite start' (default: False),\n",
      " 'mochrogaster_homolog_associated_gene_name': 'Prairie vole gene name' (default: False),\n",
      " 'mochrogaster_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mochrogaster_homolog_chrom_end': 'Prairie vole chromosome/scaffold end (bp)' (default: False),\n",
      " 'mochrogaster_homolog_chrom_start': 'Prairie vole chromosome/scaffold start (bp)' (default: False),\n",
      " 'mochrogaster_homolog_chromosome': 'Prairie vole chromosome/scaffold name' (default: False),\n",
      " 'mochrogaster_homolog_ensembl_gene': 'Prairie vole gene stable ID' (default: False),\n",
      " 'mochrogaster_homolog_ensembl_peptide': 'Prairie vole protein or transcript stable ID' (default: False),\n",
      " 'mochrogaster_homolog_goc_score': 'Prairie vole Gene-order conservation score' (default: False),\n",
      " 'mochrogaster_homolog_orthology_confidence': 'Prairie vole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mochrogaster_homolog_orthology_type': 'Prairie vole homology type' (default: False),\n",
      " 'mochrogaster_homolog_perc_id': '%id. target Prairie vole gene identical to query gene' (default: False),\n",
      " 'mochrogaster_homolog_perc_id_r1': '%id. query gene identical to target Prairie vole gene' (default: False),\n",
      " 'mochrogaster_homolog_subtype': 'Last common ancestor with Prairie vole' (default: False),\n",
      " 'mochrogaster_homolog_wga_coverage': 'Prairie vole Whole-genome alignment coverage' (default: False),\n",
      " 'mpahari_homolog_associated_gene_name': 'Shrew mouse gene name' (default: False),\n",
      " 'mpahari_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mpahari_homolog_chrom_end': 'Shrew mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mpahari_homolog_chrom_start': 'Shrew mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mpahari_homolog_chromosome': 'Shrew mouse chromosome/scaffold name' (default: False),\n",
      " 'mpahari_homolog_ensembl_gene': 'Shrew mouse gene stable ID' (default: False),\n",
      " 'mpahari_homolog_ensembl_peptide': 'Shrew mouse protein or transcript stable ID' (default: False),\n",
      " 'mpahari_homolog_goc_score': 'Shrew mouse Gene-order conservation score' (default: False),\n",
      " 'mpahari_homolog_orthology_confidence': 'Shrew mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mpahari_homolog_orthology_type': 'Shrew mouse homology type' (default: False),\n",
      " 'mpahari_homolog_perc_id': '%id. target Shrew mouse gene identical to query gene' (default: False),\n",
      " 'mpahari_homolog_perc_id_r1': '%id. query gene identical to target Shrew mouse gene' (default: False),\n",
      " 'mpahari_homolog_subtype': 'Last common ancestor with Shrew mouse' (default: False),\n",
      " 'mpahari_homolog_wga_coverage': 'Shrew mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mpfuro_homolog_associated_gene_name': 'Ferret gene name' (default: False),\n",
      " 'mpfuro_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mpfuro_homolog_chrom_end': 'Ferret chromosome/scaffold end (bp)' (default: False),\n",
      " 'mpfuro_homolog_chrom_start': 'Ferret chromosome/scaffold start (bp)' (default: False),\n",
      " 'mpfuro_homolog_chromosome': 'Ferret chromosome/scaffold name' (default: False),\n",
      " 'mpfuro_homolog_ensembl_gene': 'Ferret gene stable ID' (default: False),\n",
      " 'mpfuro_homolog_ensembl_peptide': 'Ferret protein or transcript stable ID' (default: False),\n",
      " 'mpfuro_homolog_goc_score': 'Ferret Gene-order conservation score' (default: False),\n",
      " 'mpfuro_homolog_orthology_confidence': 'Ferret orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mpfuro_homolog_orthology_type': 'Ferret homology type' (default: False),\n",
      " 'mpfuro_homolog_perc_id': '%id. target Ferret gene identical to query gene' (default: False),\n",
      " 'mpfuro_homolog_perc_id_r1': '%id. query gene identical to target Ferret gene' (default: False),\n",
      " 'mpfuro_homolog_subtype': 'Last common ancestor with Ferret' (default: False),\n",
      " 'mpfuro_homolog_wga_coverage': 'Ferret Whole-genome alignment coverage' (default: False),\n",
      " 'mspicilegus_homolog_associated_gene_name': 'Steppe mouse gene name' (default: False),\n",
      " 'mspicilegus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mspicilegus_homolog_chrom_end': 'Steppe mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mspicilegus_homolog_chrom_start': 'Steppe mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mspicilegus_homolog_chromosome': 'Steppe mouse chromosome/scaffold name' (default: False),\n",
      " 'mspicilegus_homolog_ensembl_gene': 'Steppe mouse gene stable ID' (default: False),\n",
      " 'mspicilegus_homolog_ensembl_peptide': 'Steppe mouse protein or transcript stable ID' (default: False),\n",
      " 'mspicilegus_homolog_goc_score': 'Steppe mouse Gene-order conservation score' (default: False),\n",
      " 'mspicilegus_homolog_orthology_confidence': 'Steppe mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mspicilegus_homolog_orthology_type': 'Steppe mouse homology type' (default: False),\n",
      " 'mspicilegus_homolog_perc_id': '%id. target Steppe mouse gene identical to query gene' (default: False),\n",
      " 'mspicilegus_homolog_perc_id_r1': '%id. query gene identical to target Steppe mouse gene' (default: False),\n",
      " 'mspicilegus_homolog_subtype': 'Last common ancestor with Steppe mouse' (default: False),\n",
      " 'mspicilegus_homolog_wga_coverage': 'Steppe mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mspretus_homolog_associated_gene_name': 'Algerian mouse gene name' (default: False),\n",
      " 'mspretus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mspretus_homolog_chrom_end': 'Algerian mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mspretus_homolog_chrom_start': 'Algerian mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mspretus_homolog_chromosome': 'Algerian mouse chromosome/scaffold name' (default: False),\n",
      " 'mspretus_homolog_ensembl_gene': 'Algerian mouse gene stable ID' (default: False),\n",
      " 'mspretus_homolog_ensembl_peptide': 'Algerian mouse protein or transcript stable ID' (default: False),\n",
      " 'mspretus_homolog_goc_score': 'Algerian mouse Gene-order conservation score' (default: False),\n",
      " 'mspretus_homolog_orthology_confidence': 'Algerian mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mspretus_homolog_orthology_type': 'Algerian mouse homology type' (default: False),\n",
      " 'mspretus_homolog_perc_id': '%id. target Algerian mouse gene identical to query gene' (default: False),\n",
      " 'mspretus_homolog_perc_id_r1': '%id. query gene identical to target Algerian mouse gene' (default: False),\n",
      " 'mspretus_homolog_subtype': 'Last common ancestor with Algerian mouse' (default: False),\n",
      " 'mspretus_homolog_wga_coverage': 'Algerian mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mzebra_homolog_associated_gene_name': 'Zebra mbuna gene name' (default: False),\n",
      " 'mzebra_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mzebra_homolog_chrom_end': 'Zebra mbuna chromosome/scaffold end (bp)' (default: False),\n",
      " 'mzebra_homolog_chrom_start': 'Zebra mbuna chromosome/scaffold start (bp)' (default: False),\n",
      " 'mzebra_homolog_chromosome': 'Zebra mbuna chromosome/scaffold name' (default: False),\n",
      " 'mzebra_homolog_ensembl_gene': 'Zebra mbuna gene stable ID' (default: False),\n",
      " 'mzebra_homolog_ensembl_peptide': 'Zebra mbuna protein or transcript stable ID' (default: False),\n",
      " 'mzebra_homolog_goc_score': 'Zebra mbuna Gene-order conservation score' (default: False),\n",
      " 'mzebra_homolog_orthology_confidence': 'Zebra mbuna orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mzebra_homolog_orthology_type': 'Zebra mbuna homology type' (default: False),\n",
      " 'mzebra_homolog_perc_id': '%id. target Zebra mbuna gene identical to query gene' (default: False),\n",
      " 'mzebra_homolog_perc_id_r1': '%id. query gene identical to target Zebra mbuna gene' (default: False),\n",
      " 'mzebra_homolog_subtype': 'Last common ancestor with Zebra mbuna' (default: False),\n",
      " 'mzebra_homolog_wga_coverage': 'Zebra mbuna Whole-genome alignment coverage' (default: False),\n",
      " 'name_1006': 'GO term name' (default: False),\n",
      " 'namespace_1003': 'GO domain' (default: False),\n",
      " 'nbrichardi_homolog_associated_gene_name': 'Lyretail cichlid gene name' (default: False),\n",
      " 'nbrichardi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nbrichardi_homolog_chrom_end': 'Lyretail cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'nbrichardi_homolog_chrom_start': 'Lyretail cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'nbrichardi_homolog_chromosome': 'Lyretail cichlid chromosome/scaffold name' (default: False),\n",
      " 'nbrichardi_homolog_ensembl_gene': 'Lyretail cichlid gene stable ID' (default: False),\n",
      " 'nbrichardi_homolog_ensembl_peptide': 'Lyretail cichlid protein or transcript stable ID' (default: False),\n",
      " 'nbrichardi_homolog_goc_score': 'Lyretail cichlid Gene-order conservation score' (default: False),\n",
      " 'nbrichardi_homolog_orthology_confidence': 'Lyretail cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nbrichardi_homolog_orthology_type': 'Lyretail cichlid homology type' (default: False),\n",
      " 'nbrichardi_homolog_perc_id': '%id. target Lyretail cichlid gene identical to query gene' (default: False),\n",
      " 'nbrichardi_homolog_perc_id_r1': '%id. query gene identical to target Lyretail cichlid gene' (default: False),\n",
      " 'nbrichardi_homolog_subtype': 'Last common ancestor with Lyretail cichlid' (default: False),\n",
      " 'nbrichardi_homolog_wga_coverage': 'Lyretail cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'ncoils': 'Coiled-coils (Ncoils)' (default: False),\n",
      " 'ncoils_end': 'Coiled-coils (Ncoils) end' (default: False),\n",
      " 'ncoils_start': 'Coiled-coils (Ncoils) start' (default: False),\n",
      " 'neugenii_homolog_associated_gene_name': 'Wallaby gene name' (default: False),\n",
      " 'neugenii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'neugenii_homolog_chrom_end': 'Wallaby chromosome/scaffold end (bp)' (default: False),\n",
      " 'neugenii_homolog_chrom_start': 'Wallaby chromosome/scaffold start (bp)' (default: False),\n",
      " 'neugenii_homolog_chromosome': 'Wallaby chromosome/scaffold name' (default: False),\n",
      " 'neugenii_homolog_ensembl_gene': 'Wallaby gene stable ID' (default: False),\n",
      " 'neugenii_homolog_ensembl_peptide': 'Wallaby protein or transcript stable ID' (default: False),\n",
      " 'neugenii_homolog_goc_score': 'Wallaby Gene-order conservation score' (default: False),\n",
      " 'neugenii_homolog_orthology_confidence': 'Wallaby orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'neugenii_homolog_orthology_type': 'Wallaby homology type' (default: False),\n",
      " 'neugenii_homolog_perc_id': '%id. target Wallaby gene identical to query gene' (default: False),\n",
      " 'neugenii_homolog_perc_id_r1': '%id. query gene identical to target Wallaby gene' (default: False),\n",
      " 'neugenii_homolog_subtype': 'Last common ancestor with Wallaby' (default: False),\n",
      " 'neugenii_homolog_wga_coverage': 'Wallaby Whole-genome alignment coverage' (default: False),\n",
      " 'nfurzeri_homolog_associated_gene_name': 'Turquoise killifish gene name' (default: False),\n",
      " 'nfurzeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nfurzeri_homolog_chrom_end': 'Turquoise killifish chromosome/scaffold end (bp)' (default: False),\n",
      " 'nfurzeri_homolog_chrom_start': 'Turquoise killifish chromosome/scaffold start (bp)' (default: False),\n",
      " 'nfurzeri_homolog_chromosome': 'Turquoise killifish chromosome/scaffold name' (default: False),\n",
      " 'nfurzeri_homolog_ensembl_gene': 'Turquoise killifish gene stable ID' (default: False),\n",
      " 'nfurzeri_homolog_ensembl_peptide': 'Turquoise killifish protein or transcript stable ID' (default: False),\n",
      " 'nfurzeri_homolog_goc_score': 'Turquoise killifish Gene-order conservation score' (default: False),\n",
      " 'nfurzeri_homolog_orthology_confidence': 'Turquoise killifish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nfurzeri_homolog_orthology_type': 'Turquoise killifish homology type' (default: False),\n",
      " 'nfurzeri_homolog_perc_id': '%id. target Turquoise killifish gene identical to query gene' (default: False),\n",
      " 'nfurzeri_homolog_perc_id_r1': '%id. query gene identical to target Turquoise killifish gene' (default: False),\n",
      " 'nfurzeri_homolog_subtype': 'Last common ancestor with Turquoise killifish' (default: False),\n",
      " 'nfurzeri_homolog_wga_coverage': 'Turquoise killifish Whole-genome alignment coverage' (default: False),\n",
      " 'ngalili_homolog_associated_gene_name': 'Upper Galilee mountains blind mole rat gene name' (default: False),\n",
      " 'ngalili_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ngalili_homolog_chrom_end': 'Upper Galilee mountains blind mole rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'ngalili_homolog_chrom_start': 'Upper Galilee mountains blind mole rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'ngalili_homolog_chromosome': 'Upper Galilee mountains blind mole rat chromosome/scaffold name' (default: False),\n",
      " 'ngalili_homolog_ensembl_gene': 'Upper Galilee mountains blind mole rat gene stable ID' (default: False),\n",
      " 'ngalili_homolog_ensembl_peptide': 'Upper Galilee mountains blind mole rat protein or transcript stable ID' (default: False),\n",
      " 'ngalili_homolog_goc_score': 'Upper Galilee mountains blind mole rat Gene-order conservation score' (default: False),\n",
      " 'ngalili_homolog_orthology_confidence': 'Upper Galilee mountains blind mole rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ngalili_homolog_orthology_type': 'Upper Galilee mountains blind mole rat homology type' (default: False),\n",
      " 'ngalili_homolog_perc_id': '%id. target Upper Galilee mountains blind mole rat gene identical to query gene' (default: False),\n",
      " 'ngalili_homolog_perc_id_r1': '%id. query gene identical to target Upper Galilee mountains blind mole rat gene' (default: False),\n",
      " 'ngalili_homolog_subtype': 'Last common ancestor with Upper Galilee mountains blind mole rat' (default: False),\n",
      " 'ngalili_homolog_wga_coverage': 'Upper Galilee mountains blind mole rat Whole-genome alignment coverage' (default: False),\n",
      " 'nleucogenys_homolog_associated_gene_name': 'Gibbon gene name' (default: False),\n",
      " 'nleucogenys_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nleucogenys_homolog_chrom_end': 'Gibbon chromosome/scaffold end (bp)' (default: False),\n",
      " 'nleucogenys_homolog_chrom_start': 'Gibbon chromosome/scaffold start (bp)' (default: False),\n",
      " 'nleucogenys_homolog_chromosome': 'Gibbon chromosome/scaffold name' (default: False),\n",
      " 'nleucogenys_homolog_ensembl_gene': 'Gibbon gene stable ID' (default: False),\n",
      " 'nleucogenys_homolog_ensembl_peptide': 'Gibbon protein or transcript stable ID' (default: False),\n",
      " 'nleucogenys_homolog_goc_score': 'Gibbon Gene-order conservation score' (default: False),\n",
      " 'nleucogenys_homolog_orthology_confidence': 'Gibbon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nleucogenys_homolog_orthology_type': 'Gibbon homology type' (default: False),\n",
      " 'nleucogenys_homolog_perc_id': '%id. target Gibbon gene identical to query gene' (default: False),\n",
      " 'nleucogenys_homolog_perc_id_r1': '%id. query gene identical to target Gibbon gene' (default: False),\n",
      " 'nleucogenys_homolog_subtype': 'Last common ancestor with Gibbon' (default: False),\n",
      " 'nleucogenys_homolog_wga_coverage': 'Gibbon Whole-genome alignment coverage' (default: False),\n",
      " 'nnaja_homolog_associated_gene_name': 'Indian cobra gene name' (default: False),\n",
      " 'nnaja_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nnaja_homolog_chrom_end': 'Indian cobra chromosome/scaffold end (bp)' (default: False),\n",
      " 'nnaja_homolog_chrom_start': 'Indian cobra chromosome/scaffold start (bp)' (default: False),\n",
      " 'nnaja_homolog_chromosome': 'Indian cobra chromosome/scaffold name' (default: False),\n",
      " 'nnaja_homolog_ensembl_gene': 'Indian cobra gene stable ID' (default: False),\n",
      " 'nnaja_homolog_ensembl_peptide': 'Indian cobra protein or transcript stable ID' (default: False),\n",
      " 'nnaja_homolog_goc_score': 'Indian cobra Gene-order conservation score' (default: False),\n",
      " 'nnaja_homolog_orthology_confidence': 'Indian cobra orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nnaja_homolog_orthology_type': 'Indian cobra homology type' (default: False),\n",
      " 'nnaja_homolog_perc_id': '%id. target Indian cobra gene identical to query gene' (default: False),\n",
      " 'nnaja_homolog_perc_id_r1': '%id. query gene identical to target Indian cobra gene' (default: False),\n",
      " 'nnaja_homolog_subtype': 'Last common ancestor with Indian cobra' (default: False),\n",
      " 'nnaja_homolog_wga_coverage': 'Indian cobra Whole-genome alignment coverage' (default: False),\n",
      " 'nscutatus_homolog_associated_gene_name': 'Mainland tiger snake gene name' (default: False),\n",
      " 'nscutatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nscutatus_homolog_chrom_end': 'Mainland tiger snake chromosome/scaffold end (bp)' (default: False),\n",
      " 'nscutatus_homolog_chrom_start': 'Mainland tiger snake chromosome/scaffold start (bp)' (default: False),\n",
      " 'nscutatus_homolog_chromosome': 'Mainland tiger snake chromosome/scaffold name' (default: False),\n",
      " 'nscutatus_homolog_ensembl_gene': 'Mainland tiger snake gene stable ID' (default: False),\n",
      " 'nscutatus_homolog_ensembl_peptide': 'Mainland tiger snake protein or transcript stable ID' (default: False),\n",
      " 'nscutatus_homolog_goc_score': 'Mainland tiger snake Gene-order conservation score' (default: False),\n",
      " 'nscutatus_homolog_orthology_confidence': 'Mainland tiger snake orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nscutatus_homolog_orthology_type': 'Mainland tiger snake homology type' (default: False),\n",
      " 'nscutatus_homolog_perc_id': '%id. target Mainland tiger snake gene identical to query gene' (default: False),\n",
      " 'nscutatus_homolog_perc_id_r1': '%id. query gene identical to target Mainland tiger snake gene' (default: False),\n",
      " 'nscutatus_homolog_subtype': 'Last common ancestor with Mainland tiger snake' (default: False),\n",
      " 'nscutatus_homolog_wga_coverage': 'Mainland tiger snake Whole-genome alignment coverage' (default: False),\n",
      " 'nvison_homolog_associated_gene_name': 'American mink gene name' (default: False),\n",
      " 'nvison_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nvison_homolog_chrom_end': 'American mink chromosome/scaffold end (bp)' (default: False),\n",
      " 'nvison_homolog_chrom_start': 'American mink chromosome/scaffold start (bp)' (default: False),\n",
      " 'nvison_homolog_chromosome': 'American mink chromosome/scaffold name' (default: False),\n",
      " 'nvison_homolog_ensembl_gene': 'American mink gene stable ID' (default: False),\n",
      " 'nvison_homolog_ensembl_peptide': 'American mink protein or transcript stable ID' (default: False),\n",
      " 'nvison_homolog_goc_score': 'American mink Gene-order conservation score' (default: False),\n",
      " 'nvison_homolog_orthology_confidence': 'American mink orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nvison_homolog_orthology_type': 'American mink homology type' (default: False),\n",
      " 'nvison_homolog_perc_id': '%id. target American mink gene identical to query gene' (default: False),\n",
      " 'nvison_homolog_perc_id_r1': '%id. query gene identical to target American mink gene' (default: False),\n",
      " 'nvison_homolog_subtype': 'Last common ancestor with American mink' (default: False),\n",
      " 'nvison_homolog_wga_coverage': 'American mink Whole-genome alignment coverage' (default: False),\n",
      " 'oanatinus_homolog_associated_gene_name': 'Platypus gene name' (default: False),\n",
      " 'oanatinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oanatinus_homolog_chrom_end': 'Platypus chromosome/scaffold end (bp)' (default: False),\n",
      " 'oanatinus_homolog_chrom_start': 'Platypus chromosome/scaffold start (bp)' (default: False),\n",
      " 'oanatinus_homolog_chromosome': 'Platypus chromosome/scaffold name' (default: False),\n",
      " 'oanatinus_homolog_ensembl_gene': 'Platypus gene stable ID' (default: False),\n",
      " 'oanatinus_homolog_ensembl_peptide': 'Platypus protein or transcript stable ID' (default: False),\n",
      " 'oanatinus_homolog_goc_score': 'Platypus Gene-order conservation score' (default: False),\n",
      " 'oanatinus_homolog_orthology_confidence': 'Platypus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oanatinus_homolog_orthology_type': 'Platypus homology type' (default: False),\n",
      " 'oanatinus_homolog_perc_id': '%id. target Platypus gene identical to query gene' (default: False),\n",
      " 'oanatinus_homolog_perc_id_r1': '%id. query gene identical to target Platypus gene' (default: False),\n",
      " 'oanatinus_homolog_subtype': 'Last common ancestor with Platypus' (default: False),\n",
      " 'oanatinus_homolog_wga_coverage': 'Platypus Whole-genome alignment coverage' (default: False),\n",
      " 'oarambouillet_homolog_associated_gene_name': 'Sheep gene name' (default: False),\n",
      " 'oarambouillet_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oarambouillet_homolog_chrom_end': 'Sheep chromosome/scaffold end (bp)' (default: False),\n",
      " 'oarambouillet_homolog_chrom_start': 'Sheep chromosome/scaffold start (bp)' (default: False),\n",
      " 'oarambouillet_homolog_chromosome': 'Sheep chromosome/scaffold name' (default: False),\n",
      " 'oarambouillet_homolog_ensembl_gene': 'Sheep gene stable ID' (default: False),\n",
      " 'oarambouillet_homolog_ensembl_peptide': 'Sheep protein or transcript stable ID' (default: False),\n",
      " 'oarambouillet_homolog_goc_score': 'Sheep Gene-order conservation score' (default: False),\n",
      " 'oarambouillet_homolog_orthology_confidence': 'Sheep orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oarambouillet_homolog_orthology_type': 'Sheep homology type' (default: False),\n",
      " 'oarambouillet_homolog_perc_id': '%id. target Sheep gene identical to query gene' (default: False),\n",
      " 'oarambouillet_homolog_perc_id_r1': '%id. query gene identical to target Sheep gene' (default: False),\n",
      " 'oarambouillet_homolog_subtype': 'Last common ancestor with Sheep' (default: False),\n",
      " 'oarambouillet_homolog_wga_coverage': 'Sheep Whole-genome alignment coverage' (default: False),\n",
      " 'ocuniculus_homolog_associated_gene_name': 'Rabbit gene name' (default: False),\n",
      " 'ocuniculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ocuniculus_homolog_chrom_end': 'Rabbit chromosome/scaffold end (bp)' (default: False),\n",
      " 'ocuniculus_homolog_chrom_start': 'Rabbit chromosome/scaffold start (bp)' (default: False),\n",
      " 'ocuniculus_homolog_chromosome': 'Rabbit chromosome/scaffold name' (default: False),\n",
      " 'ocuniculus_homolog_ensembl_gene': 'Rabbit gene stable ID' (default: False),\n",
      " 'ocuniculus_homolog_ensembl_peptide': 'Rabbit protein or transcript stable ID' (default: False),\n",
      " 'ocuniculus_homolog_goc_score': 'Rabbit Gene-order conservation score' (default: False),\n",
      " 'ocuniculus_homolog_orthology_confidence': 'Rabbit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ocuniculus_homolog_orthology_type': 'Rabbit homology type' (default: False),\n",
      " 'ocuniculus_homolog_perc_id': '%id. target Rabbit gene identical to query gene' (default: False),\n",
      " 'ocuniculus_homolog_perc_id_r1': '%id. query gene identical to target Rabbit gene' (default: False),\n",
      " 'ocuniculus_homolog_subtype': 'Last common ancestor with Rabbit' (default: False),\n",
      " 'ocuniculus_homolog_wga_coverage': 'Rabbit Whole-genome alignment coverage' (default: False),\n",
      " 'odegus_homolog_associated_gene_name': 'Degu gene name' (default: False),\n",
      " 'odegus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'odegus_homolog_chrom_end': 'Degu chromosome/scaffold end (bp)' (default: False),\n",
      " 'odegus_homolog_chrom_start': 'Degu chromosome/scaffold start (bp)' (default: False),\n",
      " 'odegus_homolog_chromosome': 'Degu chromosome/scaffold name' (default: False),\n",
      " 'odegus_homolog_ensembl_gene': 'Degu gene stable ID' (default: False),\n",
      " 'odegus_homolog_ensembl_peptide': 'Degu protein or transcript stable ID' (default: False),\n",
      " 'odegus_homolog_goc_score': 'Degu Gene-order conservation score' (default: False),\n",
      " 'odegus_homolog_orthology_confidence': 'Degu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'odegus_homolog_orthology_type': 'Degu homology type' (default: False),\n",
      " 'odegus_homolog_perc_id': '%id. target Degu gene identical to query gene' (default: False),\n",
      " 'odegus_homolog_perc_id_r1': '%id. query gene identical to target Degu gene' (default: False),\n",
      " 'odegus_homolog_subtype': 'Last common ancestor with Degu' (default: False),\n",
      " 'odegus_homolog_wga_coverage': 'Degu Whole-genome alignment coverage' (default: False),\n",
      " 'ogarnettii_homolog_associated_gene_name': 'Bushbaby gene name' (default: False),\n",
      " 'ogarnettii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ogarnettii_homolog_chrom_end': 'Bushbaby chromosome/scaffold end (bp)' (default: False),\n",
      " 'ogarnettii_homolog_chrom_start': 'Bushbaby chromosome/scaffold start (bp)' (default: False),\n",
      " 'ogarnettii_homolog_chromosome': 'Bushbaby chromosome/scaffold name' (default: False),\n",
      " 'ogarnettii_homolog_ensembl_gene': 'Bushbaby gene stable ID' (default: False),\n",
      " 'ogarnettii_homolog_ensembl_peptide': 'Bushbaby protein or transcript stable ID' (default: False),\n",
      " 'ogarnettii_homolog_goc_score': 'Bushbaby Gene-order conservation score' (default: False),\n",
      " 'ogarnettii_homolog_orthology_confidence': 'Bushbaby orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ogarnettii_homolog_orthology_type': 'Bushbaby homology type' (default: False),\n",
      " 'ogarnettii_homolog_perc_id': '%id. target Bushbaby gene identical to query gene' (default: False),\n",
      " 'ogarnettii_homolog_perc_id_r1': '%id. query gene identical to target Bushbaby gene' (default: False),\n",
      " 'ogarnettii_homolog_subtype': 'Last common ancestor with Bushbaby' (default: False),\n",
      " 'ogarnettii_homolog_wga_coverage': 'Bushbaby Whole-genome alignment coverage' (default: False),\n",
      " 'ojavanicus_homolog_associated_gene_name': 'Javanese ricefish gene name' (default: False),\n",
      " 'ojavanicus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ojavanicus_homolog_chrom_end': 'Javanese ricefish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ojavanicus_homolog_chrom_start': 'Javanese ricefish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ojavanicus_homolog_chromosome': 'Javanese ricefish chromosome/scaffold name' (default: False),\n",
      " 'ojavanicus_homolog_ensembl_gene': 'Javanese ricefish gene stable ID' (default: False),\n",
      " 'ojavanicus_homolog_ensembl_peptide': 'Javanese ricefish protein or transcript stable ID' (default: False),\n",
      " 'ojavanicus_homolog_goc_score': 'Javanese ricefish Gene-order conservation score' (default: False),\n",
      " 'ojavanicus_homolog_orthology_confidence': 'Javanese ricefish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ojavanicus_homolog_orthology_type': 'Javanese ricefish homology type' (default: False),\n",
      " 'ojavanicus_homolog_perc_id': '%id. target Javanese ricefish gene identical to query gene' (default: False),\n",
      " 'ojavanicus_homolog_perc_id_r1': '%id. query gene identical to target Javanese ricefish gene' (default: False),\n",
      " 'ojavanicus_homolog_subtype': 'Last common ancestor with Javanese ricefish' (default: False),\n",
      " 'ojavanicus_homolog_wga_coverage': 'Javanese ricefish Whole-genome alignment coverage' (default: False),\n",
      " 'okisutch_homolog_associated_gene_name': 'Coho salmon gene name' (default: False),\n",
      " 'okisutch_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'okisutch_homolog_chrom_end': 'Coho salmon chromosome/scaffold end (bp)' (default: False),\n",
      " 'okisutch_homolog_chrom_start': 'Coho salmon chromosome/scaffold start (bp)' (default: False),\n",
      " 'okisutch_homolog_chromosome': 'Coho salmon chromosome/scaffold name' (default: False),\n",
      " 'okisutch_homolog_ensembl_gene': 'Coho salmon gene stable ID' (default: False),\n",
      " 'okisutch_homolog_ensembl_peptide': 'Coho salmon protein or transcript stable ID' (default: False),\n",
      " 'okisutch_homolog_goc_score': 'Coho salmon Gene-order conservation score' (default: False),\n",
      " 'okisutch_homolog_orthology_confidence': 'Coho salmon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'okisutch_homolog_orthology_type': 'Coho salmon homology type' (default: False),\n",
      " 'okisutch_homolog_perc_id': '%id. target Coho salmon gene identical to query gene' (default: False),\n",
      " 'okisutch_homolog_perc_id_r1': '%id. query gene identical to target Coho salmon gene' (default: False),\n",
      " 'okisutch_homolog_subtype': 'Last common ancestor with Coho salmon' (default: False),\n",
      " 'okisutch_homolog_wga_coverage': 'Coho salmon Whole-genome alignment coverage' (default: False),\n",
      " 'olatipes_homolog_associated_gene_name': 'Japanese medaka HdrR gene name' (default: False),\n",
      " 'olatipes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'olatipes_homolog_chrom_end': 'Japanese medaka HdrR chromosome/scaffold end (bp)' (default: False),\n",
      " 'olatipes_homolog_chrom_start': 'Japanese medaka HdrR chromosome/scaffold start (bp)' (default: False),\n",
      " 'olatipes_homolog_chromosome': 'Japanese medaka HdrR chromosome/scaffold name' (default: False),\n",
      " 'olatipes_homolog_ensembl_gene': 'Japanese medaka HdrR gene stable ID' (default: False),\n",
      " 'olatipes_homolog_ensembl_peptide': 'Japanese medaka HdrR protein or transcript stable ID' (default: False),\n",
      " 'olatipes_homolog_goc_score': 'Japanese medaka HdrR Gene-order conservation score' (default: False),\n",
      " 'olatipes_homolog_orthology_confidence': 'Japanese medaka HdrR orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'olatipes_homolog_orthology_type': 'Japanese medaka HdrR homology type' (default: False),\n",
      " 'olatipes_homolog_perc_id': '%id. target Japanese medaka HdrR gene identical to query gene' (default: False),\n",
      " 'olatipes_homolog_perc_id_r1': '%id. query gene identical to target Japanese medaka HdrR gene' (default: False),\n",
      " 'olatipes_homolog_subtype': 'Last common ancestor with Japanese medaka HdrR' (default: False),\n",
      " 'olatipes_homolog_wga_coverage': 'Japanese medaka HdrR Whole-genome alignment coverage' (default: False),\n",
      " 'omelastigma_homolog_associated_gene_name': 'Indian medaka gene name' (default: False),\n",
      " 'omelastigma_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'omelastigma_homolog_chrom_end': 'Indian medaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'omelastigma_homolog_chrom_start': 'Indian medaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'omelastigma_homolog_chromosome': 'Indian medaka chromosome/scaffold name' (default: False),\n",
      " 'omelastigma_homolog_ensembl_gene': 'Indian medaka gene stable ID' (default: False),\n",
      " 'omelastigma_homolog_ensembl_peptide': 'Indian medaka protein or transcript stable ID' (default: False),\n",
      " 'omelastigma_homolog_goc_score': 'Indian medaka Gene-order conservation score' (default: False),\n",
      " 'omelastigma_homolog_orthology_confidence': 'Indian medaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'omelastigma_homolog_orthology_type': 'Indian medaka homology type' (default: False),\n",
      " 'omelastigma_homolog_perc_id': '%id. target Indian medaka gene identical to query gene' (default: False),\n",
      " 'omelastigma_homolog_perc_id_r1': '%id. query gene identical to target Indian medaka gene' (default: False),\n",
      " 'omelastigma_homolog_subtype': 'Last common ancestor with Indian medaka' (default: False),\n",
      " 'omelastigma_homolog_wga_coverage': 'Indian medaka Whole-genome alignment coverage' (default: False),\n",
      " 'omykiss_homolog_associated_gene_name': 'Rainbow trout gene name' (default: False),\n",
      " 'omykiss_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'omykiss_homolog_chrom_end': 'Rainbow trout chromosome/scaffold end (bp)' (default: False),\n",
      " 'omykiss_homolog_chrom_start': 'Rainbow trout chromosome/scaffold start (bp)' (default: False),\n",
      " 'omykiss_homolog_chromosome': 'Rainbow trout chromosome/scaffold name' (default: False),\n",
      " 'omykiss_homolog_ensembl_gene': 'Rainbow trout gene stable ID' (default: False),\n",
      " 'omykiss_homolog_ensembl_peptide': 'Rainbow trout protein or transcript stable ID' (default: False),\n",
      " 'omykiss_homolog_goc_score': 'Rainbow trout Gene-order conservation score' (default: False),\n",
      " 'omykiss_homolog_orthology_confidence': 'Rainbow trout orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'omykiss_homolog_orthology_type': 'Rainbow trout homology type' (default: False),\n",
      " 'omykiss_homolog_perc_id': '%id. target Rainbow trout gene identical to query gene' (default: False),\n",
      " 'omykiss_homolog_perc_id_r1': '%id. query gene identical to target Rainbow trout gene' (default: False),\n",
      " 'omykiss_homolog_subtype': 'Last common ancestor with Rainbow trout' (default: False),\n",
      " 'oniloticus_homolog_associated_gene_name': 'Nile tilapia gene name' (default: False),\n",
      " 'oniloticus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oniloticus_homolog_chrom_end': 'Nile tilapia chromosome/scaffold end (bp)' (default: False),\n",
      " 'oniloticus_homolog_chrom_start': 'Nile tilapia chromosome/scaffold start (bp)' (default: False),\n",
      " 'oniloticus_homolog_chromosome': 'Nile tilapia chromosome/scaffold name' (default: False),\n",
      " 'oniloticus_homolog_ensembl_gene': 'Nile tilapia gene stable ID' (default: False),\n",
      " 'oniloticus_homolog_ensembl_peptide': 'Nile tilapia protein or transcript stable ID' (default: False),\n",
      " 'oniloticus_homolog_goc_score': 'Nile tilapia Gene-order conservation score' (default: False),\n",
      " 'oniloticus_homolog_orthology_confidence': 'Nile tilapia orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oniloticus_homolog_orthology_type': 'Nile tilapia homology type' (default: False),\n",
      " 'oniloticus_homolog_perc_id': '%id. target Nile tilapia gene identical to query gene' (default: False),\n",
      " 'oniloticus_homolog_perc_id_r1': '%id. query gene identical to target Nile tilapia gene' (default: False),\n",
      " 'oniloticus_homolog_subtype': 'Last common ancestor with Nile tilapia' (default: False),\n",
      " 'oniloticus_homolog_wga_coverage': 'Nile tilapia Whole-genome alignment coverage' (default: False),\n",
      " 'oprinceps_homolog_associated_gene_name': 'Pika gene name' (default: False),\n",
      " 'oprinceps_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oprinceps_homolog_chrom_end': 'Pika chromosome/scaffold end (bp)' (default: False),\n",
      " 'oprinceps_homolog_chrom_start': 'Pika chromosome/scaffold start (bp)' (default: False),\n",
      " 'oprinceps_homolog_chromosome': 'Pika chromosome/scaffold name' (default: False),\n",
      " 'oprinceps_homolog_ensembl_gene': 'Pika gene stable ID' (default: False),\n",
      " 'oprinceps_homolog_ensembl_peptide': 'Pika protein or transcript stable ID' (default: False),\n",
      " 'oprinceps_homolog_goc_score': 'Pika Gene-order conservation score' (default: False),\n",
      " 'oprinceps_homolog_orthology_confidence': 'Pika orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oprinceps_homolog_orthology_type': 'Pika homology type' (default: False),\n",
      " 'oprinceps_homolog_perc_id': '%id. target Pika gene identical to query gene' (default: False),\n",
      " 'oprinceps_homolog_perc_id_r1': '%id. query gene identical to target Pika gene' (default: False),\n",
      " 'oprinceps_homolog_subtype': 'Last common ancestor with Pika' (default: False),\n",
      " 'oprinceps_homolog_wga_coverage': 'Pika Whole-genome alignment coverage' (default: False),\n",
      " 'osinensis_homolog_associated_gene_name': 'Chinese medaka gene name' (default: False),\n",
      " 'osinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'osinensis_homolog_chrom_end': 'Chinese medaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'osinensis_homolog_chrom_start': 'Chinese medaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'osinensis_homolog_chromosome': 'Chinese medaka chromosome/scaffold name' (default: False),\n",
      " 'osinensis_homolog_ensembl_gene': 'Chinese medaka gene stable ID' (default: False),\n",
      " 'osinensis_homolog_ensembl_peptide': 'Chinese medaka protein or transcript stable ID' (default: False),\n",
      " 'osinensis_homolog_goc_score': 'Chinese medaka Gene-order conservation score' (default: False),\n",
      " 'osinensis_homolog_orthology_confidence': 'Chinese medaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'osinensis_homolog_orthology_type': 'Chinese medaka homology type' (default: False),\n",
      " 'osinensis_homolog_perc_id': '%id. target Chinese medaka gene identical to query gene' (default: False),\n",
      " 'osinensis_homolog_perc_id_r1': '%id. query gene identical to target Chinese medaka gene' (default: False),\n",
      " 'osinensis_homolog_subtype': 'Last common ancestor with Chinese medaka' (default: False),\n",
      " 'osinensis_homolog_wga_coverage': 'Chinese medaka Whole-genome alignment coverage' (default: False),\n",
      " 'otshawytscha_homolog_associated_gene_name': 'Chinook salmon gene name' (default: False),\n",
      " 'otshawytscha_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'otshawytscha_homolog_chrom_end': 'Chinook salmon chromosome/scaffold end (bp)' (default: False),\n",
      " 'otshawytscha_homolog_chrom_start': 'Chinook salmon chromosome/scaffold start (bp)' (default: False),\n",
      " 'otshawytscha_homolog_chromosome': 'Chinook salmon chromosome/scaffold name' (default: False),\n",
      " 'otshawytscha_homolog_ensembl_gene': 'Chinook salmon gene stable ID' (default: False),\n",
      " 'otshawytscha_homolog_ensembl_peptide': 'Chinook salmon protein or transcript stable ID' (default: False),\n",
      " 'otshawytscha_homolog_goc_score': 'Chinook salmon Gene-order conservation score' (default: False),\n",
      " 'otshawytscha_homolog_orthology_confidence': 'Chinook salmon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'otshawytscha_homolog_orthology_type': 'Chinook salmon homology type' (default: False),\n",
      " 'otshawytscha_homolog_perc_id': '%id. target Chinook salmon gene identical to query gene' (default: False),\n",
      " 'otshawytscha_homolog_perc_id_r1': '%id. query gene identical to target Chinook salmon gene' (default: False),\n",
      " 'otshawytscha_homolog_subtype': 'Last common ancestor with Chinook salmon' (default: False),\n",
      " 'otshawytscha_homolog_wga_coverage': 'Chinook salmon Whole-genome alignment coverage' (default: False),\n",
      " 'p_value': 'P value' (default: False),\n",
      " 'pabelii_homolog_associated_gene_name': 'Sumatran orangutan gene name' (default: False),\n",
      " 'pabelii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pabelii_homolog_chrom_end': 'Sumatran orangutan chromosome/scaffold end (bp)' (default: False),\n",
      " 'pabelii_homolog_chrom_start': 'Sumatran orangutan chromosome/scaffold start (bp)' (default: False),\n",
      " 'pabelii_homolog_chromosome': 'Sumatran orangutan chromosome/scaffold name' (default: False),\n",
      " 'pabelii_homolog_ensembl_gene': 'Sumatran orangutan gene stable ID' (default: False),\n",
      " 'pabelii_homolog_ensembl_peptide': 'Sumatran orangutan protein or transcript stable ID' (default: False),\n",
      " 'pabelii_homolog_goc_score': 'Sumatran orangutan Gene-order conservation score' (default: False),\n",
      " 'pabelii_homolog_orthology_confidence': 'Sumatran orangutan orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pabelii_homolog_orthology_type': 'Sumatran orangutan homology type' (default: False),\n",
      " 'pabelii_homolog_perc_id': '%id. target Sumatran orangutan gene identical to query gene' (default: False),\n",
      " 'pabelii_homolog_perc_id_r1': '%id. query gene identical to target Sumatran orangutan gene' (default: False),\n",
      " 'pabelii_homolog_subtype': 'Last common ancestor with Sumatran orangutan' (default: False),\n",
      " 'pabelii_homolog_wga_coverage': 'Sumatran orangutan Whole-genome alignment coverage' (default: False),\n",
      " 'panubis_homolog_associated_gene_name': 'Olive baboon gene name' (default: False),\n",
      " 'panubis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'panubis_homolog_chrom_end': 'Olive baboon chromosome/scaffold end (bp)' (default: False),\n",
      " 'panubis_homolog_chrom_start': 'Olive baboon chromosome/scaffold start (bp)' (default: False),\n",
      " 'panubis_homolog_chromosome': 'Olive baboon chromosome/scaffold name' (default: False),\n",
      " 'panubis_homolog_ensembl_gene': 'Olive baboon gene stable ID' (default: False),\n",
      " 'panubis_homolog_ensembl_peptide': 'Olive baboon protein or transcript stable ID' (default: False),\n",
      " 'panubis_homolog_goc_score': 'Olive baboon Gene-order conservation score' (default: False),\n",
      " 'panubis_homolog_orthology_confidence': 'Olive baboon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'panubis_homolog_orthology_type': 'Olive baboon homology type' (default: False),\n",
      " 'panubis_homolog_perc_id': '%id. target Olive baboon gene identical to query gene' (default: False),\n",
      " 'panubis_homolog_perc_id_r1': '%id. query gene identical to target Olive baboon gene' (default: False),\n",
      " 'panubis_homolog_subtype': 'Last common ancestor with Olive baboon' (default: False),\n",
      " 'panubis_homolog_wga_coverage': 'Olive baboon Whole-genome alignment coverage' (default: False),\n",
      " 'pcapensis_homolog_associated_gene_name': 'Hyrax gene name' (default: False),\n",
      " 'pcapensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcapensis_homolog_chrom_end': 'Hyrax chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcapensis_homolog_chrom_start': 'Hyrax chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcapensis_homolog_chromosome': 'Hyrax chromosome/scaffold name' (default: False),\n",
      " 'pcapensis_homolog_ensembl_gene': 'Hyrax gene stable ID' (default: False),\n",
      " 'pcapensis_homolog_ensembl_peptide': 'Hyrax protein or transcript stable ID' (default: False),\n",
      " 'pcapensis_homolog_goc_score': 'Hyrax Gene-order conservation score' (default: False),\n",
      " 'pcapensis_homolog_orthology_confidence': 'Hyrax orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcapensis_homolog_orthology_type': 'Hyrax homology type' (default: False),\n",
      " 'pcapensis_homolog_perc_id': '%id. target Hyrax gene identical to query gene' (default: False),\n",
      " 'pcapensis_homolog_perc_id_r1': '%id. query gene identical to target Hyrax gene' (default: False),\n",
      " 'pcapensis_homolog_subtype': 'Last common ancestor with Hyrax' (default: False),\n",
      " 'pcapensis_homolog_wga_coverage': 'Hyrax Whole-genome alignment coverage' (default: False),\n",
      " 'pcatodon_homolog_associated_gene_name': 'Sperm whale gene name' (default: False),\n",
      " 'pcatodon_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcatodon_homolog_chrom_end': 'Sperm whale chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcatodon_homolog_chrom_start': 'Sperm whale chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcatodon_homolog_chromosome': 'Sperm whale chromosome/scaffold name' (default: False),\n",
      " 'pcatodon_homolog_ensembl_gene': 'Sperm whale gene stable ID' (default: False),\n",
      " 'pcatodon_homolog_ensembl_peptide': 'Sperm whale protein or transcript stable ID' (default: False),\n",
      " 'pcatodon_homolog_goc_score': 'Sperm whale Gene-order conservation score' (default: False),\n",
      " 'pcatodon_homolog_orthology_confidence': 'Sperm whale orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcatodon_homolog_orthology_type': 'Sperm whale homology type' (default: False),\n",
      " 'pcatodon_homolog_perc_id': '%id. target Sperm whale gene identical to query gene' (default: False),\n",
      " 'pcatodon_homolog_perc_id_r1': '%id. query gene identical to target Sperm whale gene' (default: False),\n",
      " 'pcatodon_homolog_subtype': 'Last common ancestor with Sperm whale' (default: False),\n",
      " 'pcatodon_homolog_wga_coverage': 'Sperm whale Whole-genome alignment coverage' (default: False),\n",
      " 'pcinereus_homolog_associated_gene_name': 'Koala gene name' (default: False),\n",
      " 'pcinereus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcinereus_homolog_chrom_end': 'Koala chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcinereus_homolog_chrom_start': 'Koala chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcinereus_homolog_chromosome': 'Koala chromosome/scaffold name' (default: False),\n",
      " 'pcinereus_homolog_ensembl_gene': 'Koala gene stable ID' (default: False),\n",
      " 'pcinereus_homolog_ensembl_peptide': 'Koala protein or transcript stable ID' (default: False),\n",
      " 'pcinereus_homolog_goc_score': 'Koala Gene-order conservation score' (default: False),\n",
      " 'pcinereus_homolog_orthology_confidence': 'Koala orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcinereus_homolog_orthology_type': 'Koala homology type' (default: False),\n",
      " 'pcinereus_homolog_perc_id': '%id. target Koala gene identical to query gene' (default: False),\n",
      " 'pcinereus_homolog_perc_id_r1': '%id. query gene identical to target Koala gene' (default: False),\n",
      " 'pcinereus_homolog_subtype': 'Last common ancestor with Koala' (default: False),\n",
      " 'pcinereus_homolog_wga_coverage': 'Koala Whole-genome alignment coverage' (default: False),\n",
      " 'pcoquereli_homolog_associated_gene_name': 'Coquerel's sifaka gene name' (default: False),\n",
      " 'pcoquereli_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcoquereli_homolog_chrom_end': 'Coquerel's sifaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcoquereli_homolog_chrom_start': 'Coquerel's sifaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcoquereli_homolog_chromosome': 'Coquerel's sifaka chromosome/scaffold name' (default: False),\n",
      " 'pcoquereli_homolog_ensembl_gene': 'Coquerel's sifaka gene stable ID' (default: False),\n",
      " 'pcoquereli_homolog_ensembl_peptide': 'Coquerel's sifaka protein or transcript stable ID' (default: False),\n",
      " 'pcoquereli_homolog_goc_score': 'Coquerel's sifaka Gene-order conservation score' (default: False),\n",
      " 'pcoquereli_homolog_orthology_confidence': 'Coquerel's sifaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcoquereli_homolog_orthology_type': 'Coquerel's sifaka homology type' (default: False),\n",
      " 'pcoquereli_homolog_perc_id': '%id. target Coquerel's sifaka gene identical to query gene' (default: False),\n",
      " 'pcoquereli_homolog_perc_id_r1': '%id. query gene identical to target Coquerel's sifaka gene' (default: False),\n",
      " 'pcoquereli_homolog_subtype': 'Last common ancestor with Coquerel's sifaka' (default: False),\n",
      " 'pcoquereli_homolog_wga_coverage': 'Coquerel's sifaka Whole-genome alignment coverage' (default: False),\n",
      " 'pdb': 'PDB ID' (default: False),\n",
      " 'peptide': 'Peptide' (default: False),\n",
      " 'peptide_location': 'Protein location (aa)' (default: False),\n",
      " 'peptide_shift': 'Protein allele' (default: False),\n",
      " 'peptide_version': 'Version (protein)' (default: False),\n",
      " 'percentage_gene_gc_content': 'Gene % GC content' (default: False),\n",
      " 'pfam': 'Pfam ID' (default: False),\n",
      " 'pfam_end': 'Pfam end' (default: False),\n",
      " 'pfam_start': 'Pfam start' (default: False),\n",
      " 'pformosa_homolog_associated_gene_name': 'Amazon molly gene name' (default: False),\n",
      " 'pformosa_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pformosa_homolog_chrom_end': 'Amazon molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'pformosa_homolog_chrom_start': 'Amazon molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'pformosa_homolog_chromosome': 'Amazon molly chromosome/scaffold name' (default: False),\n",
      " 'pformosa_homolog_ensembl_gene': 'Amazon molly gene stable ID' (default: False),\n",
      " 'pformosa_homolog_ensembl_peptide': 'Amazon molly protein or transcript stable ID' (default: False),\n",
      " 'pformosa_homolog_goc_score': 'Amazon molly Gene-order conservation score' (default: False),\n",
      " 'pformosa_homolog_orthology_confidence': 'Amazon molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pformosa_homolog_orthology_type': 'Amazon molly homology type' (default: False),\n",
      " 'pformosa_homolog_perc_id': '%id. target Amazon molly gene identical to query gene' (default: False),\n",
      " 'pformosa_homolog_perc_id_r1': '%id. query gene identical to target Amazon molly gene' (default: False),\n",
      " 'pformosa_homolog_subtype': 'Last common ancestor with Amazon molly' (default: False),\n",
      " 'pformosa_homolog_wga_coverage': 'Amazon molly Whole-genome alignment coverage' (default: False),\n",
      " 'pfscan': 'PROSITE profiles ID' (default: False),\n",
      " 'pfscan_end': 'PROSITE profiles end' (default: False),\n",
      " 'pfscan_start': 'PROSITE profiles start' (default: False),\n",
      " 'phalanx_onearray': 'PHALANX OneArray probe' (default: False),\n",
      " 'phase': 'Start phase' (default: False),\n",
      " 'phenotype_description': 'Phenotype description' (default: False),\n",
      " 'pirsf': 'PIRSF ID' (default: False),\n",
      " 'pirsf_end': 'PIRSF end' (default: False),\n",
      " 'pirsf_start': 'PIRSF start' (default: False),\n",
      " 'pkingsleyae_homolog_associated_gene_name': 'Paramormyrops kingsleyae gene name' (default: False),\n",
      " 'pkingsleyae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pkingsleyae_homolog_chrom_end': 'Paramormyrops kingsleyae chromosome/scaffold end (bp)' (default: False),\n",
      " 'pkingsleyae_homolog_chrom_start': 'Paramormyrops kingsleyae chromosome/scaffold start (bp)' (default: False),\n",
      " 'pkingsleyae_homolog_chromosome': 'Paramormyrops kingsleyae chromosome/scaffold name' (default: False),\n",
      " 'pkingsleyae_homolog_ensembl_gene': 'Paramormyrops kingsleyae gene stable ID' (default: False),\n",
      " 'pkingsleyae_homolog_ensembl_peptide': 'Paramormyrops kingsleyae protein or transcript stable ID' (default: False),\n",
      " 'pkingsleyae_homolog_goc_score': 'Paramormyrops kingsleyae Gene-order conservation score' (default: False),\n",
      " 'pkingsleyae_homolog_orthology_confidence': 'Paramormyrops kingsleyae orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pkingsleyae_homolog_orthology_type': 'Paramormyrops kingsleyae homology type' (default: False),\n",
      " 'pkingsleyae_homolog_perc_id': '%id. target Paramormyrops kingsleyae gene identical to query gene' (default: False),\n",
      " 'pkingsleyae_homolog_perc_id_r1': '%id. query gene identical to target Paramormyrops kingsleyae gene' (default: False),\n",
      " 'pkingsleyae_homolog_subtype': 'Last common ancestor with Paramormyrops kingsleyae' (default: False),\n",
      " 'pkingsleyae_homolog_wga_coverage': 'Paramormyrops kingsleyae Whole-genome alignment coverage' (default: False),\n",
      " 'platipinna_homolog_associated_gene_name': 'Sailfin molly gene name' (default: False),\n",
      " 'platipinna_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'platipinna_homolog_chrom_end': 'Sailfin molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'platipinna_homolog_chrom_start': 'Sailfin molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'platipinna_homolog_chromosome': 'Sailfin molly chromosome/scaffold name' (default: False),\n",
      " 'platipinna_homolog_ensembl_gene': 'Sailfin molly gene stable ID' (default: False),\n",
      " 'platipinna_homolog_ensembl_peptide': 'Sailfin molly protein or transcript stable ID' (default: False),\n",
      " 'platipinna_homolog_goc_score': 'Sailfin molly Gene-order conservation score' (default: False),\n",
      " 'platipinna_homolog_orthology_confidence': 'Sailfin molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'platipinna_homolog_orthology_type': 'Sailfin molly homology type' (default: False),\n",
      " 'platipinna_homolog_perc_id': '%id. target Sailfin molly gene identical to query gene' (default: False),\n",
      " 'platipinna_homolog_perc_id_r1': '%id. query gene identical to target Sailfin molly gene' (default: False),\n",
      " 'platipinna_homolog_subtype': 'Last common ancestor with Sailfin molly' (default: False),\n",
      " 'platipinna_homolog_wga_coverage': 'Sailfin molly Whole-genome alignment coverage' (default: False),\n",
      " 'pleo_homolog_associated_gene_name': 'Lion gene name' (default: False),\n",
      " 'pleo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pleo_homolog_chrom_end': 'Lion chromosome/scaffold end (bp)' (default: False),\n",
      " 'pleo_homolog_chrom_start': 'Lion chromosome/scaffold start (bp)' (default: False),\n",
      " 'pleo_homolog_chromosome': 'Lion chromosome/scaffold name' (default: False),\n",
      " 'pleo_homolog_ensembl_gene': 'Lion gene stable ID' (default: False),\n",
      " 'pleo_homolog_ensembl_peptide': 'Lion protein or transcript stable ID' (default: False),\n",
      " 'pleo_homolog_goc_score': 'Lion Gene-order conservation score' (default: False),\n",
      " 'pleo_homolog_orthology_confidence': 'Lion orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pleo_homolog_orthology_type': 'Lion homology type' (default: False),\n",
      " 'pleo_homolog_perc_id': '%id. target Lion gene identical to query gene' (default: False),\n",
      " 'pleo_homolog_perc_id_r1': '%id. query gene identical to target Lion gene' (default: False),\n",
      " 'pleo_homolog_subtype': 'Last common ancestor with Lion' (default: False),\n",
      " 'pleo_homolog_wga_coverage': 'Lion Whole-genome alignment coverage' (default: False),\n",
      " 'pmajor_homolog_associated_gene_name': 'Great Tit gene name' (default: False),\n",
      " 'pmajor_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmajor_homolog_chrom_end': 'Great Tit chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmajor_homolog_chrom_start': 'Great Tit chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmajor_homolog_chromosome': 'Great Tit chromosome/scaffold name' (default: False),\n",
      " 'pmajor_homolog_ensembl_gene': 'Great Tit gene stable ID' (default: False),\n",
      " 'pmajor_homolog_ensembl_peptide': 'Great Tit protein or transcript stable ID' (default: False),\n",
      " 'pmajor_homolog_goc_score': 'Great Tit Gene-order conservation score' (default: False),\n",
      " 'pmajor_homolog_orthology_confidence': 'Great Tit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmajor_homolog_orthology_type': 'Great Tit homology type' (default: False),\n",
      " 'pmajor_homolog_perc_id': '%id. target Great Tit gene identical to query gene' (default: False),\n",
      " 'pmajor_homolog_perc_id_r1': '%id. query gene identical to target Great Tit gene' (default: False),\n",
      " 'pmajor_homolog_subtype': 'Last common ancestor with Great Tit' (default: False),\n",
      " 'pmajor_homolog_wga_coverage': 'Great Tit Whole-genome alignment coverage' (default: False),\n",
      " 'pmarinus_homolog_associated_gene_name': 'Lamprey gene name' (default: False),\n",
      " 'pmarinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmarinus_homolog_chrom_end': 'Lamprey chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmarinus_homolog_chrom_start': 'Lamprey chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmarinus_homolog_chromosome': 'Lamprey chromosome/scaffold name' (default: False),\n",
      " 'pmarinus_homolog_ensembl_gene': 'Lamprey gene stable ID' (default: False),\n",
      " 'pmarinus_homolog_ensembl_peptide': 'Lamprey protein or transcript stable ID' (default: False),\n",
      " 'pmarinus_homolog_orthology_confidence': 'Lamprey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmarinus_homolog_orthology_type': 'Lamprey homology type' (default: False),\n",
      " 'pmarinus_homolog_perc_id': '%id. target Lamprey gene identical to query gene' (default: False),\n",
      " 'pmarinus_homolog_perc_id_r1': '%id. query gene identical to target Lamprey gene' (default: False),\n",
      " 'pmarinus_homolog_subtype': 'Last common ancestor with Lamprey' (default: False),\n",
      " 'pmarinus_homolog_wga_coverage': 'Lamprey Whole-genome alignment coverage' (default: False),\n",
      " 'pmbairdii_homolog_associated_gene_name': 'Northern American deer mouse gene name' (default: False),\n",
      " 'pmbairdii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmbairdii_homolog_chrom_end': 'Northern American deer mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmbairdii_homolog_chrom_start': 'Northern American deer mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmbairdii_homolog_chromosome': 'Northern American deer mouse chromosome/scaffold name' (default: False),\n",
      " 'pmbairdii_homolog_ensembl_gene': 'Northern American deer mouse gene stable ID' (default: False),\n",
      " 'pmbairdii_homolog_ensembl_peptide': 'Northern American deer mouse protein or transcript stable ID' (default: False),\n",
      " 'pmbairdii_homolog_goc_score': 'Northern American deer mouse Gene-order conservation score' (default: False),\n",
      " 'pmbairdii_homolog_orthology_confidence': 'Northern American deer mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmbairdii_homolog_orthology_type': 'Northern American deer mouse homology type' (default: False),\n",
      " 'pmbairdii_homolog_perc_id': '%id. target Northern American deer mouse gene identical to query gene' (default: False),\n",
      " 'pmbairdii_homolog_perc_id_r1': '%id. query gene identical to target Northern American deer mouse gene' (default: False),\n",
      " 'pmbairdii_homolog_subtype': 'Last common ancestor with Northern American deer mouse' (default: False),\n",
      " 'pmbairdii_homolog_wga_coverage': 'Northern American deer mouse Whole-genome alignment coverage' (default: False),\n",
      " 'pmuralis_homolog_associated_gene_name': 'Common wall lizard gene name' (default: False),\n",
      " 'pmuralis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmuralis_homolog_chrom_end': 'Common wall lizard chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmuralis_homolog_chrom_start': 'Common wall lizard chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmuralis_homolog_chromosome': 'Common wall lizard chromosome/scaffold name' (default: False),\n",
      " 'pmuralis_homolog_ensembl_gene': 'Common wall lizard gene stable ID' (default: False),\n",
      " 'pmuralis_homolog_ensembl_peptide': 'Common wall lizard protein or transcript stable ID' (default: False),\n",
      " 'pmuralis_homolog_goc_score': 'Common wall lizard Gene-order conservation score' (default: False),\n",
      " 'pmuralis_homolog_orthology_confidence': 'Common wall lizard orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmuralis_homolog_orthology_type': 'Common wall lizard homology type' (default: False),\n",
      " 'pmuralis_homolog_perc_id': '%id. target Common wall lizard gene identical to query gene' (default: False),\n",
      " 'pmuralis_homolog_perc_id_r1': '%id. query gene identical to target Common wall lizard gene' (default: False),\n",
      " 'pmuralis_homolog_subtype': 'Last common ancestor with Common wall lizard' (default: False),\n",
      " 'pmuralis_homolog_wga_coverage': 'Common wall lizard Whole-genome alignment coverage' (default: False),\n",
      " 'pnattereri_homolog_associated_gene_name': 'Red-bellied piranha gene name' (default: False),\n",
      " 'pnattereri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pnattereri_homolog_chrom_end': 'Red-bellied piranha chromosome/scaffold end (bp)' (default: False),\n",
      " 'pnattereri_homolog_chrom_start': 'Red-bellied piranha chromosome/scaffold start (bp)' (default: False),\n",
      " 'pnattereri_homolog_chromosome': 'Red-bellied piranha chromosome/scaffold name' (default: False),\n",
      " 'pnattereri_homolog_ensembl_gene': 'Red-bellied piranha gene stable ID' (default: False),\n",
      " 'pnattereri_homolog_ensembl_peptide': 'Red-bellied piranha protein or transcript stable ID' (default: False),\n",
      " 'pnattereri_homolog_goc_score': 'Red-bellied piranha Gene-order conservation score' (default: False),\n",
      " 'pnattereri_homolog_orthology_confidence': 'Red-bellied piranha orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pnattereri_homolog_orthology_type': 'Red-bellied piranha homology type' (default: False),\n",
      " 'pnattereri_homolog_perc_id': '%id. target Red-bellied piranha gene identical to query gene' (default: False),\n",
      " 'pnattereri_homolog_perc_id_r1': '%id. query gene identical to target Red-bellied piranha gene' (default: False),\n",
      " 'pnattereri_homolog_subtype': 'Last common ancestor with Red-bellied piranha' (default: False),\n",
      " 'pnattereri_homolog_wga_coverage': 'Red-bellied piranha Whole-genome alignment coverage' (default: False),\n",
      " 'pnyererei_homolog_associated_gene_name': 'Makobe Island cichlid gene name' (default: False),\n",
      " 'pnyererei_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pnyererei_homolog_chrom_end': 'Makobe Island cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'pnyererei_homolog_chrom_start': 'Makobe Island cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'pnyererei_homolog_chromosome': 'Makobe Island cichlid chromosome/scaffold name' (default: False),\n",
      " 'pnyererei_homolog_ensembl_gene': 'Makobe Island cichlid gene stable ID' (default: False),\n",
      " 'pnyererei_homolog_ensembl_peptide': 'Makobe Island cichlid protein or transcript stable ID' (default: False),\n",
      " 'pnyererei_homolog_goc_score': 'Makobe Island cichlid Gene-order conservation score' (default: False),\n",
      " 'pnyererei_homolog_orthology_confidence': 'Makobe Island cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pnyererei_homolog_orthology_type': 'Makobe Island cichlid homology type' (default: False),\n",
      " 'pnyererei_homolog_perc_id': '%id. target Makobe Island cichlid gene identical to query gene' (default: False),\n",
      " 'pnyererei_homolog_perc_id_r1': '%id. query gene identical to target Makobe Island cichlid gene' (default: False),\n",
      " 'pnyererei_homolog_subtype': 'Last common ancestor with Makobe Island cichlid' (default: False),\n",
      " 'pnyererei_homolog_wga_coverage': 'Makobe Island cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'polyphen_prediction_2076': 'PolyPhen prediction' (default: False),\n",
      " 'polyphen_score_2076': 'PolyPhen score' (default: False),\n",
      " 'ppaniscus_homolog_associated_gene_name': 'Bonobo gene name' (default: False),\n",
      " 'ppaniscus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ppaniscus_homolog_chrom_end': 'Bonobo chromosome/scaffold end (bp)' (default: False),\n",
      " 'ppaniscus_homolog_chrom_start': 'Bonobo chromosome/scaffold start (bp)' (default: False),\n",
      " 'ppaniscus_homolog_chromosome': 'Bonobo chromosome/scaffold name' (default: False),\n",
      " 'ppaniscus_homolog_ensembl_gene': 'Bonobo gene stable ID' (default: False),\n",
      " 'ppaniscus_homolog_ensembl_peptide': 'Bonobo protein or transcript stable ID' (default: False),\n",
      " 'ppaniscus_homolog_goc_score': 'Bonobo Gene-order conservation score' (default: False),\n",
      " 'ppaniscus_homolog_orthology_confidence': 'Bonobo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ppaniscus_homolog_orthology_type': 'Bonobo homology type' (default: False),\n",
      " 'ppaniscus_homolog_perc_id': '%id. target Bonobo gene identical to query gene' (default: False),\n",
      " 'ppaniscus_homolog_perc_id_r1': '%id. query gene identical to target Bonobo gene' (default: False),\n",
      " 'ppaniscus_homolog_subtype': 'Last common ancestor with Bonobo' (default: False),\n",
      " 'ppaniscus_homolog_wga_coverage': 'Bonobo Whole-genome alignment coverage' (default: False),\n",
      " 'ppardus_homolog_associated_gene_name': 'Leopard gene name' (default: False),\n",
      " 'ppardus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ppardus_homolog_chrom_end': 'Leopard chromosome/scaffold end (bp)' (default: False),\n",
      " 'ppardus_homolog_chrom_start': 'Leopard chromosome/scaffold start (bp)' (default: False),\n",
      " 'ppardus_homolog_chromosome': 'Leopard chromosome/scaffold name' (default: False),\n",
      " 'ppardus_homolog_ensembl_gene': 'Leopard gene stable ID' (default: False),\n",
      " 'ppardus_homolog_ensembl_peptide': 'Leopard protein or transcript stable ID' (default: False),\n",
      " 'ppardus_homolog_goc_score': 'Leopard Gene-order conservation score' (default: False),\n",
      " 'ppardus_homolog_orthology_confidence': 'Leopard orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ppardus_homolog_orthology_type': 'Leopard homology type' (default: False),\n",
      " 'ppardus_homolog_perc_id': '%id. target Leopard gene identical to query gene' (default: False),\n",
      " 'ppardus_homolog_perc_id_r1': '%id. query gene identical to target Leopard gene' (default: False),\n",
      " 'ppardus_homolog_subtype': 'Last common ancestor with Leopard' (default: False),\n",
      " 'ppardus_homolog_wga_coverage': 'Leopard Whole-genome alignment coverage' (default: False),\n",
      " 'preticulata_homolog_associated_gene_name': 'Guppy gene name' (default: False),\n",
      " 'preticulata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'preticulata_homolog_chrom_end': 'Guppy chromosome/scaffold end (bp)' (default: False),\n",
      " 'preticulata_homolog_chrom_start': 'Guppy chromosome/scaffold start (bp)' (default: False),\n",
      " 'preticulata_homolog_chromosome': 'Guppy chromosome/scaffold name' (default: False),\n",
      " 'preticulata_homolog_ensembl_gene': 'Guppy gene stable ID' (default: False),\n",
      " 'preticulata_homolog_ensembl_peptide': 'Guppy protein or transcript stable ID' (default: False),\n",
      " 'preticulata_homolog_goc_score': 'Guppy Gene-order conservation score' (default: False),\n",
      " 'preticulata_homolog_orthology_confidence': 'Guppy orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'preticulata_homolog_orthology_type': 'Guppy homology type' (default: False),\n",
      " 'preticulata_homolog_perc_id': '%id. target Guppy gene identical to query gene' (default: False),\n",
      " 'preticulata_homolog_perc_id_r1': '%id. query gene identical to target Guppy gene' (default: False),\n",
      " 'preticulata_homolog_subtype': 'Last common ancestor with Guppy' (default: False),\n",
      " 'preticulata_homolog_wga_coverage': 'Guppy Whole-genome alignment coverage' (default: False),\n",
      " 'prints': 'Prints ID' (default: False),\n",
      " 'prints_end': 'Prints end' (default: False),\n",
      " 'prints_start': 'Prints start' (default: False),\n",
      " 'protein_id': 'INSDC protein ID' (default: False),\n",
      " 'psimus_homolog_associated_gene_name': 'Greater bamboo lemur gene name' (default: False),\n",
      " 'psimus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psimus_homolog_chrom_end': 'Greater bamboo lemur chromosome/scaffold end (bp)' (default: False),\n",
      " 'psimus_homolog_chrom_start': 'Greater bamboo lemur chromosome/scaffold start (bp)' (default: False),\n",
      " 'psimus_homolog_chromosome': 'Greater bamboo lemur chromosome/scaffold name' (default: False),\n",
      " 'psimus_homolog_ensembl_gene': 'Greater bamboo lemur gene stable ID' (default: False),\n",
      " 'psimus_homolog_ensembl_peptide': 'Greater bamboo lemur protein or transcript stable ID' (default: False),\n",
      " 'psimus_homolog_goc_score': 'Greater bamboo lemur Gene-order conservation score' (default: False),\n",
      " 'psimus_homolog_orthology_confidence': 'Greater bamboo lemur orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psimus_homolog_orthology_type': 'Greater bamboo lemur homology type' (default: False),\n",
      " 'psimus_homolog_perc_id': '%id. target Greater bamboo lemur gene identical to query gene' (default: False),\n",
      " 'psimus_homolog_perc_id_r1': '%id. query gene identical to target Greater bamboo lemur gene' (default: False),\n",
      " 'psimus_homolog_subtype': 'Last common ancestor with Greater bamboo lemur' (default: False),\n",
      " 'psimus_homolog_wga_coverage': 'Greater bamboo lemur Whole-genome alignment coverage' (default: False),\n",
      " 'psinensis_homolog_associated_gene_name': 'Chinese softshell turtle gene name' (default: False),\n",
      " 'psinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psinensis_homolog_chrom_end': 'Chinese softshell turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'psinensis_homolog_chrom_start': 'Chinese softshell turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'psinensis_homolog_chromosome': 'Chinese softshell turtle chromosome/scaffold name' (default: False),\n",
      " 'psinensis_homolog_ensembl_gene': 'Chinese softshell turtle gene stable ID' (default: False),\n",
      " 'psinensis_homolog_ensembl_peptide': 'Chinese softshell turtle protein or transcript stable ID' (default: False),\n",
      " 'psinensis_homolog_goc_score': 'Chinese softshell turtle Gene-order conservation score' (default: False),\n",
      " 'psinensis_homolog_orthology_confidence': 'Chinese softshell turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psinensis_homolog_orthology_type': 'Chinese softshell turtle homology type' (default: False),\n",
      " 'psinensis_homolog_perc_id': '%id. target Chinese softshell turtle gene identical to query gene' (default: False),\n",
      " 'psinensis_homolog_perc_id_r1': '%id. query gene identical to target Chinese softshell turtle gene' (default: False),\n",
      " 'psinensis_homolog_subtype': 'Last common ancestor with Chinese softshell turtle' (default: False),\n",
      " 'psinensis_homolog_wga_coverage': 'Chinese softshell turtle Whole-genome alignment coverage' (default: False),\n",
      " 'psinus_homolog_associated_gene_name': 'Vaquita gene name' (default: False),\n",
      " 'psinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psinus_homolog_chrom_end': 'Vaquita chromosome/scaffold end (bp)' (default: False),\n",
      " 'psinus_homolog_chrom_start': 'Vaquita chromosome/scaffold start (bp)' (default: False),\n",
      " 'psinus_homolog_chromosome': 'Vaquita chromosome/scaffold name' (default: False),\n",
      " 'psinus_homolog_ensembl_gene': 'Vaquita gene stable ID' (default: False),\n",
      " 'psinus_homolog_ensembl_peptide': 'Vaquita protein or transcript stable ID' (default: False),\n",
      " 'psinus_homolog_goc_score': 'Vaquita Gene-order conservation score' (default: False),\n",
      " 'psinus_homolog_orthology_confidence': 'Vaquita orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psinus_homolog_orthology_type': 'Vaquita homology type' (default: False),\n",
      " 'psinus_homolog_perc_id': '%id. target Vaquita gene identical to query gene' (default: False),\n",
      " 'psinus_homolog_perc_id_r1': '%id. query gene identical to target Vaquita gene' (default: False),\n",
      " 'psinus_homolog_subtype': 'Last common ancestor with Vaquita' (default: False),\n",
      " 'psinus_homolog_wga_coverage': 'Vaquita Whole-genome alignment coverage' (default: False),\n",
      " 'ptaltaica_homolog_associated_gene_name': 'Tiger gene name' (default: False),\n",
      " 'ptaltaica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptaltaica_homolog_chrom_end': 'Tiger chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptaltaica_homolog_chrom_start': 'Tiger chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptaltaica_homolog_chromosome': 'Tiger chromosome/scaffold name' (default: False),\n",
      " 'ptaltaica_homolog_ensembl_gene': 'Tiger gene stable ID' (default: False),\n",
      " 'ptaltaica_homolog_ensembl_peptide': 'Tiger protein or transcript stable ID' (default: False),\n",
      " 'ptaltaica_homolog_goc_score': 'Tiger Gene-order conservation score' (default: False),\n",
      " 'ptaltaica_homolog_orthology_confidence': 'Tiger orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptaltaica_homolog_orthology_type': 'Tiger homology type' (default: False),\n",
      " 'ptaltaica_homolog_perc_id': '%id. target Tiger gene identical to query gene' (default: False),\n",
      " 'ptaltaica_homolog_perc_id_r1': '%id. query gene identical to target Tiger gene' (default: False),\n",
      " 'ptaltaica_homolog_subtype': 'Last common ancestor with Tiger' (default: False),\n",
      " 'ptaltaica_homolog_wga_coverage': 'Tiger Whole-genome alignment coverage' (default: False),\n",
      " 'ptextilis_homolog_associated_gene_name': 'Eastern brown snake gene name' (default: False),\n",
      " 'ptextilis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptextilis_homolog_chrom_end': 'Eastern brown snake chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptextilis_homolog_chrom_start': 'Eastern brown snake chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptextilis_homolog_chromosome': 'Eastern brown snake chromosome/scaffold name' (default: False),\n",
      " 'ptextilis_homolog_ensembl_gene': 'Eastern brown snake gene stable ID' (default: False),\n",
      " 'ptextilis_homolog_ensembl_peptide': 'Eastern brown snake protein or transcript stable ID' (default: False),\n",
      " 'ptextilis_homolog_goc_score': 'Eastern brown snake Gene-order conservation score' (default: False),\n",
      " 'ptextilis_homolog_orthology_confidence': 'Eastern brown snake orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptextilis_homolog_orthology_type': 'Eastern brown snake homology type' (default: False),\n",
      " 'ptextilis_homolog_perc_id': '%id. target Eastern brown snake gene identical to query gene' (default: False),\n",
      " 'ptextilis_homolog_perc_id_r1': '%id. query gene identical to target Eastern brown snake gene' (default: False),\n",
      " 'ptextilis_homolog_subtype': 'Last common ancestor with Eastern brown snake' (default: False),\n",
      " 'ptextilis_homolog_wga_coverage': 'Eastern brown snake Whole-genome alignment coverage' (default: False),\n",
      " 'ptroglodytes_homolog_associated_gene_name': 'Chimpanzee gene name' (default: False),\n",
      " 'ptroglodytes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptroglodytes_homolog_chrom_end': 'Chimpanzee chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptroglodytes_homolog_chrom_start': 'Chimpanzee chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptroglodytes_homolog_chromosome': 'Chimpanzee chromosome/scaffold name' (default: False),\n",
      " 'ptroglodytes_homolog_ensembl_gene': 'Chimpanzee gene stable ID' (default: False),\n",
      " 'ptroglodytes_homolog_ensembl_peptide': 'Chimpanzee protein or transcript stable ID' (default: False),\n",
      " 'ptroglodytes_homolog_goc_score': 'Chimpanzee Gene-order conservation score' (default: False),\n",
      " 'ptroglodytes_homolog_orthology_confidence': 'Chimpanzee orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptroglodytes_homolog_orthology_type': 'Chimpanzee homology type' (default: False),\n",
      " 'ptroglodytes_homolog_perc_id': '%id. target Chimpanzee gene identical to query gene' (default: False),\n",
      " 'ptroglodytes_homolog_perc_id_r1': '%id. query gene identical to target Chimpanzee gene' (default: False),\n",
      " 'ptroglodytes_homolog_subtype': 'Last common ancestor with Chimpanzee' (default: False),\n",
      " 'ptroglodytes_homolog_wga_coverage': 'Chimpanzee Whole-genome alignment coverage' (default: False),\n",
      " 'pvampyrus_homolog_associated_gene_name': 'Megabat gene name' (default: False),\n",
      " 'pvampyrus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pvampyrus_homolog_chrom_end': 'Megabat chromosome/scaffold end (bp)' (default: False),\n",
      " 'pvampyrus_homolog_chrom_start': 'Megabat chromosome/scaffold start (bp)' (default: False),\n",
      " 'pvampyrus_homolog_chromosome': 'Megabat chromosome/scaffold name' (default: False),\n",
      " 'pvampyrus_homolog_ensembl_gene': 'Megabat gene stable ID' (default: False),\n",
      " 'pvampyrus_homolog_ensembl_peptide': 'Megabat protein or transcript stable ID' (default: False),\n",
      " 'pvampyrus_homolog_goc_score': 'Megabat Gene-order conservation score' (default: False),\n",
      " 'pvampyrus_homolog_orthology_confidence': 'Megabat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pvampyrus_homolog_orthology_type': 'Megabat homology type' (default: False),\n",
      " 'pvampyrus_homolog_perc_id': '%id. target Megabat gene identical to query gene' (default: False),\n",
      " 'pvampyrus_homolog_perc_id_r1': '%id. query gene identical to target Megabat gene' (default: False),\n",
      " 'pvampyrus_homolog_subtype': 'Last common ancestor with Megabat' (default: False),\n",
      " 'pvampyrus_homolog_wga_coverage': 'Megabat Whole-genome alignment coverage' (default: False),\n",
      " 'rank': 'Exon rank in transcript' (default: False),\n",
      " 'rbieti_homolog_associated_gene_name': 'Black snub-nosed monkey gene name' (default: False),\n",
      " 'rbieti_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rbieti_homolog_chrom_end': 'Black snub-nosed monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'rbieti_homolog_chrom_start': 'Black snub-nosed monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'rbieti_homolog_chromosome': 'Black snub-nosed monkey chromosome/scaffold name' (default: False),\n",
      " 'rbieti_homolog_ensembl_gene': 'Black snub-nosed monkey gene stable ID' (default: False),\n",
      " 'rbieti_homolog_ensembl_peptide': 'Black snub-nosed monkey protein or transcript stable ID' (default: False),\n",
      " 'rbieti_homolog_goc_score': 'Black snub-nosed monkey Gene-order conservation score' (default: False),\n",
      " 'rbieti_homolog_orthology_confidence': 'Black snub-nosed monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rbieti_homolog_orthology_type': 'Black snub-nosed monkey homology type' (default: False),\n",
      " 'rbieti_homolog_perc_id': '%id. target Black snub-nosed monkey gene identical to query gene' (default: False),\n",
      " 'rbieti_homolog_perc_id_r1': '%id. query gene identical to target Black snub-nosed monkey gene' (default: False),\n",
      " 'rbieti_homolog_subtype': 'Last common ancestor with Black snub-nosed monkey' (default: False),\n",
      " 'rbieti_homolog_wga_coverage': 'Black snub-nosed monkey Whole-genome alignment coverage' (default: False),\n",
      " 'reactome': 'Reactome ID' (default: False),\n",
      " 'reactome_gene': 'Reactome gene ID' (default: False),\n",
      " 'reactome_transcript': 'Reactome transcript ID' (default: False),\n",
      " 'refseq_mrna': 'RefSeq mRNA ID' (default: False),\n",
      " 'refseq_mrna_predicted': 'RefSeq mRNA predicted ID' (default: False),\n",
      " 'refseq_ncrna': 'RefSeq ncRNA ID' (default: False),\n",
      " 'refseq_ncrna_predicted': 'RefSeq ncRNA predicted ID' (default: False),\n",
      " 'refseq_peptide': 'RefSeq peptide ID' (default: False),\n",
      " 'refseq_peptide_predicted': 'RefSeq peptide predicted ID' (default: False),\n",
      " 'rfam': 'RFAM ID' (default: False),\n",
      " 'rfam_trans_name': 'RFAM transcript name ID' (default: False),\n",
      " 'rferrumequinum_homolog_associated_gene_name': 'Greater horseshoe bat gene name' (default: False),\n",
      " 'rferrumequinum_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rferrumequinum_homolog_chrom_end': 'Greater horseshoe bat chromosome/scaffold end (bp)' (default: False),\n",
      " 'rferrumequinum_homolog_chrom_start': 'Greater horseshoe bat chromosome/scaffold start (bp)' (default: False),\n",
      " 'rferrumequinum_homolog_chromosome': 'Greater horseshoe bat chromosome/scaffold name' (default: False),\n",
      " 'rferrumequinum_homolog_ensembl_gene': 'Greater horseshoe bat gene stable ID' (default: False),\n",
      " 'rferrumequinum_homolog_ensembl_peptide': 'Greater horseshoe bat protein or transcript stable ID' (default: False),\n",
      " 'rferrumequinum_homolog_goc_score': 'Greater horseshoe bat Gene-order conservation score' (default: False),\n",
      " 'rferrumequinum_homolog_orthology_confidence': 'Greater horseshoe bat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rferrumequinum_homolog_orthology_type': 'Greater horseshoe bat homology type' (default: False),\n",
      " 'rferrumequinum_homolog_perc_id': '%id. target Greater horseshoe bat gene identical to query gene' (default: False),\n",
      " 'rferrumequinum_homolog_perc_id_r1': '%id. query gene identical to target Greater horseshoe bat gene' (default: False),\n",
      " 'rferrumequinum_homolog_subtype': 'Last common ancestor with Greater horseshoe bat' (default: False),\n",
      " 'rferrumequinum_homolog_wga_coverage': 'Greater horseshoe bat Whole-genome alignment coverage' (default: False),\n",
      " 'rnacentral': 'RNAcentral ID' (default: False),\n",
      " 'rnorvegicus_homolog_associated_gene_name': 'Rat gene name' (default: False),\n",
      " 'rnorvegicus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rnorvegicus_homolog_chrom_end': 'Rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'rnorvegicus_homolog_chrom_start': 'Rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'rnorvegicus_homolog_chromosome': 'Rat chromosome/scaffold name' (default: False),\n",
      " 'rnorvegicus_homolog_ensembl_gene': 'Rat gene stable ID' (default: False),\n",
      " 'rnorvegicus_homolog_ensembl_peptide': 'Rat protein or transcript stable ID' (default: False),\n",
      " 'rnorvegicus_homolog_goc_score': 'Rat Gene-order conservation score' (default: False),\n",
      " 'rnorvegicus_homolog_orthology_confidence': 'Rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rnorvegicus_homolog_orthology_type': 'Rat homology type' (default: False),\n",
      " 'rnorvegicus_homolog_perc_id': '%id. target Rat gene identical to query gene' (default: False),\n",
      " 'rnorvegicus_homolog_perc_id_r1': '%id. query gene identical to target Rat gene' (default: False),\n",
      " 'rnorvegicus_homolog_subtype': 'Last common ancestor with Rat' (default: False),\n",
      " 'rnorvegicus_homolog_wga_coverage': 'Rat Whole-genome alignment coverage' (default: False),\n",
      " 'rroxellana_homolog_associated_gene_name': 'Golden snub-nosed monkey gene name' (default: False),\n",
      " 'rroxellana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rroxellana_homolog_chrom_end': 'Golden snub-nosed monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'rroxellana_homolog_chrom_start': 'Golden snub-nosed monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'rroxellana_homolog_chromosome': 'Golden snub-nosed monkey chromosome/scaffold name' (default: False),\n",
      " 'rroxellana_homolog_ensembl_gene': 'Golden snub-nosed monkey gene stable ID' (default: False),\n",
      " 'rroxellana_homolog_ensembl_peptide': 'Golden snub-nosed monkey protein or transcript stable ID' (default: False),\n",
      " 'rroxellana_homolog_goc_score': 'Golden snub-nosed monkey Gene-order conservation score' (default: False),\n",
      " 'rroxellana_homolog_orthology_confidence': 'Golden snub-nosed monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rroxellana_homolog_orthology_type': 'Golden snub-nosed monkey homology type' (default: False),\n",
      " 'rroxellana_homolog_perc_id': '%id. target Golden snub-nosed monkey gene identical to query gene' (default: False),\n",
      " 'rroxellana_homolog_perc_id_r1': '%id. query gene identical to target Golden snub-nosed monkey gene' (default: False),\n",
      " 'rroxellana_homolog_subtype': 'Last common ancestor with Golden snub-nosed monkey' (default: False),\n",
      " 'rroxellana_homolog_wga_coverage': 'Golden snub-nosed monkey Whole-genome alignment coverage' (default: False),\n",
      " 'saraneus_homolog_associated_gene_name': 'Shrew gene name' (default: False),\n",
      " 'saraneus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'saraneus_homolog_chrom_end': 'Shrew chromosome/scaffold end (bp)' (default: False),\n",
      " 'saraneus_homolog_chrom_start': 'Shrew chromosome/scaffold start (bp)' (default: False),\n",
      " 'saraneus_homolog_chromosome': 'Shrew chromosome/scaffold name' (default: False),\n",
      " 'saraneus_homolog_ensembl_gene': 'Shrew gene stable ID' (default: False),\n",
      " 'saraneus_homolog_ensembl_peptide': 'Shrew protein or transcript stable ID' (default: False),\n",
      " 'saraneus_homolog_goc_score': 'Shrew Gene-order conservation score' (default: False),\n",
      " 'saraneus_homolog_orthology_confidence': 'Shrew orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'saraneus_homolog_orthology_type': 'Shrew homology type' (default: False),\n",
      " 'saraneus_homolog_perc_id': '%id. target Shrew gene identical to query gene' (default: False),\n",
      " 'saraneus_homolog_perc_id_r1': '%id. query gene identical to target Shrew gene' (default: False),\n",
      " 'saraneus_homolog_subtype': 'Last common ancestor with Shrew' (default: False),\n",
      " 'saraneus_homolog_wga_coverage': 'Shrew Whole-genome alignment coverage' (default: False),\n",
      " 'saurata_homolog_associated_gene_name': 'Gilthead seabream gene name' (default: False),\n",
      " 'saurata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'saurata_homolog_chrom_end': 'Gilthead seabream chromosome/scaffold end (bp)' (default: False),\n",
      " 'saurata_homolog_chrom_start': 'Gilthead seabream chromosome/scaffold start (bp)' (default: False),\n",
      " 'saurata_homolog_chromosome': 'Gilthead seabream chromosome/scaffold name' (default: False),\n",
      " 'saurata_homolog_ensembl_gene': 'Gilthead seabream gene stable ID' (default: False),\n",
      " 'saurata_homolog_ensembl_peptide': 'Gilthead seabream protein or transcript stable ID' (default: False),\n",
      " 'saurata_homolog_goc_score': 'Gilthead seabream Gene-order conservation score' (default: False),\n",
      " 'saurata_homolog_orthology_confidence': 'Gilthead seabream orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'saurata_homolog_orthology_type': 'Gilthead seabream homology type' (default: False),\n",
      " 'saurata_homolog_perc_id': '%id. target Gilthead seabream gene identical to query gene' (default: False),\n",
      " 'saurata_homolog_perc_id_r1': '%id. query gene identical to target Gilthead seabream gene' (default: False),\n",
      " 'saurata_homolog_subtype': 'Last common ancestor with Gilthead seabream' (default: False),\n",
      " 'saurata_homolog_wga_coverage': 'Gilthead seabream Whole-genome alignment coverage' (default: False),\n",
      " 'sbboliviensis_homolog_associated_gene_name': 'Bolivian squirrel monkey gene name' (default: False),\n",
      " 'sbboliviensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sbboliviensis_homolog_chrom_end': 'Bolivian squirrel monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'sbboliviensis_homolog_chrom_start': 'Bolivian squirrel monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'sbboliviensis_homolog_chromosome': 'Bolivian squirrel monkey chromosome/scaffold name' (default: False),\n",
      " 'sbboliviensis_homolog_ensembl_gene': 'Bolivian squirrel monkey gene stable ID' (default: False),\n",
      " 'sbboliviensis_homolog_ensembl_peptide': 'Bolivian squirrel monkey protein or transcript stable ID' (default: False),\n",
      " 'sbboliviensis_homolog_goc_score': 'Bolivian squirrel monkey Gene-order conservation score' (default: False),\n",
      " 'sbboliviensis_homolog_orthology_confidence': 'Bolivian squirrel monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sbboliviensis_homolog_orthology_type': 'Bolivian squirrel monkey homology type' (default: False),\n",
      " 'sbboliviensis_homolog_perc_id': '%id. target Bolivian squirrel monkey gene identical to query gene' (default: False),\n",
      " 'sbboliviensis_homolog_perc_id_r1': '%id. query gene identical to target Bolivian squirrel monkey gene' (default: False),\n",
      " 'sbboliviensis_homolog_subtype': 'Last common ancestor with Bolivian squirrel monkey' (default: False),\n",
      " 'sbboliviensis_homolog_wga_coverage': 'Bolivian squirrel monkey Whole-genome alignment coverage' (default: False),\n",
      " 'scanaria_homolog_associated_gene_name': 'Common canary gene name' (default: False),\n",
      " 'scanaria_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scanaria_homolog_chrom_end': 'Common canary chromosome/scaffold end (bp)' (default: False),\n",
      " 'scanaria_homolog_chrom_start': 'Common canary chromosome/scaffold start (bp)' (default: False),\n",
      " 'scanaria_homolog_chromosome': 'Common canary chromosome/scaffold name' (default: False),\n",
      " 'scanaria_homolog_ensembl_gene': 'Common canary gene stable ID' (default: False),\n",
      " 'scanaria_homolog_ensembl_peptide': 'Common canary protein or transcript stable ID' (default: False),\n",
      " 'scanaria_homolog_goc_score': 'Common canary Gene-order conservation score' (default: False),\n",
      " 'scanaria_homolog_orthology_confidence': 'Common canary orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scanaria_homolog_orthology_type': 'Common canary homology type' (default: False),\n",
      " 'scanaria_homolog_perc_id': '%id. target Common canary gene identical to query gene' (default: False),\n",
      " 'scanaria_homolog_perc_id_r1': '%id. query gene identical to target Common canary gene' (default: False),\n",
      " 'scanaria_homolog_subtype': 'Last common ancestor with Common canary' (default: False),\n",
      " 'scanaria_homolog_wga_coverage': 'Common canary Whole-genome alignment coverage' (default: False),\n",
      " 'scanprosite': 'PROSITE patterns ID' (default: False),\n",
      " 'scanprosite_end': 'PROSITE patterns end' (default: False),\n",
      " 'scanprosite_start': 'PROSITE patterns start' (default: False),\n",
      " 'scaustralis_homolog_associated_gene_name': 'African ostrich gene name' (default: False),\n",
      " 'scaustralis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scaustralis_homolog_chrom_end': 'African ostrich chromosome/scaffold end (bp)' (default: False),\n",
      " 'scaustralis_homolog_chrom_start': 'African ostrich chromosome/scaffold start (bp)' (default: False),\n",
      " 'scaustralis_homolog_chromosome': 'African ostrich chromosome/scaffold name' (default: False),\n",
      " 'scaustralis_homolog_ensembl_gene': 'African ostrich gene stable ID' (default: False),\n",
      " 'scaustralis_homolog_ensembl_peptide': 'African ostrich protein or transcript stable ID' (default: False),\n",
      " 'scaustralis_homolog_goc_score': 'African ostrich Gene-order conservation score' (default: False),\n",
      " 'scaustralis_homolog_orthology_confidence': 'African ostrich orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scaustralis_homolog_orthology_type': 'African ostrich homology type' (default: False),\n",
      " 'scaustralis_homolog_perc_id': '%id. target African ostrich gene identical to query gene' (default: False),\n",
      " 'scaustralis_homolog_perc_id_r1': '%id. query gene identical to target African ostrich gene' (default: False),\n",
      " 'scaustralis_homolog_subtype': 'Last common ancestor with African ostrich' (default: False),\n",
      " 'scaustralis_homolog_wga_coverage': 'African ostrich Whole-genome alignment coverage' (default: False),\n",
      " 'scerevisiae_homolog_associated_gene_name': 'Saccharomyces cerevisiae gene name' (default: False),\n",
      " 'scerevisiae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scerevisiae_homolog_chrom_end': 'Saccharomyces cerevisiae chromosome/scaffold end (bp)' (default: False),\n",
      " 'scerevisiae_homolog_chrom_start': 'Saccharomyces cerevisiae chromosome/scaffold start (bp)' (default: False),\n",
      " 'scerevisiae_homolog_chromosome': 'Saccharomyces cerevisiae chromosome/scaffold name' (default: False),\n",
      " 'scerevisiae_homolog_ensembl_gene': 'Saccharomyces cerevisiae gene stable ID' (default: False),\n",
      " 'scerevisiae_homolog_ensembl_peptide': 'Saccharomyces cerevisiae protein or transcript stable ID' (default: False),\n",
      " 'scerevisiae_homolog_orthology_confidence': 'Saccharomyces cerevisiae orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scerevisiae_homolog_orthology_type': 'Saccharomyces cerevisiae homology type' (default: False),\n",
      " 'scerevisiae_homolog_perc_id': '%id. target Saccharomyces cerevisiae gene identical to query gene' (default: False),\n",
      " 'scerevisiae_homolog_perc_id_r1': '%id. query gene identical to target Saccharomyces cerevisiae gene' (default: False),\n",
      " 'scerevisiae_homolog_subtype': 'Last common ancestor with Saccharomyces cerevisiae' (default: False),\n",
      " 'sdumerili_homolog_associated_gene_name': 'Greater amberjack gene name' (default: False),\n",
      " 'sdumerili_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sdumerili_homolog_chrom_end': 'Greater amberjack chromosome/scaffold end (bp)' (default: False),\n",
      " 'sdumerili_homolog_chrom_start': 'Greater amberjack chromosome/scaffold start (bp)' (default: False),\n",
      " 'sdumerili_homolog_chromosome': 'Greater amberjack chromosome/scaffold name' (default: False),\n",
      " 'sdumerili_homolog_ensembl_gene': 'Greater amberjack gene stable ID' (default: False),\n",
      " 'sdumerili_homolog_ensembl_peptide': 'Greater amberjack protein or transcript stable ID' (default: False),\n",
      " 'sdumerili_homolog_goc_score': 'Greater amberjack Gene-order conservation score' (default: False),\n",
      " 'sdumerili_homolog_orthology_confidence': 'Greater amberjack orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sdumerili_homolog_orthology_type': 'Greater amberjack homology type' (default: False),\n",
      " 'sdumerili_homolog_perc_id': '%id. target Greater amberjack gene identical to query gene' (default: False),\n",
      " 'sdumerili_homolog_perc_id_r1': '%id. query gene identical to target Greater amberjack gene' (default: False),\n",
      " 'sdumerili_homolog_subtype': 'Last common ancestor with Greater amberjack' (default: False),\n",
      " 'sdumerili_homolog_wga_coverage': 'Greater amberjack Whole-genome alignment coverage' (default: False),\n",
      " 'seg': 'Low complexity (Seg)' (default: False),\n",
      " 'seg_end': 'Low complexity (Seg) end' (default: False),\n",
      " 'seg_start': 'Low complexity (Seg) start' (default: False),\n",
      " 'sfld': 'SFLD ID' (default: False),\n",
      " 'sfld_end': 'SFLD end' (default: False),\n",
      " 'sfld_start': 'SFLD start' (default: False),\n",
      " 'sformosus_homolog_associated_gene_name': 'Asian bonytongue gene name' (default: False),\n",
      " 'sformosus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sformosus_homolog_chrom_end': 'Asian bonytongue chromosome/scaffold end (bp)' (default: False),\n",
      " 'sformosus_homolog_chrom_start': 'Asian bonytongue chromosome/scaffold start (bp)' (default: False),\n",
      " 'sformosus_homolog_chromosome': 'Asian bonytongue chromosome/scaffold name' (default: False),\n",
      " 'sformosus_homolog_ensembl_gene': 'Asian bonytongue gene stable ID' (default: False),\n",
      " 'sformosus_homolog_ensembl_peptide': 'Asian bonytongue protein or transcript stable ID' (default: False),\n",
      " 'sformosus_homolog_goc_score': 'Asian bonytongue Gene-order conservation score' (default: False),\n",
      " 'sformosus_homolog_orthology_confidence': 'Asian bonytongue orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sformosus_homolog_orthology_type': 'Asian bonytongue homology type' (default: False),\n",
      " 'sformosus_homolog_perc_id': '%id. target Asian bonytongue gene identical to query gene' (default: False),\n",
      " 'sformosus_homolog_perc_id_r1': '%id. query gene identical to target Asian bonytongue gene' (default: False),\n",
      " 'sformosus_homolog_subtype': 'Last common ancestor with Asian bonytongue' (default: False),\n",
      " 'sformosus_homolog_wga_coverage': 'Asian bonytongue Whole-genome alignment coverage' (default: False),\n",
      " 'sgrahami_homolog_associated_gene_name': 'Golden-line barbel gene name' (default: False),\n",
      " 'sgrahami_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sgrahami_homolog_chrom_end': 'Golden-line barbel chromosome/scaffold end (bp)' (default: False),\n",
      " 'sgrahami_homolog_chrom_start': 'Golden-line barbel chromosome/scaffold start (bp)' (default: False),\n",
      " 'sgrahami_homolog_chromosome': 'Golden-line barbel chromosome/scaffold name' (default: False),\n",
      " 'sgrahami_homolog_ensembl_gene': 'Golden-line barbel gene stable ID' (default: False),\n",
      " 'sgrahami_homolog_ensembl_peptide': 'Golden-line barbel protein or transcript stable ID' (default: False),\n",
      " 'sgrahami_homolog_goc_score': 'Golden-line barbel Gene-order conservation score' (default: False),\n",
      " 'sgrahami_homolog_orthology_confidence': 'Golden-line barbel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sgrahami_homolog_orthology_type': 'Golden-line barbel homology type' (default: False),\n",
      " 'sgrahami_homolog_perc_id': '%id. target Golden-line barbel gene identical to query gene' (default: False),\n",
      " 'sgrahami_homolog_perc_id_r1': '%id. query gene identical to target Golden-line barbel gene' (default: False),\n",
      " 'sgrahami_homolog_subtype': 'Last common ancestor with Golden-line barbel' (default: False),\n",
      " 'sgrahami_homolog_wga_coverage': 'Golden-line barbel Whole-genome alignment coverage' (default: False),\n",
      " 'shabroptila_homolog_associated_gene_name': 'Kakapo gene name' (default: False),\n",
      " 'shabroptila_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'shabroptila_homolog_chrom_end': 'Kakapo chromosome/scaffold end (bp)' (default: False),\n",
      " 'shabroptila_homolog_chrom_start': 'Kakapo chromosome/scaffold start (bp)' (default: False),\n",
      " 'shabroptila_homolog_chromosome': 'Kakapo chromosome/scaffold name' (default: False),\n",
      " 'shabroptila_homolog_ensembl_gene': 'Kakapo gene stable ID' (default: False),\n",
      " 'shabroptila_homolog_ensembl_peptide': 'Kakapo protein or transcript stable ID' (default: False),\n",
      " 'shabroptila_homolog_goc_score': 'Kakapo Gene-order conservation score' (default: False),\n",
      " 'shabroptila_homolog_orthology_confidence': 'Kakapo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'shabroptila_homolog_orthology_type': 'Kakapo homology type' (default: False),\n",
      " 'shabroptila_homolog_perc_id': '%id. target Kakapo gene identical to query gene' (default: False),\n",
      " 'shabroptila_homolog_perc_id_r1': '%id. query gene identical to target Kakapo gene' (default: False),\n",
      " 'shabroptila_homolog_subtype': 'Last common ancestor with Kakapo' (default: False),\n",
      " 'shabroptila_homolog_wga_coverage': 'Kakapo Whole-genome alignment coverage' (default: False),\n",
      " 'sharrisii_homolog_associated_gene_name': 'Tasmanian devil gene name' (default: False),\n",
      " 'sharrisii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sharrisii_homolog_chrom_end': 'Tasmanian devil chromosome/scaffold end (bp)' (default: False),\n",
      " 'sharrisii_homolog_chrom_start': 'Tasmanian devil chromosome/scaffold start (bp)' (default: False),\n",
      " 'sharrisii_homolog_chromosome': 'Tasmanian devil chromosome/scaffold name' (default: False),\n",
      " 'sharrisii_homolog_ensembl_gene': 'Tasmanian devil gene stable ID' (default: False),\n",
      " 'sharrisii_homolog_ensembl_peptide': 'Tasmanian devil protein or transcript stable ID' (default: False),\n",
      " 'sharrisii_homolog_goc_score': 'Tasmanian devil Gene-order conservation score' (default: False),\n",
      " 'sharrisii_homolog_orthology_confidence': 'Tasmanian devil orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sharrisii_homolog_orthology_type': 'Tasmanian devil homology type' (default: False),\n",
      " 'sharrisii_homolog_perc_id': '%id. target Tasmanian devil gene identical to query gene' (default: False),\n",
      " 'sharrisii_homolog_perc_id_r1': '%id. query gene identical to target Tasmanian devil gene' (default: False),\n",
      " 'sharrisii_homolog_subtype': 'Last common ancestor with Tasmanian devil' (default: False),\n",
      " 'sift_prediction_2076': 'SIFT prediction' (default: False),\n",
      " 'sift_score_2076': 'SIFT score' (default: False),\n",
      " 'sifts_import': 'PDB-ENSP mappings' (default: False),\n",
      " 'sifts_import_end': 'PDB-ENSP mappings end' (default: False),\n",
      " 'sifts_import_start': 'PDB-ENSP mappings start' (default: False),\n",
      " 'signalp': 'Cleavage site (Signalp)' (default: False),\n",
      " 'signalp_end': 'Cleavage site (Signalp) end' (default: False),\n",
      " 'signalp_start': 'Cleavage site (Signalp) start' (default: False),\n",
      " 'sldorsalis_homolog_associated_gene_name': 'Yellowtail amberjack gene name' (default: False),\n",
      " 'sldorsalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sldorsalis_homolog_chrom_end': 'Yellowtail amberjack chromosome/scaffold end (bp)' (default: False),\n",
      " 'sldorsalis_homolog_chrom_start': 'Yellowtail amberjack chromosome/scaffold start (bp)' (default: False),\n",
      " 'sldorsalis_homolog_chromosome': 'Yellowtail amberjack chromosome/scaffold name' (default: False),\n",
      " 'sldorsalis_homolog_ensembl_gene': 'Yellowtail amberjack gene stable ID' (default: False),\n",
      " 'sldorsalis_homolog_ensembl_peptide': 'Yellowtail amberjack protein or transcript stable ID' (default: False),\n",
      " 'sldorsalis_homolog_goc_score': 'Yellowtail amberjack Gene-order conservation score' (default: False),\n",
      " 'sldorsalis_homolog_orthology_confidence': 'Yellowtail amberjack orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sldorsalis_homolog_orthology_type': 'Yellowtail amberjack homology type' (default: False),\n",
      " 'sldorsalis_homolog_perc_id': '%id. target Yellowtail amberjack gene identical to query gene' (default: False),\n",
      " 'sldorsalis_homolog_perc_id_r1': '%id. query gene identical to target Yellowtail amberjack gene' (default: False),\n",
      " 'sldorsalis_homolog_subtype': 'Last common ancestor with Yellowtail amberjack' (default: False),\n",
      " 'sldorsalis_homolog_wga_coverage': 'Yellowtail amberjack Whole-genome alignment coverage' (default: False),\n",
      " 'slucioperca_homolog_associated_gene_name': 'Pike-perch gene name' (default: False),\n",
      " 'slucioperca_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'slucioperca_homolog_chrom_end': 'Pike-perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'slucioperca_homolog_chrom_start': 'Pike-perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'slucioperca_homolog_chromosome': 'Pike-perch chromosome/scaffold name' (default: False),\n",
      " 'slucioperca_homolog_ensembl_gene': 'Pike-perch gene stable ID' (default: False),\n",
      " 'slucioperca_homolog_ensembl_peptide': 'Pike-perch protein or transcript stable ID' (default: False),\n",
      " 'slucioperca_homolog_goc_score': 'Pike-perch Gene-order conservation score' (default: False),\n",
      " 'slucioperca_homolog_orthology_confidence': 'Pike-perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'slucioperca_homolog_orthology_type': 'Pike-perch homology type' (default: False),\n",
      " 'slucioperca_homolog_perc_id': '%id. target Pike-perch gene identical to query gene' (default: False),\n",
      " 'slucioperca_homolog_perc_id_r1': '%id. query gene identical to target Pike-perch gene' (default: False),\n",
      " 'slucioperca_homolog_subtype': 'Last common ancestor with Pike-perch' (default: False),\n",
      " 'slucioperca_homolog_wga_coverage': 'Pike-perch Whole-genome alignment coverage' (default: False),\n",
      " 'smart': 'SMART ID' (default: False),\n",
      " 'smart_end': 'SMART end' (default: False),\n",
      " 'smart_start': 'SMART start' (default: False),\n",
      " 'smaximus_homolog_associated_gene_name': 'Turbot gene name' (default: False),\n",
      " 'smaximus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'smaximus_homolog_chrom_end': 'Turbot chromosome/scaffold end (bp)' (default: False),\n",
      " 'smaximus_homolog_chrom_start': 'Turbot chromosome/scaffold start (bp)' (default: False),\n",
      " 'smaximus_homolog_chromosome': 'Turbot chromosome/scaffold name' (default: False),\n",
      " 'smaximus_homolog_ensembl_gene': 'Turbot gene stable ID' (default: False),\n",
      " 'smaximus_homolog_ensembl_peptide': 'Turbot protein or transcript stable ID' (default: False),\n",
      " 'smaximus_homolog_goc_score': 'Turbot Gene-order conservation score' (default: False),\n",
      " 'smaximus_homolog_orthology_confidence': 'Turbot orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'smaximus_homolog_orthology_type': 'Turbot homology type' (default: False),\n",
      " 'smaximus_homolog_perc_id': '%id. target Turbot gene identical to query gene' (default: False),\n",
      " 'smaximus_homolog_perc_id_r1': '%id. query gene identical to target Turbot gene' (default: False),\n",
      " 'smaximus_homolog_subtype': 'Last common ancestor with Turbot' (default: False),\n",
      " 'smerianae_homolog_associated_gene_name': 'Argentine black and white tegu gene name' (default: False),\n",
      " 'smerianae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'smerianae_homolog_chrom_end': 'Argentine black and white tegu chromosome/scaffold end (bp)' (default: False),\n",
      " 'smerianae_homolog_chrom_start': 'Argentine black and white tegu chromosome/scaffold start (bp)' (default: False),\n",
      " 'smerianae_homolog_chromosome': 'Argentine black and white tegu chromosome/scaffold name' (default: False),\n",
      " 'smerianae_homolog_ensembl_gene': 'Argentine black and white tegu gene stable ID' (default: False),\n",
      " 'smerianae_homolog_ensembl_peptide': 'Argentine black and white tegu protein or transcript stable ID' (default: False),\n",
      " 'smerianae_homolog_goc_score': 'Argentine black and white tegu Gene-order conservation score' (default: False),\n",
      " 'smerianae_homolog_orthology_confidence': 'Argentine black and white tegu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'smerianae_homolog_orthology_type': 'Argentine black and white tegu homology type' (default: False),\n",
      " 'smerianae_homolog_perc_id': '%id. target Argentine black and white tegu gene identical to query gene' (default: False),\n",
      " 'smerianae_homolog_perc_id_r1': '%id. query gene identical to target Argentine black and white tegu gene' (default: False),\n",
      " 'smerianae_homolog_subtype': 'Last common ancestor with Argentine black and white tegu' (default: False),\n",
      " 'smerianae_homolog_wga_coverage': 'Argentine black and white tegu Whole-genome alignment coverage' (default: False),\n",
      " 'snp_chromosome_strand': 'Variant chromosome Strand' (default: False),\n",
      " 'somatic_allele': 'Variant alleles' (default: False),\n",
      " 'somatic_cds_end_2076': 'CDS end' (default: False),\n",
      " 'somatic_cds_start_2076': 'CDS start' (default: False),\n",
      " 'somatic_chromosome_end': 'Chromosome/scaffold position end (bp)' (default: False),\n",
      " 'somatic_chromosome_start': 'Chromosome/scaffold position start (bp)' (default: False),\n",
      " 'somatic_mapweight': 'Mapweight' (default: False),\n",
      " 'somatic_peptide_location': 'Protein location (aa)' (default: False),\n",
      " 'somatic_snp_chromosome_strand': 'Variant chromosome/scaffold strand' (default: False),\n",
      " 'somatic_source_description': 'Variant source description' (default: False),\n",
      " 'somatic_source_name': 'Variant source' (default: False),\n",
      " 'somatic_synonymous_status': 'Variant consequence' (default: False),\n",
      " 'somatic_transcript_location': 'Transcript location (bp)' (default: False),\n",
      " 'somatic_validated': 'Variant supporting evidence' (default: False),\n",
      " 'somatic_variation_name': 'Variant name' (default: True),\n",
      " 'source': 'Source (gene)' (default: False),\n",
      " 'source_description': 'Variant source description' (default: False),\n",
      " 'source_name': 'Source name' (default: False),\n",
      " 'spartitus_homolog_associated_gene_name': 'Bicolor damselfish gene name' (default: False),\n",
      " 'spartitus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'spartitus_homolog_chrom_end': 'Bicolor damselfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'spartitus_homolog_chrom_start': 'Bicolor damselfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'spartitus_homolog_chromosome': 'Bicolor damselfish chromosome/scaffold name' (default: False),\n",
      " 'spartitus_homolog_ensembl_gene': 'Bicolor damselfish gene stable ID' (default: False),\n",
      " 'spartitus_homolog_ensembl_peptide': 'Bicolor damselfish protein or transcript stable ID' (default: False),\n",
      " 'spartitus_homolog_goc_score': 'Bicolor damselfish Gene-order conservation score' (default: False),\n",
      " 'spartitus_homolog_orthology_confidence': 'Bicolor damselfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'spartitus_homolog_orthology_type': 'Bicolor damselfish homology type' (default: False),\n",
      " 'spartitus_homolog_perc_id': '%id. target Bicolor damselfish gene identical to query gene' (default: False),\n",
      " 'spartitus_homolog_perc_id_r1': '%id. query gene identical to target Bicolor damselfish gene' (default: False),\n",
      " 'spartitus_homolog_subtype': 'Last common ancestor with Bicolor damselfish' (default: False),\n",
      " 'spartitus_homolog_wga_coverage': 'Bicolor damselfish Whole-genome alignment coverage' (default: False),\n",
      " 'spunctatus_homolog_associated_gene_name': 'Tuatara gene name' (default: False),\n",
      " 'spunctatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'spunctatus_homolog_chrom_end': 'Tuatara chromosome/scaffold end (bp)' (default: False),\n",
      " 'spunctatus_homolog_chrom_start': 'Tuatara chromosome/scaffold start (bp)' (default: False),\n",
      " 'spunctatus_homolog_chromosome': 'Tuatara chromosome/scaffold name' (default: False),\n",
      " 'spunctatus_homolog_ensembl_gene': 'Tuatara gene stable ID' (default: False),\n",
      " 'spunctatus_homolog_ensembl_peptide': 'Tuatara protein or transcript stable ID' (default: False),\n",
      " 'spunctatus_homolog_goc_score': 'Tuatara Gene-order conservation score' (default: False),\n",
      " 'spunctatus_homolog_orthology_confidence': 'Tuatara orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'spunctatus_homolog_orthology_type': 'Tuatara homology type' (default: False),\n",
      " 'spunctatus_homolog_perc_id': '%id. target Tuatara gene identical to query gene' (default: False),\n",
      " 'spunctatus_homolog_perc_id_r1': '%id. query gene identical to target Tuatara gene' (default: False),\n",
      " 'spunctatus_homolog_subtype': 'Last common ancestor with Tuatara' (default: False),\n",
      " 'spunctatus_homolog_wga_coverage': 'Tuatara Whole-genome alignment coverage' (default: False),\n",
      " 'ssalar_homolog_associated_gene_name': 'Atlantic salmon gene name' (default: False),\n",
      " 'ssalar_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ssalar_homolog_chrom_end': 'Atlantic salmon chromosome/scaffold end (bp)' (default: False),\n",
      " 'ssalar_homolog_chrom_start': 'Atlantic salmon chromosome/scaffold start (bp)' (default: False),\n",
      " 'ssalar_homolog_chromosome': 'Atlantic salmon chromosome/scaffold name' (default: False),\n",
      " 'ssalar_homolog_ensembl_gene': 'Atlantic salmon gene stable ID' (default: False),\n",
      " 'ssalar_homolog_ensembl_peptide': 'Atlantic salmon protein or transcript stable ID' (default: False),\n",
      " 'ssalar_homolog_goc_score': 'Atlantic salmon Gene-order conservation score' (default: False),\n",
      " 'ssalar_homolog_orthology_confidence': 'Atlantic salmon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ssalar_homolog_orthology_type': 'Atlantic salmon homology type' (default: False),\n",
      " 'ssalar_homolog_perc_id': '%id. target Atlantic salmon gene identical to query gene' (default: False),\n",
      " 'ssalar_homolog_perc_id_r1': '%id. query gene identical to target Atlantic salmon gene' (default: False),\n",
      " 'ssalar_homolog_subtype': 'Last common ancestor with Atlantic salmon' (default: False),\n",
      " 'sscrofa_homolog_associated_gene_name': 'Pig gene name' (default: False),\n",
      " 'sscrofa_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sscrofa_homolog_chrom_end': 'Pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'sscrofa_homolog_chrom_start': 'Pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'sscrofa_homolog_chromosome': 'Pig chromosome/scaffold name' (default: False),\n",
      " 'sscrofa_homolog_ensembl_gene': 'Pig gene stable ID' (default: False),\n",
      " 'sscrofa_homolog_ensembl_peptide': 'Pig protein or transcript stable ID' (default: False),\n",
      " 'sscrofa_homolog_goc_score': 'Pig Gene-order conservation score' (default: False),\n",
      " 'sscrofa_homolog_orthology_confidence': 'Pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sscrofa_homolog_orthology_type': 'Pig homology type' (default: False),\n",
      " 'sscrofa_homolog_perc_id': '%id. target Pig gene identical to query gene' (default: False),\n",
      " 'sscrofa_homolog_perc_id_r1': '%id. query gene identical to target Pig gene' (default: False),\n",
      " 'sscrofa_homolog_subtype': 'Last common ancestor with Pig' (default: False),\n",
      " 'sscrofa_homolog_wga_coverage': 'Pig Whole-genome alignment coverage' (default: False),\n",
      " 'start_position': 'Gene start (bp)' (default: False),\n",
      " 'strain_gender': 'Strain gender' (default: False),\n",
      " 'strain_name': 'Strain name' (default: False),\n",
      " 'strand': 'Strand' (default: False),\n",
      " 'strutta_homolog_associated_gene_name': 'Brown trout gene name' (default: False),\n",
      " 'strutta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'strutta_homolog_chrom_end': 'Brown trout chromosome/scaffold end (bp)' (default: False),\n",
      " 'strutta_homolog_chrom_start': 'Brown trout chromosome/scaffold start (bp)' (default: False),\n",
      " 'strutta_homolog_chromosome': 'Brown trout chromosome/scaffold name' (default: False),\n",
      " 'strutta_homolog_ensembl_gene': 'Brown trout gene stable ID' (default: False),\n",
      " 'strutta_homolog_ensembl_peptide': 'Brown trout protein or transcript stable ID' (default: False),\n",
      " 'strutta_homolog_goc_score': 'Brown trout Gene-order conservation score' (default: False),\n",
      " 'strutta_homolog_orthology_confidence': 'Brown trout orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'strutta_homolog_orthology_type': 'Brown trout homology type' (default: False),\n",
      " 'strutta_homolog_perc_id': '%id. target Brown trout gene identical to query gene' (default: False),\n",
      " 'strutta_homolog_perc_id_r1': '%id. query gene identical to target Brown trout gene' (default: False),\n",
      " 'strutta_homolog_subtype': 'Last common ancestor with Brown trout' (default: False),\n",
      " 'strutta_homolog_wga_coverage': 'Brown trout Whole-genome alignment coverage' (default: False),\n",
      " 'study_external_id': 'Study external reference' (default: False),\n",
      " 'superfamily': 'Superfamily ID' (default: False),\n",
      " 'superfamily_end': 'Superfamily end' (default: False),\n",
      " 'superfamily_start': 'Superfamily start' (default: False),\n",
      " 'svulgaris_homolog_associated_gene_name': 'Eurasian red squirrel gene name' (default: False),\n",
      " 'svulgaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'svulgaris_homolog_chrom_end': 'Eurasian red squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'svulgaris_homolog_chrom_start': 'Eurasian red squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'svulgaris_homolog_chromosome': 'Eurasian red squirrel chromosome/scaffold name' (default: False),\n",
      " 'svulgaris_homolog_ensembl_gene': 'Eurasian red squirrel gene stable ID' (default: False),\n",
      " 'svulgaris_homolog_ensembl_peptide': 'Eurasian red squirrel protein or transcript stable ID' (default: False),\n",
      " 'svulgaris_homolog_goc_score': 'Eurasian red squirrel Gene-order conservation score' (default: False),\n",
      " 'svulgaris_homolog_orthology_confidence': 'Eurasian red squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'svulgaris_homolog_orthology_type': 'Eurasian red squirrel homology type' (default: False),\n",
      " 'svulgaris_homolog_perc_id': '%id. target Eurasian red squirrel gene identical to query gene' (default: False),\n",
      " 'svulgaris_homolog_perc_id_r1': '%id. query gene identical to target Eurasian red squirrel gene' (default: False),\n",
      " 'svulgaris_homolog_subtype': 'Last common ancestor with Eurasian red squirrel' (default: False),\n",
      " 'svulgaris_homolog_wga_coverage': 'Eurasian red squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'synonymous_status': 'Variant consequence' (default: False),\n",
      " 'tbelangeri_homolog_associated_gene_name': 'Tree Shrew gene name' (default: False),\n",
      " 'tbelangeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tbelangeri_homolog_chrom_end': 'Tree Shrew chromosome/scaffold end (bp)' (default: False),\n",
      " 'tbelangeri_homolog_chrom_start': 'Tree Shrew chromosome/scaffold start (bp)' (default: False),\n",
      " 'tbelangeri_homolog_chromosome': 'Tree Shrew chromosome/scaffold name' (default: False),\n",
      " 'tbelangeri_homolog_ensembl_gene': 'Tree Shrew gene stable ID' (default: False),\n",
      " 'tbelangeri_homolog_ensembl_peptide': 'Tree Shrew protein or transcript stable ID' (default: False),\n",
      " 'tbelangeri_homolog_goc_score': 'Tree Shrew Gene-order conservation score' (default: False),\n",
      " 'tbelangeri_homolog_orthology_confidence': 'Tree Shrew orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tbelangeri_homolog_orthology_type': 'Tree Shrew homology type' (default: False),\n",
      " 'tbelangeri_homolog_perc_id': '%id. target Tree Shrew gene identical to query gene' (default: False),\n",
      " 'tbelangeri_homolog_perc_id_r1': '%id. query gene identical to target Tree Shrew gene' (default: False),\n",
      " 'tbelangeri_homolog_subtype': 'Last common ancestor with Tree Shrew' (default: False),\n",
      " 'tbelangeri_homolog_wga_coverage': 'Tree Shrew Whole-genome alignment coverage' (default: False),\n",
      " 'tctriunguis_homolog_associated_gene_name': 'Three-toed box turtle gene name' (default: False),\n",
      " 'tctriunguis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tctriunguis_homolog_chrom_end': 'Three-toed box turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'tctriunguis_homolog_chrom_start': 'Three-toed box turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'tctriunguis_homolog_chromosome': 'Three-toed box turtle chromosome/scaffold name' (default: False),\n",
      " 'tctriunguis_homolog_ensembl_gene': 'Three-toed box turtle gene stable ID' (default: False),\n",
      " 'tctriunguis_homolog_ensembl_peptide': 'Three-toed box turtle protein or transcript stable ID' (default: False),\n",
      " 'tctriunguis_homolog_goc_score': 'Three-toed box turtle Gene-order conservation score' (default: False),\n",
      " 'tctriunguis_homolog_orthology_confidence': 'Three-toed box turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tctriunguis_homolog_orthology_type': 'Three-toed box turtle homology type' (default: False),\n",
      " 'tctriunguis_homolog_perc_id': '%id. target Three-toed box turtle gene identical to query gene' (default: False),\n",
      " 'tctriunguis_homolog_perc_id_r1': '%id. query gene identical to target Three-toed box turtle gene' (default: False),\n",
      " 'tctriunguis_homolog_subtype': 'Last common ancestor with Three-toed box turtle' (default: False),\n",
      " 'tctriunguis_homolog_wga_coverage': 'Three-toed box turtle Whole-genome alignment coverage' (default: False),\n",
      " 'tguttata_homolog_associated_gene_name': 'Zebra finch gene name' (default: False),\n",
      " 'tguttata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tguttata_homolog_chrom_end': 'Zebra finch chromosome/scaffold end (bp)' (default: False),\n",
      " 'tguttata_homolog_chrom_start': 'Zebra finch chromosome/scaffold start (bp)' (default: False),\n",
      " 'tguttata_homolog_chromosome': 'Zebra finch chromosome/scaffold name' (default: False),\n",
      " 'tguttata_homolog_ensembl_gene': 'Zebra finch gene stable ID' (default: False),\n",
      " 'tguttata_homolog_ensembl_peptide': 'Zebra finch protein or transcript stable ID' (default: False),\n",
      " 'tguttata_homolog_goc_score': 'Zebra finch Gene-order conservation score' (default: False),\n",
      " 'tguttata_homolog_orthology_confidence': 'Zebra finch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tguttata_homolog_orthology_type': 'Zebra finch homology type' (default: False),\n",
      " 'tguttata_homolog_perc_id': '%id. target Zebra finch gene identical to query gene' (default: False),\n",
      " 'tguttata_homolog_perc_id_r1': '%id. query gene identical to target Zebra finch gene' (default: False),\n",
      " 'tguttata_homolog_subtype': 'Last common ancestor with Zebra finch' (default: False),\n",
      " 'tguttata_homolog_wga_coverage': 'Zebra finch Whole-genome alignment coverage' (default: False),\n",
      " 'tigrfam': 'TIGRFAM ID' (default: False),\n",
      " 'tigrfam_end': 'TIGRFAM end' (default: False),\n",
      " 'tigrfam_start': 'TIGRFAM start' (default: False),\n",
      " 'tmhmm': 'Transmembrane helices' (default: False),\n",
      " 'tmhmm_end': 'Transmembrane helices end' (default: False),\n",
      " 'tmhmm_start': 'Transmembrane helices start' (default: False),\n",
      " 'tnigroviridis_homolog_associated_gene_name': 'Tetraodon gene name' (default: False),\n",
      " 'tnigroviridis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tnigroviridis_homolog_chrom_end': 'Tetraodon chromosome/scaffold end (bp)' (default: False),\n",
      " 'tnigroviridis_homolog_chrom_start': 'Tetraodon chromosome/scaffold start (bp)' (default: False),\n",
      " 'tnigroviridis_homolog_chromosome': 'Tetraodon chromosome/scaffold name' (default: False),\n",
      " 'tnigroviridis_homolog_ensembl_gene': 'Tetraodon gene stable ID' (default: False),\n",
      " 'tnigroviridis_homolog_ensembl_peptide': 'Tetraodon protein or transcript stable ID' (default: False),\n",
      " 'tnigroviridis_homolog_goc_score': 'Tetraodon Gene-order conservation score' (default: False),\n",
      " 'tnigroviridis_homolog_orthology_confidence': 'Tetraodon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tnigroviridis_homolog_orthology_type': 'Tetraodon homology type' (default: False),\n",
      " 'tnigroviridis_homolog_perc_id': '%id. target Tetraodon gene identical to query gene' (default: False),\n",
      " 'tnigroviridis_homolog_perc_id_r1': '%id. query gene identical to target Tetraodon gene' (default: False),\n",
      " 'tnigroviridis_homolog_subtype': 'Last common ancestor with Tetraodon' (default: False),\n",
      " 'tnigroviridis_homolog_wga_coverage': 'Tetraodon Whole-genome alignment coverage' (default: False),\n",
      " 'transcript_appris': 'APPRIS annotation' (default: False),\n",
      " 'transcript_biotype': 'Transcript type' (default: False),\n",
      " 'transcript_count': 'Transcript count' (default: False),\n",
      " 'transcript_end': 'Transcript end (bp)' (default: False),\n",
      " 'transcript_exon_intron': 'Unspliced (Transcript)' (default: False),\n",
      " 'transcript_flank': 'Flank (Transcript)' (default: False),\n",
      " 'transcript_gencode_basic': 'GENCODE basic annotation' (default: False),\n",
      " 'transcript_is_canonical': 'Ensembl Canonical' (default: False),\n",
      " 'transcript_length': 'Transcript length (including UTRs and CDS)' (default: False),\n",
      " 'transcript_location': 'Transcript location (bp)' (default: False),\n",
      " 'transcript_mane_plus_clinical': 'RefSeq match transcript (MANE Plus Clinical)' (default: False),\n",
      " 'transcript_mane_select': 'RefSeq match transcript (MANE Select)' (default: False),\n",
      " 'transcript_source': 'Source (transcript)' (default: False),\n",
      " 'transcript_start': 'Transcript start (bp)' (default: False),\n",
      " 'transcript_tsl': 'Transcript support level (TSL)' (default: False),\n",
      " 'transcript_version': 'Version (transcript)' (default: False),\n",
      " 'transcription_start_site': 'Transcription start site (TSS)' (default: False),\n",
      " 'trubripes_homolog_associated_gene_name': 'Fugu gene name' (default: False),\n",
      " 'trubripes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'trubripes_homolog_chrom_end': 'Fugu chromosome/scaffold end (bp)' (default: False),\n",
      " 'trubripes_homolog_chrom_start': 'Fugu chromosome/scaffold start (bp)' (default: False),\n",
      " 'trubripes_homolog_chromosome': 'Fugu chromosome/scaffold name' (default: False),\n",
      " 'trubripes_homolog_ensembl_gene': 'Fugu gene stable ID' (default: False),\n",
      " 'trubripes_homolog_ensembl_peptide': 'Fugu protein or transcript stable ID' (default: False),\n",
      " 'trubripes_homolog_goc_score': 'Fugu Gene-order conservation score' (default: False),\n",
      " 'trubripes_homolog_orthology_confidence': 'Fugu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'trubripes_homolog_orthology_type': 'Fugu homology type' (default: False),\n",
      " 'trubripes_homolog_perc_id': '%id. target Fugu gene identical to query gene' (default: False),\n",
      " 'trubripes_homolog_perc_id_r1': '%id. query gene identical to target Fugu gene' (default: False),\n",
      " 'trubripes_homolog_subtype': 'Last common ancestor with Fugu' (default: False),\n",
      " 'trubripes_homolog_wga_coverage': 'Fugu Whole-genome alignment coverage' (default: False),\n",
      " 'ttruncatus_homolog_associated_gene_name': 'Dolphin gene name' (default: False),\n",
      " 'ttruncatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ttruncatus_homolog_chrom_end': 'Dolphin chromosome/scaffold end (bp)' (default: False),\n",
      " 'ttruncatus_homolog_chrom_start': 'Dolphin chromosome/scaffold start (bp)' (default: False),\n",
      " 'ttruncatus_homolog_chromosome': 'Dolphin chromosome/scaffold name' (default: False),\n",
      " 'ttruncatus_homolog_ensembl_gene': 'Dolphin gene stable ID' (default: False),\n",
      " 'ttruncatus_homolog_ensembl_peptide': 'Dolphin protein or transcript stable ID' (default: False),\n",
      " 'ttruncatus_homolog_goc_score': 'Dolphin Gene-order conservation score' (default: False),\n",
      " 'ttruncatus_homolog_orthology_confidence': 'Dolphin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ttruncatus_homolog_orthology_type': 'Dolphin homology type' (default: False),\n",
      " 'ttruncatus_homolog_perc_id': '%id. target Dolphin gene identical to query gene' (default: False),\n",
      " 'ttruncatus_homolog_perc_id_r1': '%id. query gene identical to target Dolphin gene' (default: False),\n",
      " 'ttruncatus_homolog_subtype': 'Last common ancestor with Dolphin' (default: False),\n",
      " 'ttruncatus_homolog_wga_coverage': 'Dolphin Whole-genome alignment coverage' (default: False),\n",
      " 'uamericanus_homolog_associated_gene_name': 'American black bear gene name' (default: False),\n",
      " 'uamericanus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'uamericanus_homolog_chrom_end': 'American black bear chromosome/scaffold end (bp)' (default: False),\n",
      " 'uamericanus_homolog_chrom_start': 'American black bear chromosome/scaffold start (bp)' (default: False),\n",
      " 'uamericanus_homolog_chromosome': 'American black bear chromosome/scaffold name' (default: False),\n",
      " 'uamericanus_homolog_ensembl_gene': 'American black bear gene stable ID' (default: False),\n",
      " 'uamericanus_homolog_ensembl_peptide': 'American black bear protein or transcript stable ID' (default: False),\n",
      " 'uamericanus_homolog_goc_score': 'American black bear Gene-order conservation score' (default: False),\n",
      " 'uamericanus_homolog_orthology_confidence': 'American black bear orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'uamericanus_homolog_orthology_type': 'American black bear homology type' (default: False),\n",
      " 'uamericanus_homolog_perc_id': '%id. target American black bear gene identical to query gene' (default: False),\n",
      " 'uamericanus_homolog_perc_id_r1': '%id. query gene identical to target American black bear gene' (default: False),\n",
      " 'uamericanus_homolog_subtype': 'Last common ancestor with American black bear' (default: False),\n",
      " 'uamericanus_homolog_wga_coverage': 'American black bear Whole-genome alignment coverage' (default: False),\n",
      " 'ucsc': 'UCSC Stable ID' (default: False),\n",
      " 'umaritimus_homolog_associated_gene_name': 'Polar bear gene name' (default: False),\n",
      " 'umaritimus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'umaritimus_homolog_chrom_end': 'Polar bear chromosome/scaffold end (bp)' (default: False),\n",
      " 'umaritimus_homolog_chrom_start': 'Polar bear chromosome/scaffold start (bp)' (default: False),\n",
      " 'umaritimus_homolog_chromosome': 'Polar bear chromosome/scaffold name' (default: False),\n",
      " 'umaritimus_homolog_ensembl_gene': 'Polar bear gene stable ID' (default: False),\n",
      " 'umaritimus_homolog_ensembl_peptide': 'Polar bear protein or transcript stable ID' (default: False),\n",
      " 'umaritimus_homolog_goc_score': 'Polar bear Gene-order conservation score' (default: False),\n",
      " 'umaritimus_homolog_orthology_confidence': 'Polar bear orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'umaritimus_homolog_orthology_type': 'Polar bear homology type' (default: False),\n",
      " 'umaritimus_homolog_perc_id': '%id. target Polar bear gene identical to query gene' (default: False),\n",
      " 'umaritimus_homolog_perc_id_r1': '%id. query gene identical to target Polar bear gene' (default: False),\n",
      " 'umaritimus_homolog_subtype': 'Last common ancestor with Polar bear' (default: False),\n",
      " 'umaritimus_homolog_wga_coverage': 'Polar bear Whole-genome alignment coverage' (default: False),\n",
      " 'uniparc': 'UniParc ID' (default: False),\n",
      " 'uniprot_gn_id': 'UniProtKB Gene Name ID' (default: False),\n",
      " 'uniprot_gn_symbol': 'UniProtKB Gene Name symbol' (default: False),\n",
      " 'uniprot_isoform': 'UniProtKB isoform ID' (default: False),\n",
      " 'uniprotsptrembl': 'UniProtKB/TrEMBL ID' (default: False),\n",
      " 'uniprotswissprot': 'UniProtKB/Swiss-Prot ID' (default: False),\n",
      " 'uparryii_homolog_associated_gene_name': 'Arctic ground squirrel gene name' (default: False),\n",
      " 'uparryii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'uparryii_homolog_chrom_end': 'Arctic ground squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'uparryii_homolog_chrom_start': 'Arctic ground squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'uparryii_homolog_chromosome': 'Arctic ground squirrel chromosome/scaffold name' (default: False),\n",
      " 'uparryii_homolog_ensembl_gene': 'Arctic ground squirrel gene stable ID' (default: False),\n",
      " 'uparryii_homolog_ensembl_peptide': 'Arctic ground squirrel protein or transcript stable ID' (default: False),\n",
      " 'uparryii_homolog_goc_score': 'Arctic ground squirrel Gene-order conservation score' (default: False),\n",
      " 'uparryii_homolog_orthology_confidence': 'Arctic ground squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'uparryii_homolog_orthology_type': 'Arctic ground squirrel homology type' (default: False),\n",
      " 'uparryii_homolog_perc_id': '%id. target Arctic ground squirrel gene identical to query gene' (default: False),\n",
      " 'uparryii_homolog_perc_id_r1': '%id. query gene identical to target Arctic ground squirrel gene' (default: False),\n",
      " 'uparryii_homolog_subtype': 'Last common ancestor with Arctic ground squirrel' (default: False),\n",
      " 'uparryii_homolog_wga_coverage': 'Arctic ground squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'upstream_flank': 'upstream_flank' (default: False),\n",
      " 'validated': 'Variant supporting evidence' (default: False),\n",
      " 'variation_name': 'Variant name' (default: True),\n",
      " 'version': 'Version (gene)' (default: False),\n",
      " 'vpacos_homolog_associated_gene_name': 'Alpaca gene name' (default: False),\n",
      " 'vpacos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vpacos_homolog_chrom_end': 'Alpaca chromosome/scaffold end (bp)' (default: False),\n",
      " 'vpacos_homolog_chrom_start': 'Alpaca chromosome/scaffold start (bp)' (default: False),\n",
      " 'vpacos_homolog_chromosome': 'Alpaca chromosome/scaffold name' (default: False),\n",
      " 'vpacos_homolog_ensembl_gene': 'Alpaca gene stable ID' (default: False),\n",
      " 'vpacos_homolog_ensembl_peptide': 'Alpaca protein or transcript stable ID' (default: False),\n",
      " 'vpacos_homolog_goc_score': 'Alpaca Gene-order conservation score' (default: False),\n",
      " 'vpacos_homolog_orthology_confidence': 'Alpaca orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vpacos_homolog_orthology_type': 'Alpaca homology type' (default: False),\n",
      " 'vpacos_homolog_perc_id': '%id. target Alpaca gene identical to query gene' (default: False),\n",
      " 'vpacos_homolog_perc_id_r1': '%id. query gene identical to target Alpaca gene' (default: False),\n",
      " 'vpacos_homolog_subtype': 'Last common ancestor with Alpaca' (default: False),\n",
      " 'vpacos_homolog_wga_coverage': 'Alpaca Whole-genome alignment coverage' (default: False),\n",
      " 'vursinus_homolog_associated_gene_name': 'Common wombat gene name' (default: False),\n",
      " 'vursinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vursinus_homolog_chrom_end': 'Common wombat chromosome/scaffold end (bp)' (default: False),\n",
      " 'vursinus_homolog_chrom_start': 'Common wombat chromosome/scaffold start (bp)' (default: False),\n",
      " 'vursinus_homolog_chromosome': 'Common wombat chromosome/scaffold name' (default: False),\n",
      " 'vursinus_homolog_ensembl_gene': 'Common wombat gene stable ID' (default: False),\n",
      " 'vursinus_homolog_ensembl_peptide': 'Common wombat protein or transcript stable ID' (default: False),\n",
      " 'vursinus_homolog_goc_score': 'Common wombat Gene-order conservation score' (default: False),\n",
      " 'vursinus_homolog_orthology_confidence': 'Common wombat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vursinus_homolog_orthology_type': 'Common wombat homology type' (default: False),\n",
      " 'vursinus_homolog_perc_id': '%id. target Common wombat gene identical to query gene' (default: False),\n",
      " 'vursinus_homolog_perc_id_r1': '%id. query gene identical to target Common wombat gene' (default: False),\n",
      " 'vursinus_homolog_subtype': 'Last common ancestor with Common wombat' (default: False),\n",
      " 'vursinus_homolog_wga_coverage': 'Common wombat Whole-genome alignment coverage' (default: False),\n",
      " 'vvulpes_homolog_associated_gene_name': 'Red fox gene name' (default: False),\n",
      " 'vvulpes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vvulpes_homolog_chrom_end': 'Red fox chromosome/scaffold end (bp)' (default: False),\n",
      " 'vvulpes_homolog_chrom_start': 'Red fox chromosome/scaffold start (bp)' (default: False),\n",
      " 'vvulpes_homolog_chromosome': 'Red fox chromosome/scaffold name' (default: False),\n",
      " 'vvulpes_homolog_ensembl_gene': 'Red fox gene stable ID' (default: False),\n",
      " 'vvulpes_homolog_ensembl_peptide': 'Red fox protein or transcript stable ID' (default: False),\n",
      " 'vvulpes_homolog_goc_score': 'Red fox Gene-order conservation score' (default: False),\n",
      " 'vvulpes_homolog_orthology_confidence': 'Red fox orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vvulpes_homolog_orthology_type': 'Red fox homology type' (default: False),\n",
      " 'vvulpes_homolog_perc_id': '%id. target Red fox gene identical to query gene' (default: False),\n",
      " 'vvulpes_homolog_perc_id_r1': '%id. query gene identical to target Red fox gene' (default: False),\n",
      " 'vvulpes_homolog_subtype': 'Last common ancestor with Red fox' (default: False),\n",
      " 'vvulpes_homolog_wga_coverage': 'Red fox Whole-genome alignment coverage' (default: False),\n",
      " 'wikigene_description': 'WikiGene description' (default: False),\n",
      " 'wikigene_id': 'WikiGene ID' (default: False),\n",
      " 'wikigene_name': 'WikiGene name' (default: False),\n",
      " 'xmaculatus_homolog_associated_gene_name': 'Platyfish gene name' (default: False),\n",
      " 'xmaculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xmaculatus_homolog_chrom_end': 'Platyfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'xmaculatus_homolog_chrom_start': 'Platyfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'xmaculatus_homolog_chromosome': 'Platyfish chromosome/scaffold name' (default: False),\n",
      " 'xmaculatus_homolog_ensembl_gene': 'Platyfish gene stable ID' (default: False),\n",
      " 'xmaculatus_homolog_ensembl_peptide': 'Platyfish protein or transcript stable ID' (default: False),\n",
      " 'xmaculatus_homolog_goc_score': 'Platyfish Gene-order conservation score' (default: False),\n",
      " 'xmaculatus_homolog_orthology_confidence': 'Platyfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xmaculatus_homolog_orthology_type': 'Platyfish homology type' (default: False),\n",
      " 'xmaculatus_homolog_perc_id': '%id. target Platyfish gene identical to query gene' (default: False),\n",
      " 'xmaculatus_homolog_perc_id_r1': '%id. query gene identical to target Platyfish gene' (default: False),\n",
      " 'xmaculatus_homolog_subtype': 'Last common ancestor with Platyfish' (default: False),\n",
      " 'xmaculatus_homolog_wga_coverage': 'Platyfish Whole-genome alignment coverage' (default: False),\n",
      " 'xtropicalis_homolog_associated_gene_name': 'Tropical clawed frog gene name' (default: False),\n",
      " 'xtropicalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xtropicalis_homolog_chrom_end': 'Tropical clawed frog chromosome/scaffold end (bp)' (default: False),\n",
      " 'xtropicalis_homolog_chrom_start': 'Tropical clawed frog chromosome/scaffold start (bp)' (default: False),\n",
      " 'xtropicalis_homolog_chromosome': 'Tropical clawed frog chromosome/scaffold name' (default: False),\n",
      " 'xtropicalis_homolog_ensembl_gene': 'Tropical clawed frog gene stable ID' (default: False),\n",
      " 'xtropicalis_homolog_ensembl_peptide': 'Tropical clawed frog protein or transcript stable ID' (default: False),\n",
      " 'xtropicalis_homolog_goc_score': 'Tropical clawed frog Gene-order conservation score' (default: False),\n",
      " 'xtropicalis_homolog_orthology_confidence': 'Tropical clawed frog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xtropicalis_homolog_orthology_type': 'Tropical clawed frog homology type' (default: False),\n",
      " 'xtropicalis_homolog_perc_id': '%id. target Tropical clawed frog gene identical to query gene' (default: False),\n",
      " 'xtropicalis_homolog_perc_id_r1': '%id. query gene identical to target Tropical clawed frog gene' (default: False),\n",
      " 'xtropicalis_homolog_subtype': 'Last common ancestor with Tropical clawed frog' (default: False)}\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Searching using following params:\n",
      "{'attributes': ['ensembl_gene_id',\n",
      "                'hgnc_symbol',\n",
      "                'external_gene_name',\n",
      "                'chromosome_name',\n",
      "                'ensembl_transcript_id',\n",
      "                'ensembl_peptide_id',\n",
      "                'uniprotsptrembl',\n",
      "                'uniprotswissprot']}\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Fetching attributes\n",
      "[BiomartDataset] search query:\n",
      "b'<Query virtualSchemaName=\"default\" formatter=\"TSV\" header=\"1\" uniqueRows=\"1\" datasetConfigVersion=\"0.6\" count=\"\"><Dataset name=\"hsapiens_gene_ensembl\" interface=\"default\"><Attribute name=\"ensembl_gene_id\" /><Attribute name=\"hgnc_symbol\" /><Attribute name=\"external_gene_name\" /><Attribute name=\"chromosome_name\" /><Attribute name=\"ensembl_transcript_id\" /><Attribute name=\"ensembl_peptide_id\" /><Attribute name=\"uniprotsptrembl\" /><Attribute name=\"uniprotswissprot\" /></Dataset></Query>'\n"
     ]
    }
   ],
   "source": [
    "#Connect to a Biomart Server\n",
    "server = BiomartServer(\"http://useast.ensembl.org/biomart\")\n",
    "\n",
    "# set verbose to True to get some messages\n",
    "server.verbose = True\n",
    "\n",
    "# use the 'hsapiens_gene_ensembl' dataset\n",
    "hsapiens_gene_ensembl = server.datasets['hsapiens_gene_ensembl']\n",
    "\n",
    "# show all available filters and attributes of the 'hsapiens_gene_ensembl' dataset\n",
    "#hsapiens_gene_ensembl.show_filters()  # uses pprint\n",
    "hsapiens_gene_ensembl.show_attributes()  # uses print\n",
    "\n",
    "# run a search with custom filters and attributes (no header)\n",
    "response = hsapiens_gene_ensembl.search({\n",
    "  #'filters': {\n",
    "  #    'ensembl_gene_id': unique_gene_all,\n",
    "  #},\n",
    "  'attributes': [\n",
    "      'ensembl_gene_id', 'hgnc_symbol', 'external_gene_name', 'chromosome_name', 'ensembl_transcript_id', 'ensembl_peptide_id', 'uniprotsptrembl', 'uniprotswissprot'\n",
    "  ]\n",
    "}, header = 1)\n",
    "\n",
    "# i = -1\n",
    "# #response format is TSV\n",
    "# for line in response.iter_lines():\n",
    "#   i+=1\n",
    "#   line = line.decode('utf-8')\n",
    "#   #print(line.split(\"\\t\"))\n",
    "#   conv_frame.loc[i] = line.split(\"\\t\")\n",
    "\n",
    "if response.ok:\n",
    "    # Convert the response content to a file-like object\n",
    "    content = io.StringIO(response.content.decode('utf-8'))\n",
    "\n",
    "    # Read the content into a DataFrame\n",
    "    biomart = pd.read_csv(content, sep='\\t')\n",
    "else:\n",
    "    print(\"Error with the response\")"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T17:13:09.884585700Z",
     "start_time": "2023-12-13T17:09:44.001158600Z"
    }
   },
   "id": "5ded4dc0ba67e66b"
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "outputs": [
    {
     "data": {
      "text/plain": "    Gene stable ID HGNC symbol Gene name Chromosome/scaffold name  \\\n0  ENSG00000198888      MT-ND1    MT-ND1                       MT   \n1  ENSG00000198763      MT-ND2    MT-ND2                       MT   \n2  ENSG00000198804      MT-CO1    MT-CO1                       MT   \n3  ENSG00000198712      MT-CO2    MT-CO2                       MT   \n4  ENSG00000228253     MT-ATP8   MT-ATP8                       MT   \n\n  Transcript stable ID Protein stable ID UniProtKB/TrEMBL ID  \\\n0      ENST00000361390   ENSP00000354687              U5Z754   \n1      ENST00000361453   ENSP00000355046              Q7GXY9   \n2      ENST00000361624   ENSP00000354499              U5YWV7   \n3      ENST00000361739   ENSP00000354876              U5Z487   \n4      ENST00000361851   ENSP00000355265              U5YV54   \n\n  UniProtKB/Swiss-Prot ID  \n0                  P03886  \n1                  P03891  \n2                  P00395  \n3                  P00403  \n4                  P03928  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Gene stable ID</th>\n      <th>HGNC symbol</th>\n      <th>Gene name</th>\n      <th>Chromosome/scaffold name</th>\n      <th>Transcript stable ID</th>\n      <th>Protein stable ID</th>\n      <th>UniProtKB/TrEMBL ID</th>\n      <th>UniProtKB/Swiss-Prot ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>ENSG00000198888</td>\n      <td>MT-ND1</td>\n      <td>MT-ND1</td>\n      <td>MT</td>\n      <td>ENST00000361390</td>\n      <td>ENSP00000354687</td>\n      <td>U5Z754</td>\n      <td>P03886</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>ENSG00000198763</td>\n      <td>MT-ND2</td>\n      <td>MT-ND2</td>\n      <td>MT</td>\n      <td>ENST00000361453</td>\n      <td>ENSP00000355046</td>\n      <td>Q7GXY9</td>\n      <td>P03891</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>ENSG00000198804</td>\n      <td>MT-CO1</td>\n      <td>MT-CO1</td>\n      <td>MT</td>\n      <td>ENST00000361624</td>\n      <td>ENSP00000354499</td>\n      <td>U5YWV7</td>\n      <td>P00395</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>ENSG00000198712</td>\n      <td>MT-CO2</td>\n      <td>MT-CO2</td>\n      <td>MT</td>\n      <td>ENST00000361739</td>\n      <td>ENSP00000354876</td>\n      <td>U5Z487</td>\n      <td>P00403</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>ENSG00000228253</td>\n      <td>MT-ATP8</td>\n      <td>MT-ATP8</td>\n      <td>MT</td>\n      <td>ENST00000361851</td>\n      <td>ENSP00000355265</td>\n      <td>U5YV54</td>\n      <td>P03928</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biomart.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T17:13:30.984828900Z",
     "start_time": "2023-12-13T17:13:30.953555800Z"
    }
   },
   "id": "f6367c527f846d44"
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "outputs": [
    {
     "data": {
      "text/plain": "123120"
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(biomart)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T17:13:33.017465100Z",
     "start_time": "2023-12-13T17:13:33.001464600Z"
    }
   },
   "id": "9a20d423b9bcdb57"
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "outputs": [
    {
     "data": {
      "text/plain": "111338"
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "scaffolds = ['1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', 'X', 'Y']\n",
    "biomart_primary = biomart[biomart['Chromosome/scaffold name'].isin(scaffolds)]\n",
    "\n",
    "len(biomart_primary)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T17:13:41.591247400Z",
     "start_time": "2023-12-13T17:13:41.512629800Z"
    }
   },
   "id": "26c3cb8b38e02b6a"
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "outputs": [
    {
     "data": {
      "text/plain": "         Gene stable ID HGNC symbol Gene name Chromosome/scaffold name  \\\n1129    ENSG00000273167         NaN       NaN                       13   \n1687    ENSG00000288593         NaN       NaN                       21   \n1836    ENSG00000249209         NaN       NaN                       21   \n1899    ENSG00000260861         NaN       NaN                       20   \n1900    ENSG00000260861         NaN       NaN                       20   \n...                 ...         ...       ...                      ...   \n122296  ENSG00000285629         NaN       NaN                        1   \n122665  ENSG00000288678         NaN       NaN                        1   \n122666  ENSG00000288678         NaN       NaN                        1   \n122941  ENSG00000286015         NaN       NaN                        1   \n122942  ENSG00000286015         NaN       NaN                        1   \n\n       Transcript stable ID Protein stable ID UniProtKB/TrEMBL ID  \\\n1129        ENST00000382141   ENSP00000371576          A0A0A0MRY4   \n1687        ENST00000674444   ENSP00000501503                 NaN   \n1836        ENST00000429238   ENSP00000394107              H7C0C1   \n1899        ENST00000566961   ENSP00000457551              H3BUA5   \n1900        ENST00000567028   ENSP00000454437              H3BML4   \n...                     ...               ...                 ...   \n122296      ENST00000484532   ENSP00000465763              K7EKS7   \n122665      ENST00000679970   ENSP00000505765          A0A7P0T9M0   \n122666      ENST00000710329   ENSP00000518209                 NaN   \n122941      ENST00000651827   ENSP00000498451          A0A494C0A1   \n122942      ENST00000566597   ENSP00000456439          A0A075B732   \n\n       UniProtKB/Swiss-Prot ID  \n1129                       NaN  \n1687                       NaN  \n1836                       NaN  \n1899                       NaN  \n1900                       NaN  \n...                        ...  \n122296                     NaN  \n122665                     NaN  \n122666                     NaN  \n122941                     NaN  \n122942                     NaN  \n\n[982 rows x 8 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Gene stable ID</th>\n      <th>HGNC symbol</th>\n      <th>Gene name</th>\n      <th>Chromosome/scaffold name</th>\n      <th>Transcript stable ID</th>\n      <th>Protein stable ID</th>\n      <th>UniProtKB/TrEMBL ID</th>\n      <th>UniProtKB/Swiss-Prot ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>1129</th>\n      <td>ENSG00000273167</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>13</td>\n      <td>ENST00000382141</td>\n      <td>ENSP00000371576</td>\n      <td>A0A0A0MRY4</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>1687</th>\n      <td>ENSG00000288593</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>21</td>\n      <td>ENST00000674444</td>\n      <td>ENSP00000501503</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>1836</th>\n      <td>ENSG00000249209</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>21</td>\n      <td>ENST00000429238</td>\n      <td>ENSP00000394107</td>\n      <td>H7C0C1</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>1899</th>\n      <td>ENSG00000260861</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>20</td>\n      <td>ENST00000566961</td>\n      <td>ENSP00000457551</td>\n      <td>H3BUA5</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>1900</th>\n      <td>ENSG00000260861</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>20</td>\n      <td>ENST00000567028</td>\n      <td>ENSP00000454437</td>\n      <td>H3BML4</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>122296</th>\n      <td>ENSG00000285629</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1</td>\n      <td>ENST00000484532</td>\n      <td>ENSP00000465763</td>\n      <td>K7EKS7</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>122665</th>\n      <td>ENSG00000288678</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1</td>\n      <td>ENST00000679970</td>\n      <td>ENSP00000505765</td>\n      <td>A0A7P0T9M0</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>122666</th>\n      <td>ENSG00000288678</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1</td>\n      <td>ENST00000710329</td>\n      <td>ENSP00000518209</td>\n      <td>NaN</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>122941</th>\n      <td>ENSG00000286015</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1</td>\n      <td>ENST00000651827</td>\n      <td>ENSP00000498451</td>\n      <td>A0A494C0A1</td>\n      <td>NaN</td>\n    </tr>\n    <tr>\n      <th>122942</th>\n      <td>ENSG00000286015</td>\n      <td>NaN</td>\n      <td>NaN</td>\n      <td>1</td>\n      <td>ENST00000566597</td>\n      <td>ENSP00000456439</td>\n      <td>A0A075B732</td>\n      <td>NaN</td>\n    </tr>\n  </tbody>\n</table>\n<p>982 rows  8 columns</p>\n</div>"
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "biomart_duplicates = biomart_primary[biomart_primary.duplicated(subset=['HGNC symbol'], keep=False)]\n",
    "biomart_duplicates[biomart_duplicates['Gene name'].isnull()]"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T17:20:07.880558800Z",
     "start_time": "2023-12-13T17:20:07.831219400Z"
    }
   },
   "id": "d7d978284b7015b6"
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "110320\n",
      "19825\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\dan\\AppData\\Local\\Temp\\ipykernel_14824\\1955572428.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  reactome_no_duplicates.loc['PARTICIPANT_A_ENSG'] = reactome_no_duplicates['PARTICIPANT_A'].map(hgnc_to_gene_id)\n",
      "C:\\Users\\dan\\AppData\\Local\\Temp\\ipykernel_14824\\1955572428.py:27: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  reactome_no_duplicates.loc['PARTICIPANT_B_ENSG'] = reactome_no_duplicates['PARTICIPANT_B'].map(hgnc_to_gene_id)\n"
     ]
    },
    {
     "data": {
      "text/plain": "  PARTICIPANT_A INTERACTION_TYPE PARTICIPANT_B PARTICIPANT_A_ENSG  \\\n0          A1CF  in-complex-with       APOBEC2    ENSG00000148584   \n1          A1CF  in-complex-with      APOBEC3A    ENSG00000148584   \n2          A1CF  in-complex-with      APOBEC3B    ENSG00000148584   \n3          A1CF  in-complex-with      APOBEC3C    ENSG00000148584   \n4          A1CF  in-complex-with      APOBEC3H    ENSG00000148584   \n\n  PARTICIPANT_B_ENSG  \n0    ENSG00000124701  \n1    ENSG00000128383  \n2    ENSG00000179750  \n3    ENSG00000244509  \n4    ENSG00000100298  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>PARTICIPANT_A</th>\n      <th>INTERACTION_TYPE</th>\n      <th>PARTICIPANT_B</th>\n      <th>PARTICIPANT_A_ENSG</th>\n      <th>PARTICIPANT_B_ENSG</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC2</td>\n      <td>ENSG00000148584</td>\n      <td>ENSG00000124701</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3A</td>\n      <td>ENSG00000148584</td>\n      <td>ENSG00000128383</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3B</td>\n      <td>ENSG00000148584</td>\n      <td>ENSG00000179750</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3C</td>\n      <td>ENSG00000148584</td>\n      <td>ENSG00000244509</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3H</td>\n      <td>ENSG00000148584</td>\n      <td>ENSG00000100298</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Assuming that the real dataframes have the correct mappings between HGNC symbols and PARTICIPANT_A/B\n",
    "# The following code should work for your requirement:\n",
    "\n",
    "# Code to map Gene stable IDs to PARTICIPANT_A and PARTICIPANT_B\n",
    "\n",
    "# First, create a mapping dictionary from the second dataframe\n",
    "# Here we map 'HGNC symbol' to 'Gene stable ID'\n",
    "\n",
    "# Drop duplicate biomart rows and anything without an HGNC\n",
    "\n",
    "biomart_no_null = biomart_primary.dropna(subset=['HGNC symbol'])\n",
    "print(len(biomart_no_null))\n",
    "\n",
    "biomart_no_null = biomart_no_null.drop_duplicates(subset=['HGNC symbol', 'Gene stable ID'], keep='first')\n",
    "print(len(biomart_no_null))\n",
    "\n",
    "#biomart_no_null.head()\n",
    "#biomart_no_null[biomart_no_null['Gene name'].isnull()]\n",
    "\n",
    "# These duplicates are expected because there are multiple transcripts per gene\n",
    "#biomart_duplicates = biomart[biomart.duplicated(subset=['HGNC symbol'], keep=False)]\n",
    "hgnc_to_gene_id = biomart_no_null.set_index('HGNC symbol')['Gene stable ID'].to_dict()\n",
    "\n",
    "# Now, use this dictionary to map the PARTICIPANT_A and PARTICIPANT_B values\n",
    "# in the first dataframe to their corresponding Gene stable ID\n",
    "reactome_no_duplicates.loc['PARTICIPANT_A_ENSG'] = reactome_no_duplicates['PARTICIPANT_A'].map(hgnc_to_gene_id)\n",
    "reactome_no_duplicates.loc['PARTICIPANT_B_ENSG'] = reactome_no_duplicates['PARTICIPANT_B'].map(hgnc_to_gene_id)\n",
    "\n",
    "# The result is a modified version of df1 with two new columns: PARTICIPANT_A_ENSG and PARTICIPANT_B_ENSG\n",
    "# These columns contain the Gene stable ID corresponding to the HGNC symbols in PARTICIPANT_A and PARTICIPANT_B\n",
    "\n",
    "# Show the modified dataframe (for demonstration, it will show NaNs because of the sample data)\n",
    "reactome_no_duplicates.head()  # Display the first few rows of the modified dataframe\n",
    "\n"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T18:19:08.962216700Z",
     "start_time": "2023-12-13T18:19:08.623597800Z"
    }
   },
   "id": "c9548877e5383a6d"
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "10141\n",
      "329943\n"
     ]
    }
   ],
   "source": [
    "unique_nodes = set(reactome_no_duplicates['PARTICIPANT_A_ENSG'].tolist() +  reactome_no_duplicates['PARTICIPANT_B_ENSG'].tolist())\n",
    "print(len(unique_nodes))\n",
    "print(len(reactome_no_duplicates))"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T18:50:00.741143200Z",
     "start_time": "2023-12-13T18:50:00.674989100Z"
    }
   },
   "id": "258391199566b6de"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Use biomart conv to add an ENSG column for gene A and gene B \n",
    "#biomart = pd.read_csv(data_path + \"biomart/ensg_enst_ensp_hgnc_name.txt\")\n",
    "biomart.head()\n",
    "print(len(biomart))\n",
    "gname_reactome_matches = biomart[biomart['Gene name'].isin(unique_gene_all)]\n",
    "hgnc_reactome_matches = biomart[biomart['HGNC symbol'].isin(unique_gene_all)]\n",
    "\n",
    "gname_reactome_misses = biomart[~biomart['Gene name'].isin(unique_gene_all)]\n",
    "hgnc_reactome_misses = biomart[~biomart['HGNC symbol'].isin(unique_gene_all)]\n",
    "\n",
    "print(len(hgnc_reactome_matches))\n",
    "print(len(gname_reactome_matches))\n",
    "print(len(gname_reactome_matches.drop_duplicates('Gene name')))\n",
    "print(len(hgnc_reactome_matches.drop_duplicates('Gene name')))\n",
    "\n",
    "gname_reactome_misses = gname_reactome_misses['Gene name'].tolist()\n",
    "hgnc_reactome_misses = hgnc_reactome_misses['HGNC symbol'].tolist()\n",
    "\n",
    "total_misses = set(gname_reactome_misses + hgnc_reactome_misses)\n",
    "print(total_misses)\n",
    "total_misses_df = pd.DataFrame(total_misses)\n",
    "total_misses_df.to_csv(data_path + 'missing_gene_names.csv')\n",
    "#len(biomart[biomart['Gene name'].isnull()])\n",
    "#len(biomart)\n",
    "biomart.head()\n",
    "\n",
    "biomart[biomart['HGNC symbol'].isnull()]\n",
    "\n",
    "biomart_no_null = biomart.dropna(subset=['HGNC symbol', 'Gene name'], how='all')\n",
    "\n",
    "print(len(biomart_no_null))\n",
    "biomart_no_null.head()\n",
    "biomart_no_null[biomart_no_null['Gene name'].isnull()]\n",
    "\n",
    "# These duplicates are expected because there are multiple transcripts per gene\n",
    "#biomart_duplicates = biomart[biomart.duplicated(subset=['HGNC symbol'], keep=False)]\n",
    "#biomart_duplicates\n",
    "#biomart['Chromosome/scaffold name'].unique()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "start_time": "2023-12-13T17:09:13.628503800Z"
    }
   },
   "id": "9a3f9e38cb44835a"
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "outputs": [
    {
     "data": {
      "text/plain": "{'PNLIPRP3',\n 'NQO1',\n 'ZNRF1',\n 'CHEBI:16581',\n 'TSEN2',\n 'ST6GAL1',\n 'SMC1A',\n 'CHEBI:17140',\n 'ACE2',\n 'CCNC',\n 'ZNF441',\n 'GUCY2F',\n 'UBE2W',\n 'OR10A7',\n 'RIMKLA',\n 'CHEBI:7207',\n 'EEF1AKMT1',\n 'CAPZA2',\n 'EIF3M',\n 'METTL3',\n 'RPL10A',\n 'BDH1',\n 'KIF2A',\n 'HTN1',\n 'GPR27',\n 'APOB',\n 'LTA',\n 'OR6T1',\n 'ANAPC16',\n 'THEM5',\n 'ARHGDIG',\n 'KLK3',\n 'PTPN22',\n 'AGTR2',\n 'PEX14',\n 'COL1A1',\n 'OTOF',\n 'MCCC1',\n 'TNKS',\n 'CHEBI:68575',\n 'OR6C68',\n 'CHEBI:29866',\n 'HRH4',\n 'ZNF699',\n 'ESRRG',\n 'SMAD9',\n 'TPGS1',\n 'SERPINE1',\n 'NDUFA11',\n 'BARD1',\n 'HACL1',\n 'ZNF658',\n 'PIGK',\n 'TAS2R1',\n 'PTCH1',\n 'WDR12',\n 'PDIA3',\n 'OXCT2',\n 'CMC4',\n 'SEMA3E',\n 'CHEBI:32325',\n 'CHEBI:49603',\n 'VNN3',\n 'FSHR',\n 'PTPN6',\n 'LEFTY1',\n 'WDR61',\n 'HTR1B',\n 'CENPM',\n 'OR10J5',\n 'TGFBR2',\n 'DIO1',\n 'CHEBI:37643',\n 'THRSP',\n 'DUSP16',\n 'CHEBI:138673',\n 'ORC5',\n 'DNAJC5',\n 'LRFN4',\n 'FAAP100',\n 'TAF4B',\n 'AGO2',\n 'IL1F10',\n 'OR4F17',\n 'OR13D1',\n 'SCGB3A2',\n 'HIST1H3D',\n 'WNT2',\n 'GRHL1',\n 'IFNL3',\n 'NTF3',\n 'CHEBI:65408',\n 'CPN2',\n 'NFE2',\n 'RNF146',\n 'CHEBI:7577',\n 'PIGL',\n 'CHEBI:3193',\n 'USP1',\n 'XRCC1',\n 'CHEBI:16526',\n 'E2F1',\n 'RAB26',\n 'CHEBI:36313',\n 'PIGU',\n 'VPREB1',\n 'CHEBI:15531',\n 'ACADSB',\n 'TNFSF12',\n 'UPB1',\n 'CHEBI:17345',\n 'SLC39A6',\n 'EBI3',\n 'MED26',\n 'ITGAL',\n 'EIF2B2',\n 'PROKR2',\n 'SCN1A',\n 'USP17L2',\n 'CFP',\n 'ITSN1',\n 'ARHGAP11B',\n 'LPL',\n 'EDC3',\n 'HCN4',\n 'KLK7',\n 'CHEBI:63977',\n 'ARSA',\n 'LILRB4',\n 'ZNF705G',\n 'OXCT1',\n 'HLA-B',\n 'GPX2',\n 'BYSL',\n 'LCAT',\n 'PPAT',\n 'VPS37A',\n 'CHEBI:28077',\n 'CHEBI:48416',\n 'DDX17',\n 'OR14A16',\n 'APLN',\n 'RAB44',\n 'VWF',\n 'IGHE',\n 'GSPT2',\n 'RNF144A',\n 'CHEBI:17533',\n 'ZFAND6',\n 'DEFB123',\n 'WDR75',\n 'CHEBI:59888',\n 'CHRM1',\n 'CHEBI:16330',\n 'PDCD11',\n 'TLE3',\n 'HIST2H4A',\n 'SDC1',\n 'ZNF417',\n 'PPFIA2',\n 'OR52Z1',\n 'ZNF559',\n 'ID1',\n 'PDSS1',\n 'KHK',\n 'SMARCA4',\n 'SNX18',\n 'COG1',\n 'CHEBI:49040',\n 'CETP',\n 'NANOG',\n 'HPX',\n 'SRSF1',\n 'CHEBI:60640',\n 'ACLY',\n 'MSH2',\n 'PLCG1',\n 'OR2L3',\n 'ZNF701',\n 'EVC',\n 'OR10A4',\n 'ZNF773',\n 'SF3A3',\n 'UBE2Z',\n 'KLRB1',\n 'ZNF840P',\n 'PDE7B',\n 'AASS',\n 'CHEBI:71956',\n 'SLC6A5',\n 'OR51A7',\n 'NPLOC4',\n 'SF3B2',\n 'CHEBI:27997',\n 'TFAP2B',\n 'FCER1A',\n 'GAB1',\n 'ZNF554',\n 'TFG',\n 'TNFRSF14',\n 'OR2AJ1',\n 'CEP70',\n 'BHMT2',\n 'D2HGDH',\n 'HSPA8',\n 'CHEBI:28706',\n 'MED15',\n 'OR4K1',\n 'GALNT5',\n 'TTYH1',\n 'MAP2K2',\n 'LCE2B',\n 'LRRC7',\n 'FOXL2',\n 'PTPRA',\n 'TNFRSF9',\n 'TBK1',\n 'LIPJ',\n 'OR8K1',\n 'ZIM3',\n 'PLA2G4B',\n 'TYK2',\n 'ADAMTSL5',\n 'STARD3NL',\n 'CHEBI:16827',\n 'FOXO1',\n 'CCL3L1',\n 'CDK4',\n 'NRP1',\n 'CHEBI:15489',\n 'RCC2',\n 'UIMC1',\n 'MANBA',\n 'CHEBI:37550',\n 'MSMO1',\n 'NSUN4',\n 'ASPN',\n 'EIF2B5',\n 'RAB3C',\n 'CHEBI:13115',\n 'SELENOI',\n 'CHEBI:63983',\n 'LATS2',\n 'TTYH2',\n 'ADCY6',\n 'CHEBI:37623',\n 'IGHA1',\n 'NDUFB9',\n 'CHEBI:63810',\n 'OR4P4',\n 'NTS',\n 'F2RL3',\n 'CHEBI:16830',\n 'CEBPD',\n 'CHEBI:138640',\n 'RNMT',\n 'HIST2H4B',\n 'COQ2',\n 'RAB5B',\n 'PDGFRB',\n 'NMT1',\n 'TRPC7',\n 'LIG4',\n 'GALNTL5',\n 'OR51A4',\n 'CHEBI:16650',\n 'CHEBI:16290',\n 'TEX10',\n 'CS',\n 'MYL7',\n 'PHGDH',\n 'BUD31',\n 'NDUFA3',\n 'RPL27A',\n 'MOGAT2',\n 'APBA1',\n 'ADAM19',\n 'CHEBI:15354',\n 'DBH',\n 'KLHL11',\n 'PPP2R2D',\n 'TXNIP',\n 'CHEBI:15476',\n 'HHIP',\n 'MT1M',\n 'CHEBI:62746',\n 'GSTM1',\n 'XBP1',\n 'OR2AP1',\n 'HTR6',\n 'EPGN',\n 'TAF1',\n 'FPR3',\n 'EPHA10',\n 'DAB2IP',\n 'ORC1',\n 'CHEBI:64681',\n 'DAG1',\n 'ZNF776',\n 'HOXB3',\n 'RBMX',\n 'RAB39A',\n 'TET3',\n 'UBE2Q1',\n 'ZNF200',\n 'CHEBI:16349',\n 'SLC36A2',\n 'P2RY10',\n 'DEFA1',\n 'ATP5F1D',\n 'CHEBI:52449',\n 'CACNB2',\n 'CHEBI:71958',\n 'PHF20',\n 'CDC42EP2',\n 'NUDT19',\n 'ARHGEF12',\n 'CHEBI:37642',\n 'GPAT3',\n 'CHEBI:16086',\n 'AGTPBP1',\n 'CHRNA3',\n 'MATK',\n 'MED14',\n 'CARNMT1',\n 'KCNA4',\n 'CHEBI:15637',\n 'CHEBI:50583',\n 'HS6ST3',\n 'GPR65',\n 'CAPNS2',\n 'COL25A1',\n 'DCLRE1A',\n 'FYB1',\n 'SUGT1',\n 'CHEBI:17825',\n 'ALG3',\n 'FCGR1A',\n 'ENSA',\n 'TSR3',\n 'UGT2B7',\n 'MMP9',\n 'NCAPG2',\n 'IFNAR2',\n 'OR3A2',\n 'HACD1',\n 'TDGF1P3',\n 'FANCM',\n 'PIP5K1A',\n 'CASC3',\n 'CHEBI:61910',\n 'CHEBI:25676',\n 'OR5B21',\n 'ME3',\n 'ZNF791',\n 'RPL13A',\n 'HIBCH',\n 'OR2M4',\n 'IL13RA2',\n 'PJA1',\n 'ARHGAP10',\n 'CHEBI:15635',\n 'MYH9',\n 'MYOG',\n 'GCHFR',\n 'MRPS21',\n 'TMED5',\n 'PSMD9',\n 'SUPT4H1',\n 'CYP1A2',\n 'NELL2',\n 'ARPP19',\n 'ABCA9',\n 'CHMP4A',\n 'RASGEF1A',\n 'CHEBI:140256',\n 'LCMT2',\n 'IQGAP2',\n 'MSTN',\n 'CPA3',\n 'CHEBI:35422',\n 'GPR132',\n 'MRPS22',\n 'HJURP',\n 'OR2G2',\n 'EHD2',\n 'ZNF738',\n 'TAP2',\n 'COL16A1',\n 'CD3G',\n 'SHC3',\n 'SLCO2A1',\n 'PDE1A',\n 'CHEBI:131825',\n 'IL17RE',\n 'OR2C1',\n 'CES1',\n 'SRP14',\n 'SLC6A6',\n 'CHEBI:36773',\n 'COPRS',\n 'TAF11',\n 'CHEBI:62637',\n 'PSMD1',\n 'PDE2A',\n 'SLC6A7',\n 'INO80C',\n 'MRPL28',\n 'NUDT13',\n 'SLC44A2',\n 'CHEBI:91279',\n 'CHEBI:49108',\n 'EEA1',\n 'TARBP2',\n 'LCN2',\n 'CHPF',\n 'TAS2R14',\n 'MMP16',\n 'TAF6L',\n 'HPR',\n 'SAP30L',\n 'FGFR2',\n 'PAFAH1B2',\n 'EIF3H',\n 'WDR34',\n 'ATP6V0D1',\n 'MORC2',\n 'RAD51',\n 'CHEBI:140273',\n 'TUBB3',\n 'ZNF562',\n 'NT5C',\n 'CHEBI:33738',\n 'ZNF611',\n 'PPP3CA',\n 'CHEBI:9588',\n 'RECK',\n 'RSAD2',\n 'RHCG',\n 'FCGR2A',\n 'CHEBI:61360',\n 'PADI6',\n 'ARFGAP3',\n 'GGA1',\n 'CHEBI:17405',\n 'CHEBI:16595',\n 'ZP2',\n 'ATP6V0A1',\n 'VEGFB',\n 'CHEBI:63979',\n 'ARHGEF25',\n 'CRKL',\n 'GDPD3',\n 'MRPL17',\n 'SUMO3',\n 'SH2B1',\n 'MRPL53',\n 'MMP20',\n 'ITGA5',\n 'CHEBI:133907',\n 'VPREB3',\n 'PRELP',\n 'OR4C46',\n 'AMFR',\n 'HLA-DQA1',\n 'ZNF804B',\n 'CHEBI:72608',\n 'B4GAT1',\n 'TACR2',\n 'CHEBI:52592',\n 'CHEBI:78585',\n 'NDUFC1',\n 'SLC7A5',\n 'RAB12',\n 'CHEBI:16320',\n 'CAMKK2',\n 'THBS4',\n 'TTLL11',\n 'RPS6KB2',\n 'ARHGAP23',\n 'CLASP2',\n 'TTC30A',\n 'CHEBI:46631',\n 'CHEBI:17797',\n 'DRAP1',\n 'RPS27A',\n 'CHEBI:8713',\n 'CHEBI:140200',\n 'CHRM4',\n 'AUTS2',\n 'SLX4',\n 'ARHGEF2',\n 'CHEBI:15646',\n 'OR2K2',\n 'TNFSF13',\n 'OR51F1',\n 'CTPS1',\n 'UROC1',\n 'C4B',\n 'IRAK4',\n 'CDC26',\n 'GALT',\n 'SLC26A2',\n 'OR10K1',\n 'NDUFA5',\n 'OAZ2',\n 'SUMO1',\n 'IFNAR1',\n 'EXOC1',\n 'LRSAM1',\n 'SREBF2',\n 'LY6D',\n 'P2RY4',\n 'ITPKB',\n 'CHEBI:62928',\n 'EXOSC7',\n 'OR7G1',\n 'CHEBI:140911',\n 'CSF3',\n 'CHEBI:17748',\n 'TRPM4',\n 'RAB3IP',\n 'DEFB105B',\n 'HSPB11',\n 'CHEBI:63993',\n 'GJD2',\n 'ADAMTSL3',\n 'PRDX5',\n 'GGT1',\n 'MX2',\n 'CD55',\n 'CD300LF',\n 'SRMS',\n 'WNT10B',\n 'ARHGAP17',\n 'CHEBI:27398',\n 'SPHK2',\n 'SETD9',\n 'DSE',\n 'IDS',\n 'TCTEX1D2',\n 'PIK3CA',\n 'LCT',\n 'HGD',\n 'FABP9',\n 'WRAP53',\n 'NDE1',\n 'OR13F1',\n 'PRDX3',\n 'SRSF2',\n 'APOBEC3G',\n 'ZNF99',\n 'CHEBI:78584',\n 'PML',\n 'WDR4',\n 'MRPL37',\n 'PIGY',\n 'CHEBI:13705',\n 'IL3',\n 'TRMT9B',\n 'HTR2A',\n 'CHEBI:63868',\n 'CHEBI:10049',\n 'USP24',\n 'CHEBI:28497',\n 'ADORA2A',\n 'CDK7',\n 'LDLRAP1',\n 'TSC22D1',\n 'MBD3',\n 'CHEBI:28325',\n 'ACADS',\n 'CHEBI:63974',\n 'IGKV3-11',\n 'B3GNT4',\n 'HOXB1',\n 'USP17L27',\n 'AKR7L',\n 'ALDH6A1',\n 'KRT73',\n 'KIF2C',\n 'ZNF354A',\n 'CHPF2',\n 'FBXL3',\n 'SAMM50',\n 'MUS81',\n 'GP1BA',\n 'CHEBI:16905',\n 'TADA2B',\n 'NTSR2',\n 'RGS19',\n 'ZNF540',\n 'CHEBI:28654',\n 'KALRN',\n 'PSG5',\n 'MIGA1',\n 'RPL6',\n 'ARAF',\n 'BRWD1',\n 'CRABP2',\n 'OR2A5',\n 'IRF9',\n 'C8A',\n 'CHEBI:17998',\n 'P2RY11',\n 'CENPK',\n 'CHEBI:37554',\n 'RNF126',\n 'EIF3E',\n 'ACOX1',\n 'CHEBI:15652',\n 'GCG',\n 'SLX1B',\n 'PTGER2',\n 'LPGAT1',\n 'CHEBI:27760',\n 'ADCY10',\n 'FUT7',\n 'IL33',\n 'COL9A2',\n 'MIGA2',\n 'YME1L1',\n 'CHEBI:140878',\n 'CSF2RB',\n 'TPM4',\n 'GPR17',\n 'INTS10',\n 'OR8U8',\n 'CBR1',\n 'DOCK11',\n 'NUF2',\n 'MKRN1',\n 'CHMP6',\n 'USP17L4',\n 'CHEBI:17821',\n 'NAT8',\n 'ARAP3',\n 'USP3',\n 'THSD4',\n 'AQP7',\n 'MAOB',\n 'SLC31A1',\n 'NR1D1',\n 'TPX2',\n 'MRPS18A',\n 'DCAF11',\n 'MRPS30',\n 'OR51M1',\n 'RBM28',\n 'CHEBI:63980',\n 'ZNF705A',\n 'APRT',\n 'FANCI',\n 'CHEBI:135010',\n 'CHEBI:3698',\n 'CHEBI:27847',\n 'PDE4B',\n 'POLR3GL',\n 'HDAC2',\n 'ZNF420',\n 'SPON1',\n 'CHEBI:18276',\n 'U2SURP',\n 'EME1',\n 'MASP1',\n 'CHEBI:49203',\n 'RARA',\n 'NUP43',\n 'ZNF385A',\n 'RRP8',\n 'CHEBI:78539',\n 'RPS9',\n 'PDK1',\n 'ZNF566',\n 'SLC4A5',\n 'SOCS4',\n 'OR1J1',\n 'PFAS',\n 'ABCA6',\n 'GAK',\n 'SMARCA5',\n 'PSCA',\n 'POP7',\n 'MMP25',\n 'AREG',\n 'ECE2',\n 'MATN1',\n 'APH1A',\n 'TH',\n 'VNN2',\n 'KL',\n 'OR5AC1',\n 'NOTCH2',\n 'PSMD10',\n 'DEFB108A',\n 'B3GNT9',\n 'MAP3K11',\n 'SLC10A2',\n 'CHEBI:64685',\n 'NTN1',\n 'NLK',\n 'MAF1',\n 'FBF1',\n 'OR2T5',\n 'INPP4A',\n 'LCE3D',\n 'TAF3',\n 'MYSM1',\n 'ARID3A',\n 'PRPF8',\n 'DRD2',\n 'TUBGCP2',\n 'DHFR2',\n 'CHEBI:28918',\n 'PCSK5',\n 'AKR1D1',\n 'FIS1',\n 'DMBT1',\n 'ANO2',\n 'TULP4',\n 'TRAPPC8',\n 'CHEBI:65088',\n 'GP6',\n 'UQCR11',\n 'CHEBI:41001',\n 'FABP5',\n 'IZUMO4',\n 'TMPRSS6',\n 'LSAMP',\n 'HERC2',\n 'SLC45A3',\n 'CHEBI:34490',\n 'CHEBI:15533',\n 'EIF5A2',\n 'WAS',\n 'CHEBI:15534',\n 'CHEBI:65133',\n 'CHEBI:63839',\n 'SKA1',\n 'SPSB3',\n 'CHEBI:421843',\n 'SRY',\n 'ARL6',\n 'STX16',\n 'HIP1',\n 'NPC1',\n 'AKR1B15',\n 'CHEBI:16393',\n 'HS3ST6',\n 'ZBTB17',\n 'BMP10',\n 'ARHGAP45',\n 'PDX1',\n 'OXTR',\n 'SLC5A4',\n 'EIF4G1',\n 'IL1RAP',\n 'PFKP',\n 'ZNF33A',\n 'PARD6A',\n 'RETN',\n 'SH2D2A',\n 'DUOX1',\n 'CYP8B1',\n 'CHEBI:15793',\n 'NUP188',\n 'SLC4A8',\n 'OR51B4',\n 'CALU',\n 'CHEBI:65101',\n 'ENOPH1',\n 'TUSC3',\n 'NOB1',\n 'CHEBI:2274',\n 'PTPMT1',\n 'PTPRO',\n 'PTGR1',\n 'PDCD6IP',\n 'CHEBI:63803',\n 'CHEBI:27974',\n 'UST',\n 'CHEBI:16953',\n 'PCCA',\n 'CHEBI:15956',\n 'RAD23B',\n 'CHEBI:16990',\n 'CHEBI:88130',\n 'FOXK2',\n 'SLC27A4',\n 'TADA2A',\n 'CCL23',\n 'CHEBI:2700',\n 'ARPC1A',\n 'MAT1A',\n 'SH3GL2',\n 'FBXW11',\n 'GNA15',\n 'TBC1D15',\n 'CHEBI:16196',\n 'HLA-DQB2',\n 'ST3GAL3',\n 'IRF4',\n 'GATA3',\n 'NSMCE2',\n 'CHEBI:52050',\n 'CHEBI:60637',\n 'LGI2',\n 'LIPG',\n 'OR6C74',\n 'IZUMO1R',\n 'CHEBI:15765',\n 'FAHD1',\n 'IL27RA',\n 'USP15',\n 'CHEBI:17710',\n 'PLEKHA2',\n 'CAPZA1',\n 'E2F3',\n 'CHEBI:62878',\n 'CHEBI:73231',\n 'PDZD3',\n 'TRAF7',\n 'CHEBI:64689',\n 'DEFA6',\n 'HLA-DRB4',\n 'NFATC2',\n 'PDE6D',\n 'CHEBI:77363',\n 'AHSG',\n 'ADAM11',\n 'LSM11',\n 'SET',\n 'UBA3',\n 'SLC35C1',\n 'SLC35B4',\n 'CHEBI:28766',\n 'HIST1H2AL',\n 'OR10J1',\n 'ALCAM',\n 'EP300',\n 'SSTR1',\n 'CHEBI:28806',\n 'EBP',\n 'CHEBI:26355',\n 'PCTP',\n 'DAO',\n 'SLC1A7',\n 'CHEBI:64183',\n 'CNDP2',\n 'WNT7A',\n 'OR5I1',\n 'DARS',\n 'RIN2',\n 'MTHFD1',\n 'ITGA10',\n 'CHEBI:37946',\n 'HAUS6',\n 'ZNF506',\n 'TIMMDC1',\n 'WWP1',\n 'CHEBI:91280',\n 'HIKESHI',\n 'HMGCR',\n 'FBXO41',\n 'SH2B3',\n 'CHEBI:4470',\n 'FGF12',\n 'BNIP1',\n 'GJA1',\n 'TDH',\n 'CHEBI:60430',\n 'CATSPER2',\n 'LRTOMT',\n 'STK26',\n 'CHEBI:135377',\n 'CHEBI:3210',\n 'FGF6',\n 'CHEBI:16897',\n 'PIK3R6',\n 'FNBP1',\n 'ADSL',\n 'GMPS',\n 'FOLR1',\n 'JUNB',\n 'IGHD',\n 'OTUD3',\n 'BACE1',\n 'CFTR',\n 'CXCR2',\n 'KRT7',\n 'PPP2R2A',\n 'SCLT1',\n 'MPP5',\n 'TIMM23',\n 'SFTPD',\n 'PIK3C3',\n 'RTN4RL2',\n 'C1D',\n 'PARP14',\n 'DPAGT1',\n 'CHEBI:15508',\n 'TANK',\n 'RAB27B',\n 'PDPN',\n 'CDC14B',\n 'CTSD',\n 'KCNG2',\n 'FZD7',\n 'SLC22A6',\n 'CHEBI:140268',\n 'EPHA6',\n 'TRAV8-4',\n 'CHEBI:28602',\n 'CHEBI:16285',\n 'CHEBI:140919',\n 'OR4C45',\n 'CTSH',\n 'KREMEN2',\n 'MAN1A1',\n 'CHEBI:63984',\n 'ATP6V0D2',\n 'GRIP1',\n 'CFHR5',\n 'NCKAP1L',\n 'GPX6',\n 'CHEBI:18384',\n 'TUBA8',\n 'CYP2C19',\n 'G6PC3',\n 'BET1L',\n 'KIFC3',\n 'GPX5',\n 'AGER',\n 'CHEBI:6498',\n 'RP2',\n 'SGTA',\n 'CHEBI:17625',\n 'DEGS2',\n 'ELK1',\n 'ZNF471',\n 'NPPA',\n 'ZNF215',\n 'SIPA1',\n 'CHEBI:42589',\n 'ARL3',\n 'NCAM1',\n 'OR52N1',\n 'CHEBI:28790',\n 'TFAP2A',\n 'S100A7A',\n 'CLEC1B',\n 'RORA',\n 'SETD6',\n 'TRPM7',\n 'SCOC',\n 'COPS8',\n 'EARS2',\n 'CHEBI:17627',\n 'MT-CO1',\n 'PAX5',\n 'CHEBI:37154',\n 'NUP133',\n 'ATF2',\n 'CHEBI:63637',\n 'CYP11B1',\n 'CENPO',\n 'CHEBI:15554',\n 'ANKRD9',\n 'PAPOLA',\n 'SLC10A1',\n 'ZNF445',\n 'ALYREF',\n 'OR52P1P',\n 'ETNPPL',\n 'CHEBI:31399',\n 'GAS1',\n 'NOP2',\n 'RBFOX2',\n 'TNNT1',\n 'CNOT9',\n 'CD300C',\n 'B3GALT6',\n 'GCLC',\n 'PRKCSH',\n 'GUSB',\n 'SFTPB',\n 'SLC25A28',\n 'PSMD3',\n 'ORMDL2',\n 'SYNJ2',\n 'GRM7',\n 'MYH2',\n 'CHEBI:6438',\n 'RMI1',\n 'DDHD2',\n 'CHEBI:31998',\n 'COQ10B',\n 'DGKB',\n 'CHEBI:62833',\n 'INCENP',\n ...}"
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "unique_gene_all"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T14:08:07.353712600Z",
     "start_time": "2023-12-13T14:08:07.334758300Z"
    }
   },
   "id": "65fca4e16f565f1a"
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "377688\n"
     ]
    },
    {
     "data": {
      "text/plain": "  PARTICIPANT_A INTERACTION_TYPE PARTICIPANT_B Gene name   Gene stable ID\n0          A1CF  in-complex-with       APOBEC2   APOBEC2  ENSG00000124701\n1          A1CF  in-complex-with      APOBEC3A  APOBEC3A  ENSG00000262156\n3          A1CF  in-complex-with      APOBEC3A  APOBEC3A  ENSG00000128383\n5          A1CF  in-complex-with      APOBEC3B  APOBEC3B  ENSG00000179750\n9          A1CF  in-complex-with      APOBEC3C  APOBEC3C  ENSG00000244509",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>PARTICIPANT_A</th>\n      <th>INTERACTION_TYPE</th>\n      <th>PARTICIPANT_B</th>\n      <th>Gene name</th>\n      <th>Gene stable ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC2</td>\n      <td>APOBEC2</td>\n      <td>ENSG00000124701</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3A</td>\n      <td>APOBEC3A</td>\n      <td>ENSG00000262156</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3A</td>\n      <td>APOBEC3A</td>\n      <td>ENSG00000128383</td>\n    </tr>\n    <tr>\n      <th>5</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3B</td>\n      <td>APOBEC3B</td>\n      <td>ENSG00000179750</td>\n    </tr>\n    <tr>\n      <th>9</th>\n      <td>A1CF</td>\n      <td>in-complex-with</td>\n      <td>APOBEC3C</td>\n      <td>APOBEC3C</td>\n      <td>ENSG00000244509</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_ensga = pd.merge(reactome_no_duplicates, biomart_no_null[['Gene name','Gene stable ID']], left_on='PARTICIPANT_B', right_on='Gene name', how=\"left\")\n",
    "reactome_ensga.head(20)\n",
    "\n",
    "reactome_ensga = reactome_ensga.drop_duplicates()\n",
    "\n",
    "print(len(reactome_ensga))\n",
    "reactome_ensga.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T14:23:39.802740600Z",
     "start_time": "2023-12-13T14:23:38.752911600Z"
    }
   },
   "id": "b70194fbf57b7d07"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Look at gProfiler for namespace conversion\n",
    "from gprofiler import GProfiler\n",
    "\n",
    "gp = GProfiler(return_dataframe=True)\n",
    "gp_conv = gp.convert(organism='hsapiens',\n",
    "            query=list(unique_gene_all),\n",
    "            target_namespace='ENSG')\n",
    "            #target_namespace='ENTREZGENE_ACC')\n",
    "            \n",
    "gp_conv.head()\n",
    "gp_conv[gp_conv.duplicated(subset=['incoming'],keep=False)]"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "40fabf8fcac2f769"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "gp_conv.head()"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "879c5ab38334bd49"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "help(GProfiler)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "553144745558a211"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result = pd.merge(reactome_no_duplicates, gp_conv[['incoming','converted']], left_on='PARTICIPANT_A', right_on='incoming', how=\"left\")\n",
    "\n",
    "result.head()"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "978fc013bfe8ca7a"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "len(result)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "b8006b6e2059e0f4"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result = result.drop_duplicates()\n",
    "len(result)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "8547f5a66e3ec47f"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Replacing 'None' with NaN in column X\n",
    "result[['converted']] = result[['converted']].replace('None', np.nan)\n",
    "\n",
    "result[result['converted'].isnull()]"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "1f8d4c3ba3d5f82e"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result_duplicates = result.duplicated(subset=['PARTICIPANT_A','PARTICIPANT_B'])\n",
    "result[~result_duplicates]"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "22e433fb453ac7ee"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result2 = pd.merge(result, biomart_no_null[['Gene name','Gene stable ID']], left_on='PARTICIPANT_A', right_on='Gene name', how=\"left\")\n",
    "result2.head()\n",
    "\n",
    "result2[(result2['converted'].notna()) & (result2['Gene stable ID'].isnull())]"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "9d3bcd86e2c0bb6f"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "len(result2)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "d5927e16529916d7"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result2['PARTICIPANT_A_ENSG'] = np.where(result2['Gene stable ID'].notna(), result2['Gene stable ID'], result2['converted'])\n",
    "result2"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "57216ada4eb1b485"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "len(result2)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "c351a0b13648f9e2"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Drop conversion columns to use again\n",
    "result2 = result2.drop(['incoming', 'converted', 'Gene name', 'Gene stable ID'], axis=1)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "9abed455089e8d84"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result3 = pd.merge(result2, gp_conv[['incoming','converted']], left_on='PARTICIPANT_B', right_on='incoming', how=\"left\")"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "f353174990511902"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result3.head()"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "aa1d811e89e7b6c7"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "# Replacing 'None' with NaN in column X\n",
    "result3[['converted']] = result3[['converted']].replace('None', np.nan)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "a4fdb9ad3bc7b927"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result3[result3['converted'].isnull()]"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "64591be82909aebc"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result4 = pd.merge(result3, biomart_no_null[['Gene name','Gene stable ID']], left_on='PARTICIPANT_B', right_on='Gene name', how=\"left\")\n",
    "result4.head()"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "dbb90a972a975a18"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result4['PARTICIPANT_B_ENSG'] = np.where(result4['Gene stable ID'].notna(), result4['Gene stable ID'], result4['converted'])\n",
    "result4"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "a5d06f1cc11152fd"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result4 = result4.drop(['incoming', 'converted', 'Gene name', 'Gene stable ID'], axis=1)\n",
    "result4"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "7aab769e9e3f1b95"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result4[result4['PARTICIPANT_A_ENSG'].notna() & result4['PARTICIPANT_B_ENSG'].notna()]"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "fdb4fc05754ea296"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result4 = result4.drop_duplicates()\n",
    "\n",
    "len(result4)"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "d551556cdb881356"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "result4"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "e87ac4dae8e8d745"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [
    "full_results = result4[result4['PARTICIPANT_A_ENSG'].notna() & result4['PARTICIPANT_B_ENSG'].notna()]\n",
    "print(len(full_results))\n",
    "full_results"
   ],
   "metadata": {
    "collapsed": false
   },
   "id": "e6138015370be88e"
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "outputs": [],
   "source": [],
   "metadata": {
    "collapsed": false
   },
   "id": "bb5475209976a1d1"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
