{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 45,
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "from biomart import BiomartServer\n",
    "import io"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T13:14:17.221845800Z",
     "start_time": "2023-12-13T13:14:17.183936500Z"
    }
   },
   "id": "initial_id"
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "outputs": [
    {
     "data": {
      "text/plain": "['biomart',\n 'depmap',\n 'missing_gene_names.csv',\n 'pathway_commons',\n 'reactome',\n 'reactome_rows_to_drop.csv',\n 'string']"
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#os.getcwd()\n",
    "\n",
    "# Linux workstation\n",
    "#data_path = '/home/db600/phd/data/'\n",
    "\n",
    "# Laptop\n",
    "data_path = 'C:\\\\Users\\\\dan\\\\Documents\\\\phd\\\\data\\\\'\n",
    "\n",
    "os.listdir(data_path)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.214639600Z",
     "start_time": "2023-12-13T11:52:39.199101700Z"
    }
   },
   "id": "e734db8459feaf32"
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "outputs": [
    {
     "data": {
      "text/plain": "['reactome.homo_sapiens.interactions.tab-delimited_downloaded_23_10_23.txt',\n 'reactome_dropped_rows.csv',\n 'reactome_ppi.csv',\n 'test_dropped_rows.csv']"
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir(data_path +'/reactome/')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.261540200Z",
     "start_time": "2023-12-13T11:52:39.214639600Z"
    }
   },
   "id": "9bf0f08e1d21ac42"
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['# Interactor 1 uniprot id', 'Interactor 1 Ensembl gene id',\n",
      "       'Interactor 1 Entrez Gene id', 'Interactor 2 uniprot id',\n",
      "       'Interactor 2 Ensembl gene id', 'Interactor 2 Entrez Gene id',\n",
      "       'Interaction type', 'Interaction context', 'Pubmed references'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/plain": "  # Interactor 1 uniprot id  \\\n0          uniprotkb:Q9Y287   \n1          uniprotkb:P37840   \n2          uniprotkb:P0DJI8   \n3          uniprotkb:P06727   \n4          uniprotkb:P01160   \n\n                        Interactor 1 Ensembl gene id  \\\n0  ENSEMBL:ENST00000378549|ENSEMBL:ENSP0000036781...   \n1  ENSEMBL:ENST00000394989|ENSEMBL:ENST0000039498...   \n2  ENSEMBL:ENST00000356524|ENSEMBL:ENSG0000017343...   \n3  ENSEMBL:ENSG00000110244|ENSEMBL:ENST0000035778...   \n4  ENSEMBL:ENSG00000175206|ENSEMBL:ENSP0000036566...   \n\n  Interactor 1 Entrez Gene id Interactor 2 uniprot id  \\\n0                           -        uniprotkb:Q9Y287   \n1                           -        uniprotkb:P37840   \n2                           -        uniprotkb:P0DJI8   \n3                           -        uniprotkb:P06727   \n4                           -        uniprotkb:P01160   \n\n                        Interactor 2 Ensembl gene id  \\\n0  ENSEMBL:ENST00000378549|ENSEMBL:ENSP0000036781...   \n1  ENSEMBL:ENST00000394989|ENSEMBL:ENST0000039498...   \n2  ENSEMBL:ENST00000356524|ENSEMBL:ENSG0000017343...   \n3  ENSEMBL:ENSG00000110244|ENSEMBL:ENST0000035778...   \n4  ENSEMBL:ENSG00000175206|ENSEMBL:ENSP0000036566...   \n\n  Interactor 2 Entrez Gene id      Interaction type     Interaction context  \\\n0                           -  physical association   reactome:R-HSA-976871   \n1                           -  physical association  reactome:R-HSA-1247852   \n2                           -  physical association   reactome:R-HSA-976898   \n3                           -  physical association   reactome:R-HSA-976889   \n4                           -  physical association   reactome:R-HSA-976987   \n\n   Pubmed references  \n0  14690516|10391242  \n1           24243840  \n2    19393650|103558  \n3           15146166  \n4    2142465|2945573  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th># Interactor 1 uniprot id</th>\n      <th>Interactor 1 Ensembl gene id</th>\n      <th>Interactor 1 Entrez Gene id</th>\n      <th>Interactor 2 uniprot id</th>\n      <th>Interactor 2 Ensembl gene id</th>\n      <th>Interactor 2 Entrez Gene id</th>\n      <th>Interaction type</th>\n      <th>Interaction context</th>\n      <th>Pubmed references</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>uniprotkb:Q9Y287</td>\n      <td>ENSEMBL:ENST00000378549|ENSEMBL:ENSP0000036781...</td>\n      <td>-</td>\n      <td>uniprotkb:Q9Y287</td>\n      <td>ENSEMBL:ENST00000378549|ENSEMBL:ENSP0000036781...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-976871</td>\n      <td>14690516|10391242</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>uniprotkb:P37840</td>\n      <td>ENSEMBL:ENST00000394989|ENSEMBL:ENST0000039498...</td>\n      <td>-</td>\n      <td>uniprotkb:P37840</td>\n      <td>ENSEMBL:ENST00000394989|ENSEMBL:ENST0000039498...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-1247852</td>\n      <td>24243840</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>uniprotkb:P0DJI8</td>\n      <td>ENSEMBL:ENST00000356524|ENSEMBL:ENSG0000017343...</td>\n      <td>-</td>\n      <td>uniprotkb:P0DJI8</td>\n      <td>ENSEMBL:ENST00000356524|ENSEMBL:ENSG0000017343...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-976898</td>\n      <td>19393650|103558</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>uniprotkb:P06727</td>\n      <td>ENSEMBL:ENSG00000110244|ENSEMBL:ENST0000035778...</td>\n      <td>-</td>\n      <td>uniprotkb:P06727</td>\n      <td>ENSEMBL:ENSG00000110244|ENSEMBL:ENST0000035778...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-976889</td>\n      <td>15146166</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>uniprotkb:P01160</td>\n      <td>ENSEMBL:ENSG00000175206|ENSEMBL:ENSP0000036566...</td>\n      <td>-</td>\n      <td>uniprotkb:P01160</td>\n      <td>ENSEMBL:ENSG00000175206|ENSEMBL:ENSP0000036566...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-976987</td>\n      <td>2142465|2945573</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome = pd.read_csv(data_path + '/reactome/reactome.homo_sapiens.interactions.tab-delimited_downloaded_23_10_23.txt', sep=\"\\t\")\n",
    "print(reactome.columns)\n",
    "reactome.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.497182800Z",
     "start_time": "2023-12-13T11:52:39.230290500Z"
    }
   },
   "id": "911a8569b59173b9"
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['physical association' 'enzymatic reaction'\n",
      " 'oxidoreductase activity electron transfer reaction'\n",
      " 'dephosphorylation reaction' 'glycosylation reaction'\n",
      " 'phospholipase reaction' 'isomerase reaction' 'cleavage reaction'\n",
      " 'decarboxylation reaction' 'deacetylation reaction'\n",
      " 'carboxylation reaction' 'acetylation reaction'\n",
      " 'nucleoside triphosphatase reaction' 'deneddylation reaction'\n",
      " 'deubiquitination reaction' 'gtpase reaction' 'atpase reaction'\n",
      " 'demethylation reaction' 'sulfurtransfer reaction' 'amidation reaction'\n",
      " 'de-ADP-ribosylation reaction' 'phosphopantetheinylation'\n",
      " 'tyrosinylation']\n"
     ]
    }
   ],
   "source": [
    "# Take a look at the interaction types listed\n",
    "print(reactome['Interaction type'].unique())"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.544454800Z",
     "start_time": "2023-12-13T11:52:39.497182800Z"
    }
   },
   "id": "7527938f700d6de5"
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "outputs": [
    {
     "data": {
      "text/plain": "106142"
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get row count for number of edges in the dataset\n",
    "len(reactome)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.575694100Z",
     "start_time": "2023-12-13T11:52:39.514302600Z"
    }
   },
   "id": "eaa487c299371ea7"
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "outputs": [
    {
     "data": {
      "text/plain": "         extracted_ENSG 1   extracted_ENSG 2\n0       [ENSG00000136156]  [ENSG00000136156]\n1       [ENSG00000145335]  [ENSG00000145335]\n2       [ENSG00000173432]  [ENSG00000173432]\n3       [ENSG00000110244]  [ENSG00000110244]\n4       [ENSG00000175206]  [ENSG00000175206]\n...                   ...                ...\n106137  [ENSG00000043591]                 []\n106138  [ENSG00000043591]                 []\n106139  [ENSG00000188778]                 []\n106140  [ENSG00000188778]                 []\n106141  [ENSG00000188778]                 []\n\n[106142 rows x 2 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>extracted_ENSG 1</th>\n      <th>extracted_ENSG 2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>[ENSG00000136156]</td>\n      <td>[ENSG00000136156]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>[ENSG00000145335]</td>\n      <td>[ENSG00000145335]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>[ENSG00000173432]</td>\n      <td>[ENSG00000173432]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>[ENSG00000110244]</td>\n      <td>[ENSG00000110244]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>[ENSG00000175206]</td>\n      <td>[ENSG00000175206]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>106137</th>\n      <td>[ENSG00000043591]</td>\n      <td>[]</td>\n    </tr>\n    <tr>\n      <th>106138</th>\n      <td>[ENSG00000043591]</td>\n      <td>[]</td>\n    </tr>\n    <tr>\n      <th>106139</th>\n      <td>[ENSG00000188778]</td>\n      <td>[]</td>\n    </tr>\n    <tr>\n      <th>106140</th>\n      <td>[ENSG00000188778]</td>\n      <td>[]</td>\n    </tr>\n    <tr>\n      <th>106141</th>\n      <td>[ENSG00000188778]</td>\n      <td>[]</td>\n    </tr>\n  </tbody>\n</table>\n<p>106142 rows × 2 columns</p>\n</div>"
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Function to extract ENSG followed by 11 digits\n",
    "def extract_ENSG(text):\n",
    "    return re.findall(r'ENSG\\d{11}', text)\n",
    "\n",
    "# Apply the function to each cell in the column and store the result in a new column\n",
    "reactome['extracted_ENSG 1'] = reactome['Interactor 1 Ensembl gene id'].apply(extract_ENSG)\n",
    "reactome['extracted_ENSG 2'] = reactome['Interactor 2 Ensembl gene id'].apply(extract_ENSG)\n",
    "\n",
    "# See the two new columns\n",
    "reactome[['extracted_ENSG 1', 'extracted_ENSG 2']]"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.898084300Z",
     "start_time": "2023-12-13T11:52:39.544454800Z"
    }
   },
   "id": "f2b89016c842faa1"
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "outputs": [
    {
     "data": {
      "text/plain": "96071"
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Filter out any self references\n",
    "reactome = reactome[reactome['extracted_ENSG 1'] != reactome['extracted_ENSG 2']]\n",
    "len(reactome)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:39.964611100Z",
     "start_time": "2023-12-13T11:52:39.907858900Z"
    }
   },
   "id": "12507aeec4c6847"
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "outputs": [
    {
     "data": {
      "text/plain": "31980"
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get rows where either of the extracted ENSG columns is just an empty list\n",
    "rows_with_empty_list =  reactome[reactome['extracted_ENSG 1'].apply(lambda x: x == []) | reactome['extracted_ENSG 2'].apply(lambda x: x == [])] \n",
    "rows_with_empty_list\n",
    "len(rows_with_empty_list)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.031809400Z",
     "start_time": "2023-12-13T11:52:39.934955600Z"
    }
   },
   "id": "c532665e65400d28"
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "64091\n"
     ]
    },
    {
     "data": {
      "text/plain": "     # Interactor 1 uniprot id  \\\n39            uniprotkb:P08123   \n41            uniprotkb:P08123   \n45            uniprotkb:P08123   \n47            uniprotkb:P08123   \n49            uniprotkb:P08123   \n...                        ...   \n1573          uniprotkb:Q86YM7   \n1574          uniprotkb:Q86YM7   \n1575          uniprotkb:Q9NSB8   \n1576          uniprotkb:Q8N4T8   \n1579          uniprotkb:P08238   \n\n                           Interactor 1 Ensembl gene id  \\\n39    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n41    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n45    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n47    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n49    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n...                                                 ...   \n1573  ENSEMBL:ENSP00000426651|ENSEMBL:ENSG0000015241...   \n1574  ENSEMBL:ENSP00000426651|ENSEMBL:ENSG0000015241...   \n1575  ENSEMBL:ENST00000304231|ENSEMBL:ENSP0000040763...   \n1576  ENSEMBL:ENST00000306193|ENSEMBL:ENSG0000014543...   \n1579  ENSEMBL:ENST00000620073|ENSEMBL:ENSP0000036070...   \n\n     Interactor 1 Entrez Gene id Interactor 2 uniprot id  \\\n39                             -        uniprotkb:P02452   \n41                             -        uniprotkb:P02452   \n45                             -        uniprotkb:P02452   \n47                             -        uniprotkb:P02452   \n49                             -        uniprotkb:P02452   \n...                          ...                     ...   \n1573                           -        uniprotkb:Q13255   \n1574                           -        uniprotkb:P41594   \n1575                           -        uniprotkb:P41594   \n1576                           -        uniprotkb:Q92506   \n1579                           -        uniprotkb:Q15185   \n\n                           Interactor 2 Ensembl gene id  \\\n39    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n41    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n45    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n47    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n49    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n...                                                 ...   \n1573  ENSEMBL:ENST00000507907|ENSEMBL:ENST0000049280...   \n1574  ENSEMBL:ENST00000305447|ENSEMBL:ENSG0000016895...   \n1575  ENSEMBL:ENST00000305447|ENSEMBL:ENSG0000016895...   \n1576  ENSEMBL:ENSP00000406488|ENSEMBL:ENST0000037466...   \n1579  ENSEMBL:ENSP00000262033|ENSEMBL:ENST0000041427...   \n\n     Interactor 2 Entrez Gene id      Interaction type  \\\n39                             -  physical association   \n41                             -  physical association   \n45                             -  physical association   \n47                             -  physical association   \n49                             -  physical association   \n...                          ...                   ...   \n1573                           -  physical association   \n1574                           -  physical association   \n1575                           -  physical association   \n1576                           -  physical association   \n1579                           -  physical association   \n\n         Interaction context Pubmed references   extracted_ENSG 1  \\\n39    reactome:R-HSA-2428940          24243840  [ENSG00000164692]   \n41    reactome:R-HSA-2428940          24243840  [ENSG00000164692]   \n45    reactome:R-HSA-2428940          24243840  [ENSG00000164692]   \n47    reactome:R-HSA-2428940          24243840  [ENSG00000164692]   \n49    reactome:R-HSA-2428940          24243840  [ENSG00000164692]   \n...                      ...               ...                ...   \n1573  reactome:R-HSA-6794288   9808458|9069287  [ENSG00000152413]   \n1574  reactome:R-HSA-6794288   9808458|9069287  [ENSG00000152413]   \n1575  reactome:R-HSA-6794288   9808458|9069287  [ENSG00000103942]   \n1576  reactome:R-HSA-8862181          24243840  [ENSG00000145439]   \n1579  reactome:R-HSA-8937201          10409767  [ENSG00000096384]   \n\n                                       extracted_ENSG 2  \n39                                    [ENSG00000108821]  \n41                                    [ENSG00000108821]  \n45                                    [ENSG00000108821]  \n47                                    [ENSG00000108821]  \n49                                    [ENSG00000108821]  \n...                                                 ...  \n1573                                  [ENSG00000152822]  \n1574                                  [ENSG00000168959]  \n1575                                  [ENSG00000168959]  \n1576  [ENSG00000232357, ENSG00000228357, ENSG0000022...  \n1579                                  [ENSG00000110958]  \n\n[1000 rows x 11 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th># Interactor 1 uniprot id</th>\n      <th>Interactor 1 Ensembl gene id</th>\n      <th>Interactor 1 Entrez Gene id</th>\n      <th>Interactor 2 uniprot id</th>\n      <th>Interactor 2 Ensembl gene id</th>\n      <th>Interactor 2 Entrez Gene id</th>\n      <th>Interaction type</th>\n      <th>Interaction context</th>\n      <th>Pubmed references</th>\n      <th>extracted_ENSG 1</th>\n      <th>extracted_ENSG 2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>39</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>41</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>45</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>47</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>49</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>1573</th>\n      <td>uniprotkb:Q86YM7</td>\n      <td>ENSEMBL:ENSP00000426651|ENSEMBL:ENSG0000015241...</td>\n      <td>-</td>\n      <td>uniprotkb:Q13255</td>\n      <td>ENSEMBL:ENST00000507907|ENSEMBL:ENST0000049280...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-6794288</td>\n      <td>9808458|9069287</td>\n      <td>[ENSG00000152413]</td>\n      <td>[ENSG00000152822]</td>\n    </tr>\n    <tr>\n      <th>1574</th>\n      <td>uniprotkb:Q86YM7</td>\n      <td>ENSEMBL:ENSP00000426651|ENSEMBL:ENSG0000015241...</td>\n      <td>-</td>\n      <td>uniprotkb:P41594</td>\n      <td>ENSEMBL:ENST00000305447|ENSEMBL:ENSG0000016895...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-6794288</td>\n      <td>9808458|9069287</td>\n      <td>[ENSG00000152413]</td>\n      <td>[ENSG00000168959]</td>\n    </tr>\n    <tr>\n      <th>1575</th>\n      <td>uniprotkb:Q9NSB8</td>\n      <td>ENSEMBL:ENST00000304231|ENSEMBL:ENSP0000040763...</td>\n      <td>-</td>\n      <td>uniprotkb:P41594</td>\n      <td>ENSEMBL:ENST00000305447|ENSEMBL:ENSG0000016895...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-6794288</td>\n      <td>9808458|9069287</td>\n      <td>[ENSG00000103942]</td>\n      <td>[ENSG00000168959]</td>\n    </tr>\n    <tr>\n      <th>1576</th>\n      <td>uniprotkb:Q8N4T8</td>\n      <td>ENSEMBL:ENST00000306193|ENSEMBL:ENSG0000014543...</td>\n      <td>-</td>\n      <td>uniprotkb:Q92506</td>\n      <td>ENSEMBL:ENSP00000406488|ENSEMBL:ENST0000037466...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-8862181</td>\n      <td>24243840</td>\n      <td>[ENSG00000145439]</td>\n      <td>[ENSG00000232357, ENSG00000228357, ENSG0000022...</td>\n    </tr>\n    <tr>\n      <th>1579</th>\n      <td>uniprotkb:P08238</td>\n      <td>ENSEMBL:ENST00000620073|ENSEMBL:ENSP0000036070...</td>\n      <td>-</td>\n      <td>uniprotkb:Q15185</td>\n      <td>ENSEMBL:ENSP00000262033|ENSEMBL:ENST0000041427...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-8937201</td>\n      <td>10409767</td>\n      <td>[ENSG00000096384]</td>\n      <td>[ENSG00000110958]</td>\n    </tr>\n  </tbody>\n</table>\n<p>1000 rows × 11 columns</p>\n</div>"
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove the rows with empty ENSG values from the main dataset\n",
    "reactome_complete = reactome[~reactome.isin(rows_with_empty_list).all(axis=1)]\n",
    "\n",
    "print(len(reactome_complete))\n",
    "reactome_complete.head(1000)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.096174700Z",
     "start_time": "2023-12-13T11:52:39.981206800Z"
    }
   },
   "id": "24ad4cb1ad72dbbc"
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "outputs": [
    {
     "data": {
      "text/plain": "6849"
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# See how many rows where either column1 or column2 contains a list with more than one item\n",
    "len(reactome_complete[reactome_complete['extracted_ENSG 1'].apply(lambda x: isinstance(x, list) and len(x) > 1) |\n",
    "                   reactome_complete['extracted_ENSG 2'].apply(lambda x: isinstance(x, list) and len(x) > 1)])"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.145599600Z",
     "start_time": "2023-12-13T11:52:40.069359600Z"
    }
   },
   "id": "7045c8ebcdb62b0"
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from itertools import product\n",
    "\n",
    "def expand_list_pairs(df, col1, col2, dropped_rows_csv_path):\n",
    "    # Create a new DataFrame to hold the expanded rows\n",
    "    expanded_rows = pd.DataFrame(columns=df.columns)\n",
    "\n",
    "    # Track indices to drop from the original DataFrame\n",
    "    indices_to_drop = []\n",
    "\n",
    "    # Iterate through the DataFrame\n",
    "    for idx, row in df.iterrows():\n",
    "        # Check if either column has a list with more than one item\n",
    "        is_col1_list = isinstance(row[col1], list) and len(row[col1]) > 1\n",
    "        is_col2_list = isinstance(row[col2], list) and len(row[col2]) > 1\n",
    "\n",
    "        if is_col1_list or is_col2_list:\n",
    "            # Add index to the list of indices to drop\n",
    "            indices_to_drop.append(idx)\n",
    "\n",
    "            # Create combinations of items\n",
    "            items_col1 = row[col1] if is_col1_list else [row[col1]]\n",
    "            items_col2 = row[col2] if is_col2_list else [row[col2]]\n",
    "\n",
    "            for combination in product(items_col1, items_col2):\n",
    "                new_row = row.copy()\n",
    "                new_row[col1] = combination[0]\n",
    "                new_row[col2] = combination[1]\n",
    "                # Append new_row to expanded_rows using concat\n",
    "                expanded_rows = pd.concat([expanded_rows, pd.DataFrame([new_row])], ignore_index=True)\n",
    "\n",
    "    # Extract the rows to be dropped into a separate DataFrame\n",
    "    rows_to_drop = df.loc[indices_to_drop]\n",
    "\n",
    "    # Save the rows to be dropped to a CSV file\n",
    "    rows_to_drop.to_csv(dropped_rows_csv_path, index=False)\n",
    "\n",
    "    # Drop the original rows where lists had more than one item\n",
    "    df = df.drop(indices_to_drop)\n",
    "\n",
    "    # Concatenate the original DataFrame with the expanded rows\n",
    "    df = pd.concat([df, expanded_rows], ignore_index=True)\n",
    "\n",
    "    return df"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.208093500Z",
     "start_time": "2023-12-13T11:52:40.115926Z"
    }
   },
   "id": "a4cd9da56ad9a503"
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "outputs": [],
   "source": [
    "data = {\n",
    "    'column1': [[1, 2], [3], ['l'], ['a', 'b', 'c'], [5]],\n",
    "    'column2': [[2], [1, 2], ['x', 'y'], [6], ['a']]\n",
    "}\n",
    "df1 = pd.DataFrame(data)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.208093500Z",
     "start_time": "2023-12-13T11:52:40.129943300Z"
    }
   },
   "id": "4a4044d07057aa26"
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "outputs": [],
   "source": [
    "df_revised = expand_list_pairs(df1, 'column1', 'column2', data_path + '/reactome/test_dropped_rows.csv')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.208093500Z",
     "start_time": "2023-12-13T11:52:40.145599600Z"
    }
   },
   "id": "459f3f1d0116562a"
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "     column1 column2\n",
      "0     [1, 2]     [2]\n",
      "1        [3]  [1, 2]\n",
      "2        [l]  [x, y]\n",
      "3  [a, b, c]     [6]\n",
      "4        [5]     [a]\n",
      "  column1 column2\n",
      "0     [5]     [a]\n",
      "1       1     [2]\n",
      "2       2     [2]\n",
      "3     [3]       1\n",
      "4     [3]       2\n",
      "5     [l]       x\n",
      "6     [l]       y\n",
      "7       a     [6]\n",
      "8       b     [6]\n",
      "9       c     [6]\n"
     ]
    }
   ],
   "source": [
    "print(df1)\n",
    "print(df_revised)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:52:40.208093500Z",
     "start_time": "2023-12-13T11:52:40.176846300Z"
    }
   },
   "id": "52feb1c3e16fad31"
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "outputs": [],
   "source": [
    "#reactome_final = expand_list_pairs(reactome_complete, 'extracted_ENSG 1', 'extracted_ENSG 2')\n",
    "reactome_final = expand_list_pairs(reactome_complete, 'extracted_ENSG 1', 'extracted_ENSG 2', data_path + '/reactome/reactome_dropped_rows.csv')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:54.771708900Z",
     "start_time": "2023-12-13T11:52:40.192467500Z"
    }
   },
   "id": "89e43880eafb7bde"
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "outputs": [
    {
     "data": {
      "text/plain": "Empty DataFrame\nColumns: [# Interactor 1 uniprot id, Interactor 1 Ensembl gene id, Interactor 1 Entrez Gene id, Interactor 2 uniprot id, Interactor 2 Ensembl gene id, Interactor 2 Entrez Gene id, Interaction type, Interaction context, Pubmed references, extracted_ENSG 1, extracted_ENSG 2]\nIndex: []",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th># Interactor 1 uniprot id</th>\n      <th>Interactor 1 Ensembl gene id</th>\n      <th>Interactor 1 Entrez Gene id</th>\n      <th>Interactor 2 uniprot id</th>\n      <th>Interactor 2 Ensembl gene id</th>\n      <th>Interactor 2 Entrez Gene id</th>\n      <th>Interaction type</th>\n      <th>Interaction context</th>\n      <th>Pubmed references</th>\n      <th>extracted_ENSG 1</th>\n      <th>extracted_ENSG 2</th>\n    </tr>\n  </thead>\n  <tbody>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "multiple_genes_listed = reactome_final[reactome_final['extracted_ENSG 1'].apply(lambda x: isinstance(x, list) and len(x) > 1) |\n",
    "                   reactome_final['extracted_ENSG 2'].apply(lambda x: isinstance(x, list) and len(x) > 1)]\n",
    "\n",
    "multiple_genes_listed"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:54.838794800Z",
     "start_time": "2023-12-13T11:53:54.771708900Z"
    }
   },
   "id": "26b79923a744accf"
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "103760\n"
     ]
    },
    {
     "data": {
      "text/plain": "    # Interactor 1 uniprot id  \\\n0            uniprotkb:P08123   \n1            uniprotkb:P08123   \n2            uniprotkb:P08123   \n3            uniprotkb:P08123   \n4            uniprotkb:P08123   \n..                        ...   \n995          uniprotkb:P46098   \n996          uniprotkb:P46098   \n997          uniprotkb:P01877   \n998          uniprotkb:Q8IVH4   \n999          uniprotkb:Q8IVH4   \n\n                          Interactor 1 Ensembl gene id  \\\n0    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n1    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n2    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n3    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n4    ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...   \n..                                                 ...   \n995  ENSEMBL:ENSP00000347754|ENSEMBL:ENSP0000042477...   \n996  ENSEMBL:ENSP00000347754|ENSEMBL:ENSP0000042477...   \n997  ENSEMBL:ENSG00000211890|ENSEMBL:ENSP0000037498...   \n998  ENSEMBL:ENSP00000497680|ENSEMBL:ENSP0000049700...   \n999  ENSEMBL:ENSP00000497680|ENSEMBL:ENSP0000049700...   \n\n    Interactor 1 Entrez Gene id Interactor 2 uniprot id  \\\n0                             -        uniprotkb:P02452   \n1                             -        uniprotkb:P02452   \n2                             -        uniprotkb:P02452   \n3                             -        uniprotkb:P02452   \n4                             -        uniprotkb:P02452   \n..                          ...                     ...   \n995                           -        uniprotkb:Q70Z44   \n996                           -        uniprotkb:A5X5Y0   \n997                           -        uniprotkb:P01591   \n998                           -        uniprotkb:P22033   \n999                           -        uniprotkb:P22033   \n\n                          Interactor 2 Ensembl gene id  \\\n0    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n1    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n2    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n3    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n4    ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...   \n..                                                 ...   \n995  ENSEMBL:ENST00000428798|ENSEMBL:ENSP0000038926...   \n996  ENSEMBL:ENST00000425359|ENSEMBL:ENSG0000018603...   \n997  ENSEMBL:ENSP00000254801|ENSEMBL:ENSP0000042668...   \n998  ENSEMBL:ENST00000274813|ENSEMBL:ENSG0000014608...   \n999  ENSEMBL:ENST00000274813|ENSEMBL:ENSG0000014608...   \n\n    Interactor 2 Entrez Gene id      Interaction type     Interaction context  \\\n0                             -  physical association  reactome:R-HSA-2428940   \n1                             -  physical association  reactome:R-HSA-2428940   \n2                             -  physical association  reactome:R-HSA-2428940   \n3                             -  physical association  reactome:R-HSA-2428940   \n4                             -  physical association  reactome:R-HSA-2428940   \n..                          ...                   ...                     ...   \n995                           -  physical association  reactome:R-HSA-6792753   \n996                           -  physical association  reactome:R-HSA-6792754   \n997                           -  physical association  reactome:R-HSA-8858031   \n998                           -  physical association  reactome:R-HSA-5654974   \n999                           -  physical association  reactome:R-HSA-5654974   \n\n    Pubmed references   extracted_ENSG 1   extracted_ENSG 2  \n0            24243840  [ENSG00000164692]  [ENSG00000108821]  \n1            24243840  [ENSG00000164692]  [ENSG00000108821]  \n2            24243840  [ENSG00000164692]  [ENSG00000108821]  \n3            24243840  [ENSG00000164692]  [ENSG00000108821]  \n4            24243840  [ENSG00000164692]  [ENSG00000108821]  \n..                ...                ...                ...  \n995          24243840  [ENSG00000166736]  [ENSG00000186090]  \n996          24243840  [ENSG00000166736]  [ENSG00000186038]  \n997          11877257  [ENSG00000211890]  [ENSG00000132465]  \n998          24243840  [ENSG00000151611]  [ENSG00000146085]  \n999          24243840  [ENSG00000151611]  [ENSG00000146085]  \n\n[1000 rows x 11 columns]",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th># Interactor 1 uniprot id</th>\n      <th>Interactor 1 Ensembl gene id</th>\n      <th>Interactor 1 Entrez Gene id</th>\n      <th>Interactor 2 uniprot id</th>\n      <th>Interactor 2 Ensembl gene id</th>\n      <th>Interactor 2 Entrez Gene id</th>\n      <th>Interaction type</th>\n      <th>Interaction context</th>\n      <th>Pubmed references</th>\n      <th>extracted_ENSG 1</th>\n      <th>extracted_ENSG 2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>uniprotkb:P08123</td>\n      <td>ENSEMBL:ENSG00000164692|ENSEMBL:ENST0000029726...</td>\n      <td>-</td>\n      <td>uniprotkb:P02452</td>\n      <td>ENSEMBL:ENSP00000225964|ENSEMBL:ENSG0000010882...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-2428940</td>\n      <td>24243840</td>\n      <td>[ENSG00000164692]</td>\n      <td>[ENSG00000108821]</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>995</th>\n      <td>uniprotkb:P46098</td>\n      <td>ENSEMBL:ENSP00000347754|ENSEMBL:ENSP0000042477...</td>\n      <td>-</td>\n      <td>uniprotkb:Q70Z44</td>\n      <td>ENSEMBL:ENST00000428798|ENSEMBL:ENSP0000038926...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-6792753</td>\n      <td>24243840</td>\n      <td>[ENSG00000166736]</td>\n      <td>[ENSG00000186090]</td>\n    </tr>\n    <tr>\n      <th>996</th>\n      <td>uniprotkb:P46098</td>\n      <td>ENSEMBL:ENSP00000347754|ENSEMBL:ENSP0000042477...</td>\n      <td>-</td>\n      <td>uniprotkb:A5X5Y0</td>\n      <td>ENSEMBL:ENST00000425359|ENSEMBL:ENSG0000018603...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-6792754</td>\n      <td>24243840</td>\n      <td>[ENSG00000166736]</td>\n      <td>[ENSG00000186038]</td>\n    </tr>\n    <tr>\n      <th>997</th>\n      <td>uniprotkb:P01877</td>\n      <td>ENSEMBL:ENSG00000211890|ENSEMBL:ENSP0000037498...</td>\n      <td>-</td>\n      <td>uniprotkb:P01591</td>\n      <td>ENSEMBL:ENSP00000254801|ENSEMBL:ENSP0000042668...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-8858031</td>\n      <td>11877257</td>\n      <td>[ENSG00000211890]</td>\n      <td>[ENSG00000132465]</td>\n    </tr>\n    <tr>\n      <th>998</th>\n      <td>uniprotkb:Q8IVH4</td>\n      <td>ENSEMBL:ENSP00000497680|ENSEMBL:ENSP0000049700...</td>\n      <td>-</td>\n      <td>uniprotkb:P22033</td>\n      <td>ENSEMBL:ENST00000274813|ENSEMBL:ENSG0000014608...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-5654974</td>\n      <td>24243840</td>\n      <td>[ENSG00000151611]</td>\n      <td>[ENSG00000146085]</td>\n    </tr>\n    <tr>\n      <th>999</th>\n      <td>uniprotkb:Q8IVH4</td>\n      <td>ENSEMBL:ENSP00000497680|ENSEMBL:ENSP0000049700...</td>\n      <td>-</td>\n      <td>uniprotkb:P22033</td>\n      <td>ENSEMBL:ENST00000274813|ENSEMBL:ENSG0000014608...</td>\n      <td>-</td>\n      <td>physical association</td>\n      <td>reactome:R-HSA-5654974</td>\n      <td>24243840</td>\n      <td>[ENSG00000151611]</td>\n      <td>[ENSG00000146085]</td>\n    </tr>\n  </tbody>\n</table>\n<p>1000 rows × 11 columns</p>\n</div>"
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(reactome_final))\n",
    "reactome_final.head(1000)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:54.879407600Z",
     "start_time": "2023-12-13T11:53:54.838794800Z"
    }
   },
   "id": "f5d6c3189742bbb8"
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37951\n",
      "       extracted_ENSG 1     extracted_ENSG 2\n",
      "0   ['ENSG00000164692']  ['ENSG00000108821']\n",
      "18  ['ENSG00000108821']  ['ENSG00000164692']\n",
      "72  ['ENSG00000108947']  ['ENSG00000090776']\n",
      "73  ['ENSG00000125266']  ['ENSG00000108947']\n",
      "74  ['ENSG00000125266']  ['ENSG00000090776']\n",
      "75  ['ENSG00000184009']  ['ENSG00000075624']\n",
      "76  ['ENSG00000149534']  ['ENSG00000179639']\n",
      "77  ['ENSG00000149534']  ['ENSG00000158869']\n",
      "78  ['ENSG00000158869']  ['ENSG00000179639']\n",
      "79  ['ENSG00000186393']  ['ENSG00000204897']\n"
     ]
    }
   ],
   "source": [
    "# Convert list columns to strings\n",
    "reactome_final['extracted_ENSG 1'] = reactome_final['extracted_ENSG 1'].apply(str)\n",
    "reactome_final['extracted_ENSG 2'] = reactome_final['extracted_ENSG 2'].apply(str)\n",
    "\n",
    "# Now select only the columns we want and drop duplicates\n",
    "reactome_ppi = reactome_final[['extracted_ENSG 1', 'extracted_ENSG 2']] \n",
    "reactome_ppi = reactome_ppi.drop_duplicates()\n",
    "\n",
    "# Print length and head\n",
    "print(len(reactome_ppi))\n",
    "print(reactome_ppi.head(10))"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:55.021087Z",
     "start_time": "2023-12-13T11:53:54.858562600Z"
    }
   },
   "id": "7b51583b10bc178e"
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37951\n",
      "37951\n",
      "75902\n"
     ]
    },
    {
     "data": {
      "text/plain": "<Figure size 640x480 with 1 Axes>",
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAjoAAAGdCAYAAAAbudkLAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMCwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy88F64QAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAoO0lEQVR4nO3df3AUZZ7H8c8QkpCwIYiR/CA/CCvohmg8AioILmGPuAFhl9RZeHdCVOA2y/gDA2VBcbcouhuUI3IeAcS1QO7HgnpIrYJyURBQyjLBgGj2WMRAAiRmYZWEsCSQ9P1hZdYhP0hmJunJM+9X1VQ53T3d335ols8+/fTTDsuyLAEAABioj90FAAAAdBeCDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6AADAWH3tLsBuzc3NOnPmjCIiIuRwOOwuBwAAdIJlWaqrq1NcXJz69Gm/3ybgg86ZM2eUkJBgdxkAAMADlZWVio+Pb3d9wAediIgISd811IABA2yuBgAAdEZtba0SEhJc/463J+CDTsvtqgEDBhB0AADoZa417ITByAAAwFgEHQAAYCyCDgAAMFbAj9EBAKCrLMvSlStX1NTUZHcpxgoKClLfvn29nvqFoAMAQBc0NjaqqqpKFy9etLsU44WHhys2NlYhISEe7yNgg05hYaEKCwtJ4wCATmtublZ5ebmCgoIUFxenkJAQJpvtBpZlqbGxUX/6059UXl6u4cOHdzgpYEcclmVZPq6vV6mtrVVkZKTOnz/P4+UAgA5dunRJ5eXlSkpKUnh4uN3lGO/ixYs6efKkkpOT1a9fP7d1nf33m8HIAAB0kae9C+gaX7Qzf1IAAMBYBB0AANDKBx98IIfDoW+//dbuUrwSsIORAQDwlaGLd/To8U6smNqjx+vN6NEBAADGIugAABAALMvS888/r2HDhiksLExpaWl64403XOt37typESNGKCwsTBkZGTpx4kSrfbz88stKSEhQeHi4ZsyYoYKCAg0cONBtm7feekvp6enq16+fhg0bpqefflpXrlxxrX/qqaeUmJio0NBQxcXF6bHHHuuuU5bEratu1dKVSRcjAMBu//zP/6xt27Zp3bp1Gj58uPbt26cHHnhAN9xwg4YNG6bs7Gzl5ubql7/8pUpKSrRw4UK333/00UfKzc3Vc889p+nTp+u9997Tv/zLv7hts2vXLj3wwAN68cUXNWHCBB0/flz/9E//JElatmyZ3njjDb3wwgvasmWLRo4cqerqah0+fLhbz5ugAwCA4err61VQUKDdu3dr7NixkqRhw4bpww8/1EsvvaShQ4dq2LBheuGFF+RwOHTTTTfpyJEjeu6551z7+Pd//3dlZWVp0aJFkqQRI0bowIEDevvtt13b/PrXv9bixYuVk5PjOsYzzzyjJ598UsuWLVNFRYViYmL0t3/7twoODlZiYqJuv/32bj13bl0BAGC4srIyXbp0SZMnT9YPfvAD12fz5s06fvy4/vCHP+jOO+90m+W5JRC1OHr0aKtQcvX3gwcPavny5W7HmDdvnuuVGffdd5/+8pe/aNiwYZo3b57efPNNt9ta3YEeHQAADNfc3CxJ2rFjh4YMGeK2LjQ0VI8++ug192FZVqvXXVz9coXm5mY9/fTTys7ObvX7fv36KSEhQUePHlVRUZHee+89zZ8/XytXrtTevXsVHBzc1dPqFIIOAACGS0lJUWhoqCoqKvTjH/+4zfXbt293W/bxxx+7fb/55pv1ySefuC0rKSlx+z5q1CgdPXpUN954Y7u1hIWFafr06Zo+fbqcTqduvvlmHTlyRKNGjeriWXUOQQcAAMNFRERo0aJFeuKJJ9Tc3Kzx48ertrZWBw4c0A9+8APl5uZq1apVysvL0y9+8QsdPHhQmzZtctvHo48+qrvvvlsFBQWaNm2adu/erXfeecetl+dXv/qV7r33XiUkJOi+++5Tnz599Nlnn+nIkSN69tlntWnTJjU1NemOO+5QeHi4/uM//kNhYWFKSkrqtnNnjA4AAAHgmWee0a9+9Svl5+frRz/6ke655x699dZbSk5OVmJiov7nf/5Hb731ltLS0rR+/Xr95je/cfv9XXfdpfXr16ugoEBpaWl699139cQTT7i9bPOee+7R22+/raKiIo0ZM0Z33nmnCgoKXEFm4MCBevnll3XXXXfp1ltv1fvvv6+33npL119/fbedN28v78a3l/N4OQCYpeXt5W29TTsQzZs3T//3f/+n/fv3d8v+O2rvzv77za0rAADQKf/6r/+qyZMnq3///nrnnXf06quvau3atXaX1SGCDgAA6JRPPvlEzz//vOrq6jRs2DC9+OKLmjt3rt1ldYigAwAAOuW1116zu4QuYzAyAAAwFkEHAAAYi6ADAEAXBfgDyz3GF+1M0AEAoJNaXlNw8eJFmysJDC3t7M3rIRiMDABAJwUFBWngwIGqqamRJIWHh7d6/xO8Z1mWLl68qJqaGg0cOFBBQUEe74ugAwBAF8TExEiSK+yg+wwcONDV3p4i6AAA0AUOh0OxsbEaPHiwLl++bHc5xgoODvaqJ6dFrw86dXV1mjRpki5fvqympiY99thjmjdvnt1lAQAMFxQU5JN/iNG9en3QCQ8P1969exUeHq6LFy8qNTVV2dnZ3fqCMAAA0Dv0+qeugoKCFB4eLum7l381NTXx2B8AAJDkB0Fn3759mjZtmuLi4uRwOLR9+/ZW26xdu9b15tL09PRWb0n99ttvlZaWpvj4eD355JOKiorqoeoBAIA/sz3o1NfXKy0tTWvWrGlz/datW7VgwQItXbpUpaWlmjBhgrKyslRRUeHaZuDAgTp8+LDKy8v13//93/r666/bPV5DQ4Nqa2vdPgAAwEy2B52srCw9++yzys7ObnN9QUGB5syZo7lz5+pHP/qRVq9erYSEBK1bt67VttHR0br11lu1b9++do+Xn5+vyMhI1ychIcFn5wIAAPyL7UGnI42NjTp48KAyMzPdlmdmZurAgQOSpK+//trVK1NbW6t9+/bppptuanefS5Ys0fnz512fysrK7jsBAABgK79+6urs2bNqampSdHS02/Lo6GhVV1dLkk6dOqU5c+bIsixZlqVHHnlEt956a7v7DA0NVWhoaLfWDQAA/INfB50WV0+vbVmWa1l6eroOHTpkQ1UAAMDf+fWtq6ioKAUFBbl6b1rU1NS06uUBAAC4ml8HnZCQEKWnp6uoqMhteVFRkcaNG+fVvgsLC5WSkqIxY8Z4tR8AAOC/bL91deHCBX355Zeu7+Xl5Tp06JAGDRqkxMRE5eXladasWRo9erTGjh2rDRs2qKKiQrm5uV4d1+l0yul0qra2VpGRkd6eBgAA8EO2B52SkhJlZGS4vufl5UmScnJytGnTJs2cOVPnzp3T8uXLVVVVpdTUVO3cuVNJSUl2lQwAAHoJhxXg70to6dE5f/68BgwY4NN9D128Q5J0YsVUn+4XAIBA19l/v/16jA4AAIA3CDoAAMBYARt0eOoKAADzBWzQcTqdKisrU3Fxsd2lAACAbhKwQQcAAJiPoAMAAIxF0AEAAMYi6AAAAGMFbNDhqSsAAMwXsEGHp64AADBfwAYdAABgPoIOAAAwFkEHAAAYi6ADAACMRdDpYUMX79DQxTvsLgMAgIAQsEGHx8sBADBfwAYdHi8HAMB8fe0uIBBwqwoAAHsEbI8OAAAwH0EHAAAYi6ADAACMRdABAADGIugAAABjBWzQYR4dAADMF7BBh3l0AAAwX8AGHQAAYD6CDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6AADAWAEbdJgZGQAA8wVs0GFmZAAAzBewQQcAAJiPoAMAAIxF0AEAAMYi6AAAAGP1tbsAfGfo4h2u/z6xYqqNlQAAYA56dAAAgLEIOgAAwFgEHQAAYCyCDgAAMBZBBwAAGIugAwAAjBWwQYeXegIAYL6ADTq81BMAAPMFbNABAADmI+jYZOjiHW6zIQMAAN8j6AAAAGMRdAAAgLEIOgAAwFgEHQAAYCyCDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6AADAWAQdAABgLIJOL8ALQAEA8AxBBwAAGCtgg05hYaFSUlI0ZswYu0sBAADdJGCDjtPpVFlZmYqLi+0uBQAAdJOADToAAMB8BB0AAGAsgg4AADAWQQcAABiLoAMAAIzV1+4CAh0TAQIA0H3o0QEAAMYi6AAAAGMRdAAAgLEIOgAAwFgEHQAAYCyCjh8aungHT2MBAOADBB0AAGAsgg4AADAWQQcAABiLoAMAAIxF0AEAAMYi6AAAAGMRdAAAgLEIOgAAwFgEHQAAYCyCDgAAMBZBBwAAGIug08vxXiwAANrX64NOZWWlJk6cqJSUFN166616/fXX7S4JAAD4ib52F+Ctvn37avXq1brttttUU1OjUaNGacqUKerfv7/dpQEAAJv1+qATGxur2NhYSdLgwYM1aNAg/fnPfyboAAAA+29d7du3T9OmTVNcXJwcDoe2b9/eapu1a9cqOTlZ/fr1U3p6uvbv39/mvkpKStTc3KyEhIRurhoAAPQGtged+vp6paWlac2aNW2u37p1qxYsWKClS5eqtLRUEyZMUFZWlioqKty2O3funGbPnq0NGzb0RNkAAKAXsP3WVVZWlrKystpdX1BQoDlz5mju3LmSpNWrV2vXrl1at26d8vPzJUkNDQ2aMWOGlixZonHjxnV4vIaGBjU0NLi+19bW+uAsAACAP7K9R6cjjY2NOnjwoDIzM92WZ2Zm6sCBA5Iky7L04IMPatKkSZo1a9Y195mfn6/IyEjXh9tcAACYy6+DztmzZ9XU1KTo6Gi35dHR0aqurpYkffTRR9q6dau2b9+u2267TbfddpuOHDnS7j6XLFmi8+fPuz6VlZXdeg52Yo4dAECgs/3WVWc4HA6375ZluZaNHz9ezc3Nnd5XaGioQkNDfVofAADwT37doxMVFaWgoCBX702LmpqaVr08AAAAV/ProBMSEqL09HQVFRW5LS8qKrrmoGMAAADbb11duHBBX375pet7eXm5Dh06pEGDBikxMVF5eXmaNWuWRo8erbFjx2rDhg2qqKhQbm6uV8ctLCxUYWGhmpqavD2FbsP4GgAAvGN70CkpKVFGRobre15eniQpJydHmzZt0syZM3Xu3DktX75cVVVVSk1N1c6dO5WUlOTVcZ1Op5xOp2praxUZGenVvgAAgH+yPehMnDhRlmV1uM38+fM1f/78HqoIAACYwqOgU15eruTkZF/Xgmv4/q2sEyum2lgJAAC9g0eDkW+88UZlZGToP//zP3Xp0iVf1wQAAOATHgWdw4cP62/+5m+0cOFCxcTE6Be/+IU++eQTX9fWrQoLC5WSkqIxY8bYXUqPa5lIkMHOAADTeRR0UlNTVVBQoNOnT2vjxo2qrq7W+PHjNXLkSBUUFOhPf/qTr+v0OafTqbKyMhUXF9tdCgAA6CZezaPTt29fzZgxQ6+99pqee+45HT9+XIsWLVJ8fLxmz56tqqoqX9UJAADQZV4FnZKSEs2fP1+xsbEqKCjQokWLdPz4ce3evVunT5/Wz372M1/VCQAA0GUePXVVUFCgjRs36ujRo5oyZYo2b96sKVOmqE+f73JTcnKyXnrpJd18880+LRYAAKArPAo669at08MPP6yHHnpIMTExbW6TmJioV155xaviAAAAvOFR0Dl27Ng1twkJCVFOTo4nu+8RveEVEAAAwDsejdHZuHGjXn/99VbLX3/9db366qteF9UTeOoKAADzeRR0VqxYoaioqFbLBw8erN/85jdeF4VrYx4cAACuzaOgc/LkyTZfAZGUlKSKigqviwIAAPAFj4LO4MGD9dlnn7VafvjwYV1//fVeF4Wu81UPDz1FAACTeBR07r//fj322GPas2ePmpqa1NTUpN27d+vxxx/X/fff7+saAQAAPOLRU1fPPvusTp48qZ/85Cfq2/e7XTQ3N2v27NmM0QEAAH7Do6ATEhKirVu36plnntHhw4cVFhamW265RUlJSb6ur9vweDkAAObzKOi0GDFihEaMGOGrWnqU0+mU0+lUbW2tIiMj7S4HAAB0A4+CTlNTkzZt2qT3339fNTU1am5udlu/e/dunxQHAADgDY+CzuOPP65NmzZp6tSpSk1NlcPh8HVdAAAAXvMo6GzZskWvvfaapkyZ4ut60A1aHhc/sWKqzZUAANCzPHq8PCQkRDfeeKOvawEAAPApj4LOwoUL9W//9m+yLMvX9QAAAPiMR7euPvzwQ+3Zs0fvvPOORo4cqeDgYLf127Zt80lxAAAA3vAo6AwcOFAzZszwdS09inl0AAAwn0dBZ+PGjb6uo8cxjw4AAObzaIyOJF25ckXvvfeeXnrpJdXV1UmSzpw5owsXLvisOAAAAG941KNz8uRJ/fSnP1VFRYUaGho0efJkRURE6Pnnn9elS5e0fv16X9cJAADQZR716Dz++OMaPXq0vvnmG4WFhbmWz5gxQ++//77PigMAAPCGx09dffTRRwoJCXFbnpSUpNOnT/ukMAAAAG95FHSam5vbfFrp1KlTioiI8LoodI+WGZIBAAgUHt26mjx5slavXu367nA4dOHCBS1btozXQgAAAL/hUY/OCy+8oIyMDKWkpOjSpUv6h3/4Bx07dkxRUVH63e9+5+saAQAAPOJR0ImLi9OhQ4f0u9/9Tp9++qmam5s1Z84c/eM//qPb4GQAAAA7eRR0JCksLEwPP/ywHn74YV/W02OYGRkAAPN5FHQ2b97c4frZs2d7VExPMnVmZAYcAwDwVx4Fnccff9zt++XLl3Xx4kWFhIQoPDy8VwQdAABgPo+euvrmm2/cPhcuXNDRo0c1fvx4BiPDJ4Yu3kHvFADAax6/6+pqw4cP14oVK1r19iAwEEwAAP7IZ0FHkoKCgnTmzBlf7hJ+hDADAOhtPBqj8/vf/97tu2VZqqqq0po1a3TXXXf5pDAAAABveRR0fv7zn7t9dzgcuuGGGzRp0iStWrXKF3UBAAB4zeN3XQGd1XK768SKqTZXAgAIND4dowMAAOBPPOrRycvL6/S2BQUFnhwCfoQByACA3sqjoFNaWqpPP/1UV65c0U033SRJ+uMf/6igoCCNGjXKtZ3D4fBNlQAAAB7wKOhMmzZNERERevXVV3XddddJ+m4SwYceekgTJkzQwoULfVokAACAJzwao7Nq1Srl5+e7Qo4kXXfddXr22Wd7zVNXhYWFSklJ0ZgxY+wuBQAAdBOPgk5tba2+/vrrVstrampUV1fndVE9wel0qqysTMXFxXaXAgAAuolHQWfGjBl66KGH9MYbb+jUqVM6deqU3njjDc2ZM0fZ2dm+rhEAAMAjHo3RWb9+vRYtWqQHHnhAly9f/m5Hfftqzpw5WrlypU8LBAAA8JRHQSc8PFxr167VypUrdfz4cVmWpRtvvFH9+/f3dX0AAAAe82rCwKqqKlVVVWnEiBHq37+/LMvyVV0AAABe8yjonDt3Tj/5yU80YsQITZkyRVVVVZKkuXPn8mg5AADwGx4FnSeeeELBwcGqqKhQeHi4a/nMmTP17rvv+qw4AAAAb3g0Rud///d/tWvXLsXHx7stHz58uE6ePOmTwgAAALzlUY9OfX29W09Oi7Nnzyo0NNTrogAAAHzBox6du+++W5s3b9Yzzzwj6bt3WjU3N2vlypXKyMjwaYEw1/dfFnpixVQbKwHaxjUK9H4eBZ2VK1dq4sSJKikpUWNjo5588kl98cUX+vOf/6yPPvrI1zUCAAB4xKNbVykpKfrss890++23a/Lkyaqvr1d2drZKS0v1wx/+0Nc1AgAAeKTLPTqXL19WZmamXnrpJT399NPdURMAAIBPdLlHJzg4WJ9//rkcDkd31IMeNnTxDrdxCAAAmMSjW1ezZ8/WK6+84utaAAAAfMqjwciNjY367W9/q6KiIo0ePbrVO64KCgp8UhwAAIA3uhR0vvrqKw0dOlSff/65Ro0aJUn64x//6LYNt7QAAIC/6FLQGT58uKqqqrRnzx5J373y4cUXX1R0dHS3FAcAAOCNLo3Rufrt5O+8847q6+t9WlBPKSwsVEpKisaMGWN3KQAAoJt4NBi5xdXBpzdxOp0qKytTcXGx3aUAAIBu0qVbVw6Ho9UYHMbkmOHqR8x55BwAYIIuBR3LsvTggw+6Xtx56dIl5ebmtnrqatu2bb6rEAAAwENdCjo5OTlu3x944AGfFgMAAOBLXQo6Gzdu7K46AAAAfM6rwcgAAAD+zKOZkYGe8v1B0SdWTLWxEgBAb0SPDgAAMBZBBwAAGIuggy4bunhHu/PsdLQOAICeRtABAADGIugAAABj8dQVPNbVW1S+vqXVsj+exgIAtIceHQAAYCyCDgAAMBZBB/AQT5gBQPv85X8jCToAAMBYBB0AAGAsgg6M4i9dpQAA/0DQAQAAxiLoAAAAYxF0AACAsZgZGd2CcTIAAH9Ajw4AADCWEUFnxowZuu666/R3f/d3dpcCAAD8iBFB57HHHtPmzZvtLgMAAPgZI4JORkaGIiIi7C4DvQRz7QBA4LA96Ozbt0/Tpk1TXFycHA6Htm/f3mqbtWvXKjk5Wf369VN6err279/f84UCAIBex/agU19fr7S0NK1Zs6bN9Vu3btWCBQu0dOlSlZaWasKECcrKylJFRUUPVwoAAHob2x8vz8rKUlZWVrvrCwoKNGfOHM2dO1eStHr1au3atUvr1q1Tfn5+l4/X0NCghoYG1/fa2tquFw0AAHoF23t0OtLY2KiDBw8qMzPTbXlmZqYOHDjg0T7z8/MVGRnp+iQkJPiiVHiJcTMIFFzrQM/y66Bz9uxZNTU1KTo62m15dHS0qqurXd/vuece3Xfffdq5c6fi4+NVXFzc7j6XLFmi8+fPuz6VlZXdVj8AALCX7beuOsPhcLh9tyzLbdmuXbs6va/Q0FCFhob6rDYAAOC//LpHJyoqSkFBQW69N5JUU1PTqpcHAADgan4ddEJCQpSenq6ioiK35UVFRRo3bpxNVQEAgN7C9ltXFy5c0Jdffun6Xl5erkOHDmnQoEFKTExUXl6eZs2apdGjR2vs2LHasGGDKioqlJub69VxCwsLVVhYqKamJm9PAT7EIE0AgC/ZHnRKSkqUkZHh+p6XlydJysnJ0aZNmzRz5kydO3dOy5cvV1VVlVJTU7Vz504lJSV5dVyn0ymn06na2lpFRkZ6tS8AAOCfbA86EydOlGVZHW4zf/58zZ8/v4cqAgAApvDrMToAAADeIOgAAABj2X7ryi4MRu69PBmwzCBnAAhMAduj43Q6VVZW1uEsygAAoHcL2KADAADMR9ABAADGIugAAABjEXQAAICxeOqKp656PU+fqLr6dydWTPVFOQAAPxKwPTo8dQUAgPkCNugAAADzEXQAAICxCDoAAMBYBB0AAGAsgg4AADAWj5fzeDk60PIIuumPnn//UXvTzxVAYAnYHh0eLwcAwHwBG3QAAID5CDoAAMBYBB0AAGAsgg4AADAWQQcAABiLoAMAAIxF0AEAAMZiwkAmDOw1vj+pXXfv/+pJ87r72NcSKBMXom38+QOeC9geHSYMBADAfAEbdAAAgPkIOgAAwFgEHQAAYCyCDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYzEzMjMjG83uGY074+oavZ399upZdDua8bkzv2+rTmboBdBbBGyPDjMjAwBgvoANOgAAwHwEHQAAYCyCDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6AADAWAQdAABgLIIOAAAwFi/15KWe8JGOXiB69Qs2O3opZkf76cwx4K4zbW7isQF8J2B7dHipJwAA5gvYoAMAAMxH0AEAAMYi6AAAAGMRdAAAgLEIOgAAwFgEHQAAYCyCDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6AADAWAQdAABgLIIOAAAwFkEHAAAYq6/dBdilsLBQhYWFampqsrsUdIOhi3fYXUKHeqK+to5x9bITK6Z6tJ/ObNOy75Z1nTmWtzqqtSfr6El2n5fdx/dH378OaRf7BWyPjtPpVFlZmYqLi+0uBQAAdJOADToAAMB8BB0AAGAsgg4AADAWQQcAABiLoAMAAIxF0AEAAMYi6AAAAGMRdAAAgLEIOgAAwFgEHQAAYCyCDgAAMBZBBwAAGIugAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6AADAWAQdAABgLIIOAAAwFkEHAAAYi6ADAACMRdABAADGIugAAABjEXQAAICxjAg6b7/9tm666SYNHz5cv/3tb+0uBwAA+Im+dhfgrStXrigvL0979uzRgAEDNGrUKGVnZ2vQoEF2lwYAAGzW63t0PvnkE40cOVJDhgxRRESEpkyZol27dtldFgAA8AO2B519+/Zp2rRpiouLk8Ph0Pbt21tts3btWiUnJ6tfv35KT0/X/v37XevOnDmjIUOGuL7Hx8fr9OnTPVE6AADwc7YHnfr6eqWlpWnNmjVtrt+6dasWLFigpUuXqrS0VBMmTFBWVpYqKiokSZZltfqNw+Fo93gNDQ2qra11+wAAADPZPkYnKytLWVlZ7a4vKCjQnDlzNHfuXEnS6tWrtWvXLq1bt075+fkaMmSIWw/OqVOndMcdd7S7v/z8fD399NO+OwEYaejiHd2yrSk8Oefv/+bEiqlt7qdl+ffXfX9ZT/H1n2lPXCNtHaMzbXd1O3vb7h393s4/U3/T1t8HdA/be3Q60tjYqIMHDyozM9NteWZmpg4cOCBJuv322/X555/r9OnTqqur086dO3XPPfe0u88lS5bo/Pnzrk9lZWW3ngMAALCP7T06HTl79qyampoUHR3ttjw6OlrV1dWSpL59+2rVqlXKyMhQc3OznnzySV1//fXt7jM0NFShoaHdWjcAAPAPfh10Wlw95sayLLdl06dP1/Tp03u6LAAA4Of8+tZVVFSUgoKCXL03LWpqalr18gAAAFzNr4NOSEiI0tPTVVRU5La8qKhI48aN82rfhYWFSklJ0ZgxY7zaDwAA8F+237q6cOGCvvzyS9f38vJyHTp0SIMGDVJiYqLy8vI0a9YsjR49WmPHjtWGDRtUUVGh3Nxcr47rdDrldDpVW1uryMhIb08DAAD4IduDTklJiTIyMlzf8/LyJEk5OTnatGmTZs6cqXPnzmn58uWqqqpSamqqdu7cqaSkJLtKBgAAvYTtQWfixIltTvr3ffPnz9f8+fN7qCIAAGAKvx6jAwAA4A2CDgAAMFbABh2eugIAwHwBG3ScTqfKyspUXFxsdykAAKCbBGzQAQAA5iPoAAAAYxF0AACAsWyfR8duLXP41NbW+nzfzQ0Xfb5P9E4t11d3XhOeHKOj33hbsyf7/v7fw5Z1Xfm72ZlaO7O/jmruio7q8eS82vqNt8e4et9dbfeu/N6TP9Pe6Pt/Ju2da2e26e26+8+7Zb/XmovPYV1rC0MVFhaqsLBQjY2NOn78uN3lAAAAD1RWVio+Pr7d9QEbdFo0NzfrzJkzioiIkMPh8Hp/tbW1SkhIUGVlpQYMGOCDCkGb+hbt6Xu0qW/Rnr5nYptalqW6ujrFxcWpT5/2R+IE/K2rPn36dJgEPTVgwABjLiZ/QZv6Fu3pe7Spb9Gevmdam3bmpdwMRgYAAMYi6AAAAGMRdHwsNDRUy5YtU2hoqN2lGIM29S3a0/doU9+iPX0vkNs04AcjAwAAc9GjAwAAjEXQAQAAxiLoAAAAYxF0AACAsQg6PrZ27VolJyerX79+Sk9P1/79++0uqVd46qmn5HA43D4xMTGu9ZZl6amnnlJcXJzCwsI0ceJEffHFFzZW7H/27dunadOmKS4uTg6HQ9u3b3db35k2bGho0KOPPqqoqCj1799f06dP16lTp3rwLPzHtdrzwQcfbHXN3nnnnW7b0J5/lZ+frzFjxigiIkKDBw/Wz3/+cx09etRtG67RzutMe3KNfoeg40Nbt27VggULtHTpUpWWlmrChAnKyspSRUWF3aX1CiNHjlRVVZXrc+TIEde6559/XgUFBVqzZo2Ki4sVExOjyZMnq66uzsaK/Ut9fb3S0tK0Zs2aNtd3pg0XLFigN998U1u2bNGHH36oCxcu6N5771VTU1NPnYbfuFZ7StJPf/pTt2t2586dbutpz7/au3evnE6nPv74YxUVFenKlSvKzMxUfX29axuu0c7rTHtKXKOSJAs+c/vtt1u5ubluy26++WZr8eLFNlXUeyxbtsxKS0trc11zc7MVExNjrVixwrXs0qVLVmRkpLV+/foeqrB3kWS9+eabru+dacNvv/3WCg4OtrZs2eLa5vTp01afPn2sd999t8dq90dXt6dlWVZOTo71s5/9rN3f0J4dq6mpsSRZe/futSyLa9RbV7enZXGNtqBHx0caGxt18OBBZWZmui3PzMzUgQMHbKqqdzl27Jji4uKUnJys+++/X1999ZUkqby8XNXV1W5tGxoaqh//+Me0bSd1pg0PHjyoy5cvu20TFxen1NRU2rkdH3zwgQYPHqwRI0Zo3rx5qqmpca2jPTt2/vx5SdKgQYMkcY166+r2bME1yq0rnzl79qyampoUHR3ttjw6OlrV1dU2VdV73HHHHdq8ebN27dqll19+WdXV1Ro3bpzOnTvnaj/a1nOdacPq6mqFhITouuuua3cb/FVWVpb+67/+S7t379aqVatUXFysSZMmqaGhQRLt2RHLspSXl6fx48crNTVVEteoN9pqT4lrtEXAv73c1xwOh9t3y7JaLUNrWVlZrv++5ZZbNHbsWP3whz/Uq6++6ho8R9t6z5M2pJ3bNnPmTNd/p6amavTo0UpKStKOHTuUnZ3d7u9oT+mRRx7RZ599pg8//LDVOq7RrmuvPblGv0OPjo9ERUUpKCioVQquqalp9f9QcG39+/fXLbfcomPHjrmevqJtPdeZNoyJiVFjY6O++eabdrdB+2JjY5WUlKRjx45Joj3b8+ijj+r3v/+99uzZo/j4eNdyrlHPtNeebQnUa5Sg4yMhISFKT09XUVGR2/KioiKNGzfOpqp6r4aGBv3hD39QbGyskpOTFRMT49a2jY2N2rt3L23bSZ1pw/T0dAUHB7ttU1VVpc8//5x27oRz586psrJSsbGxkmjPq1mWpUceeUTbtm3T7t27lZyc7Laea7RrrtWebQnYa9SeMdBm2rJlixUcHGy98sorVllZmbVgwQKrf//+1okTJ+wuze8tXLjQ+uCDD6yvvvrK+vjjj617773XioiIcLXdihUrrMjISGvbtm3WkSNHrL//+7+3YmNjrdraWpsr9x91dXVWaWmpVVpaakmyCgoKrNLSUuvkyZOWZXWuDXNzc634+Hjrvffesz799FNr0qRJVlpamnXlyhW7Tss2HbVnXV2dtXDhQuvAgQNWeXm5tWfPHmvs2LHWkCFDaM92/PKXv7QiIyOtDz74wKqqqnJ9Ll686NqGa7TzrtWeXKN/RdDxscLCQispKckKCQmxRo0a5faoH9o3c+ZMKzY21goODrbi4uKs7Oxs64svvnCtb25utpYtW2bFxMRYoaGh1t13320dOXLExor9z549eyxJrT45OTmWZXWuDf/yl79YjzzyiDVo0CArLCzMuvfee62KigobzsZ+HbXnxYsXrczMTOuGG26wgoODrcTERCsnJ6dVW9Gef9VWW0qyNm7c6NqGa7TzrtWeXKN/5bAsy+q5/iMAAICewxgdAABgLIIOAAAwFkEHAAAYi6ADAACMRdABAADGIugAAABjEXQAAICxCDoAAMBYBB0AAGAsgg4AADAWQQcAABiLoAMAAIz1/9dTtLVMuSsaAAAAAElFTkSuQmCC\n"
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Get number of nodes and edges and plot histogram of edges showing number of nodes with X edges \n",
    "total_gene_a = reactome_ppi['extracted_ENSG 1'].tolist()\n",
    "total_gene_b = reactome_ppi['extracted_ENSG 2'].tolist()\n",
    "total_gene = total_gene_a + total_gene_b\n",
    "\n",
    "total_gene_df = pd.DataFrame(total_gene, columns=['gene'])\n",
    "\n",
    "total_gene_df['edges'] = total_gene_df['gene'].map(total_gene_df['gene'].value_counts())\n",
    "\n",
    "print(len(total_gene_a))\n",
    "print(len(total_gene_b))\n",
    "print(len(total_gene))\n",
    "\n",
    "total_gene_df.head()\n",
    "\n",
    "unique_gene_df = total_gene_df.drop_duplicates('gene') \n",
    "unique_gene_df.sort_values(by='edges', ascending=False)\n",
    "\n",
    "fig = unique_gene_df.plot(kind='hist', bins=200)\n",
    "\n",
    "plt.yscale('log')\n",
    "plt.show()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:55.787797800Z",
     "start_time": "2023-12-13T11:53:54.975096800Z"
    }
   },
   "id": "2fa02ebdb6baf5dd"
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4277\n",
      "4263\n",
      "5478\n"
     ]
    }
   ],
   "source": [
    "# See how many unique nodes there are in total\n",
    "unique_gene_a = reactome_ppi['extracted_ENSG 1'].unique().tolist()\n",
    "unique_gene_b = reactome_ppi['extracted_ENSG 2'].unique().tolist()\n",
    "unique_gene_all = set(unique_gene_a + unique_gene_b)\n",
    "\n",
    "print(len(unique_gene_a))\n",
    "print(len(unique_gene_b))\n",
    "print(len(unique_gene_all))\n",
    "\n",
    "#df = pd.DataFrame.from_dict(count, orient='index')\n",
    "#df.plot(kind='bar')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:55.870843100Z",
     "start_time": "2023-12-13T11:53:55.788401300Z"
    }
   },
   "id": "d7c55fd4c10e2704"
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "outputs": [
    {
     "data": {
      "text/plain": "                  gene1                gene2\n0   ['ENSG00000164692']  ['ENSG00000108821']\n18  ['ENSG00000108821']  ['ENSG00000164692']\n72  ['ENSG00000108947']  ['ENSG00000090776']\n73  ['ENSG00000125266']  ['ENSG00000108947']\n74  ['ENSG00000125266']  ['ENSG00000090776']",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>gene1</th>\n      <th>gene2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>['ENSG00000164692']</td>\n      <td>['ENSG00000108821']</td>\n    </tr>\n    <tr>\n      <th>18</th>\n      <td>['ENSG00000108821']</td>\n      <td>['ENSG00000164692']</td>\n    </tr>\n    <tr>\n      <th>72</th>\n      <td>['ENSG00000108947']</td>\n      <td>['ENSG00000090776']</td>\n    </tr>\n    <tr>\n      <th>73</th>\n      <td>['ENSG00000125266']</td>\n      <td>['ENSG00000108947']</td>\n    </tr>\n    <tr>\n      <th>74</th>\n      <td>['ENSG00000125266']</td>\n      <td>['ENSG00000090776']</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_ppi.columns = ['gene1', 'gene2']\n",
    "reactome_ppi.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:55.881842800Z",
     "start_time": "2023-12-13T11:53:55.814165500Z"
    }
   },
   "id": "6a1164cf16d7d04f"
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "outputs": [
    {
     "data": {
      "text/plain": "              gene1            gene2\n0   ENSG00000164692  ENSG00000108821\n18  ENSG00000108821  ENSG00000164692\n72  ENSG00000108947  ENSG00000090776\n73  ENSG00000125266  ENSG00000108947\n74  ENSG00000125266  ENSG00000090776",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>gene1</th>\n      <th>gene2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>ENSG00000164692</td>\n      <td>ENSG00000108821</td>\n    </tr>\n    <tr>\n      <th>18</th>\n      <td>ENSG00000108821</td>\n      <td>ENSG00000164692</td>\n    </tr>\n    <tr>\n      <th>72</th>\n      <td>ENSG00000108947</td>\n      <td>ENSG00000090776</td>\n    </tr>\n    <tr>\n      <th>73</th>\n      <td>ENSG00000125266</td>\n      <td>ENSG00000108947</td>\n    </tr>\n    <tr>\n      <th>74</th>\n      <td>ENSG00000125266</td>\n      <td>ENSG00000090776</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reactome_ppi['gene1'] = reactome_ppi['gene1'].str.replace(\"\\['\", \"\", regex=True).str.replace(\"']\", \"\", regex=True)\n",
    "reactome_ppi['gene2'] = reactome_ppi['gene2'].str.replace(\"\\['\", \"\", regex=True).str.replace(\"']\", \"\", regex=True)\n",
    "\n",
    "reactome_ppi.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:55.954738400Z",
     "start_time": "2023-12-13T11:53:55.823052400Z"
    }
   },
   "id": "aed3c534d8340336"
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "outputs": [],
   "source": [
    "reactome_ppi.to_csv(data_path + '/reactome/reactome_ppi.csv', index=False)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:56.132135300Z",
     "start_time": "2023-12-13T11:53:55.924778200Z"
    }
   },
   "id": "c8667355c8aa85e0"
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "outputs": [],
   "source": [
    "gene_converter = pd.read_csv(data_path + '/biomart/gene_multi_converter.csv')"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:56.512994400Z",
     "start_time": "2023-12-13T11:53:55.990553200Z"
    }
   },
   "id": "3f6493337865561f"
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "outputs": [
    {
     "data": {
      "text/plain": "   Unnamed: 0  Entrez gene ID  \\\n0           0            4535   \n1           1            4536   \n2           2            4512   \n3           3            4513   \n4           4            4509   \n\n                                               Fasta Gene name  \\\n0  MPMANLLLLIVPILIAMAFLMLTERKILGYMQLRKGPNVVGPYGLL...    MT-ND1   \n1  INPLAQPVIYSTIFAGTLITALSSHWFFTWVGLEMNMLAFIPVLTK...    MT-ND2   \n2  MFADRWLFSTNHKDIGTLYLLFGAWAGVLGTALSLLIRAELGQPGN...    MT-CO1   \n3  MAHAAQVGLQDATSPIMEELITFHDHALMIIFLICFLVLYALFLTL...    MT-CO2   \n4  MPQLNTTVWPTMITPMLLTLFLITQLKMLNTNYHLPPSPKPMKMKN...   MT-ATP8   \n\n    Gene stable ID Protein stable ID RefSeq peptide ID Transcript stable ID  \\\n0  ENSG00000198888   ENSP00000354687      YP_003024026      ENST00000361390   \n1  ENSG00000198763   ENSP00000355046      YP_003024027      ENST00000361453   \n2  ENSG00000198804   ENSP00000354499      YP_003024028      ENST00000361624   \n3  ENSG00000198712   ENSP00000354876      YP_003024029      ENST00000361739   \n4  ENSG00000228253   ENSP00000355265      YP_003024030      ENST00000361851   \n\n  UniProtKB/Swiss-Prot ID UniProtKB/TrEMBL ID  \n0                  P03886              U5Z754  \n1                  P03891              Q7GXY9  \n2                  P00395              U5YWV7  \n3                  P00403              U5Z487  \n4                  P03928              U5YV54  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Unnamed: 0</th>\n      <th>Entrez gene ID</th>\n      <th>Fasta</th>\n      <th>Gene name</th>\n      <th>Gene stable ID</th>\n      <th>Protein stable ID</th>\n      <th>RefSeq peptide ID</th>\n      <th>Transcript stable ID</th>\n      <th>UniProtKB/Swiss-Prot ID</th>\n      <th>UniProtKB/TrEMBL ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>4535</td>\n      <td>MPMANLLLLIVPILIAMAFLMLTERKILGYMQLRKGPNVVGPYGLL...</td>\n      <td>MT-ND1</td>\n      <td>ENSG00000198888</td>\n      <td>ENSP00000354687</td>\n      <td>YP_003024026</td>\n      <td>ENST00000361390</td>\n      <td>P03886</td>\n      <td>U5Z754</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>1</td>\n      <td>4536</td>\n      <td>INPLAQPVIYSTIFAGTLITALSSHWFFTWVGLEMNMLAFIPVLTK...</td>\n      <td>MT-ND2</td>\n      <td>ENSG00000198763</td>\n      <td>ENSP00000355046</td>\n      <td>YP_003024027</td>\n      <td>ENST00000361453</td>\n      <td>P03891</td>\n      <td>Q7GXY9</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>2</td>\n      <td>4512</td>\n      <td>MFADRWLFSTNHKDIGTLYLLFGAWAGVLGTALSLLIRAELGQPGN...</td>\n      <td>MT-CO1</td>\n      <td>ENSG00000198804</td>\n      <td>ENSP00000354499</td>\n      <td>YP_003024028</td>\n      <td>ENST00000361624</td>\n      <td>P00395</td>\n      <td>U5YWV7</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>3</td>\n      <td>4513</td>\n      <td>MAHAAQVGLQDATSPIMEELITFHDHALMIIFLICFLVLYALFLTL...</td>\n      <td>MT-CO2</td>\n      <td>ENSG00000198712</td>\n      <td>ENSP00000354876</td>\n      <td>YP_003024029</td>\n      <td>ENST00000361739</td>\n      <td>P00403</td>\n      <td>U5Z487</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>4</td>\n      <td>4509</td>\n      <td>MPQLNTTVWPTMITPMLLTLFLITQLKMLNTNYHLPPSPKPMKMKN...</td>\n      <td>MT-ATP8</td>\n      <td>ENSG00000228253</td>\n      <td>ENSP00000355265</td>\n      <td>YP_003024030</td>\n      <td>ENST00000361851</td>\n      <td>P03928</td>\n      <td>U5YV54</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_converter.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:56.529246700Z",
     "start_time": "2023-12-13T11:53:56.513994100Z"
    }
   },
   "id": "6e66e792aad24228"
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "outputs": [
    {
     "data": {
      "text/plain": "3678"
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(gene_converter[gene_converter['Gene stable ID'].isin(unique_gene_all)])"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T11:53:56.602283700Z",
     "start_time": "2023-12-13T11:53:56.531247Z"
    }
   },
   "id": "3045bb6f3600acf9"
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[BiomartServer:'http://useast.ensembl.org/biomart/martservice'] Fetching datasets\n",
      "[BiomartServer:'http://useast.ensembl.org/biomart/martservice'] Fetching databases\n",
      "[BiomartDatabase:'Ensembl Genes 110'] Fetching datasets\n",
      "[BiomartDatabase:'Mouse strains 110'] Fetching datasets\n",
      "[BiomartDatabase:'Sequence'] Fetching datasets\n",
      "[BiomartDatabase:'Ontology'] Fetching datasets\n",
      "[BiomartDatabase:'Genomic features 110'] Fetching datasets\n",
      "[BiomartDatabase:'Ensembl Variation 110'] Fetching datasets\n",
      "[BiomartDatabase:'Ensembl Regulation 110'] Fetching datasets\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Fetching attributes\n",
      "{'3_utr_end': '3' UTR end' (default: False),\n",
      " '3_utr_start': '3' UTR start' (default: False),\n",
      " '3utr': '3' UTR' (default: False),\n",
      " '5_utr_end': '5' UTR end' (default: False),\n",
      " '5_utr_start': '5' UTR start' (default: False),\n",
      " '5utr': '5' UTR' (default: False),\n",
      " 'abrachyrhynchus_homolog_associated_gene_name': 'Pink-footed goose gene name' (default: False),\n",
      " 'abrachyrhynchus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_chrom_end': 'Pink-footed goose chromosome/scaffold end (bp)' (default: False),\n",
      " 'abrachyrhynchus_homolog_chrom_start': 'Pink-footed goose chromosome/scaffold start (bp)' (default: False),\n",
      " 'abrachyrhynchus_homolog_chromosome': 'Pink-footed goose chromosome/scaffold name' (default: False),\n",
      " 'abrachyrhynchus_homolog_ensembl_gene': 'Pink-footed goose gene stable ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_ensembl_peptide': 'Pink-footed goose protein or transcript stable ID' (default: False),\n",
      " 'abrachyrhynchus_homolog_goc_score': 'Pink-footed goose Gene-order conservation score' (default: False),\n",
      " 'abrachyrhynchus_homolog_orthology_confidence': 'Pink-footed goose orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'abrachyrhynchus_homolog_orthology_type': 'Pink-footed goose homology type' (default: False),\n",
      " 'abrachyrhynchus_homolog_perc_id': '%id. target Pink-footed goose gene identical to query gene' (default: False),\n",
      " 'abrachyrhynchus_homolog_perc_id_r1': '%id. query gene identical to target Pink-footed goose gene' (default: False),\n",
      " 'abrachyrhynchus_homolog_subtype': 'Last common ancestor with Pink-footed goose' (default: False),\n",
      " 'abrachyrhynchus_homolog_wga_coverage': 'Pink-footed goose Whole-genome alignment coverage' (default: False),\n",
      " 'acalliptera_homolog_associated_gene_name': 'Eastern happy gene name' (default: False),\n",
      " 'acalliptera_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acalliptera_homolog_chrom_end': 'Eastern happy chromosome/scaffold end (bp)' (default: False),\n",
      " 'acalliptera_homolog_chrom_start': 'Eastern happy chromosome/scaffold start (bp)' (default: False),\n",
      " 'acalliptera_homolog_chromosome': 'Eastern happy chromosome/scaffold name' (default: False),\n",
      " 'acalliptera_homolog_ensembl_gene': 'Eastern happy gene stable ID' (default: False),\n",
      " 'acalliptera_homolog_ensembl_peptide': 'Eastern happy protein or transcript stable ID' (default: False),\n",
      " 'acalliptera_homolog_goc_score': 'Eastern happy Gene-order conservation score' (default: False),\n",
      " 'acalliptera_homolog_orthology_confidence': 'Eastern happy orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acalliptera_homolog_orthology_type': 'Eastern happy homology type' (default: False),\n",
      " 'acalliptera_homolog_perc_id': '%id. target Eastern happy gene identical to query gene' (default: False),\n",
      " 'acalliptera_homolog_perc_id_r1': '%id. query gene identical to target Eastern happy gene' (default: False),\n",
      " 'acalliptera_homolog_subtype': 'Last common ancestor with Eastern happy' (default: False),\n",
      " 'acalliptera_homolog_wga_coverage': 'Eastern happy Whole-genome alignment coverage' (default: False),\n",
      " 'acarolinensis_homolog_associated_gene_name': 'Green anole gene name' (default: False),\n",
      " 'acarolinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acarolinensis_homolog_chrom_end': 'Green anole chromosome/scaffold end (bp)' (default: False),\n",
      " 'acarolinensis_homolog_chrom_start': 'Green anole chromosome/scaffold start (bp)' (default: False),\n",
      " 'acarolinensis_homolog_chromosome': 'Green anole chromosome/scaffold name' (default: False),\n",
      " 'acarolinensis_homolog_ensembl_gene': 'Green anole gene stable ID' (default: False),\n",
      " 'acarolinensis_homolog_ensembl_peptide': 'Green anole protein or transcript stable ID' (default: False),\n",
      " 'acarolinensis_homolog_goc_score': 'Green anole Gene-order conservation score' (default: False),\n",
      " 'acarolinensis_homolog_orthology_confidence': 'Green anole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acarolinensis_homolog_orthology_type': 'Green anole homology type' (default: False),\n",
      " 'acarolinensis_homolog_perc_id': '%id. target Green anole gene identical to query gene' (default: False),\n",
      " 'acarolinensis_homolog_perc_id_r1': '%id. query gene identical to target Green anole gene' (default: False),\n",
      " 'acarolinensis_homolog_subtype': 'Last common ancestor with Green anole' (default: False),\n",
      " 'acchrysaetos_homolog_associated_gene_name': 'Golden eagle gene name' (default: False),\n",
      " 'acchrysaetos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acchrysaetos_homolog_chrom_end': 'Golden eagle chromosome/scaffold end (bp)' (default: False),\n",
      " 'acchrysaetos_homolog_chrom_start': 'Golden eagle chromosome/scaffold start (bp)' (default: False),\n",
      " 'acchrysaetos_homolog_chromosome': 'Golden eagle chromosome/scaffold name' (default: False),\n",
      " 'acchrysaetos_homolog_ensembl_gene': 'Golden eagle gene stable ID' (default: False),\n",
      " 'acchrysaetos_homolog_ensembl_peptide': 'Golden eagle protein or transcript stable ID' (default: False),\n",
      " 'acchrysaetos_homolog_goc_score': 'Golden eagle Gene-order conservation score' (default: False),\n",
      " 'acchrysaetos_homolog_orthology_confidence': 'Golden eagle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acchrysaetos_homolog_orthology_type': 'Golden eagle homology type' (default: False),\n",
      " 'acchrysaetos_homolog_perc_id': '%id. target Golden eagle gene identical to query gene' (default: False),\n",
      " 'acchrysaetos_homolog_perc_id_r1': '%id. query gene identical to target Golden eagle gene' (default: False),\n",
      " 'acchrysaetos_homolog_subtype': 'Last common ancestor with Golden eagle' (default: False),\n",
      " 'acchrysaetos_homolog_wga_coverage': 'Golden eagle Whole-genome alignment coverage' (default: False),\n",
      " 'acitrinellus_homolog_associated_gene_name': 'Midas cichlid gene name' (default: False),\n",
      " 'acitrinellus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'acitrinellus_homolog_chrom_end': 'Midas cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'acitrinellus_homolog_chrom_start': 'Midas cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'acitrinellus_homolog_chromosome': 'Midas cichlid chromosome/scaffold name' (default: False),\n",
      " 'acitrinellus_homolog_ensembl_gene': 'Midas cichlid gene stable ID' (default: False),\n",
      " 'acitrinellus_homolog_ensembl_peptide': 'Midas cichlid protein or transcript stable ID' (default: False),\n",
      " 'acitrinellus_homolog_goc_score': 'Midas cichlid Gene-order conservation score' (default: False),\n",
      " 'acitrinellus_homolog_orthology_confidence': 'Midas cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'acitrinellus_homolog_orthology_type': 'Midas cichlid homology type' (default: False),\n",
      " 'acitrinellus_homolog_perc_id': '%id. target Midas cichlid gene identical to query gene' (default: False),\n",
      " 'acitrinellus_homolog_perc_id_r1': '%id. query gene identical to target Midas cichlid gene' (default: False),\n",
      " 'acitrinellus_homolog_subtype': 'Last common ancestor with Midas cichlid' (default: False),\n",
      " 'acitrinellus_homolog_wga_coverage': 'Midas cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'affy_hc_g110': 'AFFY HC G110 probe' (default: False),\n",
      " 'affy_hg_focus': 'AFFY HG Focus probe' (default: False),\n",
      " 'affy_hg_u133_plus_2': 'AFFY HG U133 Plus 2 probe' (default: False),\n",
      " 'affy_hg_u133a_2': 'AFFY HG U133A 2 probe' (default: False),\n",
      " 'affy_hg_u133b': 'AFFY HG U133B probe' (default: False),\n",
      " 'affy_hg_u95a': 'AFFY HG U95A probe' (default: False),\n",
      " 'affy_hg_u95av2': 'AFFY HG U95Av2 probe' (default: False),\n",
      " 'affy_hg_u95b': 'AFFY HG U95B probe' (default: False),\n",
      " 'affy_hg_u95c': 'AFFY HG U95C probe' (default: False),\n",
      " 'affy_hg_u95d': 'AFFY HG U95D probe' (default: False),\n",
      " 'affy_hg_u95e': 'AFFY HG U95E probe' (default: False),\n",
      " 'affy_ht_hg_u133_plus_pm': 'AFFY HT HG U133 Plus PM probe' (default: False),\n",
      " 'affy_hta_2_0': 'AFFY HTA 2 0 probe' (default: False),\n",
      " 'affy_huex_1_0_st_v2': 'AFFY HuEx 1 0 st v2 probe' (default: False),\n",
      " 'affy_hugene_1_0_st_v1': 'AFFY HuGene 1 0 st v1 probe' (default: False),\n",
      " 'affy_hugene_2_0_st_v1': 'AFFY HuGene 2 0 st v1 probe' (default: False),\n",
      " 'affy_hugene_2_1_st_v1': 'AFFY HuGene 2 1 st v1 probe' (default: False),\n",
      " 'affy_hugenefl': 'AFFY HuGeneFL probe' (default: False),\n",
      " 'affy_primeview': 'AFFY PrimeView probe' (default: False),\n",
      " 'affy_u133_x3p': 'AFFY U133 X3P probe' (default: False),\n",
      " 'agilent_cgh_44b': 'AGILENT CGH 44b probe' (default: False),\n",
      " 'agilent_gpl19072': 'AGILENT GPL19072 probe' (default: False),\n",
      " 'agilent_gpl26966': 'AGILENT GPL26966 probe' (default: False),\n",
      " 'agilent_gpl6848': 'AGILENT GPL6848 probe' (default: False),\n",
      " 'agilent_sureprint_g3_ge_8x60k': 'AGILENT SurePrint G3 GE 8x60k probe' (default: False),\n",
      " 'agilent_sureprint_g3_ge_8x60k_v2': 'AGILENT SurePrint G3 GE 8x60k v2 probe' (default: False),\n",
      " 'agilent_wholegenome': 'AGILENT WholeGenome probe' (default: False),\n",
      " 'agilent_wholegenome_4x44k_v1': 'AGILENT WholeGenome 4x44k v1 probe' (default: False),\n",
      " 'agilent_wholegenome_4x44k_v2': 'AGILENT WholeGenome 4x44k v2 probe' (default: False),\n",
      " 'allele': 'Variant alleles' (default: False),\n",
      " 'allele_string_2076': 'Consequence specific allele' (default: False),\n",
      " 'alphafold': 'AFDB-ENSP mappings' (default: False),\n",
      " 'alphafold_end': 'AFDB-ENSP mappings end' (default: False),\n",
      " 'alphafold_start': 'AFDB-ENSP mappings start' (default: False),\n",
      " 'amelanoleuca_homolog_associated_gene_name': 'Giant panda gene name' (default: False),\n",
      " 'amelanoleuca_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'amelanoleuca_homolog_chrom_end': 'Giant panda chromosome/scaffold end (bp)' (default: False),\n",
      " 'amelanoleuca_homolog_chrom_start': 'Giant panda chromosome/scaffold start (bp)' (default: False),\n",
      " 'amelanoleuca_homolog_chromosome': 'Giant panda chromosome/scaffold name' (default: False),\n",
      " 'amelanoleuca_homolog_ensembl_gene': 'Giant panda gene stable ID' (default: False),\n",
      " 'amelanoleuca_homolog_ensembl_peptide': 'Giant panda protein or transcript stable ID' (default: False),\n",
      " 'amelanoleuca_homolog_goc_score': 'Giant panda Gene-order conservation score' (default: False),\n",
      " 'amelanoleuca_homolog_orthology_confidence': 'Giant panda orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'amelanoleuca_homolog_orthology_type': 'Giant panda homology type' (default: False),\n",
      " 'amelanoleuca_homolog_perc_id': '%id. target Giant panda gene identical to query gene' (default: False),\n",
      " 'amelanoleuca_homolog_perc_id_r1': '%id. query gene identical to target Giant panda gene' (default: False),\n",
      " 'amelanoleuca_homolog_subtype': 'Last common ancestor with Giant panda' (default: False),\n",
      " 'amexicanus_homolog_associated_gene_name': 'Mexican tetra gene name' (default: False),\n",
      " 'amexicanus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'amexicanus_homolog_chrom_end': 'Mexican tetra chromosome/scaffold end (bp)' (default: False),\n",
      " 'amexicanus_homolog_chrom_start': 'Mexican tetra chromosome/scaffold start (bp)' (default: False),\n",
      " 'amexicanus_homolog_chromosome': 'Mexican tetra chromosome/scaffold name' (default: False),\n",
      " 'amexicanus_homolog_ensembl_gene': 'Mexican tetra gene stable ID' (default: False),\n",
      " 'amexicanus_homolog_ensembl_peptide': 'Mexican tetra protein or transcript stable ID' (default: False),\n",
      " 'amexicanus_homolog_goc_score': 'Mexican tetra Gene-order conservation score' (default: False),\n",
      " 'amexicanus_homolog_orthology_confidence': 'Mexican tetra orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'amexicanus_homolog_orthology_type': 'Mexican tetra homology type' (default: False),\n",
      " 'amexicanus_homolog_perc_id': '%id. target Mexican tetra gene identical to query gene' (default: False),\n",
      " 'amexicanus_homolog_perc_id_r1': '%id. query gene identical to target Mexican tetra gene' (default: False),\n",
      " 'amexicanus_homolog_subtype': 'Last common ancestor with Mexican tetra' (default: False),\n",
      " 'amexicanus_homolog_wga_coverage': 'Mexican tetra Whole-genome alignment coverage' (default: False),\n",
      " 'anancymaae_homolog_associated_gene_name': 'Ma's night monkey gene name' (default: False),\n",
      " 'anancymaae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'anancymaae_homolog_chrom_end': 'Ma's night monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'anancymaae_homolog_chrom_start': 'Ma's night monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'anancymaae_homolog_chromosome': 'Ma's night monkey chromosome/scaffold name' (default: False),\n",
      " 'anancymaae_homolog_ensembl_gene': 'Ma's night monkey gene stable ID' (default: False),\n",
      " 'anancymaae_homolog_ensembl_peptide': 'Ma's night monkey protein or transcript stable ID' (default: False),\n",
      " 'anancymaae_homolog_goc_score': 'Ma's night monkey Gene-order conservation score' (default: False),\n",
      " 'anancymaae_homolog_orthology_confidence': 'Ma's night monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'anancymaae_homolog_orthology_type': 'Ma's night monkey homology type' (default: False),\n",
      " 'anancymaae_homolog_perc_id': '%id. target Ma's night monkey gene identical to query gene' (default: False),\n",
      " 'anancymaae_homolog_perc_id_r1': '%id. query gene identical to target Ma's night monkey gene' (default: False),\n",
      " 'anancymaae_homolog_subtype': 'Last common ancestor with Ma's night monkey' (default: False),\n",
      " 'anancymaae_homolog_wga_coverage': 'Ma's night monkey Whole-genome alignment coverage' (default: False),\n",
      " 'aocellaris_homolog_associated_gene_name': 'Clown anemonefish gene name' (default: False),\n",
      " 'aocellaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'aocellaris_homolog_chrom_end': 'Clown anemonefish chromosome/scaffold end (bp)' (default: False),\n",
      " 'aocellaris_homolog_chrom_start': 'Clown anemonefish chromosome/scaffold start (bp)' (default: False),\n",
      " 'aocellaris_homolog_chromosome': 'Clown anemonefish chromosome/scaffold name' (default: False),\n",
      " 'aocellaris_homolog_ensembl_gene': 'Clown anemonefish gene stable ID' (default: False),\n",
      " 'aocellaris_homolog_ensembl_peptide': 'Clown anemonefish protein or transcript stable ID' (default: False),\n",
      " 'aocellaris_homolog_goc_score': 'Clown anemonefish Gene-order conservation score' (default: False),\n",
      " 'aocellaris_homolog_orthology_confidence': 'Clown anemonefish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'aocellaris_homolog_orthology_type': 'Clown anemonefish homology type' (default: False),\n",
      " 'aocellaris_homolog_perc_id': '%id. target Clown anemonefish gene identical to query gene' (default: False),\n",
      " 'aocellaris_homolog_perc_id_r1': '%id. query gene identical to target Clown anemonefish gene' (default: False),\n",
      " 'aocellaris_homolog_subtype': 'Last common ancestor with Clown anemonefish' (default: False),\n",
      " 'aocellaris_homolog_wga_coverage': 'Clown anemonefish Whole-genome alignment coverage' (default: False),\n",
      " 'apercula_homolog_associated_gene_name': 'Orange clownfish gene name' (default: False),\n",
      " 'apercula_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'apercula_homolog_chrom_end': 'Orange clownfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'apercula_homolog_chrom_start': 'Orange clownfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'apercula_homolog_chromosome': 'Orange clownfish chromosome/scaffold name' (default: False),\n",
      " 'apercula_homolog_ensembl_gene': 'Orange clownfish gene stable ID' (default: False),\n",
      " 'apercula_homolog_ensembl_peptide': 'Orange clownfish protein or transcript stable ID' (default: False),\n",
      " 'apercula_homolog_goc_score': 'Orange clownfish Gene-order conservation score' (default: False),\n",
      " 'apercula_homolog_orthology_confidence': 'Orange clownfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'apercula_homolog_orthology_type': 'Orange clownfish homology type' (default: False),\n",
      " 'apercula_homolog_perc_id': '%id. target Orange clownfish gene identical to query gene' (default: False),\n",
      " 'apercula_homolog_perc_id_r1': '%id. query gene identical to target Orange clownfish gene' (default: False),\n",
      " 'apercula_homolog_subtype': 'Last common ancestor with Orange clownfish' (default: False),\n",
      " 'apercula_homolog_wga_coverage': 'Orange clownfish Whole-genome alignment coverage' (default: False),\n",
      " 'apolyacanthus_homolog_associated_gene_name': 'Spiny chromis gene name' (default: False),\n",
      " 'apolyacanthus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'apolyacanthus_homolog_chrom_end': 'Spiny chromis chromosome/scaffold end (bp)' (default: False),\n",
      " 'apolyacanthus_homolog_chrom_start': 'Spiny chromis chromosome/scaffold start (bp)' (default: False),\n",
      " 'apolyacanthus_homolog_chromosome': 'Spiny chromis chromosome/scaffold name' (default: False),\n",
      " 'apolyacanthus_homolog_ensembl_gene': 'Spiny chromis gene stable ID' (default: False),\n",
      " 'apolyacanthus_homolog_ensembl_peptide': 'Spiny chromis protein or transcript stable ID' (default: False),\n",
      " 'apolyacanthus_homolog_goc_score': 'Spiny chromis Gene-order conservation score' (default: False),\n",
      " 'apolyacanthus_homolog_orthology_confidence': 'Spiny chromis orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'apolyacanthus_homolog_orthology_type': 'Spiny chromis homology type' (default: False),\n",
      " 'apolyacanthus_homolog_perc_id': '%id. target Spiny chromis gene identical to query gene' (default: False),\n",
      " 'apolyacanthus_homolog_perc_id_r1': '%id. query gene identical to target Spiny chromis gene' (default: False),\n",
      " 'apolyacanthus_homolog_subtype': 'Last common ancestor with Spiny chromis' (default: False),\n",
      " 'apolyacanthus_homolog_wga_coverage': 'Spiny chromis Whole-genome alignment coverage' (default: False),\n",
      " 'applatyrhynchos_homolog_associated_gene_name': 'Duck gene name' (default: False),\n",
      " 'applatyrhynchos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'applatyrhynchos_homolog_chrom_end': 'Duck chromosome/scaffold end (bp)' (default: False),\n",
      " 'applatyrhynchos_homolog_chrom_start': 'Duck chromosome/scaffold start (bp)' (default: False),\n",
      " 'applatyrhynchos_homolog_chromosome': 'Duck chromosome/scaffold name' (default: False),\n",
      " 'applatyrhynchos_homolog_ensembl_gene': 'Duck gene stable ID' (default: False),\n",
      " 'applatyrhynchos_homolog_ensembl_peptide': 'Duck protein or transcript stable ID' (default: False),\n",
      " 'applatyrhynchos_homolog_goc_score': 'Duck Gene-order conservation score' (default: False),\n",
      " 'applatyrhynchos_homolog_orthology_confidence': 'Duck orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'applatyrhynchos_homolog_orthology_type': 'Duck homology type' (default: False),\n",
      " 'applatyrhynchos_homolog_perc_id': '%id. target Duck gene identical to query gene' (default: False),\n",
      " 'applatyrhynchos_homolog_perc_id_r1': '%id. query gene identical to target Duck gene' (default: False),\n",
      " 'applatyrhynchos_homolog_subtype': 'Last common ancestor with Duck' (default: False),\n",
      " 'applatyrhynchos_homolog_wga_coverage': 'Duck Whole-genome alignment coverage' (default: False),\n",
      " 'arrayexpress': 'Expression Atlas ID' (default: False),\n",
      " 'atestudineus_homolog_associated_gene_name': 'Climbing perch gene name' (default: False),\n",
      " 'atestudineus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'atestudineus_homolog_chrom_end': 'Climbing perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'atestudineus_homolog_chrom_start': 'Climbing perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'atestudineus_homolog_chromosome': 'Climbing perch chromosome/scaffold name' (default: False),\n",
      " 'atestudineus_homolog_ensembl_gene': 'Climbing perch gene stable ID' (default: False),\n",
      " 'atestudineus_homolog_ensembl_peptide': 'Climbing perch protein or transcript stable ID' (default: False),\n",
      " 'atestudineus_homolog_goc_score': 'Climbing perch Gene-order conservation score' (default: False),\n",
      " 'atestudineus_homolog_orthology_confidence': 'Climbing perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'atestudineus_homolog_orthology_type': 'Climbing perch homology type' (default: False),\n",
      " 'atestudineus_homolog_perc_id': '%id. target Climbing perch gene identical to query gene' (default: False),\n",
      " 'atestudineus_homolog_perc_id_r1': '%id. query gene identical to target Climbing perch gene' (default: False),\n",
      " 'atestudineus_homolog_subtype': 'Last common ancestor with Climbing perch' (default: False),\n",
      " 'band': 'Karyotype band' (default: False),\n",
      " 'bbbison_homolog_associated_gene_name': 'American bison gene name' (default: False),\n",
      " 'bbbison_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bbbison_homolog_chrom_end': 'American bison chromosome/scaffold end (bp)' (default: False),\n",
      " 'bbbison_homolog_chrom_start': 'American bison chromosome/scaffold start (bp)' (default: False),\n",
      " 'bbbison_homolog_chromosome': 'American bison chromosome/scaffold name' (default: False),\n",
      " 'bbbison_homolog_ensembl_gene': 'American bison gene stable ID' (default: False),\n",
      " 'bbbison_homolog_ensembl_peptide': 'American bison protein or transcript stable ID' (default: False),\n",
      " 'bbbison_homolog_goc_score': 'American bison Gene-order conservation score' (default: False),\n",
      " 'bbbison_homolog_orthology_confidence': 'American bison orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bbbison_homolog_orthology_type': 'American bison homology type' (default: False),\n",
      " 'bbbison_homolog_perc_id': '%id. target American bison gene identical to query gene' (default: False),\n",
      " 'bbbison_homolog_perc_id_r1': '%id. query gene identical to target American bison gene' (default: False),\n",
      " 'bbbison_homolog_subtype': 'Last common ancestor with American bison' (default: False),\n",
      " 'bbbison_homolog_wga_coverage': 'American bison Whole-genome alignment coverage' (default: False),\n",
      " 'bgrunniens_homolog_associated_gene_name': 'Domestic yak gene name' (default: False),\n",
      " 'bgrunniens_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bgrunniens_homolog_chrom_end': 'Domestic yak chromosome/scaffold end (bp)' (default: False),\n",
      " 'bgrunniens_homolog_chrom_start': 'Domestic yak chromosome/scaffold start (bp)' (default: False),\n",
      " 'bgrunniens_homolog_chromosome': 'Domestic yak chromosome/scaffold name' (default: False),\n",
      " 'bgrunniens_homolog_ensembl_gene': 'Domestic yak gene stable ID' (default: False),\n",
      " 'bgrunniens_homolog_ensembl_peptide': 'Domestic yak protein or transcript stable ID' (default: False),\n",
      " 'bgrunniens_homolog_goc_score': 'Domestic yak Gene-order conservation score' (default: False),\n",
      " 'bgrunniens_homolog_orthology_confidence': 'Domestic yak orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bgrunniens_homolog_orthology_type': 'Domestic yak homology type' (default: False),\n",
      " 'bgrunniens_homolog_perc_id': '%id. target Domestic yak gene identical to query gene' (default: False),\n",
      " 'bgrunniens_homolog_perc_id_r1': '%id. query gene identical to target Domestic yak gene' (default: False),\n",
      " 'bgrunniens_homolog_subtype': 'Last common ancestor with Domestic yak' (default: False),\n",
      " 'bgrunniens_homolog_wga_coverage': 'Domestic yak Whole-genome alignment coverage' (default: False),\n",
      " 'bihybrid_homolog_associated_gene_name': 'Hybrid - Bos Indicus gene name' (default: False),\n",
      " 'bihybrid_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bihybrid_homolog_chrom_end': 'Hybrid - Bos Indicus chromosome/scaffold end (bp)' (default: False),\n",
      " 'bihybrid_homolog_chrom_start': 'Hybrid - Bos Indicus chromosome/scaffold start (bp)' (default: False),\n",
      " 'bihybrid_homolog_chromosome': 'Hybrid - Bos Indicus chromosome/scaffold name' (default: False),\n",
      " 'bihybrid_homolog_ensembl_gene': 'Hybrid - Bos Indicus gene stable ID' (default: False),\n",
      " 'bihybrid_homolog_ensembl_peptide': 'Hybrid - Bos Indicus protein or transcript stable ID' (default: False),\n",
      " 'bihybrid_homolog_goc_score': 'Hybrid - Bos Indicus Gene-order conservation score' (default: False),\n",
      " 'bihybrid_homolog_orthology_confidence': 'Hybrid - Bos Indicus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bihybrid_homolog_orthology_type': 'Hybrid - Bos Indicus homology type' (default: False),\n",
      " 'bihybrid_homolog_perc_id': '%id. target Hybrid - Bos Indicus gene identical to query gene' (default: False),\n",
      " 'bihybrid_homolog_perc_id_r1': '%id. query gene identical to target Hybrid - Bos Indicus gene' (default: False),\n",
      " 'bihybrid_homolog_subtype': 'Last common ancestor with Hybrid - Bos Indicus' (default: False),\n",
      " 'bihybrid_homolog_wga_coverage': 'Hybrid - Bos Indicus Whole-genome alignment coverage' (default: False),\n",
      " 'biogrid': 'BioGRID Interaction data, The General Repository for Interaction Datasets ID' (default: False),\n",
      " 'bmusculus_homolog_associated_gene_name': 'Blue whale gene name' (default: False),\n",
      " 'bmusculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bmusculus_homolog_chrom_end': 'Blue whale chromosome/scaffold end (bp)' (default: False),\n",
      " 'bmusculus_homolog_chrom_start': 'Blue whale chromosome/scaffold start (bp)' (default: False),\n",
      " 'bmusculus_homolog_chromosome': 'Blue whale chromosome/scaffold name' (default: False),\n",
      " 'bmusculus_homolog_ensembl_gene': 'Blue whale gene stable ID' (default: False),\n",
      " 'bmusculus_homolog_ensembl_peptide': 'Blue whale protein or transcript stable ID' (default: False),\n",
      " 'bmusculus_homolog_goc_score': 'Blue whale Gene-order conservation score' (default: False),\n",
      " 'bmusculus_homolog_orthology_confidence': 'Blue whale orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bmusculus_homolog_orthology_type': 'Blue whale homology type' (default: False),\n",
      " 'bmusculus_homolog_perc_id': '%id. target Blue whale gene identical to query gene' (default: False),\n",
      " 'bmusculus_homolog_perc_id_r1': '%id. query gene identical to target Blue whale gene' (default: False),\n",
      " 'bmusculus_homolog_subtype': 'Last common ancestor with Blue whale' (default: False),\n",
      " 'bmusculus_homolog_wga_coverage': 'Blue whale Whole-genome alignment coverage' (default: False),\n",
      " 'bmutus_homolog_associated_gene_name': 'Wild yak gene name' (default: False),\n",
      " 'bmutus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bmutus_homolog_chrom_end': 'Wild yak chromosome/scaffold end (bp)' (default: False),\n",
      " 'bmutus_homolog_chrom_start': 'Wild yak chromosome/scaffold start (bp)' (default: False),\n",
      " 'bmutus_homolog_chromosome': 'Wild yak chromosome/scaffold name' (default: False),\n",
      " 'bmutus_homolog_ensembl_gene': 'Wild yak gene stable ID' (default: False),\n",
      " 'bmutus_homolog_ensembl_peptide': 'Wild yak protein or transcript stable ID' (default: False),\n",
      " 'bmutus_homolog_goc_score': 'Wild yak Gene-order conservation score' (default: False),\n",
      " 'bmutus_homolog_orthology_confidence': 'Wild yak orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bmutus_homolog_orthology_type': 'Wild yak homology type' (default: False),\n",
      " 'bmutus_homolog_perc_id': '%id. target Wild yak gene identical to query gene' (default: False),\n",
      " 'bmutus_homolog_perc_id_r1': '%id. query gene identical to target Wild yak gene' (default: False),\n",
      " 'bmutus_homolog_subtype': 'Last common ancestor with Wild yak' (default: False),\n",
      " 'bmutus_homolog_wga_coverage': 'Wild yak Whole-genome alignment coverage' (default: False),\n",
      " 'bsplendens_homolog_associated_gene_name': 'Siamese fighting fish gene name' (default: False),\n",
      " 'bsplendens_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'bsplendens_homolog_chrom_end': 'Siamese fighting fish chromosome/scaffold end (bp)' (default: False),\n",
      " 'bsplendens_homolog_chrom_start': 'Siamese fighting fish chromosome/scaffold start (bp)' (default: False),\n",
      " 'bsplendens_homolog_chromosome': 'Siamese fighting fish chromosome/scaffold name' (default: False),\n",
      " 'bsplendens_homolog_ensembl_gene': 'Siamese fighting fish gene stable ID' (default: False),\n",
      " 'bsplendens_homolog_ensembl_peptide': 'Siamese fighting fish protein or transcript stable ID' (default: False),\n",
      " 'bsplendens_homolog_goc_score': 'Siamese fighting fish Gene-order conservation score' (default: False),\n",
      " 'bsplendens_homolog_orthology_confidence': 'Siamese fighting fish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'bsplendens_homolog_orthology_type': 'Siamese fighting fish homology type' (default: False),\n",
      " 'bsplendens_homolog_perc_id': '%id. target Siamese fighting fish gene identical to query gene' (default: False),\n",
      " 'bsplendens_homolog_perc_id_r1': '%id. query gene identical to target Siamese fighting fish gene' (default: False),\n",
      " 'bsplendens_homolog_subtype': 'Last common ancestor with Siamese fighting fish' (default: False),\n",
      " 'bsplendens_homolog_wga_coverage': 'Siamese fighting fish Whole-genome alignment coverage' (default: False),\n",
      " 'btaurus_homolog_associated_gene_name': 'Cow gene name' (default: False),\n",
      " 'btaurus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'btaurus_homolog_chrom_end': 'Cow chromosome/scaffold end (bp)' (default: False),\n",
      " 'btaurus_homolog_chrom_start': 'Cow chromosome/scaffold start (bp)' (default: False),\n",
      " 'btaurus_homolog_chromosome': 'Cow chromosome/scaffold name' (default: False),\n",
      " 'btaurus_homolog_ensembl_gene': 'Cow gene stable ID' (default: False),\n",
      " 'btaurus_homolog_ensembl_peptide': 'Cow protein or transcript stable ID' (default: False),\n",
      " 'btaurus_homolog_goc_score': 'Cow Gene-order conservation score' (default: False),\n",
      " 'btaurus_homolog_orthology_confidence': 'Cow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'btaurus_homolog_orthology_type': 'Cow homology type' (default: False),\n",
      " 'btaurus_homolog_perc_id': '%id. target Cow gene identical to query gene' (default: False),\n",
      " 'btaurus_homolog_perc_id_r1': '%id. query gene identical to target Cow gene' (default: False),\n",
      " 'btaurus_homolog_subtype': 'Last common ancestor with Cow' (default: False),\n",
      " 'btaurus_homolog_wga_coverage': 'Cow Whole-genome alignment coverage' (default: False),\n",
      " 'cabingdonii_homolog_associated_gene_name': 'Abingdon island giant tortoise gene name' (default: False),\n",
      " 'cabingdonii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cabingdonii_homolog_chrom_end': 'Abingdon island giant tortoise chromosome/scaffold end (bp)' (default: False),\n",
      " 'cabingdonii_homolog_chrom_start': 'Abingdon island giant tortoise chromosome/scaffold start (bp)' (default: False),\n",
      " 'cabingdonii_homolog_chromosome': 'Abingdon island giant tortoise chromosome/scaffold name' (default: False),\n",
      " 'cabingdonii_homolog_ensembl_gene': 'Abingdon island giant tortoise gene stable ID' (default: False),\n",
      " 'cabingdonii_homolog_ensembl_peptide': 'Abingdon island giant tortoise protein or transcript stable ID' (default: False),\n",
      " 'cabingdonii_homolog_goc_score': 'Abingdon island giant tortoise Gene-order conservation score' (default: False),\n",
      " 'cabingdonii_homolog_orthology_confidence': 'Abingdon island giant tortoise orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cabingdonii_homolog_orthology_type': 'Abingdon island giant tortoise homology type' (default: False),\n",
      " 'cabingdonii_homolog_perc_id': '%id. target Abingdon island giant tortoise gene identical to query gene' (default: False),\n",
      " 'cabingdonii_homolog_perc_id_r1': '%id. query gene identical to target Abingdon island giant tortoise gene' (default: False),\n",
      " 'cabingdonii_homolog_subtype': 'Last common ancestor with Abingdon island giant tortoise' (default: False),\n",
      " 'cabingdonii_homolog_wga_coverage': 'Abingdon island giant tortoise Whole-genome alignment coverage' (default: False),\n",
      " 'catys_homolog_associated_gene_name': 'Sooty mangabey gene name' (default: False),\n",
      " 'catys_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'catys_homolog_chrom_end': 'Sooty mangabey chromosome/scaffold end (bp)' (default: False),\n",
      " 'catys_homolog_chrom_start': 'Sooty mangabey chromosome/scaffold start (bp)' (default: False),\n",
      " 'catys_homolog_chromosome': 'Sooty mangabey chromosome/scaffold name' (default: False),\n",
      " 'catys_homolog_ensembl_gene': 'Sooty mangabey gene stable ID' (default: False),\n",
      " 'catys_homolog_ensembl_peptide': 'Sooty mangabey protein or transcript stable ID' (default: False),\n",
      " 'catys_homolog_goc_score': 'Sooty mangabey Gene-order conservation score' (default: False),\n",
      " 'catys_homolog_orthology_confidence': 'Sooty mangabey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'catys_homolog_orthology_type': 'Sooty mangabey homology type' (default: False),\n",
      " 'catys_homolog_perc_id': '%id. target Sooty mangabey gene identical to query gene' (default: False),\n",
      " 'catys_homolog_perc_id_r1': '%id. query gene identical to target Sooty mangabey gene' (default: False),\n",
      " 'catys_homolog_subtype': 'Last common ancestor with Sooty mangabey' (default: False),\n",
      " 'catys_homolog_wga_coverage': 'Sooty mangabey Whole-genome alignment coverage' (default: False),\n",
      " 'cauratus_homolog_associated_gene_name': 'Goldfish gene name' (default: False),\n",
      " 'cauratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cauratus_homolog_chrom_end': 'Goldfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'cauratus_homolog_chrom_start': 'Goldfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'cauratus_homolog_chromosome': 'Goldfish chromosome/scaffold name' (default: False),\n",
      " 'cauratus_homolog_ensembl_gene': 'Goldfish gene stable ID' (default: False),\n",
      " 'cauratus_homolog_ensembl_peptide': 'Goldfish protein or transcript stable ID' (default: False),\n",
      " 'cauratus_homolog_goc_score': 'Goldfish Gene-order conservation score' (default: False),\n",
      " 'cauratus_homolog_orthology_confidence': 'Goldfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cauratus_homolog_orthology_type': 'Goldfish homology type' (default: False),\n",
      " 'cauratus_homolog_perc_id': '%id. target Goldfish gene identical to query gene' (default: False),\n",
      " 'cauratus_homolog_perc_id_r1': '%id. query gene identical to target Goldfish gene' (default: False),\n",
      " 'cauratus_homolog_subtype': 'Last common ancestor with Goldfish' (default: False),\n",
      " 'cauratus_homolog_wga_coverage': 'Goldfish Whole-genome alignment coverage' (default: False),\n",
      " 'cccarpio_homolog_associated_gene_name': 'Common carp gene name' (default: False),\n",
      " 'cccarpio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cccarpio_homolog_chrom_end': 'Common carp chromosome/scaffold end (bp)' (default: False),\n",
      " 'cccarpio_homolog_chrom_start': 'Common carp chromosome/scaffold start (bp)' (default: False),\n",
      " 'cccarpio_homolog_chromosome': 'Common carp chromosome/scaffold name' (default: False),\n",
      " 'cccarpio_homolog_ensembl_gene': 'Common carp gene stable ID' (default: False),\n",
      " 'cccarpio_homolog_ensembl_peptide': 'Common carp protein or transcript stable ID' (default: False),\n",
      " 'cccarpio_homolog_goc_score': 'Common carp Gene-order conservation score' (default: False),\n",
      " 'cccarpio_homolog_orthology_confidence': 'Common carp orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cccarpio_homolog_orthology_type': 'Common carp homology type' (default: False),\n",
      " 'cccarpio_homolog_perc_id': '%id. target Common carp gene identical to query gene' (default: False),\n",
      " 'cccarpio_homolog_perc_id_r1': '%id. query gene identical to target Common carp gene' (default: False),\n",
      " 'cccarpio_homolog_subtype': 'Last common ancestor with Common carp' (default: False),\n",
      " 'ccds': 'CCDS ID' (default: False),\n",
      " 'cdd': 'CDD ID' (default: False),\n",
      " 'cdd_end': 'CDD end' (default: False),\n",
      " 'cdd_start': 'CDD start' (default: False),\n",
      " 'cdna': 'cDNA sequences' (default: False),\n",
      " 'cdna_coding_end': 'cDNA coding end' (default: False),\n",
      " 'cdna_coding_start': 'cDNA coding start' (default: False),\n",
      " 'cdromedarius_homolog_associated_gene_name': 'Arabian camel gene name' (default: False),\n",
      " 'cdromedarius_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cdromedarius_homolog_chrom_end': 'Arabian camel chromosome/scaffold end (bp)' (default: False),\n",
      " 'cdromedarius_homolog_chrom_start': 'Arabian camel chromosome/scaffold start (bp)' (default: False),\n",
      " 'cdromedarius_homolog_chromosome': 'Arabian camel chromosome/scaffold name' (default: False),\n",
      " 'cdromedarius_homolog_ensembl_gene': 'Arabian camel gene stable ID' (default: False),\n",
      " 'cdromedarius_homolog_ensembl_peptide': 'Arabian camel protein or transcript stable ID' (default: False),\n",
      " 'cdromedarius_homolog_goc_score': 'Arabian camel Gene-order conservation score' (default: False),\n",
      " 'cdromedarius_homolog_orthology_confidence': 'Arabian camel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cdromedarius_homolog_orthology_type': 'Arabian camel homology type' (default: False),\n",
      " 'cdromedarius_homolog_perc_id': '%id. target Arabian camel gene identical to query gene' (default: False),\n",
      " 'cdromedarius_homolog_perc_id_r1': '%id. query gene identical to target Arabian camel gene' (default: False),\n",
      " 'cdromedarius_homolog_subtype': 'Last common ancestor with Arabian camel' (default: False),\n",
      " 'cdromedarius_homolog_wga_coverage': 'Arabian camel Whole-genome alignment coverage' (default: False),\n",
      " 'cds_end': 'CDS end' (default: False),\n",
      " 'cds_end_2076': 'CDS end' (default: False),\n",
      " 'cds_length': 'CDS Length' (default: False),\n",
      " 'cds_start': 'CDS start' (default: False),\n",
      " 'cds_start_2076': 'CDS start' (default: False),\n",
      " 'celegans_homolog_associated_gene_name': 'Caenorhabditis elegans (PRJNA13758) gene name' (default: False),\n",
      " 'celegans_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'celegans_homolog_chrom_end': 'Caenorhabditis elegans (PRJNA13758) chromosome/scaffold end (bp)' (default: False),\n",
      " 'celegans_homolog_chrom_start': 'Caenorhabditis elegans (PRJNA13758) chromosome/scaffold start (bp)' (default: False),\n",
      " 'celegans_homolog_chromosome': 'Caenorhabditis elegans (PRJNA13758) chromosome/scaffold name' (default: False),\n",
      " 'celegans_homolog_ensembl_gene': 'Caenorhabditis elegans (PRJNA13758) gene stable ID' (default: False),\n",
      " 'celegans_homolog_ensembl_peptide': 'Caenorhabditis elegans (PRJNA13758) protein or transcript stable ID' (default: False),\n",
      " 'celegans_homolog_orthology_confidence': 'Caenorhabditis elegans (PRJNA13758) orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'celegans_homolog_orthology_type': 'Caenorhabditis elegans (PRJNA13758) homology type' (default: False),\n",
      " 'celegans_homolog_perc_id': '%id. target Caenorhabditis elegans (PRJNA13758) gene identical to query gene' (default: False),\n",
      " 'celegans_homolog_perc_id_r1': '%id. query gene identical to target Caenorhabditis elegans (PRJNA13758) gene' (default: False),\n",
      " 'celegans_homolog_subtype': 'Last common ancestor with Caenorhabditis elegans (PRJNA13758)' (default: False),\n",
      " 'cgchok1gshd_homolog_associated_gene_name': 'Chinese hamster CHOK1GS gene name' (default: False),\n",
      " 'cgchok1gshd_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgchok1gshd_homolog_chrom_end': 'Chinese hamster CHOK1GS chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgchok1gshd_homolog_chrom_start': 'Chinese hamster CHOK1GS chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgchok1gshd_homolog_chromosome': 'Chinese hamster CHOK1GS chromosome/scaffold name' (default: False),\n",
      " 'cgchok1gshd_homolog_ensembl_gene': 'Chinese hamster CHOK1GS gene stable ID' (default: False),\n",
      " 'cgchok1gshd_homolog_ensembl_peptide': 'Chinese hamster CHOK1GS protein or transcript stable ID' (default: False),\n",
      " 'cgchok1gshd_homolog_goc_score': 'Chinese hamster CHOK1GS Gene-order conservation score' (default: False),\n",
      " 'cgchok1gshd_homolog_orthology_confidence': 'Chinese hamster CHOK1GS orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgchok1gshd_homolog_orthology_type': 'Chinese hamster CHOK1GS homology type' (default: False),\n",
      " 'cgchok1gshd_homolog_perc_id': '%id. target Chinese hamster CHOK1GS gene identical to query gene' (default: False),\n",
      " 'cgchok1gshd_homolog_perc_id_r1': '%id. query gene identical to target Chinese hamster CHOK1GS gene' (default: False),\n",
      " 'cgchok1gshd_homolog_subtype': 'Last common ancestor with Chinese hamster CHOK1GS' (default: False),\n",
      " 'cgchok1gshd_homolog_wga_coverage': 'Chinese hamster CHOK1GS Whole-genome alignment coverage' (default: False),\n",
      " 'cgobio_homolog_associated_gene_name': 'Channel bull blenny gene name' (default: False),\n",
      " 'cgobio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cgobio_homolog_chrom_end': 'Channel bull blenny chromosome/scaffold end (bp)' (default: False),\n",
      " 'cgobio_homolog_chrom_start': 'Channel bull blenny chromosome/scaffold start (bp)' (default: False),\n",
      " 'cgobio_homolog_chromosome': 'Channel bull blenny chromosome/scaffold name' (default: False),\n",
      " 'cgobio_homolog_ensembl_gene': 'Channel bull blenny gene stable ID' (default: False),\n",
      " 'cgobio_homolog_ensembl_peptide': 'Channel bull blenny protein or transcript stable ID' (default: False),\n",
      " 'cgobio_homolog_goc_score': 'Channel bull blenny Gene-order conservation score' (default: False),\n",
      " 'cgobio_homolog_orthology_confidence': 'Channel bull blenny orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cgobio_homolog_orthology_type': 'Channel bull blenny homology type' (default: False),\n",
      " 'cgobio_homolog_perc_id': '%id. target Channel bull blenny gene identical to query gene' (default: False),\n",
      " 'cgobio_homolog_perc_id_r1': '%id. query gene identical to target Channel bull blenny gene' (default: False),\n",
      " 'cgobio_homolog_subtype': 'Last common ancestor with Channel bull blenny' (default: False),\n",
      " 'cgobio_homolog_wga_coverage': 'Channel bull blenny Whole-genome alignment coverage' (default: False),\n",
      " 'charengus_homolog_associated_gene_name': 'Atlantic herring gene name' (default: False),\n",
      " 'charengus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'charengus_homolog_chrom_end': 'Atlantic herring chromosome/scaffold end (bp)' (default: False),\n",
      " 'charengus_homolog_chrom_start': 'Atlantic herring chromosome/scaffold start (bp)' (default: False),\n",
      " 'charengus_homolog_chromosome': 'Atlantic herring chromosome/scaffold name' (default: False),\n",
      " 'charengus_homolog_ensembl_gene': 'Atlantic herring gene stable ID' (default: False),\n",
      " 'charengus_homolog_ensembl_peptide': 'Atlantic herring protein or transcript stable ID' (default: False),\n",
      " 'charengus_homolog_goc_score': 'Atlantic herring Gene-order conservation score' (default: False),\n",
      " 'charengus_homolog_orthology_confidence': 'Atlantic herring orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'charengus_homolog_orthology_type': 'Atlantic herring homology type' (default: False),\n",
      " 'charengus_homolog_perc_id': '%id. target Atlantic herring gene identical to query gene' (default: False),\n",
      " 'charengus_homolog_perc_id_r1': '%id. query gene identical to target Atlantic herring gene' (default: False),\n",
      " 'charengus_homolog_subtype': 'Last common ancestor with Atlantic herring' (default: False),\n",
      " 'charengus_homolog_wga_coverage': 'Atlantic herring Whole-genome alignment coverage' (default: False),\n",
      " 'chembl': 'ChEMBL ID' (default: False),\n",
      " 'chircus_homolog_associated_gene_name': 'Goat gene name' (default: False),\n",
      " 'chircus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'chircus_homolog_chrom_end': 'Goat chromosome/scaffold end (bp)' (default: False),\n",
      " 'chircus_homolog_chrom_start': 'Goat chromosome/scaffold start (bp)' (default: False),\n",
      " 'chircus_homolog_chromosome': 'Goat chromosome/scaffold name' (default: False),\n",
      " 'chircus_homolog_ensembl_gene': 'Goat gene stable ID' (default: False),\n",
      " 'chircus_homolog_ensembl_peptide': 'Goat protein or transcript stable ID' (default: False),\n",
      " 'chircus_homolog_goc_score': 'Goat Gene-order conservation score' (default: False),\n",
      " 'chircus_homolog_orthology_confidence': 'Goat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'chircus_homolog_orthology_type': 'Goat homology type' (default: False),\n",
      " 'chircus_homolog_perc_id': '%id. target Goat gene identical to query gene' (default: False),\n",
      " 'chircus_homolog_perc_id_r1': '%id. query gene identical to target Goat gene' (default: False),\n",
      " 'chircus_homolog_subtype': 'Last common ancestor with Goat' (default: False),\n",
      " 'chircus_homolog_wga_coverage': 'Goat Whole-genome alignment coverage' (default: False),\n",
      " 'choffmanni_homolog_associated_gene_name': 'Sloth gene name' (default: False),\n",
      " 'choffmanni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'choffmanni_homolog_chrom_end': 'Sloth chromosome/scaffold end (bp)' (default: False),\n",
      " 'choffmanni_homolog_chrom_start': 'Sloth chromosome/scaffold start (bp)' (default: False),\n",
      " 'choffmanni_homolog_chromosome': 'Sloth chromosome/scaffold name' (default: False),\n",
      " 'choffmanni_homolog_ensembl_gene': 'Sloth gene stable ID' (default: False),\n",
      " 'choffmanni_homolog_ensembl_peptide': 'Sloth protein or transcript stable ID' (default: False),\n",
      " 'choffmanni_homolog_goc_score': 'Sloth Gene-order conservation score' (default: False),\n",
      " 'choffmanni_homolog_orthology_confidence': 'Sloth orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'choffmanni_homolog_orthology_type': 'Sloth homology type' (default: False),\n",
      " 'choffmanni_homolog_perc_id': '%id. target Sloth gene identical to query gene' (default: False),\n",
      " 'choffmanni_homolog_perc_id_r1': '%id. query gene identical to target Sloth gene' (default: False),\n",
      " 'choffmanni_homolog_subtype': 'Last common ancestor with Sloth' (default: False),\n",
      " 'choffmanni_homolog_wga_coverage': 'Sloth Whole-genome alignment coverage' (default: False),\n",
      " 'chromosome_end': 'Chromosome/scaffold position end (bp)' (default: False),\n",
      " 'chromosome_name': 'Chromosome/scaffold name' (default: False),\n",
      " 'chromosome_start': 'chromosome/scaffold position start (bp)' (default: False),\n",
      " 'chyarkandensis_homolog_associated_gene_name': 'Yarkand deer gene name' (default: False),\n",
      " 'chyarkandensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'chyarkandensis_homolog_chrom_end': 'Yarkand deer chromosome/scaffold end (bp)' (default: False),\n",
      " 'chyarkandensis_homolog_chrom_start': 'Yarkand deer chromosome/scaffold start (bp)' (default: False),\n",
      " 'chyarkandensis_homolog_chromosome': 'Yarkand deer chromosome/scaffold name' (default: False),\n",
      " 'chyarkandensis_homolog_ensembl_gene': 'Yarkand deer gene stable ID' (default: False),\n",
      " 'chyarkandensis_homolog_ensembl_peptide': 'Yarkand deer protein or transcript stable ID' (default: False),\n",
      " 'chyarkandensis_homolog_goc_score': 'Yarkand deer Gene-order conservation score' (default: False),\n",
      " 'chyarkandensis_homolog_orthology_confidence': 'Yarkand deer orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'chyarkandensis_homolog_orthology_type': 'Yarkand deer homology type' (default: False),\n",
      " 'chyarkandensis_homolog_perc_id': '%id. target Yarkand deer gene identical to query gene' (default: False),\n",
      " 'chyarkandensis_homolog_perc_id_r1': '%id. query gene identical to target Yarkand deer gene' (default: False),\n",
      " 'chyarkandensis_homolog_subtype': 'Last common ancestor with Yarkand deer' (default: False),\n",
      " 'chyarkandensis_homolog_wga_coverage': 'Yarkand deer Whole-genome alignment coverage' (default: False),\n",
      " 'cimitator_homolog_associated_gene_name': 'Panamanian white-faced capuchin gene name' (default: False),\n",
      " 'cimitator_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cimitator_homolog_chrom_end': 'Panamanian white-faced capuchin chromosome/scaffold end (bp)' (default: False),\n",
      " 'cimitator_homolog_chrom_start': 'Panamanian white-faced capuchin chromosome/scaffold start (bp)' (default: False),\n",
      " 'cimitator_homolog_chromosome': 'Panamanian white-faced capuchin chromosome/scaffold name' (default: False),\n",
      " 'cimitator_homolog_ensembl_gene': 'Panamanian white-faced capuchin gene stable ID' (default: False),\n",
      " 'cimitator_homolog_ensembl_peptide': 'Panamanian white-faced capuchin protein or transcript stable ID' (default: False),\n",
      " 'cimitator_homolog_goc_score': 'Panamanian white-faced capuchin Gene-order conservation score' (default: False),\n",
      " 'cimitator_homolog_orthology_confidence': 'Panamanian white-faced capuchin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cimitator_homolog_orthology_type': 'Panamanian white-faced capuchin homology type' (default: False),\n",
      " 'cimitator_homolog_perc_id': '%id. target Panamanian white-faced capuchin gene identical to query gene' (default: False),\n",
      " 'cimitator_homolog_perc_id_r1': '%id. query gene identical to target Panamanian white-faced capuchin gene' (default: False),\n",
      " 'cimitator_homolog_subtype': 'Last common ancestor with Panamanian white-faced capuchin' (default: False),\n",
      " 'cimitator_homolog_wga_coverage': 'Panamanian white-faced capuchin Whole-genome alignment coverage' (default: False),\n",
      " 'cintestinalis_homolog_associated_gene_name': 'C.intestinalis gene name' (default: False),\n",
      " 'cintestinalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cintestinalis_homolog_chrom_end': 'C.intestinalis chromosome/scaffold end (bp)' (default: False),\n",
      " 'cintestinalis_homolog_chrom_start': 'C.intestinalis chromosome/scaffold start (bp)' (default: False),\n",
      " 'cintestinalis_homolog_chromosome': 'C.intestinalis chromosome/scaffold name' (default: False),\n",
      " 'cintestinalis_homolog_ensembl_gene': 'C.intestinalis gene stable ID' (default: False),\n",
      " 'cintestinalis_homolog_ensembl_peptide': 'C.intestinalis protein or transcript stable ID' (default: False),\n",
      " 'cintestinalis_homolog_orthology_confidence': 'C.intestinalis orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cintestinalis_homolog_orthology_type': 'C.intestinalis homology type' (default: False),\n",
      " 'cintestinalis_homolog_perc_id': '%id. target C.intestinalis gene identical to query gene' (default: False),\n",
      " 'cintestinalis_homolog_perc_id_r1': '%id. query gene identical to target C.intestinalis gene' (default: False),\n",
      " 'cintestinalis_homolog_subtype': 'Last common ancestor with C.intestinalis' (default: False),\n",
      " 'cintestinalis_homolog_wga_coverage': 'C.intestinalis Whole-genome alignment coverage' (default: False),\n",
      " 'cjacchus_homolog_associated_gene_name': 'White-tufted-ear marmoset gene name' (default: False),\n",
      " 'cjacchus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cjacchus_homolog_chrom_end': 'White-tufted-ear marmoset chromosome/scaffold end (bp)' (default: False),\n",
      " 'cjacchus_homolog_chrom_start': 'White-tufted-ear marmoset chromosome/scaffold start (bp)' (default: False),\n",
      " 'cjacchus_homolog_chromosome': 'White-tufted-ear marmoset chromosome/scaffold name' (default: False),\n",
      " 'cjacchus_homolog_ensembl_gene': 'White-tufted-ear marmoset gene stable ID' (default: False),\n",
      " 'cjacchus_homolog_ensembl_peptide': 'White-tufted-ear marmoset protein or transcript stable ID' (default: False),\n",
      " 'cjacchus_homolog_goc_score': 'White-tufted-ear marmoset Gene-order conservation score' (default: False),\n",
      " 'cjacchus_homolog_orthology_confidence': 'White-tufted-ear marmoset orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cjacchus_homolog_orthology_type': 'White-tufted-ear marmoset homology type' (default: False),\n",
      " 'cjacchus_homolog_perc_id': '%id. target White-tufted-ear marmoset gene identical to query gene' (default: False),\n",
      " 'cjacchus_homolog_perc_id_r1': '%id. query gene identical to target White-tufted-ear marmoset gene' (default: False),\n",
      " 'cjacchus_homolog_subtype': 'Last common ancestor with White-tufted-ear marmoset' (default: False),\n",
      " 'cjacchus_homolog_wga_coverage': 'White-tufted-ear marmoset Whole-genome alignment coverage' (default: False),\n",
      " 'cjaponica_homolog_associated_gene_name': 'Japanese quail gene name' (default: False),\n",
      " 'cjaponica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cjaponica_homolog_chrom_end': 'Japanese quail chromosome/scaffold end (bp)' (default: False),\n",
      " 'cjaponica_homolog_chrom_start': 'Japanese quail chromosome/scaffold start (bp)' (default: False),\n",
      " 'cjaponica_homolog_chromosome': 'Japanese quail chromosome/scaffold name' (default: False),\n",
      " 'cjaponica_homolog_ensembl_gene': 'Japanese quail gene stable ID' (default: False),\n",
      " 'cjaponica_homolog_ensembl_peptide': 'Japanese quail protein or transcript stable ID' (default: False),\n",
      " 'cjaponica_homolog_goc_score': 'Japanese quail Gene-order conservation score' (default: False),\n",
      " 'cjaponica_homolog_orthology_confidence': 'Japanese quail orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cjaponica_homolog_orthology_type': 'Japanese quail homology type' (default: False),\n",
      " 'cjaponica_homolog_perc_id': '%id. target Japanese quail gene identical to query gene' (default: False),\n",
      " 'cjaponica_homolog_perc_id_r1': '%id. query gene identical to target Japanese quail gene' (default: False),\n",
      " 'cjaponica_homolog_subtype': 'Last common ancestor with Japanese quail' (default: False),\n",
      " 'cjaponica_homolog_wga_coverage': 'Japanese quail Whole-genome alignment coverage' (default: False),\n",
      " 'clanigera_homolog_associated_gene_name': 'Long-tailed chinchilla gene name' (default: False),\n",
      " 'clanigera_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clanigera_homolog_chrom_end': 'Long-tailed chinchilla chromosome/scaffold end (bp)' (default: False),\n",
      " 'clanigera_homolog_chrom_start': 'Long-tailed chinchilla chromosome/scaffold start (bp)' (default: False),\n",
      " 'clanigera_homolog_chromosome': 'Long-tailed chinchilla chromosome/scaffold name' (default: False),\n",
      " 'clanigera_homolog_ensembl_gene': 'Long-tailed chinchilla gene stable ID' (default: False),\n",
      " 'clanigera_homolog_ensembl_peptide': 'Long-tailed chinchilla protein or transcript stable ID' (default: False),\n",
      " 'clanigera_homolog_goc_score': 'Long-tailed chinchilla Gene-order conservation score' (default: False),\n",
      " 'clanigera_homolog_orthology_confidence': 'Long-tailed chinchilla orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clanigera_homolog_orthology_type': 'Long-tailed chinchilla homology type' (default: False),\n",
      " 'clanigera_homolog_perc_id': '%id. target Long-tailed chinchilla gene identical to query gene' (default: False),\n",
      " 'clanigera_homolog_perc_id_r1': '%id. query gene identical to target Long-tailed chinchilla gene' (default: False),\n",
      " 'clanigera_homolog_subtype': 'Last common ancestor with Long-tailed chinchilla' (default: False),\n",
      " 'clanigera_homolog_wga_coverage': 'Long-tailed chinchilla Whole-genome alignment coverage' (default: False),\n",
      " 'cldingo_homolog_associated_gene_name': 'Dingo gene name' (default: False),\n",
      " 'cldingo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cldingo_homolog_chrom_end': 'Dingo chromosome/scaffold end (bp)' (default: False),\n",
      " 'cldingo_homolog_chrom_start': 'Dingo chromosome/scaffold start (bp)' (default: False),\n",
      " 'cldingo_homolog_chromosome': 'Dingo chromosome/scaffold name' (default: False),\n",
      " 'cldingo_homolog_ensembl_gene': 'Dingo gene stable ID' (default: False),\n",
      " 'cldingo_homolog_ensembl_peptide': 'Dingo protein or transcript stable ID' (default: False),\n",
      " 'cldingo_homolog_goc_score': 'Dingo Gene-order conservation score' (default: False),\n",
      " 'cldingo_homolog_orthology_confidence': 'Dingo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cldingo_homolog_orthology_type': 'Dingo homology type' (default: False),\n",
      " 'cldingo_homolog_perc_id': '%id. target Dingo gene identical to query gene' (default: False),\n",
      " 'cldingo_homolog_perc_id_r1': '%id. query gene identical to target Dingo gene' (default: False),\n",
      " 'cldingo_homolog_subtype': 'Last common ancestor with Dingo' (default: False),\n",
      " 'cldingo_homolog_wga_coverage': 'Dingo Whole-genome alignment coverage' (default: False),\n",
      " 'clfamiliaris_homolog_associated_gene_name': 'Dog gene name' (default: False),\n",
      " 'clfamiliaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clfamiliaris_homolog_chrom_end': 'Dog chromosome/scaffold end (bp)' (default: False),\n",
      " 'clfamiliaris_homolog_chrom_start': 'Dog chromosome/scaffold start (bp)' (default: False),\n",
      " 'clfamiliaris_homolog_chromosome': 'Dog chromosome/scaffold name' (default: False),\n",
      " 'clfamiliaris_homolog_ensembl_gene': 'Dog gene stable ID' (default: False),\n",
      " 'clfamiliaris_homolog_ensembl_peptide': 'Dog protein or transcript stable ID' (default: False),\n",
      " 'clfamiliaris_homolog_goc_score': 'Dog Gene-order conservation score' (default: False),\n",
      " 'clfamiliaris_homolog_orthology_confidence': 'Dog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clfamiliaris_homolog_orthology_type': 'Dog homology type' (default: False),\n",
      " 'clfamiliaris_homolog_perc_id': '%id. target Dog gene identical to query gene' (default: False),\n",
      " 'clfamiliaris_homolog_perc_id_r1': '%id. query gene identical to target Dog gene' (default: False),\n",
      " 'clfamiliaris_homolog_subtype': 'Last common ancestor with Dog' (default: False),\n",
      " 'clfamiliaris_homolog_wga_coverage': 'Dog Whole-genome alignment coverage' (default: False),\n",
      " 'clinical_significance': 'Clinical significance' (default: False),\n",
      " 'clumpus_homolog_associated_gene_name': 'Lumpfish gene name' (default: False),\n",
      " 'clumpus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'clumpus_homolog_chrom_end': 'Lumpfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'clumpus_homolog_chrom_start': 'Lumpfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'clumpus_homolog_chromosome': 'Lumpfish chromosome/scaffold name' (default: False),\n",
      " 'clumpus_homolog_ensembl_gene': 'Lumpfish gene stable ID' (default: False),\n",
      " 'clumpus_homolog_ensembl_peptide': 'Lumpfish protein or transcript stable ID' (default: False),\n",
      " 'clumpus_homolog_goc_score': 'Lumpfish Gene-order conservation score' (default: False),\n",
      " 'clumpus_homolog_orthology_confidence': 'Lumpfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'clumpus_homolog_orthology_type': 'Lumpfish homology type' (default: False),\n",
      " 'clumpus_homolog_perc_id': '%id. target Lumpfish gene identical to query gene' (default: False),\n",
      " 'clumpus_homolog_perc_id_r1': '%id. query gene identical to target Lumpfish gene' (default: False),\n",
      " 'clumpus_homolog_subtype': 'Last common ancestor with Lumpfish' (default: False),\n",
      " 'clumpus_homolog_wga_coverage': 'Lumpfish Whole-genome alignment coverage' (default: False),\n",
      " 'cmilii_homolog_associated_gene_name': 'Elephant shark gene name' (default: False),\n",
      " 'cmilii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cmilii_homolog_chrom_end': 'Elephant shark chromosome/scaffold end (bp)' (default: False),\n",
      " 'cmilii_homolog_chrom_start': 'Elephant shark chromosome/scaffold start (bp)' (default: False),\n",
      " 'cmilii_homolog_chromosome': 'Elephant shark chromosome/scaffold name' (default: False),\n",
      " 'cmilii_homolog_ensembl_gene': 'Elephant shark gene stable ID' (default: False),\n",
      " 'cmilii_homolog_ensembl_peptide': 'Elephant shark protein or transcript stable ID' (default: False),\n",
      " 'cmilii_homolog_orthology_confidence': 'Elephant shark orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cmilii_homolog_orthology_type': 'Elephant shark homology type' (default: False),\n",
      " 'cmilii_homolog_perc_id': '%id. target Elephant shark gene identical to query gene' (default: False),\n",
      " 'cmilii_homolog_perc_id_r1': '%id. query gene identical to target Elephant shark gene' (default: False),\n",
      " 'cmilii_homolog_subtype': 'Last common ancestor with Elephant shark' (default: False),\n",
      " 'cmilii_homolog_wga_coverage': 'Elephant shark Whole-genome alignment coverage' (default: False),\n",
      " 'codelink_codelink': 'CODELINK CODELINK probe' (default: False),\n",
      " 'coding': 'Coding sequence' (default: False),\n",
      " 'coding_gene_flank': 'Flank-coding region (Gene)' (default: False),\n",
      " 'coding_transcript_flank': 'Flank-coding region (Transcript)' (default: False),\n",
      " 'cpbellii_homolog_associated_gene_name': 'Painted turtle gene name' (default: False),\n",
      " 'cpbellii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cpbellii_homolog_chrom_end': 'Painted turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'cpbellii_homolog_chrom_start': 'Painted turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'cpbellii_homolog_chromosome': 'Painted turtle chromosome/scaffold name' (default: False),\n",
      " 'cpbellii_homolog_ensembl_gene': 'Painted turtle gene stable ID' (default: False),\n",
      " 'cpbellii_homolog_ensembl_peptide': 'Painted turtle protein or transcript stable ID' (default: False),\n",
      " 'cpbellii_homolog_goc_score': 'Painted turtle Gene-order conservation score' (default: False),\n",
      " 'cpbellii_homolog_orthology_confidence': 'Painted turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cpbellii_homolog_orthology_type': 'Painted turtle homology type' (default: False),\n",
      " 'cpbellii_homolog_perc_id': '%id. target Painted turtle gene identical to query gene' (default: False),\n",
      " 'cpbellii_homolog_perc_id_r1': '%id. query gene identical to target Painted turtle gene' (default: False),\n",
      " 'cpbellii_homolog_subtype': 'Last common ancestor with Painted turtle' (default: False),\n",
      " 'cpbellii_homolog_wga_coverage': 'Painted turtle Whole-genome alignment coverage' (default: False),\n",
      " 'cporcellus_homolog_associated_gene_name': 'Guinea Pig gene name' (default: False),\n",
      " 'cporcellus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cporcellus_homolog_chrom_end': 'Guinea Pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'cporcellus_homolog_chrom_start': 'Guinea Pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'cporcellus_homolog_chromosome': 'Guinea Pig chromosome/scaffold name' (default: False),\n",
      " 'cporcellus_homolog_ensembl_gene': 'Guinea Pig gene stable ID' (default: False),\n",
      " 'cporcellus_homolog_ensembl_peptide': 'Guinea Pig protein or transcript stable ID' (default: False),\n",
      " 'cporcellus_homolog_goc_score': 'Guinea Pig Gene-order conservation score' (default: False),\n",
      " 'cporcellus_homolog_orthology_confidence': 'Guinea Pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cporcellus_homolog_orthology_type': 'Guinea Pig homology type' (default: False),\n",
      " 'cporcellus_homolog_perc_id': '%id. target Guinea Pig gene identical to query gene' (default: False),\n",
      " 'cporcellus_homolog_perc_id_r1': '%id. query gene identical to target Guinea Pig gene' (default: False),\n",
      " 'cporcellus_homolog_subtype': 'Last common ancestor with Guinea Pig' (default: False),\n",
      " 'cporcellus_homolog_wga_coverage': 'Guinea Pig Whole-genome alignment coverage' (default: False),\n",
      " 'cporosus_homolog_associated_gene_name': 'Australian saltwater crocodile gene name' (default: False),\n",
      " 'cporosus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cporosus_homolog_chrom_end': 'Australian saltwater crocodile chromosome/scaffold end (bp)' (default: False),\n",
      " 'cporosus_homolog_chrom_start': 'Australian saltwater crocodile chromosome/scaffold start (bp)' (default: False),\n",
      " 'cporosus_homolog_chromosome': 'Australian saltwater crocodile chromosome/scaffold name' (default: False),\n",
      " 'cporosus_homolog_ensembl_gene': 'Australian saltwater crocodile gene stable ID' (default: False),\n",
      " 'cporosus_homolog_ensembl_peptide': 'Australian saltwater crocodile protein or transcript stable ID' (default: False),\n",
      " 'cporosus_homolog_goc_score': 'Australian saltwater crocodile Gene-order conservation score' (default: False),\n",
      " 'cporosus_homolog_orthology_confidence': 'Australian saltwater crocodile orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cporosus_homolog_orthology_type': 'Australian saltwater crocodile homology type' (default: False),\n",
      " 'cporosus_homolog_perc_id': '%id. target Australian saltwater crocodile gene identical to query gene' (default: False),\n",
      " 'cporosus_homolog_perc_id_r1': '%id. query gene identical to target Australian saltwater crocodile gene' (default: False),\n",
      " 'cporosus_homolog_subtype': 'Last common ancestor with Australian saltwater crocodile' (default: False),\n",
      " 'cporosus_homolog_wga_coverage': 'Australian saltwater crocodile Whole-genome alignment coverage' (default: False),\n",
      " 'csabaeus_homolog_associated_gene_name': 'Vervet-AGM gene name' (default: False),\n",
      " 'csabaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csabaeus_homolog_chrom_end': 'Vervet-AGM chromosome/scaffold end (bp)' (default: False),\n",
      " 'csabaeus_homolog_chrom_start': 'Vervet-AGM chromosome/scaffold start (bp)' (default: False),\n",
      " 'csabaeus_homolog_chromosome': 'Vervet-AGM chromosome/scaffold name' (default: False),\n",
      " 'csabaeus_homolog_ensembl_gene': 'Vervet-AGM gene stable ID' (default: False),\n",
      " 'csabaeus_homolog_ensembl_peptide': 'Vervet-AGM protein or transcript stable ID' (default: False),\n",
      " 'csabaeus_homolog_goc_score': 'Vervet-AGM Gene-order conservation score' (default: False),\n",
      " 'csabaeus_homolog_orthology_confidence': 'Vervet-AGM orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csabaeus_homolog_orthology_type': 'Vervet-AGM homology type' (default: False),\n",
      " 'csabaeus_homolog_perc_id': '%id. target Vervet-AGM gene identical to query gene' (default: False),\n",
      " 'csabaeus_homolog_perc_id_r1': '%id. query gene identical to target Vervet-AGM gene' (default: False),\n",
      " 'csabaeus_homolog_subtype': 'Last common ancestor with Vervet-AGM' (default: False),\n",
      " 'csabaeus_homolog_wga_coverage': 'Vervet-AGM Whole-genome alignment coverage' (default: False),\n",
      " 'csavignyi_homolog_associated_gene_name': 'C.savignyi gene name' (default: False),\n",
      " 'csavignyi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csavignyi_homolog_chrom_end': 'C.savignyi chromosome/scaffold end (bp)' (default: False),\n",
      " 'csavignyi_homolog_chrom_start': 'C.savignyi chromosome/scaffold start (bp)' (default: False),\n",
      " 'csavignyi_homolog_chromosome': 'C.savignyi chromosome/scaffold name' (default: False),\n",
      " 'csavignyi_homolog_ensembl_gene': 'C.savignyi gene stable ID' (default: False),\n",
      " 'csavignyi_homolog_ensembl_peptide': 'C.savignyi protein or transcript stable ID' (default: False),\n",
      " 'csavignyi_homolog_orthology_confidence': 'C.savignyi orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csavignyi_homolog_orthology_type': 'C.savignyi homology type' (default: False),\n",
      " 'csavignyi_homolog_perc_id': '%id. target C.savignyi gene identical to query gene' (default: False),\n",
      " 'csavignyi_homolog_perc_id_r1': '%id. query gene identical to target C.savignyi gene' (default: False),\n",
      " 'csavignyi_homolog_subtype': 'Last common ancestor with C.savignyi' (default: False),\n",
      " 'csavignyi_homolog_wga_coverage': 'C.savignyi Whole-genome alignment coverage' (default: False),\n",
      " 'csemilaevis_homolog_associated_gene_name': 'Tongue sole gene name' (default: False),\n",
      " 'csemilaevis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csemilaevis_homolog_chrom_end': 'Tongue sole chromosome/scaffold end (bp)' (default: False),\n",
      " 'csemilaevis_homolog_chrom_start': 'Tongue sole chromosome/scaffold start (bp)' (default: False),\n",
      " 'csemilaevis_homolog_chromosome': 'Tongue sole chromosome/scaffold name' (default: False),\n",
      " 'csemilaevis_homolog_ensembl_gene': 'Tongue sole gene stable ID' (default: False),\n",
      " 'csemilaevis_homolog_ensembl_peptide': 'Tongue sole protein or transcript stable ID' (default: False),\n",
      " 'csemilaevis_homolog_goc_score': 'Tongue sole Gene-order conservation score' (default: False),\n",
      " 'csemilaevis_homolog_orthology_confidence': 'Tongue sole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csemilaevis_homolog_orthology_type': 'Tongue sole homology type' (default: False),\n",
      " 'csemilaevis_homolog_perc_id': '%id. target Tongue sole gene identical to query gene' (default: False),\n",
      " 'csemilaevis_homolog_perc_id_r1': '%id. query gene identical to target Tongue sole gene' (default: False),\n",
      " 'csemilaevis_homolog_subtype': 'Last common ancestor with Tongue sole' (default: False),\n",
      " 'csemilaevis_homolog_wga_coverage': 'Tongue sole Whole-genome alignment coverage' (default: False),\n",
      " 'csyrichta_homolog_associated_gene_name': 'Tarsier gene name' (default: False),\n",
      " 'csyrichta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'csyrichta_homolog_chrom_end': 'Tarsier chromosome/scaffold end (bp)' (default: False),\n",
      " 'csyrichta_homolog_chrom_start': 'Tarsier chromosome/scaffold start (bp)' (default: False),\n",
      " 'csyrichta_homolog_chromosome': 'Tarsier chromosome/scaffold name' (default: False),\n",
      " 'csyrichta_homolog_ensembl_gene': 'Tarsier gene stable ID' (default: False),\n",
      " 'csyrichta_homolog_ensembl_peptide': 'Tarsier protein or transcript stable ID' (default: False),\n",
      " 'csyrichta_homolog_goc_score': 'Tarsier Gene-order conservation score' (default: False),\n",
      " 'csyrichta_homolog_orthology_confidence': 'Tarsier orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'csyrichta_homolog_orthology_type': 'Tarsier homology type' (default: False),\n",
      " 'csyrichta_homolog_perc_id': '%id. target Tarsier gene identical to query gene' (default: False),\n",
      " 'csyrichta_homolog_perc_id_r1': '%id. query gene identical to target Tarsier gene' (default: False),\n",
      " 'csyrichta_homolog_subtype': 'Last common ancestor with Tarsier' (default: False),\n",
      " 'csyrichta_homolog_wga_coverage': 'Tarsier Whole-genome alignment coverage' (default: False),\n",
      " 'cvariegatus_homolog_associated_gene_name': 'Sheepshead minnow gene name' (default: False),\n",
      " 'cvariegatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cvariegatus_homolog_chrom_end': 'Sheepshead minnow chromosome/scaffold end (bp)' (default: False),\n",
      " 'cvariegatus_homolog_chrom_start': 'Sheepshead minnow chromosome/scaffold start (bp)' (default: False),\n",
      " 'cvariegatus_homolog_chromosome': 'Sheepshead minnow chromosome/scaffold name' (default: False),\n",
      " 'cvariegatus_homolog_ensembl_gene': 'Sheepshead minnow gene stable ID' (default: False),\n",
      " 'cvariegatus_homolog_ensembl_peptide': 'Sheepshead minnow protein or transcript stable ID' (default: False),\n",
      " 'cvariegatus_homolog_goc_score': 'Sheepshead minnow Gene-order conservation score' (default: False),\n",
      " 'cvariegatus_homolog_orthology_confidence': 'Sheepshead minnow orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cvariegatus_homolog_orthology_type': 'Sheepshead minnow homology type' (default: False),\n",
      " 'cvariegatus_homolog_perc_id': '%id. target Sheepshead minnow gene identical to query gene' (default: False),\n",
      " 'cvariegatus_homolog_perc_id_r1': '%id. query gene identical to target Sheepshead minnow gene' (default: False),\n",
      " 'cvariegatus_homolog_subtype': 'Last common ancestor with Sheepshead minnow' (default: False),\n",
      " 'cvariegatus_homolog_wga_coverage': 'Sheepshead minnow Whole-genome alignment coverage' (default: False),\n",
      " 'cwagneri_homolog_associated_gene_name': 'Chacoan peccary gene name' (default: False),\n",
      " 'cwagneri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'cwagneri_homolog_chrom_end': 'Chacoan peccary chromosome/scaffold end (bp)' (default: False),\n",
      " 'cwagneri_homolog_chrom_start': 'Chacoan peccary chromosome/scaffold start (bp)' (default: False),\n",
      " 'cwagneri_homolog_chromosome': 'Chacoan peccary chromosome/scaffold name' (default: False),\n",
      " 'cwagneri_homolog_ensembl_gene': 'Chacoan peccary gene stable ID' (default: False),\n",
      " 'cwagneri_homolog_ensembl_peptide': 'Chacoan peccary protein or transcript stable ID' (default: False),\n",
      " 'cwagneri_homolog_goc_score': 'Chacoan peccary Gene-order conservation score' (default: False),\n",
      " 'cwagneri_homolog_orthology_confidence': 'Chacoan peccary orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'cwagneri_homolog_orthology_type': 'Chacoan peccary homology type' (default: False),\n",
      " 'cwagneri_homolog_perc_id': '%id. target Chacoan peccary gene identical to query gene' (default: False),\n",
      " 'cwagneri_homolog_perc_id_r1': '%id. query gene identical to target Chacoan peccary gene' (default: False),\n",
      " 'cwagneri_homolog_subtype': 'Last common ancestor with Chacoan peccary' (default: False),\n",
      " 'cwagneri_homolog_wga_coverage': 'Chacoan peccary Whole-genome alignment coverage' (default: False),\n",
      " 'dbass3_id': 'DataBase of Aberrant 3' Splice Sites ID' (default: False),\n",
      " 'dbass3_name': 'DataBase of Aberrant 3' Splice Sites name' (default: False),\n",
      " 'dbass5_id': 'DataBase of Aberrant 5' Splice Sites ID' (default: False),\n",
      " 'dbass5_name': 'DataBase of Aberrant 5' Splice Sites name' (default: False),\n",
      " 'dclupeoides_homolog_associated_gene_name': 'Denticle herring gene name' (default: False),\n",
      " 'dclupeoides_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dclupeoides_homolog_chrom_end': 'Denticle herring chromosome/scaffold end (bp)' (default: False),\n",
      " 'dclupeoides_homolog_chrom_start': 'Denticle herring chromosome/scaffold start (bp)' (default: False),\n",
      " 'dclupeoides_homolog_chromosome': 'Denticle herring chromosome/scaffold name' (default: False),\n",
      " 'dclupeoides_homolog_ensembl_gene': 'Denticle herring gene stable ID' (default: False),\n",
      " 'dclupeoides_homolog_ensembl_peptide': 'Denticle herring protein or transcript stable ID' (default: False),\n",
      " 'dclupeoides_homolog_goc_score': 'Denticle herring Gene-order conservation score' (default: False),\n",
      " 'dclupeoides_homolog_orthology_confidence': 'Denticle herring orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dclupeoides_homolog_orthology_type': 'Denticle herring homology type' (default: False),\n",
      " 'dclupeoides_homolog_perc_id': '%id. target Denticle herring gene identical to query gene' (default: False),\n",
      " 'dclupeoides_homolog_perc_id_r1': '%id. query gene identical to target Denticle herring gene' (default: False),\n",
      " 'dclupeoides_homolog_subtype': 'Last common ancestor with Denticle herring' (default: False),\n",
      " 'dclupeoides_homolog_wga_coverage': 'Denticle herring Whole-genome alignment coverage' (default: False),\n",
      " 'definition_1006': 'GO term definition' (default: False),\n",
      " 'description': 'Gene description' (default: False),\n",
      " 'distance_to_transcript_2076': 'Distance to transcript' (default: False),\n",
      " 'dlabrax_homolog_associated_gene_name': 'European seabass gene name' (default: False),\n",
      " 'dlabrax_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dlabrax_homolog_chrom_end': 'European seabass chromosome/scaffold end (bp)' (default: False),\n",
      " 'dlabrax_homolog_chrom_start': 'European seabass chromosome/scaffold start (bp)' (default: False),\n",
      " 'dlabrax_homolog_chromosome': 'European seabass chromosome/scaffold name' (default: False),\n",
      " 'dlabrax_homolog_ensembl_gene': 'European seabass gene stable ID' (default: False),\n",
      " 'dlabrax_homolog_ensembl_peptide': 'European seabass protein or transcript stable ID' (default: False),\n",
      " 'dlabrax_homolog_goc_score': 'European seabass Gene-order conservation score' (default: False),\n",
      " 'dlabrax_homolog_orthology_confidence': 'European seabass orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dlabrax_homolog_orthology_type': 'European seabass homology type' (default: False),\n",
      " 'dlabrax_homolog_perc_id': '%id. target European seabass gene identical to query gene' (default: False),\n",
      " 'dlabrax_homolog_perc_id_r1': '%id. query gene identical to target European seabass gene' (default: False),\n",
      " 'dlabrax_homolog_subtype': 'Last common ancestor with European seabass' (default: False),\n",
      " 'dleucas_homolog_associated_gene_name': 'Beluga whale gene name' (default: False),\n",
      " 'dleucas_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dleucas_homolog_chrom_end': 'Beluga whale chromosome/scaffold end (bp)' (default: False),\n",
      " 'dleucas_homolog_chrom_start': 'Beluga whale chromosome/scaffold start (bp)' (default: False),\n",
      " 'dleucas_homolog_chromosome': 'Beluga whale chromosome/scaffold name' (default: False),\n",
      " 'dleucas_homolog_ensembl_gene': 'Beluga whale gene stable ID' (default: False),\n",
      " 'dleucas_homolog_ensembl_peptide': 'Beluga whale protein or transcript stable ID' (default: False),\n",
      " 'dleucas_homolog_goc_score': 'Beluga whale Gene-order conservation score' (default: False),\n",
      " 'dleucas_homolog_orthology_confidence': 'Beluga whale orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dleucas_homolog_orthology_type': 'Beluga whale homology type' (default: False),\n",
      " 'dleucas_homolog_perc_id': '%id. target Beluga whale gene identical to query gene' (default: False),\n",
      " 'dleucas_homolog_perc_id_r1': '%id. query gene identical to target Beluga whale gene' (default: False),\n",
      " 'dleucas_homolog_subtype': 'Last common ancestor with Beluga whale' (default: False),\n",
      " 'dleucas_homolog_wga_coverage': 'Beluga whale Whole-genome alignment coverage' (default: False),\n",
      " 'dmelanogaster_homolog_associated_gene_name': 'Drosophila melanogaster (Fruit fly) gene name' (default: False),\n",
      " 'dmelanogaster_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dmelanogaster_homolog_chrom_end': 'Drosophila melanogaster (Fruit fly) chromosome/scaffold end (bp)' (default: False),\n",
      " 'dmelanogaster_homolog_chrom_start': 'Drosophila melanogaster (Fruit fly) chromosome/scaffold start (bp)' (default: False),\n",
      " 'dmelanogaster_homolog_chromosome': 'Drosophila melanogaster (Fruit fly) chromosome/scaffold name' (default: False),\n",
      " 'dmelanogaster_homolog_ensembl_gene': 'Drosophila melanogaster (Fruit fly) gene stable ID' (default: False),\n",
      " 'dmelanogaster_homolog_ensembl_peptide': 'Drosophila melanogaster (Fruit fly) protein or transcript stable ID' (default: False),\n",
      " 'dmelanogaster_homolog_orthology_confidence': 'Drosophila melanogaster (Fruit fly) orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dmelanogaster_homolog_orthology_type': 'Drosophila melanogaster (Fruit fly) homology type' (default: False),\n",
      " 'dmelanogaster_homolog_perc_id': '%id. target Drosophila melanogaster (Fruit fly) gene identical to query gene' (default: False),\n",
      " 'dmelanogaster_homolog_perc_id_r1': '%id. query gene identical to target Drosophila melanogaster (Fruit fly) gene' (default: False),\n",
      " 'dmelanogaster_homolog_subtype': 'Last common ancestor with Drosophila melanogaster (Fruit fly)' (default: False),\n",
      " 'dnovemcinctus_homolog_associated_gene_name': 'Armadillo gene name' (default: False),\n",
      " 'dnovemcinctus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dnovemcinctus_homolog_chrom_end': 'Armadillo chromosome/scaffold end (bp)' (default: False),\n",
      " 'dnovemcinctus_homolog_chrom_start': 'Armadillo chromosome/scaffold start (bp)' (default: False),\n",
      " 'dnovemcinctus_homolog_chromosome': 'Armadillo chromosome/scaffold name' (default: False),\n",
      " 'dnovemcinctus_homolog_ensembl_gene': 'Armadillo gene stable ID' (default: False),\n",
      " 'dnovemcinctus_homolog_ensembl_peptide': 'Armadillo protein or transcript stable ID' (default: False),\n",
      " 'dnovemcinctus_homolog_goc_score': 'Armadillo Gene-order conservation score' (default: False),\n",
      " 'dnovemcinctus_homolog_orthology_confidence': 'Armadillo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dnovemcinctus_homolog_orthology_type': 'Armadillo homology type' (default: False),\n",
      " 'dnovemcinctus_homolog_perc_id': '%id. target Armadillo gene identical to query gene' (default: False),\n",
      " 'dnovemcinctus_homolog_perc_id_r1': '%id. query gene identical to target Armadillo gene' (default: False),\n",
      " 'dnovemcinctus_homolog_subtype': 'Last common ancestor with Armadillo' (default: False),\n",
      " 'dnovemcinctus_homolog_wga_coverage': 'Armadillo Whole-genome alignment coverage' (default: False),\n",
      " 'dordii_homolog_associated_gene_name': 'Kangaroo rat gene name' (default: False),\n",
      " 'dordii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'dordii_homolog_chrom_end': 'Kangaroo rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'dordii_homolog_chrom_start': 'Kangaroo rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'dordii_homolog_chromosome': 'Kangaroo rat chromosome/scaffold name' (default: False),\n",
      " 'dordii_homolog_ensembl_gene': 'Kangaroo rat gene stable ID' (default: False),\n",
      " 'dordii_homolog_ensembl_peptide': 'Kangaroo rat protein or transcript stable ID' (default: False),\n",
      " 'dordii_homolog_goc_score': 'Kangaroo rat Gene-order conservation score' (default: False),\n",
      " 'dordii_homolog_orthology_confidence': 'Kangaroo rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'dordii_homolog_orthology_type': 'Kangaroo rat homology type' (default: False),\n",
      " 'dordii_homolog_perc_id': '%id. target Kangaroo rat gene identical to query gene' (default: False),\n",
      " 'dordii_homolog_perc_id_r1': '%id. query gene identical to target Kangaroo rat gene' (default: False),\n",
      " 'dordii_homolog_subtype': 'Last common ancestor with Kangaroo rat' (default: False),\n",
      " 'dordii_homolog_wga_coverage': 'Kangaroo rat Whole-genome alignment coverage' (default: False),\n",
      " 'downstream_flank': 'downstream_flank' (default: False),\n",
      " 'drerio_homolog_associated_gene_name': 'Zebrafish gene name' (default: False),\n",
      " 'drerio_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'drerio_homolog_chrom_end': 'Zebrafish chromosome/scaffold end (bp)' (default: False),\n",
      " 'drerio_homolog_chrom_start': 'Zebrafish chromosome/scaffold start (bp)' (default: False),\n",
      " 'drerio_homolog_chromosome': 'Zebrafish chromosome/scaffold name' (default: False),\n",
      " 'drerio_homolog_ensembl_gene': 'Zebrafish gene stable ID' (default: False),\n",
      " 'drerio_homolog_ensembl_peptide': 'Zebrafish protein or transcript stable ID' (default: False),\n",
      " 'drerio_homolog_goc_score': 'Zebrafish Gene-order conservation score' (default: False),\n",
      " 'drerio_homolog_orthology_confidence': 'Zebrafish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'drerio_homolog_orthology_type': 'Zebrafish homology type' (default: False),\n",
      " 'drerio_homolog_perc_id': '%id. target Zebrafish gene identical to query gene' (default: False),\n",
      " 'drerio_homolog_perc_id_r1': '%id. query gene identical to target Zebrafish gene' (default: False),\n",
      " 'drerio_homolog_subtype': 'Last common ancestor with Zebrafish' (default: False),\n",
      " 'drerio_homolog_wga_coverage': 'Zebrafish Whole-genome alignment coverage' (default: False),\n",
      " 'easinus_homolog_associated_gene_name': 'Donkey gene name' (default: False),\n",
      " 'easinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'easinus_homolog_chrom_end': 'Donkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'easinus_homolog_chrom_start': 'Donkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'easinus_homolog_chromosome': 'Donkey chromosome/scaffold name' (default: False),\n",
      " 'easinus_homolog_ensembl_gene': 'Donkey gene stable ID' (default: False),\n",
      " 'easinus_homolog_ensembl_peptide': 'Donkey protein or transcript stable ID' (default: False),\n",
      " 'easinus_homolog_goc_score': 'Donkey Gene-order conservation score' (default: False),\n",
      " 'easinus_homolog_orthology_confidence': 'Donkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'easinus_homolog_orthology_type': 'Donkey homology type' (default: False),\n",
      " 'easinus_homolog_perc_id': '%id. target Donkey gene identical to query gene' (default: False),\n",
      " 'easinus_homolog_perc_id_r1': '%id. query gene identical to target Donkey gene' (default: False),\n",
      " 'easinus_homolog_subtype': 'Last common ancestor with Donkey' (default: False),\n",
      " 'eburgeri_homolog_associated_gene_name': 'Hagfish gene name' (default: False),\n",
      " 'eburgeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eburgeri_homolog_chrom_end': 'Hagfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'eburgeri_homolog_chrom_start': 'Hagfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'eburgeri_homolog_chromosome': 'Hagfish chromosome/scaffold name' (default: False),\n",
      " 'eburgeri_homolog_ensembl_gene': 'Hagfish gene stable ID' (default: False),\n",
      " 'eburgeri_homolog_ensembl_peptide': 'Hagfish protein or transcript stable ID' (default: False),\n",
      " 'eburgeri_homolog_orthology_confidence': 'Hagfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eburgeri_homolog_orthology_type': 'Hagfish homology type' (default: False),\n",
      " 'eburgeri_homolog_perc_id': '%id. target Hagfish gene identical to query gene' (default: False),\n",
      " 'eburgeri_homolog_perc_id_r1': '%id. query gene identical to target Hagfish gene' (default: False),\n",
      " 'eburgeri_homolog_subtype': 'Last common ancestor with Hagfish' (default: False),\n",
      " 'eburgeri_homolog_wga_coverage': 'Hagfish Whole-genome alignment coverage' (default: False),\n",
      " 'ecaballus_homolog_associated_gene_name': 'Horse gene name' (default: False),\n",
      " 'ecaballus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ecaballus_homolog_chrom_end': 'Horse chromosome/scaffold end (bp)' (default: False),\n",
      " 'ecaballus_homolog_chrom_start': 'Horse chromosome/scaffold start (bp)' (default: False),\n",
      " 'ecaballus_homolog_chromosome': 'Horse chromosome/scaffold name' (default: False),\n",
      " 'ecaballus_homolog_ensembl_gene': 'Horse gene stable ID' (default: False),\n",
      " 'ecaballus_homolog_ensembl_peptide': 'Horse protein or transcript stable ID' (default: False),\n",
      " 'ecaballus_homolog_goc_score': 'Horse Gene-order conservation score' (default: False),\n",
      " 'ecaballus_homolog_orthology_confidence': 'Horse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ecaballus_homolog_orthology_type': 'Horse homology type' (default: False),\n",
      " 'ecaballus_homolog_perc_id': '%id. target Horse gene identical to query gene' (default: False),\n",
      " 'ecaballus_homolog_perc_id_r1': '%id. query gene identical to target Horse gene' (default: False),\n",
      " 'ecaballus_homolog_subtype': 'Last common ancestor with Horse' (default: False),\n",
      " 'ecaballus_homolog_wga_coverage': 'Horse Whole-genome alignment coverage' (default: False),\n",
      " 'ecalabaricus_homolog_associated_gene_name': 'Reedfish gene name' (default: False),\n",
      " 'ecalabaricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ecalabaricus_homolog_chrom_end': 'Reedfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ecalabaricus_homolog_chrom_start': 'Reedfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ecalabaricus_homolog_chromosome': 'Reedfish chromosome/scaffold name' (default: False),\n",
      " 'ecalabaricus_homolog_ensembl_gene': 'Reedfish gene stable ID' (default: False),\n",
      " 'ecalabaricus_homolog_ensembl_peptide': 'Reedfish protein or transcript stable ID' (default: False),\n",
      " 'ecalabaricus_homolog_goc_score': 'Reedfish Gene-order conservation score' (default: False),\n",
      " 'ecalabaricus_homolog_orthology_confidence': 'Reedfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ecalabaricus_homolog_orthology_type': 'Reedfish homology type' (default: False),\n",
      " 'ecalabaricus_homolog_perc_id': '%id. target Reedfish gene identical to query gene' (default: False),\n",
      " 'ecalabaricus_homolog_perc_id_r1': '%id. query gene identical to target Reedfish gene' (default: False),\n",
      " 'ecalabaricus_homolog_subtype': 'Last common ancestor with Reedfish' (default: False),\n",
      " 'ecalabaricus_homolog_wga_coverage': 'Reedfish Whole-genome alignment coverage' (default: False),\n",
      " 'eelectricus_homolog_associated_gene_name': 'Electric eel gene name' (default: False),\n",
      " 'eelectricus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eelectricus_homolog_chrom_end': 'Electric eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'eelectricus_homolog_chrom_start': 'Electric eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'eelectricus_homolog_chromosome': 'Electric eel chromosome/scaffold name' (default: False),\n",
      " 'eelectricus_homolog_ensembl_gene': 'Electric eel gene stable ID' (default: False),\n",
      " 'eelectricus_homolog_ensembl_peptide': 'Electric eel protein or transcript stable ID' (default: False),\n",
      " 'eelectricus_homolog_goc_score': 'Electric eel Gene-order conservation score' (default: False),\n",
      " 'eelectricus_homolog_orthology_confidence': 'Electric eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eelectricus_homolog_orthology_type': 'Electric eel homology type' (default: False),\n",
      " 'eelectricus_homolog_perc_id': '%id. target Electric eel gene identical to query gene' (default: False),\n",
      " 'eelectricus_homolog_perc_id_r1': '%id. query gene identical to target Electric eel gene' (default: False),\n",
      " 'eelectricus_homolog_subtype': 'Last common ancestor with Electric eel' (default: False),\n",
      " 'eelectricus_homolog_wga_coverage': 'Electric eel Whole-genome alignment coverage' (default: False),\n",
      " 'eeuropaeus_homolog_associated_gene_name': 'Hedgehog gene name' (default: False),\n",
      " 'eeuropaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'eeuropaeus_homolog_chrom_end': 'Hedgehog chromosome/scaffold end (bp)' (default: False),\n",
      " 'eeuropaeus_homolog_chrom_start': 'Hedgehog chromosome/scaffold start (bp)' (default: False),\n",
      " 'eeuropaeus_homolog_chromosome': 'Hedgehog chromosome/scaffold name' (default: False),\n",
      " 'eeuropaeus_homolog_ensembl_gene': 'Hedgehog gene stable ID' (default: False),\n",
      " 'eeuropaeus_homolog_ensembl_peptide': 'Hedgehog protein or transcript stable ID' (default: False),\n",
      " 'eeuropaeus_homolog_goc_score': 'Hedgehog Gene-order conservation score' (default: False),\n",
      " 'eeuropaeus_homolog_orthology_confidence': 'Hedgehog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'eeuropaeus_homolog_orthology_type': 'Hedgehog homology type' (default: False),\n",
      " 'eeuropaeus_homolog_perc_id': '%id. target Hedgehog gene identical to query gene' (default: False),\n",
      " 'eeuropaeus_homolog_perc_id_r1': '%id. query gene identical to target Hedgehog gene' (default: False),\n",
      " 'eeuropaeus_homolog_subtype': 'Last common ancestor with Hedgehog' (default: False),\n",
      " 'eeuropaeus_homolog_wga_coverage': 'Hedgehog Whole-genome alignment coverage' (default: False),\n",
      " 'elucius_homolog_associated_gene_name': 'Northern pike gene name' (default: False),\n",
      " 'elucius_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'elucius_homolog_chrom_end': 'Northern pike chromosome/scaffold end (bp)' (default: False),\n",
      " 'elucius_homolog_chrom_start': 'Northern pike chromosome/scaffold start (bp)' (default: False),\n",
      " 'elucius_homolog_chromosome': 'Northern pike chromosome/scaffold name' (default: False),\n",
      " 'elucius_homolog_ensembl_gene': 'Northern pike gene stable ID' (default: False),\n",
      " 'elucius_homolog_ensembl_peptide': 'Northern pike protein or transcript stable ID' (default: False),\n",
      " 'elucius_homolog_goc_score': 'Northern pike Gene-order conservation score' (default: False),\n",
      " 'elucius_homolog_orthology_confidence': 'Northern pike orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'elucius_homolog_orthology_type': 'Northern pike homology type' (default: False),\n",
      " 'elucius_homolog_perc_id': '%id. target Northern pike gene identical to query gene' (default: False),\n",
      " 'elucius_homolog_perc_id_r1': '%id. query gene identical to target Northern pike gene' (default: False),\n",
      " 'elucius_homolog_subtype': 'Last common ancestor with Northern pike' (default: False),\n",
      " 'elucius_homolog_wga_coverage': 'Northern pike Whole-genome alignment coverage' (default: False),\n",
      " 'embl': 'European Nucleotide Archive ID' (default: False),\n",
      " 'end_phase': 'End phase' (default: False),\n",
      " 'end_position': 'Gene end (bp)' (default: False),\n",
      " 'ens_lrg_gene': 'LRG display in Ensembl gene ID' (default: False),\n",
      " 'ens_lrg_transcript': 'LRG display in Ensembl transcript ID' (default: False),\n",
      " 'ensembl_exon_id': 'Exon stable ID' (default: False),\n",
      " 'ensembl_gene_id': 'Gene stable ID' (default: False),\n",
      " 'ensembl_gene_id_version': 'Gene stable ID version' (default: False),\n",
      " 'ensembl_peptide_id': 'Protein stable ID' (default: False),\n",
      " 'ensembl_peptide_id_version': 'Protein stable ID version' (default: False),\n",
      " 'ensembl_transcript_id': 'Transcript stable ID' (default: False),\n",
      " 'ensembl_transcript_id_version': 'Transcript stable ID version' (default: False),\n",
      " 'entrezgene_accession': 'NCBI gene (formerly Entrezgene) accession' (default: False),\n",
      " 'entrezgene_description': 'NCBI gene (formerly Entrezgene) description' (default: False),\n",
      " 'entrezgene_id': 'NCBI gene (formerly Entrezgene) ID' (default: False),\n",
      " 'entrezgene_trans_name': 'EntrezGene transcript name ID' (default: False),\n",
      " 'etelfairi_homolog_associated_gene_name': 'Lesser hedgehog tenrec gene name' (default: False),\n",
      " 'etelfairi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'etelfairi_homolog_chrom_end': 'Lesser hedgehog tenrec chromosome/scaffold end (bp)' (default: False),\n",
      " 'etelfairi_homolog_chrom_start': 'Lesser hedgehog tenrec chromosome/scaffold start (bp)' (default: False),\n",
      " 'etelfairi_homolog_chromosome': 'Lesser hedgehog tenrec chromosome/scaffold name' (default: False),\n",
      " 'etelfairi_homolog_ensembl_gene': 'Lesser hedgehog tenrec gene stable ID' (default: False),\n",
      " 'etelfairi_homolog_ensembl_peptide': 'Lesser hedgehog tenrec protein or transcript stable ID' (default: False),\n",
      " 'etelfairi_homolog_goc_score': 'Lesser hedgehog tenrec Gene-order conservation score' (default: False),\n",
      " 'etelfairi_homolog_orthology_confidence': 'Lesser hedgehog tenrec orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'etelfairi_homolog_orthology_type': 'Lesser hedgehog tenrec homology type' (default: False),\n",
      " 'etelfairi_homolog_perc_id': '%id. target Lesser hedgehog tenrec gene identical to query gene' (default: False),\n",
      " 'etelfairi_homolog_perc_id_r1': '%id. query gene identical to target Lesser hedgehog tenrec gene' (default: False),\n",
      " 'etelfairi_homolog_subtype': 'Last common ancestor with Lesser hedgehog tenrec' (default: False),\n",
      " 'etelfairi_homolog_wga_coverage': 'Lesser hedgehog tenrec Whole-genome alignment coverage' (default: False),\n",
      " 'exon_chrom_end': 'Exon region end (bp)' (default: False),\n",
      " 'exon_chrom_start': 'Exon region start (bp)' (default: False),\n",
      " 'external_gene_name': 'Gene name' (default: False),\n",
      " 'external_gene_source': 'Source of gene name' (default: False),\n",
      " 'external_synonym': 'Gene Synonym' (default: False),\n",
      " 'external_transcript_name': 'Transcript name' (default: False),\n",
      " 'external_transcript_source_name': 'Source of transcript name' (default: False),\n",
      " 'falbicollis_homolog_associated_gene_name': 'Collared flycatcher gene name' (default: False),\n",
      " 'falbicollis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'falbicollis_homolog_chrom_end': 'Collared flycatcher chromosome/scaffold end (bp)' (default: False),\n",
      " 'falbicollis_homolog_chrom_start': 'Collared flycatcher chromosome/scaffold start (bp)' (default: False),\n",
      " 'falbicollis_homolog_chromosome': 'Collared flycatcher chromosome/scaffold name' (default: False),\n",
      " 'falbicollis_homolog_ensembl_gene': 'Collared flycatcher gene stable ID' (default: False),\n",
      " 'falbicollis_homolog_ensembl_peptide': 'Collared flycatcher protein or transcript stable ID' (default: False),\n",
      " 'falbicollis_homolog_goc_score': 'Collared flycatcher Gene-order conservation score' (default: False),\n",
      " 'falbicollis_homolog_orthology_confidence': 'Collared flycatcher orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'falbicollis_homolog_orthology_type': 'Collared flycatcher homology type' (default: False),\n",
      " 'falbicollis_homolog_perc_id': '%id. target Collared flycatcher gene identical to query gene' (default: False),\n",
      " 'falbicollis_homolog_perc_id_r1': '%id. query gene identical to target Collared flycatcher gene' (default: False),\n",
      " 'falbicollis_homolog_subtype': 'Last common ancestor with Collared flycatcher' (default: False),\n",
      " 'fcatus_homolog_associated_gene_name': 'Cat gene name' (default: False),\n",
      " 'fcatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fcatus_homolog_chrom_end': 'Cat chromosome/scaffold end (bp)' (default: False),\n",
      " 'fcatus_homolog_chrom_start': 'Cat chromosome/scaffold start (bp)' (default: False),\n",
      " 'fcatus_homolog_chromosome': 'Cat chromosome/scaffold name' (default: False),\n",
      " 'fcatus_homolog_ensembl_gene': 'Cat gene stable ID' (default: False),\n",
      " 'fcatus_homolog_ensembl_peptide': 'Cat protein or transcript stable ID' (default: False),\n",
      " 'fcatus_homolog_goc_score': 'Cat Gene-order conservation score' (default: False),\n",
      " 'fcatus_homolog_orthology_confidence': 'Cat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fcatus_homolog_orthology_type': 'Cat homology type' (default: False),\n",
      " 'fcatus_homolog_perc_id': '%id. target Cat gene identical to query gene' (default: False),\n",
      " 'fcatus_homolog_perc_id_r1': '%id. query gene identical to target Cat gene' (default: False),\n",
      " 'fcatus_homolog_subtype': 'Last common ancestor with Cat' (default: False),\n",
      " 'fcatus_homolog_wga_coverage': 'Cat Whole-genome alignment coverage' (default: False),\n",
      " 'fheteroclitus_homolog_associated_gene_name': 'Mummichog gene name' (default: False),\n",
      " 'fheteroclitus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'fheteroclitus_homolog_chrom_end': 'Mummichog chromosome/scaffold end (bp)' (default: False),\n",
      " 'fheteroclitus_homolog_chrom_start': 'Mummichog chromosome/scaffold start (bp)' (default: False),\n",
      " 'fheteroclitus_homolog_chromosome': 'Mummichog chromosome/scaffold name' (default: False),\n",
      " 'fheteroclitus_homolog_ensembl_gene': 'Mummichog gene stable ID' (default: False),\n",
      " 'fheteroclitus_homolog_ensembl_peptide': 'Mummichog protein or transcript stable ID' (default: False),\n",
      " 'fheteroclitus_homolog_goc_score': 'Mummichog Gene-order conservation score' (default: False),\n",
      " 'fheteroclitus_homolog_orthology_confidence': 'Mummichog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'fheteroclitus_homolog_orthology_type': 'Mummichog homology type' (default: False),\n",
      " 'fheteroclitus_homolog_perc_id': '%id. target Mummichog gene identical to query gene' (default: False),\n",
      " 'fheteroclitus_homolog_perc_id_r1': '%id. query gene identical to target Mummichog gene' (default: False),\n",
      " 'fheteroclitus_homolog_subtype': 'Last common ancestor with Mummichog' (default: False),\n",
      " 'fheteroclitus_homolog_wga_coverage': 'Mummichog Whole-genome alignment coverage' (default: False),\n",
      " 'gaculeatus_homolog_associated_gene_name': 'Stickleback gene name' (default: False),\n",
      " 'gaculeatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gaculeatus_homolog_chrom_end': 'Stickleback chromosome/scaffold end (bp)' (default: False),\n",
      " 'gaculeatus_homolog_chrom_start': 'Stickleback chromosome/scaffold start (bp)' (default: False),\n",
      " 'gaculeatus_homolog_chromosome': 'Stickleback chromosome/scaffold name' (default: False),\n",
      " 'gaculeatus_homolog_ensembl_gene': 'Stickleback gene stable ID' (default: False),\n",
      " 'gaculeatus_homolog_ensembl_peptide': 'Stickleback protein or transcript stable ID' (default: False),\n",
      " 'gaculeatus_homolog_goc_score': 'Stickleback Gene-order conservation score' (default: False),\n",
      " 'gaculeatus_homolog_orthology_confidence': 'Stickleback orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gaculeatus_homolog_orthology_type': 'Stickleback homology type' (default: False),\n",
      " 'gaculeatus_homolog_perc_id': '%id. target Stickleback gene identical to query gene' (default: False),\n",
      " 'gaculeatus_homolog_perc_id_r1': '%id. query gene identical to target Stickleback gene' (default: False),\n",
      " 'gaculeatus_homolog_subtype': 'Last common ancestor with Stickleback' (default: False),\n",
      " 'gaculeatus_homolog_wga_coverage': 'Stickleback Whole-genome alignment coverage' (default: False),\n",
      " 'gene3d': 'Gene3D ID' (default: False),\n",
      " 'gene3d_end': 'Gene3D end' (default: False),\n",
      " 'gene3d_start': 'Gene3D start' (default: False),\n",
      " 'gene_biotype': 'Gene type' (default: False),\n",
      " 'gene_exon': 'Exon sequences' (default: False),\n",
      " 'gene_exon_intron': 'Unspliced (Gene)' (default: False),\n",
      " 'gene_flank': 'Flank (Gene)' (default: False),\n",
      " 'genecards': 'GeneCards ID' (default: False),\n",
      " 'genomic_coding_end': 'Genomic coding end' (default: False),\n",
      " 'genomic_coding_start': 'Genomic coding start' (default: False),\n",
      " 'germ_line_variation_source': 'Variant source' (default: False),\n",
      " 'gevgoodei_homolog_associated_gene_name': 'Goodes thornscrub tortoise gene name' (default: False),\n",
      " 'gevgoodei_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gevgoodei_homolog_chrom_end': 'Goodes thornscrub tortoise chromosome/scaffold end (bp)' (default: False),\n",
      " 'gevgoodei_homolog_chrom_start': 'Goodes thornscrub tortoise chromosome/scaffold start (bp)' (default: False),\n",
      " 'gevgoodei_homolog_chromosome': 'Goodes thornscrub tortoise chromosome/scaffold name' (default: False),\n",
      " 'gevgoodei_homolog_ensembl_gene': 'Goodes thornscrub tortoise gene stable ID' (default: False),\n",
      " 'gevgoodei_homolog_ensembl_peptide': 'Goodes thornscrub tortoise protein or transcript stable ID' (default: False),\n",
      " 'gevgoodei_homolog_goc_score': 'Goodes thornscrub tortoise Gene-order conservation score' (default: False),\n",
      " 'gevgoodei_homolog_orthology_confidence': 'Goodes thornscrub tortoise orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gevgoodei_homolog_orthology_type': 'Goodes thornscrub tortoise homology type' (default: False),\n",
      " 'gevgoodei_homolog_perc_id': '%id. target Goodes thornscrub tortoise gene identical to query gene' (default: False),\n",
      " 'gevgoodei_homolog_perc_id_r1': '%id. query gene identical to target Goodes thornscrub tortoise gene' (default: False),\n",
      " 'gevgoodei_homolog_subtype': 'Last common ancestor with Goodes thornscrub tortoise' (default: False),\n",
      " 'gevgoodei_homolog_wga_coverage': 'Goodes thornscrub tortoise Whole-genome alignment coverage' (default: False),\n",
      " 'gfortis_homolog_associated_gene_name': 'Medium ground-finch gene name' (default: False),\n",
      " 'gfortis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gfortis_homolog_chrom_end': 'Medium ground-finch chromosome/scaffold end (bp)' (default: False),\n",
      " 'gfortis_homolog_chrom_start': 'Medium ground-finch chromosome/scaffold start (bp)' (default: False),\n",
      " 'gfortis_homolog_chromosome': 'Medium ground-finch chromosome/scaffold name' (default: False),\n",
      " 'gfortis_homolog_ensembl_gene': 'Medium ground-finch gene stable ID' (default: False),\n",
      " 'gfortis_homolog_ensembl_peptide': 'Medium ground-finch protein or transcript stable ID' (default: False),\n",
      " 'gfortis_homolog_goc_score': 'Medium ground-finch Gene-order conservation score' (default: False),\n",
      " 'gfortis_homolog_orthology_confidence': 'Medium ground-finch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gfortis_homolog_orthology_type': 'Medium ground-finch homology type' (default: False),\n",
      " 'gfortis_homolog_perc_id': '%id. target Medium ground-finch gene identical to query gene' (default: False),\n",
      " 'gfortis_homolog_perc_id_r1': '%id. query gene identical to target Medium ground-finch gene' (default: False),\n",
      " 'gfortis_homolog_subtype': 'Last common ancestor with Medium ground-finch' (default: False),\n",
      " 'gfortis_homolog_wga_coverage': 'Medium ground-finch Whole-genome alignment coverage' (default: False),\n",
      " 'ggallus_homolog_associated_gene_name': 'Chicken gene name' (default: False),\n",
      " 'ggallus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ggallus_homolog_chrom_end': 'Chicken chromosome/scaffold end (bp)' (default: False),\n",
      " 'ggallus_homolog_chrom_start': 'Chicken chromosome/scaffold start (bp)' (default: False),\n",
      " 'ggallus_homolog_chromosome': 'Chicken chromosome/scaffold name' (default: False),\n",
      " 'ggallus_homolog_ensembl_gene': 'Chicken gene stable ID' (default: False),\n",
      " 'ggallus_homolog_ensembl_peptide': 'Chicken protein or transcript stable ID' (default: False),\n",
      " 'ggallus_homolog_goc_score': 'Chicken Gene-order conservation score' (default: False),\n",
      " 'ggallus_homolog_orthology_confidence': 'Chicken orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ggallus_homolog_orthology_type': 'Chicken homology type' (default: False),\n",
      " 'ggallus_homolog_perc_id': '%id. target Chicken gene identical to query gene' (default: False),\n",
      " 'ggallus_homolog_perc_id_r1': '%id. query gene identical to target Chicken gene' (default: False),\n",
      " 'ggallus_homolog_subtype': 'Last common ancestor with Chicken' (default: False),\n",
      " 'ggallus_homolog_wga_coverage': 'Chicken Whole-genome alignment coverage' (default: False),\n",
      " 'ggorilla_homolog_associated_gene_name': 'Gorilla gene name' (default: False),\n",
      " 'ggorilla_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ggorilla_homolog_chrom_end': 'Gorilla chromosome/scaffold end (bp)' (default: False),\n",
      " 'ggorilla_homolog_chrom_start': 'Gorilla chromosome/scaffold start (bp)' (default: False),\n",
      " 'ggorilla_homolog_chromosome': 'Gorilla chromosome/scaffold name' (default: False),\n",
      " 'ggorilla_homolog_ensembl_gene': 'Gorilla gene stable ID' (default: False),\n",
      " 'ggorilla_homolog_ensembl_peptide': 'Gorilla protein or transcript stable ID' (default: False),\n",
      " 'ggorilla_homolog_goc_score': 'Gorilla Gene-order conservation score' (default: False),\n",
      " 'ggorilla_homolog_orthology_confidence': 'Gorilla orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ggorilla_homolog_orthology_type': 'Gorilla homology type' (default: False),\n",
      " 'ggorilla_homolog_perc_id': '%id. target Gorilla gene identical to query gene' (default: False),\n",
      " 'ggorilla_homolog_perc_id_r1': '%id. query gene identical to target Gorilla gene' (default: False),\n",
      " 'ggorilla_homolog_subtype': 'Last common ancestor with Gorilla' (default: False),\n",
      " 'ggorilla_homolog_wga_coverage': 'Gorilla Whole-genome alignment coverage' (default: False),\n",
      " 'gmorhua_homolog_associated_gene_name': 'Atlantic cod gene name' (default: False),\n",
      " 'gmorhua_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'gmorhua_homolog_chrom_end': 'Atlantic cod chromosome/scaffold end (bp)' (default: False),\n",
      " 'gmorhua_homolog_chrom_start': 'Atlantic cod chromosome/scaffold start (bp)' (default: False),\n",
      " 'gmorhua_homolog_chromosome': 'Atlantic cod chromosome/scaffold name' (default: False),\n",
      " 'gmorhua_homolog_ensembl_gene': 'Atlantic cod gene stable ID' (default: False),\n",
      " 'gmorhua_homolog_ensembl_peptide': 'Atlantic cod protein or transcript stable ID' (default: False),\n",
      " 'gmorhua_homolog_goc_score': 'Atlantic cod Gene-order conservation score' (default: False),\n",
      " 'gmorhua_homolog_orthology_confidence': 'Atlantic cod orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'gmorhua_homolog_orthology_type': 'Atlantic cod homology type' (default: False),\n",
      " 'gmorhua_homolog_perc_id': '%id. target Atlantic cod gene identical to query gene' (default: False),\n",
      " 'gmorhua_homolog_perc_id_r1': '%id. query gene identical to target Atlantic cod gene' (default: False),\n",
      " 'gmorhua_homolog_subtype': 'Last common ancestor with Atlantic cod' (default: False),\n",
      " 'go_id': 'GO term accession' (default: False),\n",
      " 'go_linkage_type': 'GO term evidence code' (default: False),\n",
      " 'goslim_goa_accession': 'GOSlim GOA Accession(s)' (default: False),\n",
      " 'goslim_goa_description': 'GOSlim GOA Description' (default: False),\n",
      " 'hamap': 'HAMAP ID' (default: False),\n",
      " 'hamap_end': 'HAMAP end' (default: False),\n",
      " 'hamap_start': 'HAMAP start' (default: False),\n",
      " 'hburtoni_homolog_associated_gene_name': 'Burton's mouthbrooder gene name' (default: False),\n",
      " 'hburtoni_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hburtoni_homolog_chrom_end': 'Burton's mouthbrooder chromosome/scaffold end (bp)' (default: False),\n",
      " 'hburtoni_homolog_chrom_start': 'Burton's mouthbrooder chromosome/scaffold start (bp)' (default: False),\n",
      " 'hburtoni_homolog_chromosome': 'Burton's mouthbrooder chromosome/scaffold name' (default: False),\n",
      " 'hburtoni_homolog_ensembl_gene': 'Burton's mouthbrooder gene stable ID' (default: False),\n",
      " 'hburtoni_homolog_ensembl_peptide': 'Burton's mouthbrooder protein or transcript stable ID' (default: False),\n",
      " 'hburtoni_homolog_goc_score': 'Burton's mouthbrooder Gene-order conservation score' (default: False),\n",
      " 'hburtoni_homolog_orthology_confidence': 'Burton's mouthbrooder orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hburtoni_homolog_orthology_type': 'Burton's mouthbrooder homology type' (default: False),\n",
      " 'hburtoni_homolog_perc_id': '%id. target Burton's mouthbrooder gene identical to query gene' (default: False),\n",
      " 'hburtoni_homolog_perc_id_r1': '%id. query gene identical to target Burton's mouthbrooder gene' (default: False),\n",
      " 'hburtoni_homolog_subtype': 'Last common ancestor with Burton's mouthbrooder' (default: False),\n",
      " 'hburtoni_homolog_wga_coverage': 'Burton's mouthbrooder Whole-genome alignment coverage' (default: False),\n",
      " 'hcomes_homolog_associated_gene_name': 'Tiger tail seahorse gene name' (default: False),\n",
      " 'hcomes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hcomes_homolog_chrom_end': 'Tiger tail seahorse chromosome/scaffold end (bp)' (default: False),\n",
      " 'hcomes_homolog_chrom_start': 'Tiger tail seahorse chromosome/scaffold start (bp)' (default: False),\n",
      " 'hcomes_homolog_chromosome': 'Tiger tail seahorse chromosome/scaffold name' (default: False),\n",
      " 'hcomes_homolog_ensembl_gene': 'Tiger tail seahorse gene stable ID' (default: False),\n",
      " 'hcomes_homolog_ensembl_peptide': 'Tiger tail seahorse protein or transcript stable ID' (default: False),\n",
      " 'hcomes_homolog_goc_score': 'Tiger tail seahorse Gene-order conservation score' (default: False),\n",
      " 'hcomes_homolog_orthology_confidence': 'Tiger tail seahorse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hcomes_homolog_orthology_type': 'Tiger tail seahorse homology type' (default: False),\n",
      " 'hcomes_homolog_perc_id': '%id. target Tiger tail seahorse gene identical to query gene' (default: False),\n",
      " 'hcomes_homolog_perc_id_r1': '%id. query gene identical to target Tiger tail seahorse gene' (default: False),\n",
      " 'hcomes_homolog_subtype': 'Last common ancestor with Tiger tail seahorse' (default: False),\n",
      " 'hcomes_homolog_wga_coverage': 'Tiger tail seahorse Whole-genome alignment coverage' (default: False),\n",
      " 'hgfemale_homolog_associated_gene_name': 'Naked mole-rat female gene name' (default: False),\n",
      " 'hgfemale_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hgfemale_homolog_chrom_end': 'Naked mole-rat female chromosome/scaffold end (bp)' (default: False),\n",
      " 'hgfemale_homolog_chrom_start': 'Naked mole-rat female chromosome/scaffold start (bp)' (default: False),\n",
      " 'hgfemale_homolog_chromosome': 'Naked mole-rat female chromosome/scaffold name' (default: False),\n",
      " 'hgfemale_homolog_ensembl_gene': 'Naked mole-rat female gene stable ID' (default: False),\n",
      " 'hgfemale_homolog_ensembl_peptide': 'Naked mole-rat female protein or transcript stable ID' (default: False),\n",
      " 'hgfemale_homolog_goc_score': 'Naked mole-rat female Gene-order conservation score' (default: False),\n",
      " 'hgfemale_homolog_orthology_confidence': 'Naked mole-rat female orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hgfemale_homolog_orthology_type': 'Naked mole-rat female homology type' (default: False),\n",
      " 'hgfemale_homolog_perc_id': '%id. target Naked mole-rat female gene identical to query gene' (default: False),\n",
      " 'hgfemale_homolog_perc_id_r1': '%id. query gene identical to target Naked mole-rat female gene' (default: False),\n",
      " 'hgfemale_homolog_subtype': 'Last common ancestor with Naked mole-rat female' (default: False),\n",
      " 'hgnc_id': 'HGNC ID' (default: False),\n",
      " 'hgnc_symbol': 'HGNC symbol' (default: False),\n",
      " 'hgnc_trans_name': 'Transcript name ID' (default: False),\n",
      " 'hhucho_homolog_associated_gene_name': 'Huchen gene name' (default: False),\n",
      " 'hhucho_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'hhucho_homolog_chrom_end': 'Huchen chromosome/scaffold end (bp)' (default: False),\n",
      " 'hhucho_homolog_chrom_start': 'Huchen chromosome/scaffold start (bp)' (default: False),\n",
      " 'hhucho_homolog_chromosome': 'Huchen chromosome/scaffold name' (default: False),\n",
      " 'hhucho_homolog_ensembl_gene': 'Huchen gene stable ID' (default: False),\n",
      " 'hhucho_homolog_ensembl_peptide': 'Huchen protein or transcript stable ID' (default: False),\n",
      " 'hhucho_homolog_goc_score': 'Huchen Gene-order conservation score' (default: False),\n",
      " 'hhucho_homolog_orthology_confidence': 'Huchen orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'hhucho_homolog_orthology_type': 'Huchen homology type' (default: False),\n",
      " 'hhucho_homolog_perc_id': '%id. target Huchen gene identical to query gene' (default: False),\n",
      " 'hhucho_homolog_perc_id_r1': '%id. query gene identical to target Huchen gene' (default: False),\n",
      " 'hhucho_homolog_subtype': 'Last common ancestor with Huchen' (default: False),\n",
      " 'hhucho_homolog_wga_coverage': 'Huchen Whole-genome alignment coverage' (default: False),\n",
      " 'hmmpanther': 'PANTHER ID' (default: False),\n",
      " 'hmmpanther_end': 'PANTHER end' (default: False),\n",
      " 'hmmpanther_start': 'PANTHER start' (default: False),\n",
      " 'hpa_accession': 'Human Protein Atlas accession' (default: False),\n",
      " 'hpa_id': 'Human Protein Atlas ID' (default: False),\n",
      " 'hsapiens_paralog_associated_gene_name': 'Human paralogue associated gene name' (default: False),\n",
      " 'hsapiens_paralog_canonical_transcript_protein': 'Paralogue query protein or transcript ID' (default: False),\n",
      " 'hsapiens_paralog_chrom_end': 'Human paralogue chromosome/scaffold end (bp)' (default: False),\n",
      " 'hsapiens_paralog_chrom_start': 'Human paralogue chromosome/scaffold start (bp)' (default: False),\n",
      " 'hsapiens_paralog_chromosome': 'Human paralogue chromosome/scaffold name' (default: False),\n",
      " 'hsapiens_paralog_ensembl_gene': 'Human paralogue gene stable ID' (default: False),\n",
      " 'hsapiens_paralog_ensembl_peptide': 'Human paralogue protein or transcript ID' (default: False),\n",
      " 'hsapiens_paralog_orthology_type': 'Human paralogue homology type' (default: False),\n",
      " 'hsapiens_paralog_perc_id': 'Paralogue %id. target Human gene identical to query gene' (default: False),\n",
      " 'hsapiens_paralog_perc_id_r1': 'Paralogue %id. query gene identical to target Human gene' (default: False),\n",
      " 'hsapiens_paralog_subtype': 'Paralogue last common ancestor with Human' (default: False),\n",
      " 'illumina_humanref_8_v3': 'ILLUMINA HumanRef 8 V3 probe' (default: False),\n",
      " 'illumina_humanwg_6_v3': 'ILLUMINA HumanWG 6 V3 probe' (default: False),\n",
      " 'interpro': 'Interpro ID' (default: False),\n",
      " 'interpro_description': 'Interpro Description' (default: False),\n",
      " 'interpro_end': 'Interpro end' (default: False),\n",
      " 'interpro_short_description': 'Interpro Short Description' (default: False),\n",
      " 'interpro_start': 'Interpro start' (default: False),\n",
      " 'ipunctatus_homolog_associated_gene_name': 'Channel catfish gene name' (default: False),\n",
      " 'ipunctatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ipunctatus_homolog_chrom_end': 'Channel catfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ipunctatus_homolog_chrom_start': 'Channel catfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ipunctatus_homolog_chromosome': 'Channel catfish chromosome/scaffold name' (default: False),\n",
      " 'ipunctatus_homolog_ensembl_gene': 'Channel catfish gene stable ID' (default: False),\n",
      " 'ipunctatus_homolog_ensembl_peptide': 'Channel catfish protein or transcript stable ID' (default: False),\n",
      " 'ipunctatus_homolog_goc_score': 'Channel catfish Gene-order conservation score' (default: False),\n",
      " 'ipunctatus_homolog_orthology_confidence': 'Channel catfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ipunctatus_homolog_orthology_type': 'Channel catfish homology type' (default: False),\n",
      " 'ipunctatus_homolog_perc_id': '%id. target Channel catfish gene identical to query gene' (default: False),\n",
      " 'ipunctatus_homolog_perc_id_r1': '%id. query gene identical to target Channel catfish gene' (default: False),\n",
      " 'ipunctatus_homolog_subtype': 'Last common ancestor with Channel catfish' (default: False),\n",
      " 'ipunctatus_homolog_wga_coverage': 'Channel catfish Whole-genome alignment coverage' (default: False),\n",
      " 'is_constitutive': 'Constitutive exon' (default: False),\n",
      " 'itridecemlineatus_homolog_associated_gene_name': 'Squirrel gene name' (default: False),\n",
      " 'itridecemlineatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'itridecemlineatus_homolog_chrom_end': 'Squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'itridecemlineatus_homolog_chrom_start': 'Squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'itridecemlineatus_homolog_chromosome': 'Squirrel chromosome/scaffold name' (default: False),\n",
      " 'itridecemlineatus_homolog_ensembl_gene': 'Squirrel gene stable ID' (default: False),\n",
      " 'itridecemlineatus_homolog_ensembl_peptide': 'Squirrel protein or transcript stable ID' (default: False),\n",
      " 'itridecemlineatus_homolog_goc_score': 'Squirrel Gene-order conservation score' (default: False),\n",
      " 'itridecemlineatus_homolog_orthology_confidence': 'Squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'itridecemlineatus_homolog_orthology_type': 'Squirrel homology type' (default: False),\n",
      " 'itridecemlineatus_homolog_perc_id': '%id. target Squirrel gene identical to query gene' (default: False),\n",
      " 'itridecemlineatus_homolog_perc_id_r1': '%id. query gene identical to target Squirrel gene' (default: False),\n",
      " 'itridecemlineatus_homolog_subtype': 'Last common ancestor with Squirrel' (default: False),\n",
      " 'itridecemlineatus_homolog_wga_coverage': 'Squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'jjaculus_homolog_associated_gene_name': 'Lesser Egyptian jerboa gene name' (default: False),\n",
      " 'jjaculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'jjaculus_homolog_chrom_end': 'Lesser Egyptian jerboa chromosome/scaffold end (bp)' (default: False),\n",
      " 'jjaculus_homolog_chrom_start': 'Lesser Egyptian jerboa chromosome/scaffold start (bp)' (default: False),\n",
      " 'jjaculus_homolog_chromosome': 'Lesser Egyptian jerboa chromosome/scaffold name' (default: False),\n",
      " 'jjaculus_homolog_ensembl_gene': 'Lesser Egyptian jerboa gene stable ID' (default: False),\n",
      " 'jjaculus_homolog_ensembl_peptide': 'Lesser Egyptian jerboa protein or transcript stable ID' (default: False),\n",
      " 'jjaculus_homolog_goc_score': 'Lesser Egyptian jerboa Gene-order conservation score' (default: False),\n",
      " 'jjaculus_homolog_orthology_confidence': 'Lesser Egyptian jerboa orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'jjaculus_homolog_orthology_type': 'Lesser Egyptian jerboa homology type' (default: False),\n",
      " 'jjaculus_homolog_perc_id': '%id. target Lesser Egyptian jerboa gene identical to query gene' (default: False),\n",
      " 'jjaculus_homolog_perc_id_r1': '%id. query gene identical to target Lesser Egyptian jerboa gene' (default: False),\n",
      " 'jjaculus_homolog_subtype': 'Last common ancestor with Lesser Egyptian jerboa' (default: False),\n",
      " 'jjaculus_homolog_wga_coverage': 'Lesser Egyptian jerboa Whole-genome alignment coverage' (default: False),\n",
      " 'kmarmoratus_homolog_associated_gene_name': 'Mangrove rivulus gene name' (default: False),\n",
      " 'kmarmoratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'kmarmoratus_homolog_chrom_end': 'Mangrove rivulus chromosome/scaffold end (bp)' (default: False),\n",
      " 'kmarmoratus_homolog_chrom_start': 'Mangrove rivulus chromosome/scaffold start (bp)' (default: False),\n",
      " 'kmarmoratus_homolog_chromosome': 'Mangrove rivulus chromosome/scaffold name' (default: False),\n",
      " 'kmarmoratus_homolog_ensembl_gene': 'Mangrove rivulus gene stable ID' (default: False),\n",
      " 'kmarmoratus_homolog_ensembl_peptide': 'Mangrove rivulus protein or transcript stable ID' (default: False),\n",
      " 'kmarmoratus_homolog_goc_score': 'Mangrove rivulus Gene-order conservation score' (default: False),\n",
      " 'kmarmoratus_homolog_orthology_confidence': 'Mangrove rivulus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'kmarmoratus_homolog_orthology_type': 'Mangrove rivulus homology type' (default: False),\n",
      " 'kmarmoratus_homolog_perc_id': '%id. target Mangrove rivulus gene identical to query gene' (default: False),\n",
      " 'kmarmoratus_homolog_perc_id_r1': '%id. query gene identical to target Mangrove rivulus gene' (default: False),\n",
      " 'kmarmoratus_homolog_subtype': 'Last common ancestor with Mangrove rivulus' (default: False),\n",
      " 'kmarmoratus_homolog_wga_coverage': 'Mangrove rivulus Whole-genome alignment coverage' (default: False),\n",
      " 'lafricana_homolog_associated_gene_name': 'Elephant gene name' (default: False),\n",
      " 'lafricana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lafricana_homolog_chrom_end': 'Elephant chromosome/scaffold end (bp)' (default: False),\n",
      " 'lafricana_homolog_chrom_start': 'Elephant chromosome/scaffold start (bp)' (default: False),\n",
      " 'lafricana_homolog_chromosome': 'Elephant chromosome/scaffold name' (default: False),\n",
      " 'lafricana_homolog_ensembl_gene': 'Elephant gene stable ID' (default: False),\n",
      " 'lafricana_homolog_ensembl_peptide': 'Elephant protein or transcript stable ID' (default: False),\n",
      " 'lafricana_homolog_goc_score': 'Elephant Gene-order conservation score' (default: False),\n",
      " 'lafricana_homolog_orthology_confidence': 'Elephant orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lafricana_homolog_orthology_type': 'Elephant homology type' (default: False),\n",
      " 'lafricana_homolog_perc_id': '%id. target Elephant gene identical to query gene' (default: False),\n",
      " 'lafricana_homolog_perc_id_r1': '%id. query gene identical to target Elephant gene' (default: False),\n",
      " 'lafricana_homolog_subtype': 'Last common ancestor with Elephant' (default: False),\n",
      " 'lafricana_homolog_wga_coverage': 'Elephant Whole-genome alignment coverage' (default: False),\n",
      " 'lbergylta_homolog_associated_gene_name': 'Ballan wrasse gene name' (default: False),\n",
      " 'lbergylta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lbergylta_homolog_chrom_end': 'Ballan wrasse chromosome/scaffold end (bp)' (default: False),\n",
      " 'lbergylta_homolog_chrom_start': 'Ballan wrasse chromosome/scaffold start (bp)' (default: False),\n",
      " 'lbergylta_homolog_chromosome': 'Ballan wrasse chromosome/scaffold name' (default: False),\n",
      " 'lbergylta_homolog_ensembl_gene': 'Ballan wrasse gene stable ID' (default: False),\n",
      " 'lbergylta_homolog_ensembl_peptide': 'Ballan wrasse protein or transcript stable ID' (default: False),\n",
      " 'lbergylta_homolog_goc_score': 'Ballan wrasse Gene-order conservation score' (default: False),\n",
      " 'lbergylta_homolog_orthology_confidence': 'Ballan wrasse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lbergylta_homolog_orthology_type': 'Ballan wrasse homology type' (default: False),\n",
      " 'lbergylta_homolog_perc_id': '%id. target Ballan wrasse gene identical to query gene' (default: False),\n",
      " 'lbergylta_homolog_perc_id_r1': '%id. query gene identical to target Ballan wrasse gene' (default: False),\n",
      " 'lbergylta_homolog_subtype': 'Last common ancestor with Ballan wrasse' (default: False),\n",
      " 'lbergylta_homolog_wga_coverage': 'Ballan wrasse Whole-genome alignment coverage' (default: False),\n",
      " 'lcalcarifer_homolog_associated_gene_name': 'Barramundi perch gene name' (default: False),\n",
      " 'lcalcarifer_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcalcarifer_homolog_chrom_end': 'Barramundi perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcalcarifer_homolog_chrom_start': 'Barramundi perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcalcarifer_homolog_chromosome': 'Barramundi perch chromosome/scaffold name' (default: False),\n",
      " 'lcalcarifer_homolog_ensembl_gene': 'Barramundi perch gene stable ID' (default: False),\n",
      " 'lcalcarifer_homolog_ensembl_peptide': 'Barramundi perch protein or transcript stable ID' (default: False),\n",
      " 'lcalcarifer_homolog_goc_score': 'Barramundi perch Gene-order conservation score' (default: False),\n",
      " 'lcalcarifer_homolog_orthology_confidence': 'Barramundi perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcalcarifer_homolog_orthology_type': 'Barramundi perch homology type' (default: False),\n",
      " 'lcalcarifer_homolog_perc_id': '%id. target Barramundi perch gene identical to query gene' (default: False),\n",
      " 'lcalcarifer_homolog_perc_id_r1': '%id. query gene identical to target Barramundi perch gene' (default: False),\n",
      " 'lcalcarifer_homolog_subtype': 'Last common ancestor with Barramundi perch' (default: False),\n",
      " 'lcalcarifer_homolog_wga_coverage': 'Barramundi perch Whole-genome alignment coverage' (default: False),\n",
      " 'lchalumnae_homolog_associated_gene_name': 'Coelacanth gene name' (default: False),\n",
      " 'lchalumnae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lchalumnae_homolog_chrom_end': 'Coelacanth chromosome/scaffold end (bp)' (default: False),\n",
      " 'lchalumnae_homolog_chrom_start': 'Coelacanth chromosome/scaffold start (bp)' (default: False),\n",
      " 'lchalumnae_homolog_chromosome': 'Coelacanth chromosome/scaffold name' (default: False),\n",
      " 'lchalumnae_homolog_ensembl_gene': 'Coelacanth gene stable ID' (default: False),\n",
      " 'lchalumnae_homolog_ensembl_peptide': 'Coelacanth protein or transcript stable ID' (default: False),\n",
      " 'lchalumnae_homolog_goc_score': 'Coelacanth Gene-order conservation score' (default: False),\n",
      " 'lchalumnae_homolog_orthology_confidence': 'Coelacanth orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lchalumnae_homolog_orthology_type': 'Coelacanth homology type' (default: False),\n",
      " 'lchalumnae_homolog_perc_id': '%id. target Coelacanth gene identical to query gene' (default: False),\n",
      " 'lchalumnae_homolog_perc_id_r1': '%id. query gene identical to target Coelacanth gene' (default: False),\n",
      " 'lchalumnae_homolog_subtype': 'Last common ancestor with Coelacanth' (default: False),\n",
      " 'lchalumnae_homolog_wga_coverage': 'Coelacanth Whole-genome alignment coverage' (default: False),\n",
      " 'lcrocea_homolog_associated_gene_name': 'Large yellow croaker gene name' (default: False),\n",
      " 'lcrocea_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lcrocea_homolog_chrom_end': 'Large yellow croaker chromosome/scaffold end (bp)' (default: False),\n",
      " 'lcrocea_homolog_chrom_start': 'Large yellow croaker chromosome/scaffold start (bp)' (default: False),\n",
      " 'lcrocea_homolog_chromosome': 'Large yellow croaker chromosome/scaffold name' (default: False),\n",
      " 'lcrocea_homolog_ensembl_gene': 'Large yellow croaker gene stable ID' (default: False),\n",
      " 'lcrocea_homolog_ensembl_peptide': 'Large yellow croaker protein or transcript stable ID' (default: False),\n",
      " 'lcrocea_homolog_goc_score': 'Large yellow croaker Gene-order conservation score' (default: False),\n",
      " 'lcrocea_homolog_orthology_confidence': 'Large yellow croaker orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lcrocea_homolog_orthology_type': 'Large yellow croaker homology type' (default: False),\n",
      " 'lcrocea_homolog_perc_id': '%id. target Large yellow croaker gene identical to query gene' (default: False),\n",
      " 'lcrocea_homolog_perc_id_r1': '%id. query gene identical to target Large yellow croaker gene' (default: False),\n",
      " 'lcrocea_homolog_subtype': 'Last common ancestor with Large yellow croaker' (default: False),\n",
      " 'lcrocea_homolog_wga_coverage': 'Large yellow croaker Whole-genome alignment coverage' (default: False),\n",
      " 'llaticaudata_homolog_associated_gene_name': 'Blue-ringed sea krait gene name' (default: False),\n",
      " 'llaticaudata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'llaticaudata_homolog_chrom_end': 'Blue-ringed sea krait chromosome/scaffold end (bp)' (default: False),\n",
      " 'llaticaudata_homolog_chrom_start': 'Blue-ringed sea krait chromosome/scaffold start (bp)' (default: False),\n",
      " 'llaticaudata_homolog_chromosome': 'Blue-ringed sea krait chromosome/scaffold name' (default: False),\n",
      " 'llaticaudata_homolog_ensembl_gene': 'Blue-ringed sea krait gene stable ID' (default: False),\n",
      " 'llaticaudata_homolog_ensembl_peptide': 'Blue-ringed sea krait protein or transcript stable ID' (default: False),\n",
      " 'llaticaudata_homolog_goc_score': 'Blue-ringed sea krait Gene-order conservation score' (default: False),\n",
      " 'llaticaudata_homolog_orthology_confidence': 'Blue-ringed sea krait orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'llaticaudata_homolog_orthology_type': 'Blue-ringed sea krait homology type' (default: False),\n",
      " 'llaticaudata_homolog_perc_id': '%id. target Blue-ringed sea krait gene identical to query gene' (default: False),\n",
      " 'llaticaudata_homolog_perc_id_r1': '%id. query gene identical to target Blue-ringed sea krait gene' (default: False),\n",
      " 'llaticaudata_homolog_subtype': 'Last common ancestor with Blue-ringed sea krait' (default: False),\n",
      " 'llaticaudata_homolog_wga_coverage': 'Blue-ringed sea krait Whole-genome alignment coverage' (default: False),\n",
      " 'lleishanense_homolog_associated_gene_name': 'Leishan spiny toad gene name' (default: False),\n",
      " 'lleishanense_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'lleishanense_homolog_chrom_end': 'Leishan spiny toad chromosome/scaffold end (bp)' (default: False),\n",
      " 'lleishanense_homolog_chrom_start': 'Leishan spiny toad chromosome/scaffold start (bp)' (default: False),\n",
      " 'lleishanense_homolog_chromosome': 'Leishan spiny toad chromosome/scaffold name' (default: False),\n",
      " 'lleishanense_homolog_ensembl_gene': 'Leishan spiny toad gene stable ID' (default: False),\n",
      " 'lleishanense_homolog_ensembl_peptide': 'Leishan spiny toad protein or transcript stable ID' (default: False),\n",
      " 'lleishanense_homolog_goc_score': 'Leishan spiny toad Gene-order conservation score' (default: False),\n",
      " 'lleishanense_homolog_orthology_confidence': 'Leishan spiny toad orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'lleishanense_homolog_orthology_type': 'Leishan spiny toad homology type' (default: False),\n",
      " 'lleishanense_homolog_perc_id': '%id. target Leishan spiny toad gene identical to query gene' (default: False),\n",
      " 'lleishanense_homolog_perc_id_r1': '%id. query gene identical to target Leishan spiny toad gene' (default: False),\n",
      " 'lleishanense_homolog_subtype': 'Last common ancestor with Leishan spiny toad' (default: False),\n",
      " 'lleishanense_homolog_wga_coverage': 'Leishan spiny toad Whole-genome alignment coverage' (default: False),\n",
      " 'loculatus_homolog_associated_gene_name': 'Spotted gar gene name' (default: False),\n",
      " 'loculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'loculatus_homolog_chrom_end': 'Spotted gar chromosome/scaffold end (bp)' (default: False),\n",
      " 'loculatus_homolog_chrom_start': 'Spotted gar chromosome/scaffold start (bp)' (default: False),\n",
      " 'loculatus_homolog_chromosome': 'Spotted gar chromosome/scaffold name' (default: False),\n",
      " 'loculatus_homolog_ensembl_gene': 'Spotted gar gene stable ID' (default: False),\n",
      " 'loculatus_homolog_ensembl_peptide': 'Spotted gar protein or transcript stable ID' (default: False),\n",
      " 'loculatus_homolog_goc_score': 'Spotted gar Gene-order conservation score' (default: False),\n",
      " 'loculatus_homolog_orthology_confidence': 'Spotted gar orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'loculatus_homolog_orthology_type': 'Spotted gar homology type' (default: False),\n",
      " 'loculatus_homolog_perc_id': '%id. target Spotted gar gene identical to query gene' (default: False),\n",
      " 'loculatus_homolog_perc_id_r1': '%id. query gene identical to target Spotted gar gene' (default: False),\n",
      " 'loculatus_homolog_subtype': 'Last common ancestor with Spotted gar' (default: False),\n",
      " 'loculatus_homolog_wga_coverage': 'Spotted gar Whole-genome alignment coverage' (default: False),\n",
      " 'mapweight': 'Mapweight' (default: False),\n",
      " 'marmatus_homolog_associated_gene_name': 'Zig-zag eel gene name' (default: False),\n",
      " 'marmatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'marmatus_homolog_chrom_end': 'Zig-zag eel chromosome/scaffold end (bp)' (default: False),\n",
      " 'marmatus_homolog_chrom_start': 'Zig-zag eel chromosome/scaffold start (bp)' (default: False),\n",
      " 'marmatus_homolog_chromosome': 'Zig-zag eel chromosome/scaffold name' (default: False),\n",
      " 'marmatus_homolog_ensembl_gene': 'Zig-zag eel gene stable ID' (default: False),\n",
      " 'marmatus_homolog_ensembl_peptide': 'Zig-zag eel protein or transcript stable ID' (default: False),\n",
      " 'marmatus_homolog_goc_score': 'Zig-zag eel Gene-order conservation score' (default: False),\n",
      " 'marmatus_homolog_orthology_confidence': 'Zig-zag eel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'marmatus_homolog_orthology_type': 'Zig-zag eel homology type' (default: False),\n",
      " 'marmatus_homolog_perc_id': '%id. target Zig-zag eel gene identical to query gene' (default: False),\n",
      " 'marmatus_homolog_perc_id_r1': '%id. query gene identical to target Zig-zag eel gene' (default: False),\n",
      " 'marmatus_homolog_subtype': 'Last common ancestor with Zig-zag eel' (default: False),\n",
      " 'mart_transcript_variation_som__dm_allele_string_2076': 'Consequence specific allele' (default: False),\n",
      " 'mart_transcript_variation_som__dm_distance_to_transcript_2076': 'Distance to transcript' (default: False),\n",
      " 'mauratus_homolog_associated_gene_name': 'Golden Hamster gene name' (default: False),\n",
      " 'mauratus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mauratus_homolog_chrom_end': 'Golden Hamster chromosome/scaffold end (bp)' (default: False),\n",
      " 'mauratus_homolog_chrom_start': 'Golden Hamster chromosome/scaffold start (bp)' (default: False),\n",
      " 'mauratus_homolog_chromosome': 'Golden Hamster chromosome/scaffold name' (default: False),\n",
      " 'mauratus_homolog_ensembl_gene': 'Golden Hamster gene stable ID' (default: False),\n",
      " 'mauratus_homolog_ensembl_peptide': 'Golden Hamster protein or transcript stable ID' (default: False),\n",
      " 'mauratus_homolog_goc_score': 'Golden Hamster Gene-order conservation score' (default: False),\n",
      " 'mauratus_homolog_orthology_confidence': 'Golden Hamster orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mauratus_homolog_orthology_type': 'Golden Hamster homology type' (default: False),\n",
      " 'mauratus_homolog_perc_id': '%id. target Golden Hamster gene identical to query gene' (default: False),\n",
      " 'mauratus_homolog_perc_id_r1': '%id. query gene identical to target Golden Hamster gene' (default: False),\n",
      " 'mauratus_homolog_subtype': 'Last common ancestor with Golden Hamster' (default: False),\n",
      " 'mauratus_homolog_wga_coverage': 'Golden Hamster Whole-genome alignment coverage' (default: False),\n",
      " 'mcaroli_homolog_associated_gene_name': 'Ryukyu mouse gene name' (default: False),\n",
      " 'mcaroli_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mcaroli_homolog_chrom_end': 'Ryukyu mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mcaroli_homolog_chrom_start': 'Ryukyu mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mcaroli_homolog_chromosome': 'Ryukyu mouse chromosome/scaffold name' (default: False),\n",
      " 'mcaroli_homolog_ensembl_gene': 'Ryukyu mouse gene stable ID' (default: False),\n",
      " 'mcaroli_homolog_ensembl_peptide': 'Ryukyu mouse protein or transcript stable ID' (default: False),\n",
      " 'mcaroli_homolog_goc_score': 'Ryukyu mouse Gene-order conservation score' (default: False),\n",
      " 'mcaroli_homolog_orthology_confidence': 'Ryukyu mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mcaroli_homolog_orthology_type': 'Ryukyu mouse homology type' (default: False),\n",
      " 'mcaroli_homolog_perc_id': '%id. target Ryukyu mouse gene identical to query gene' (default: False),\n",
      " 'mcaroli_homolog_perc_id_r1': '%id. query gene identical to target Ryukyu mouse gene' (default: False),\n",
      " 'mcaroli_homolog_subtype': 'Last common ancestor with Ryukyu mouse' (default: False),\n",
      " 'mcaroli_homolog_wga_coverage': 'Ryukyu mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mdomestica_homolog_associated_gene_name': 'Opossum gene name' (default: False),\n",
      " 'mdomestica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mdomestica_homolog_chrom_end': 'Opossum chromosome/scaffold end (bp)' (default: False),\n",
      " 'mdomestica_homolog_chrom_start': 'Opossum chromosome/scaffold start (bp)' (default: False),\n",
      " 'mdomestica_homolog_chromosome': 'Opossum chromosome/scaffold name' (default: False),\n",
      " 'mdomestica_homolog_ensembl_gene': 'Opossum gene stable ID' (default: False),\n",
      " 'mdomestica_homolog_ensembl_peptide': 'Opossum protein or transcript stable ID' (default: False),\n",
      " 'mdomestica_homolog_goc_score': 'Opossum Gene-order conservation score' (default: False),\n",
      " 'mdomestica_homolog_orthology_confidence': 'Opossum orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mdomestica_homolog_orthology_type': 'Opossum homology type' (default: False),\n",
      " 'mdomestica_homolog_perc_id': '%id. target Opossum gene identical to query gene' (default: False),\n",
      " 'mdomestica_homolog_perc_id_r1': '%id. query gene identical to target Opossum gene' (default: False),\n",
      " 'mdomestica_homolog_subtype': 'Last common ancestor with Opossum' (default: False),\n",
      " 'mdomestica_homolog_wga_coverage': 'Opossum Whole-genome alignment coverage' (default: False),\n",
      " 'merops': 'MEROPS - the Peptidase Database ID' (default: False),\n",
      " 'mfascicularis_homolog_associated_gene_name': 'Crab-eating macaque gene name' (default: False),\n",
      " 'mfascicularis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mfascicularis_homolog_chrom_end': 'Crab-eating macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mfascicularis_homolog_chrom_start': 'Crab-eating macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mfascicularis_homolog_chromosome': 'Crab-eating macaque chromosome/scaffold name' (default: False),\n",
      " 'mfascicularis_homolog_ensembl_gene': 'Crab-eating macaque gene stable ID' (default: False),\n",
      " 'mfascicularis_homolog_ensembl_peptide': 'Crab-eating macaque protein or transcript stable ID' (default: False),\n",
      " 'mfascicularis_homolog_goc_score': 'Crab-eating macaque Gene-order conservation score' (default: False),\n",
      " 'mfascicularis_homolog_orthology_confidence': 'Crab-eating macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mfascicularis_homolog_orthology_type': 'Crab-eating macaque homology type' (default: False),\n",
      " 'mfascicularis_homolog_perc_id': '%id. target Crab-eating macaque gene identical to query gene' (default: False),\n",
      " 'mfascicularis_homolog_perc_id_r1': '%id. query gene identical to target Crab-eating macaque gene' (default: False),\n",
      " 'mfascicularis_homolog_subtype': 'Last common ancestor with Crab-eating macaque' (default: False),\n",
      " 'mfascicularis_homolog_wga_coverage': 'Crab-eating macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mgallopavo_homolog_associated_gene_name': 'Turkey gene name' (default: False),\n",
      " 'mgallopavo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mgallopavo_homolog_chrom_end': 'Turkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'mgallopavo_homolog_chrom_start': 'Turkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'mgallopavo_homolog_chromosome': 'Turkey chromosome/scaffold name' (default: False),\n",
      " 'mgallopavo_homolog_ensembl_gene': 'Turkey gene stable ID' (default: False),\n",
      " 'mgallopavo_homolog_ensembl_peptide': 'Turkey protein or transcript stable ID' (default: False),\n",
      " 'mgallopavo_homolog_goc_score': 'Turkey Gene-order conservation score' (default: False),\n",
      " 'mgallopavo_homolog_orthology_confidence': 'Turkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mgallopavo_homolog_orthology_type': 'Turkey homology type' (default: False),\n",
      " 'mgallopavo_homolog_perc_id': '%id. target Turkey gene identical to query gene' (default: False),\n",
      " 'mgallopavo_homolog_perc_id_r1': '%id. query gene identical to target Turkey gene' (default: False),\n",
      " 'mgallopavo_homolog_subtype': 'Last common ancestor with Turkey' (default: False),\n",
      " 'mim_gene_accession': 'MIM gene accession' (default: False),\n",
      " 'mim_gene_description': 'MIM gene description' (default: False),\n",
      " 'mim_morbid_accession': 'MIM morbid accession' (default: False),\n",
      " 'mim_morbid_description': 'MIM morbid description' (default: False),\n",
      " 'minor_allele': 'Minor allele' (default: False),\n",
      " 'minor_allele_count': 'Minor allele count' (default: False),\n",
      " 'minor_allele_freq': 'Minor allele frequency' (default: False),\n",
      " 'mirbase_accession': 'miRBase accession' (default: False),\n",
      " 'mirbase_id': 'miRBase ID' (default: False),\n",
      " 'mirbase_trans_name': 'miRBase transcript name ID' (default: False),\n",
      " 'mleucophaeus_homolog_associated_gene_name': 'Drill gene name' (default: False),\n",
      " 'mleucophaeus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mleucophaeus_homolog_chrom_end': 'Drill chromosome/scaffold end (bp)' (default: False),\n",
      " 'mleucophaeus_homolog_chrom_start': 'Drill chromosome/scaffold start (bp)' (default: False),\n",
      " 'mleucophaeus_homolog_chromosome': 'Drill chromosome/scaffold name' (default: False),\n",
      " 'mleucophaeus_homolog_ensembl_gene': 'Drill gene stable ID' (default: False),\n",
      " 'mleucophaeus_homolog_ensembl_peptide': 'Drill protein or transcript stable ID' (default: False),\n",
      " 'mleucophaeus_homolog_goc_score': 'Drill Gene-order conservation score' (default: False),\n",
      " 'mleucophaeus_homolog_orthology_confidence': 'Drill orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mleucophaeus_homolog_orthology_type': 'Drill homology type' (default: False),\n",
      " 'mleucophaeus_homolog_perc_id': '%id. target Drill gene identical to query gene' (default: False),\n",
      " 'mleucophaeus_homolog_perc_id_r1': '%id. query gene identical to target Drill gene' (default: False),\n",
      " 'mleucophaeus_homolog_subtype': 'Last common ancestor with Drill' (default: False),\n",
      " 'mleucophaeus_homolog_wga_coverage': 'Drill Whole-genome alignment coverage' (default: False),\n",
      " 'mlucifugus_homolog_associated_gene_name': 'Microbat gene name' (default: False),\n",
      " 'mlucifugus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mlucifugus_homolog_chrom_end': 'Microbat chromosome/scaffold end (bp)' (default: False),\n",
      " 'mlucifugus_homolog_chrom_start': 'Microbat chromosome/scaffold start (bp)' (default: False),\n",
      " 'mlucifugus_homolog_chromosome': 'Microbat chromosome/scaffold name' (default: False),\n",
      " 'mlucifugus_homolog_ensembl_gene': 'Microbat gene stable ID' (default: False),\n",
      " 'mlucifugus_homolog_ensembl_peptide': 'Microbat protein or transcript stable ID' (default: False),\n",
      " 'mlucifugus_homolog_goc_score': 'Microbat Gene-order conservation score' (default: False),\n",
      " 'mlucifugus_homolog_orthology_confidence': 'Microbat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mlucifugus_homolog_orthology_type': 'Microbat homology type' (default: False),\n",
      " 'mlucifugus_homolog_perc_id': '%id. target Microbat gene identical to query gene' (default: False),\n",
      " 'mlucifugus_homolog_perc_id_r1': '%id. query gene identical to target Microbat gene' (default: False),\n",
      " 'mlucifugus_homolog_subtype': 'Last common ancestor with Microbat' (default: False),\n",
      " 'mlucifugus_homolog_wga_coverage': 'Microbat Whole-genome alignment coverage' (default: False),\n",
      " 'mmmarmota_homolog_associated_gene_name': 'Alpine marmot gene name' (default: False),\n",
      " 'mmmarmota_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmmarmota_homolog_chrom_end': 'Alpine marmot chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmmarmota_homolog_chrom_start': 'Alpine marmot chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmmarmota_homolog_chromosome': 'Alpine marmot chromosome/scaffold name' (default: False),\n",
      " 'mmmarmota_homolog_ensembl_gene': 'Alpine marmot gene stable ID' (default: False),\n",
      " 'mmmarmota_homolog_ensembl_peptide': 'Alpine marmot protein or transcript stable ID' (default: False),\n",
      " 'mmmarmota_homolog_goc_score': 'Alpine marmot Gene-order conservation score' (default: False),\n",
      " 'mmmarmota_homolog_orthology_confidence': 'Alpine marmot orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmmarmota_homolog_orthology_type': 'Alpine marmot homology type' (default: False),\n",
      " 'mmmarmota_homolog_perc_id': '%id. target Alpine marmot gene identical to query gene' (default: False),\n",
      " 'mmmarmota_homolog_perc_id_r1': '%id. query gene identical to target Alpine marmot gene' (default: False),\n",
      " 'mmmarmota_homolog_subtype': 'Last common ancestor with Alpine marmot' (default: False),\n",
      " 'mmmarmota_homolog_wga_coverage': 'Alpine marmot Whole-genome alignment coverage' (default: False),\n",
      " 'mmonoceros_homolog_associated_gene_name': 'Narwhal gene name' (default: False),\n",
      " 'mmonoceros_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmonoceros_homolog_chrom_end': 'Narwhal chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmonoceros_homolog_chrom_start': 'Narwhal chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmonoceros_homolog_chromosome': 'Narwhal chromosome/scaffold name' (default: False),\n",
      " 'mmonoceros_homolog_ensembl_gene': 'Narwhal gene stable ID' (default: False),\n",
      " 'mmonoceros_homolog_ensembl_peptide': 'Narwhal protein or transcript stable ID' (default: False),\n",
      " 'mmonoceros_homolog_goc_score': 'Narwhal Gene-order conservation score' (default: False),\n",
      " 'mmonoceros_homolog_orthology_confidence': 'Narwhal orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmonoceros_homolog_orthology_type': 'Narwhal homology type' (default: False),\n",
      " 'mmonoceros_homolog_perc_id': '%id. target Narwhal gene identical to query gene' (default: False),\n",
      " 'mmonoceros_homolog_perc_id_r1': '%id. query gene identical to target Narwhal gene' (default: False),\n",
      " 'mmonoceros_homolog_subtype': 'Last common ancestor with Narwhal' (default: False),\n",
      " 'mmonoceros_homolog_wga_coverage': 'Narwhal Whole-genome alignment coverage' (default: False),\n",
      " 'mmoschiferus_homolog_associated_gene_name': 'Siberian musk deer gene name' (default: False),\n",
      " 'mmoschiferus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmoschiferus_homolog_chrom_end': 'Siberian musk deer chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmoschiferus_homolog_chrom_start': 'Siberian musk deer chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmoschiferus_homolog_chromosome': 'Siberian musk deer chromosome/scaffold name' (default: False),\n",
      " 'mmoschiferus_homolog_ensembl_gene': 'Siberian musk deer gene stable ID' (default: False),\n",
      " 'mmoschiferus_homolog_ensembl_peptide': 'Siberian musk deer protein or transcript stable ID' (default: False),\n",
      " 'mmoschiferus_homolog_goc_score': 'Siberian musk deer Gene-order conservation score' (default: False),\n",
      " 'mmoschiferus_homolog_orthology_confidence': 'Siberian musk deer orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmoschiferus_homolog_orthology_type': 'Siberian musk deer homology type' (default: False),\n",
      " 'mmoschiferus_homolog_perc_id': '%id. target Siberian musk deer gene identical to query gene' (default: False),\n",
      " 'mmoschiferus_homolog_perc_id_r1': '%id. query gene identical to target Siberian musk deer gene' (default: False),\n",
      " 'mmoschiferus_homolog_subtype': 'Last common ancestor with Siberian musk deer' (default: False),\n",
      " 'mmoschiferus_homolog_wga_coverage': 'Siberian musk deer Whole-genome alignment coverage' (default: False),\n",
      " 'mmulatta_homolog_associated_gene_name': 'Macaque gene name' (default: False),\n",
      " 'mmulatta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmulatta_homolog_chrom_end': 'Macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmulatta_homolog_chrom_start': 'Macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmulatta_homolog_chromosome': 'Macaque chromosome/scaffold name' (default: False),\n",
      " 'mmulatta_homolog_ensembl_gene': 'Macaque gene stable ID' (default: False),\n",
      " 'mmulatta_homolog_ensembl_peptide': 'Macaque protein or transcript stable ID' (default: False),\n",
      " 'mmulatta_homolog_goc_score': 'Macaque Gene-order conservation score' (default: False),\n",
      " 'mmulatta_homolog_orthology_confidence': 'Macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmulatta_homolog_orthology_type': 'Macaque homology type' (default: False),\n",
      " 'mmulatta_homolog_perc_id': '%id. target Macaque gene identical to query gene' (default: False),\n",
      " 'mmulatta_homolog_perc_id_r1': '%id. query gene identical to target Macaque gene' (default: False),\n",
      " 'mmulatta_homolog_subtype': 'Last common ancestor with Macaque' (default: False),\n",
      " 'mmulatta_homolog_wga_coverage': 'Macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mmurdjan_homolog_associated_gene_name': 'Pinecone soldierfish gene name' (default: False),\n",
      " 'mmurdjan_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmurdjan_homolog_chrom_end': 'Pinecone soldierfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmurdjan_homolog_chrom_start': 'Pinecone soldierfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmurdjan_homolog_chromosome': 'Pinecone soldierfish chromosome/scaffold name' (default: False),\n",
      " 'mmurdjan_homolog_ensembl_gene': 'Pinecone soldierfish gene stable ID' (default: False),\n",
      " 'mmurdjan_homolog_ensembl_peptide': 'Pinecone soldierfish protein or transcript stable ID' (default: False),\n",
      " 'mmurdjan_homolog_goc_score': 'Pinecone soldierfish Gene-order conservation score' (default: False),\n",
      " 'mmurdjan_homolog_orthology_confidence': 'Pinecone soldierfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmurdjan_homolog_orthology_type': 'Pinecone soldierfish homology type' (default: False),\n",
      " 'mmurdjan_homolog_perc_id': '%id. target Pinecone soldierfish gene identical to query gene' (default: False),\n",
      " 'mmurdjan_homolog_perc_id_r1': '%id. query gene identical to target Pinecone soldierfish gene' (default: False),\n",
      " 'mmurdjan_homolog_subtype': 'Last common ancestor with Pinecone soldierfish' (default: False),\n",
      " 'mmurdjan_homolog_wga_coverage': 'Pinecone soldierfish Whole-genome alignment coverage' (default: False),\n",
      " 'mmurinus_homolog_associated_gene_name': 'Mouse Lemur gene name' (default: False),\n",
      " 'mmurinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmurinus_homolog_chrom_end': 'Mouse Lemur chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmurinus_homolog_chrom_start': 'Mouse Lemur chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmurinus_homolog_chromosome': 'Mouse Lemur chromosome/scaffold name' (default: False),\n",
      " 'mmurinus_homolog_ensembl_gene': 'Mouse Lemur gene stable ID' (default: False),\n",
      " 'mmurinus_homolog_ensembl_peptide': 'Mouse Lemur protein or transcript stable ID' (default: False),\n",
      " 'mmurinus_homolog_goc_score': 'Mouse Lemur Gene-order conservation score' (default: False),\n",
      " 'mmurinus_homolog_orthology_confidence': 'Mouse Lemur orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmurinus_homolog_orthology_type': 'Mouse Lemur homology type' (default: False),\n",
      " 'mmurinus_homolog_perc_id': '%id. target Mouse Lemur gene identical to query gene' (default: False),\n",
      " 'mmurinus_homolog_perc_id_r1': '%id. query gene identical to target Mouse Lemur gene' (default: False),\n",
      " 'mmurinus_homolog_subtype': 'Last common ancestor with Mouse Lemur' (default: False),\n",
      " 'mmurinus_homolog_wga_coverage': 'Mouse Lemur Whole-genome alignment coverage' (default: False),\n",
      " 'mmusculus_homolog_associated_gene_name': 'Mouse gene name' (default: False),\n",
      " 'mmusculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mmusculus_homolog_chrom_end': 'Mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mmusculus_homolog_chrom_start': 'Mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mmusculus_homolog_chromosome': 'Mouse chromosome/scaffold name' (default: False),\n",
      " 'mmusculus_homolog_ensembl_gene': 'Mouse gene stable ID' (default: False),\n",
      " 'mmusculus_homolog_ensembl_peptide': 'Mouse protein or transcript stable ID' (default: False),\n",
      " 'mmusculus_homolog_goc_score': 'Mouse Gene-order conservation score' (default: False),\n",
      " 'mmusculus_homolog_orthology_confidence': 'Mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mmusculus_homolog_orthology_type': 'Mouse homology type' (default: False),\n",
      " 'mmusculus_homolog_perc_id': '%id. target Mouse gene identical to query gene' (default: False),\n",
      " 'mmusculus_homolog_perc_id_r1': '%id. query gene identical to target Mouse gene' (default: False),\n",
      " 'mmusculus_homolog_subtype': 'Last common ancestor with Mouse' (default: False),\n",
      " 'mmusculus_homolog_wga_coverage': 'Mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mnemestrina_homolog_associated_gene_name': 'Pig-tailed macaque gene name' (default: False),\n",
      " 'mnemestrina_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mnemestrina_homolog_chrom_end': 'Pig-tailed macaque chromosome/scaffold end (bp)' (default: False),\n",
      " 'mnemestrina_homolog_chrom_start': 'Pig-tailed macaque chromosome/scaffold start (bp)' (default: False),\n",
      " 'mnemestrina_homolog_chromosome': 'Pig-tailed macaque chromosome/scaffold name' (default: False),\n",
      " 'mnemestrina_homolog_ensembl_gene': 'Pig-tailed macaque gene stable ID' (default: False),\n",
      " 'mnemestrina_homolog_ensembl_peptide': 'Pig-tailed macaque protein or transcript stable ID' (default: False),\n",
      " 'mnemestrina_homolog_goc_score': 'Pig-tailed macaque Gene-order conservation score' (default: False),\n",
      " 'mnemestrina_homolog_orthology_confidence': 'Pig-tailed macaque orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mnemestrina_homolog_orthology_type': 'Pig-tailed macaque homology type' (default: False),\n",
      " 'mnemestrina_homolog_perc_id': '%id. target Pig-tailed macaque gene identical to query gene' (default: False),\n",
      " 'mnemestrina_homolog_perc_id_r1': '%id. query gene identical to target Pig-tailed macaque gene' (default: False),\n",
      " 'mnemestrina_homolog_subtype': 'Last common ancestor with Pig-tailed macaque' (default: False),\n",
      " 'mnemestrina_homolog_wga_coverage': 'Pig-tailed macaque Whole-genome alignment coverage' (default: False),\n",
      " 'mobidblite': 'MobiDB lite' (default: False),\n",
      " 'mobidblite_end': 'MobiDB lite end' (default: False),\n",
      " 'mobidblite_start': 'MobiDB lite start' (default: False),\n",
      " 'mochrogaster_homolog_associated_gene_name': 'Prairie vole gene name' (default: False),\n",
      " 'mochrogaster_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mochrogaster_homolog_chrom_end': 'Prairie vole chromosome/scaffold end (bp)' (default: False),\n",
      " 'mochrogaster_homolog_chrom_start': 'Prairie vole chromosome/scaffold start (bp)' (default: False),\n",
      " 'mochrogaster_homolog_chromosome': 'Prairie vole chromosome/scaffold name' (default: False),\n",
      " 'mochrogaster_homolog_ensembl_gene': 'Prairie vole gene stable ID' (default: False),\n",
      " 'mochrogaster_homolog_ensembl_peptide': 'Prairie vole protein or transcript stable ID' (default: False),\n",
      " 'mochrogaster_homolog_goc_score': 'Prairie vole Gene-order conservation score' (default: False),\n",
      " 'mochrogaster_homolog_orthology_confidence': 'Prairie vole orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mochrogaster_homolog_orthology_type': 'Prairie vole homology type' (default: False),\n",
      " 'mochrogaster_homolog_perc_id': '%id. target Prairie vole gene identical to query gene' (default: False),\n",
      " 'mochrogaster_homolog_perc_id_r1': '%id. query gene identical to target Prairie vole gene' (default: False),\n",
      " 'mochrogaster_homolog_subtype': 'Last common ancestor with Prairie vole' (default: False),\n",
      " 'mochrogaster_homolog_wga_coverage': 'Prairie vole Whole-genome alignment coverage' (default: False),\n",
      " 'mpahari_homolog_associated_gene_name': 'Shrew mouse gene name' (default: False),\n",
      " 'mpahari_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mpahari_homolog_chrom_end': 'Shrew mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mpahari_homolog_chrom_start': 'Shrew mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mpahari_homolog_chromosome': 'Shrew mouse chromosome/scaffold name' (default: False),\n",
      " 'mpahari_homolog_ensembl_gene': 'Shrew mouse gene stable ID' (default: False),\n",
      " 'mpahari_homolog_ensembl_peptide': 'Shrew mouse protein or transcript stable ID' (default: False),\n",
      " 'mpahari_homolog_goc_score': 'Shrew mouse Gene-order conservation score' (default: False),\n",
      " 'mpahari_homolog_orthology_confidence': 'Shrew mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mpahari_homolog_orthology_type': 'Shrew mouse homology type' (default: False),\n",
      " 'mpahari_homolog_perc_id': '%id. target Shrew mouse gene identical to query gene' (default: False),\n",
      " 'mpahari_homolog_perc_id_r1': '%id. query gene identical to target Shrew mouse gene' (default: False),\n",
      " 'mpahari_homolog_subtype': 'Last common ancestor with Shrew mouse' (default: False),\n",
      " 'mpahari_homolog_wga_coverage': 'Shrew mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mpfuro_homolog_associated_gene_name': 'Ferret gene name' (default: False),\n",
      " 'mpfuro_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mpfuro_homolog_chrom_end': 'Ferret chromosome/scaffold end (bp)' (default: False),\n",
      " 'mpfuro_homolog_chrom_start': 'Ferret chromosome/scaffold start (bp)' (default: False),\n",
      " 'mpfuro_homolog_chromosome': 'Ferret chromosome/scaffold name' (default: False),\n",
      " 'mpfuro_homolog_ensembl_gene': 'Ferret gene stable ID' (default: False),\n",
      " 'mpfuro_homolog_ensembl_peptide': 'Ferret protein or transcript stable ID' (default: False),\n",
      " 'mpfuro_homolog_goc_score': 'Ferret Gene-order conservation score' (default: False),\n",
      " 'mpfuro_homolog_orthology_confidence': 'Ferret orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mpfuro_homolog_orthology_type': 'Ferret homology type' (default: False),\n",
      " 'mpfuro_homolog_perc_id': '%id. target Ferret gene identical to query gene' (default: False),\n",
      " 'mpfuro_homolog_perc_id_r1': '%id. query gene identical to target Ferret gene' (default: False),\n",
      " 'mpfuro_homolog_subtype': 'Last common ancestor with Ferret' (default: False),\n",
      " 'mpfuro_homolog_wga_coverage': 'Ferret Whole-genome alignment coverage' (default: False),\n",
      " 'mspicilegus_homolog_associated_gene_name': 'Steppe mouse gene name' (default: False),\n",
      " 'mspicilegus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mspicilegus_homolog_chrom_end': 'Steppe mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mspicilegus_homolog_chrom_start': 'Steppe mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mspicilegus_homolog_chromosome': 'Steppe mouse chromosome/scaffold name' (default: False),\n",
      " 'mspicilegus_homolog_ensembl_gene': 'Steppe mouse gene stable ID' (default: False),\n",
      " 'mspicilegus_homolog_ensembl_peptide': 'Steppe mouse protein or transcript stable ID' (default: False),\n",
      " 'mspicilegus_homolog_goc_score': 'Steppe mouse Gene-order conservation score' (default: False),\n",
      " 'mspicilegus_homolog_orthology_confidence': 'Steppe mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mspicilegus_homolog_orthology_type': 'Steppe mouse homology type' (default: False),\n",
      " 'mspicilegus_homolog_perc_id': '%id. target Steppe mouse gene identical to query gene' (default: False),\n",
      " 'mspicilegus_homolog_perc_id_r1': '%id. query gene identical to target Steppe mouse gene' (default: False),\n",
      " 'mspicilegus_homolog_subtype': 'Last common ancestor with Steppe mouse' (default: False),\n",
      " 'mspicilegus_homolog_wga_coverage': 'Steppe mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mspretus_homolog_associated_gene_name': 'Algerian mouse gene name' (default: False),\n",
      " 'mspretus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mspretus_homolog_chrom_end': 'Algerian mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'mspretus_homolog_chrom_start': 'Algerian mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'mspretus_homolog_chromosome': 'Algerian mouse chromosome/scaffold name' (default: False),\n",
      " 'mspretus_homolog_ensembl_gene': 'Algerian mouse gene stable ID' (default: False),\n",
      " 'mspretus_homolog_ensembl_peptide': 'Algerian mouse protein or transcript stable ID' (default: False),\n",
      " 'mspretus_homolog_goc_score': 'Algerian mouse Gene-order conservation score' (default: False),\n",
      " 'mspretus_homolog_orthology_confidence': 'Algerian mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mspretus_homolog_orthology_type': 'Algerian mouse homology type' (default: False),\n",
      " 'mspretus_homolog_perc_id': '%id. target Algerian mouse gene identical to query gene' (default: False),\n",
      " 'mspretus_homolog_perc_id_r1': '%id. query gene identical to target Algerian mouse gene' (default: False),\n",
      " 'mspretus_homolog_subtype': 'Last common ancestor with Algerian mouse' (default: False),\n",
      " 'mspretus_homolog_wga_coverage': 'Algerian mouse Whole-genome alignment coverage' (default: False),\n",
      " 'mzebra_homolog_associated_gene_name': 'Zebra mbuna gene name' (default: False),\n",
      " 'mzebra_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'mzebra_homolog_chrom_end': 'Zebra mbuna chromosome/scaffold end (bp)' (default: False),\n",
      " 'mzebra_homolog_chrom_start': 'Zebra mbuna chromosome/scaffold start (bp)' (default: False),\n",
      " 'mzebra_homolog_chromosome': 'Zebra mbuna chromosome/scaffold name' (default: False),\n",
      " 'mzebra_homolog_ensembl_gene': 'Zebra mbuna gene stable ID' (default: False),\n",
      " 'mzebra_homolog_ensembl_peptide': 'Zebra mbuna protein or transcript stable ID' (default: False),\n",
      " 'mzebra_homolog_goc_score': 'Zebra mbuna Gene-order conservation score' (default: False),\n",
      " 'mzebra_homolog_orthology_confidence': 'Zebra mbuna orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'mzebra_homolog_orthology_type': 'Zebra mbuna homology type' (default: False),\n",
      " 'mzebra_homolog_perc_id': '%id. target Zebra mbuna gene identical to query gene' (default: False),\n",
      " 'mzebra_homolog_perc_id_r1': '%id. query gene identical to target Zebra mbuna gene' (default: False),\n",
      " 'mzebra_homolog_subtype': 'Last common ancestor with Zebra mbuna' (default: False),\n",
      " 'mzebra_homolog_wga_coverage': 'Zebra mbuna Whole-genome alignment coverage' (default: False),\n",
      " 'name_1006': 'GO term name' (default: False),\n",
      " 'namespace_1003': 'GO domain' (default: False),\n",
      " 'nbrichardi_homolog_associated_gene_name': 'Lyretail cichlid gene name' (default: False),\n",
      " 'nbrichardi_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nbrichardi_homolog_chrom_end': 'Lyretail cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'nbrichardi_homolog_chrom_start': 'Lyretail cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'nbrichardi_homolog_chromosome': 'Lyretail cichlid chromosome/scaffold name' (default: False),\n",
      " 'nbrichardi_homolog_ensembl_gene': 'Lyretail cichlid gene stable ID' (default: False),\n",
      " 'nbrichardi_homolog_ensembl_peptide': 'Lyretail cichlid protein or transcript stable ID' (default: False),\n",
      " 'nbrichardi_homolog_goc_score': 'Lyretail cichlid Gene-order conservation score' (default: False),\n",
      " 'nbrichardi_homolog_orthology_confidence': 'Lyretail cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nbrichardi_homolog_orthology_type': 'Lyretail cichlid homology type' (default: False),\n",
      " 'nbrichardi_homolog_perc_id': '%id. target Lyretail cichlid gene identical to query gene' (default: False),\n",
      " 'nbrichardi_homolog_perc_id_r1': '%id. query gene identical to target Lyretail cichlid gene' (default: False),\n",
      " 'nbrichardi_homolog_subtype': 'Last common ancestor with Lyretail cichlid' (default: False),\n",
      " 'nbrichardi_homolog_wga_coverage': 'Lyretail cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'ncoils': 'Coiled-coils (Ncoils)' (default: False),\n",
      " 'ncoils_end': 'Coiled-coils (Ncoils) end' (default: False),\n",
      " 'ncoils_start': 'Coiled-coils (Ncoils) start' (default: False),\n",
      " 'neugenii_homolog_associated_gene_name': 'Wallaby gene name' (default: False),\n",
      " 'neugenii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'neugenii_homolog_chrom_end': 'Wallaby chromosome/scaffold end (bp)' (default: False),\n",
      " 'neugenii_homolog_chrom_start': 'Wallaby chromosome/scaffold start (bp)' (default: False),\n",
      " 'neugenii_homolog_chromosome': 'Wallaby chromosome/scaffold name' (default: False),\n",
      " 'neugenii_homolog_ensembl_gene': 'Wallaby gene stable ID' (default: False),\n",
      " 'neugenii_homolog_ensembl_peptide': 'Wallaby protein or transcript stable ID' (default: False),\n",
      " 'neugenii_homolog_goc_score': 'Wallaby Gene-order conservation score' (default: False),\n",
      " 'neugenii_homolog_orthology_confidence': 'Wallaby orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'neugenii_homolog_orthology_type': 'Wallaby homology type' (default: False),\n",
      " 'neugenii_homolog_perc_id': '%id. target Wallaby gene identical to query gene' (default: False),\n",
      " 'neugenii_homolog_perc_id_r1': '%id. query gene identical to target Wallaby gene' (default: False),\n",
      " 'neugenii_homolog_subtype': 'Last common ancestor with Wallaby' (default: False),\n",
      " 'neugenii_homolog_wga_coverage': 'Wallaby Whole-genome alignment coverage' (default: False),\n",
      " 'nfurzeri_homolog_associated_gene_name': 'Turquoise killifish gene name' (default: False),\n",
      " 'nfurzeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nfurzeri_homolog_chrom_end': 'Turquoise killifish chromosome/scaffold end (bp)' (default: False),\n",
      " 'nfurzeri_homolog_chrom_start': 'Turquoise killifish chromosome/scaffold start (bp)' (default: False),\n",
      " 'nfurzeri_homolog_chromosome': 'Turquoise killifish chromosome/scaffold name' (default: False),\n",
      " 'nfurzeri_homolog_ensembl_gene': 'Turquoise killifish gene stable ID' (default: False),\n",
      " 'nfurzeri_homolog_ensembl_peptide': 'Turquoise killifish protein or transcript stable ID' (default: False),\n",
      " 'nfurzeri_homolog_goc_score': 'Turquoise killifish Gene-order conservation score' (default: False),\n",
      " 'nfurzeri_homolog_orthology_confidence': 'Turquoise killifish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nfurzeri_homolog_orthology_type': 'Turquoise killifish homology type' (default: False),\n",
      " 'nfurzeri_homolog_perc_id': '%id. target Turquoise killifish gene identical to query gene' (default: False),\n",
      " 'nfurzeri_homolog_perc_id_r1': '%id. query gene identical to target Turquoise killifish gene' (default: False),\n",
      " 'nfurzeri_homolog_subtype': 'Last common ancestor with Turquoise killifish' (default: False),\n",
      " 'nfurzeri_homolog_wga_coverage': 'Turquoise killifish Whole-genome alignment coverage' (default: False),\n",
      " 'ngalili_homolog_associated_gene_name': 'Upper Galilee mountains blind mole rat gene name' (default: False),\n",
      " 'ngalili_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ngalili_homolog_chrom_end': 'Upper Galilee mountains blind mole rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'ngalili_homolog_chrom_start': 'Upper Galilee mountains blind mole rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'ngalili_homolog_chromosome': 'Upper Galilee mountains blind mole rat chromosome/scaffold name' (default: False),\n",
      " 'ngalili_homolog_ensembl_gene': 'Upper Galilee mountains blind mole rat gene stable ID' (default: False),\n",
      " 'ngalili_homolog_ensembl_peptide': 'Upper Galilee mountains blind mole rat protein or transcript stable ID' (default: False),\n",
      " 'ngalili_homolog_goc_score': 'Upper Galilee mountains blind mole rat Gene-order conservation score' (default: False),\n",
      " 'ngalili_homolog_orthology_confidence': 'Upper Galilee mountains blind mole rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ngalili_homolog_orthology_type': 'Upper Galilee mountains blind mole rat homology type' (default: False),\n",
      " 'ngalili_homolog_perc_id': '%id. target Upper Galilee mountains blind mole rat gene identical to query gene' (default: False),\n",
      " 'ngalili_homolog_perc_id_r1': '%id. query gene identical to target Upper Galilee mountains blind mole rat gene' (default: False),\n",
      " 'ngalili_homolog_subtype': 'Last common ancestor with Upper Galilee mountains blind mole rat' (default: False),\n",
      " 'ngalili_homolog_wga_coverage': 'Upper Galilee mountains blind mole rat Whole-genome alignment coverage' (default: False),\n",
      " 'nleucogenys_homolog_associated_gene_name': 'Gibbon gene name' (default: False),\n",
      " 'nleucogenys_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nleucogenys_homolog_chrom_end': 'Gibbon chromosome/scaffold end (bp)' (default: False),\n",
      " 'nleucogenys_homolog_chrom_start': 'Gibbon chromosome/scaffold start (bp)' (default: False),\n",
      " 'nleucogenys_homolog_chromosome': 'Gibbon chromosome/scaffold name' (default: False),\n",
      " 'nleucogenys_homolog_ensembl_gene': 'Gibbon gene stable ID' (default: False),\n",
      " 'nleucogenys_homolog_ensembl_peptide': 'Gibbon protein or transcript stable ID' (default: False),\n",
      " 'nleucogenys_homolog_goc_score': 'Gibbon Gene-order conservation score' (default: False),\n",
      " 'nleucogenys_homolog_orthology_confidence': 'Gibbon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nleucogenys_homolog_orthology_type': 'Gibbon homology type' (default: False),\n",
      " 'nleucogenys_homolog_perc_id': '%id. target Gibbon gene identical to query gene' (default: False),\n",
      " 'nleucogenys_homolog_perc_id_r1': '%id. query gene identical to target Gibbon gene' (default: False),\n",
      " 'nleucogenys_homolog_subtype': 'Last common ancestor with Gibbon' (default: False),\n",
      " 'nleucogenys_homolog_wga_coverage': 'Gibbon Whole-genome alignment coverage' (default: False),\n",
      " 'nnaja_homolog_associated_gene_name': 'Indian cobra gene name' (default: False),\n",
      " 'nnaja_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nnaja_homolog_chrom_end': 'Indian cobra chromosome/scaffold end (bp)' (default: False),\n",
      " 'nnaja_homolog_chrom_start': 'Indian cobra chromosome/scaffold start (bp)' (default: False),\n",
      " 'nnaja_homolog_chromosome': 'Indian cobra chromosome/scaffold name' (default: False),\n",
      " 'nnaja_homolog_ensembl_gene': 'Indian cobra gene stable ID' (default: False),\n",
      " 'nnaja_homolog_ensembl_peptide': 'Indian cobra protein or transcript stable ID' (default: False),\n",
      " 'nnaja_homolog_goc_score': 'Indian cobra Gene-order conservation score' (default: False),\n",
      " 'nnaja_homolog_orthology_confidence': 'Indian cobra orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nnaja_homolog_orthology_type': 'Indian cobra homology type' (default: False),\n",
      " 'nnaja_homolog_perc_id': '%id. target Indian cobra gene identical to query gene' (default: False),\n",
      " 'nnaja_homolog_perc_id_r1': '%id. query gene identical to target Indian cobra gene' (default: False),\n",
      " 'nnaja_homolog_subtype': 'Last common ancestor with Indian cobra' (default: False),\n",
      " 'nnaja_homolog_wga_coverage': 'Indian cobra Whole-genome alignment coverage' (default: False),\n",
      " 'nscutatus_homolog_associated_gene_name': 'Mainland tiger snake gene name' (default: False),\n",
      " 'nscutatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nscutatus_homolog_chrom_end': 'Mainland tiger snake chromosome/scaffold end (bp)' (default: False),\n",
      " 'nscutatus_homolog_chrom_start': 'Mainland tiger snake chromosome/scaffold start (bp)' (default: False),\n",
      " 'nscutatus_homolog_chromosome': 'Mainland tiger snake chromosome/scaffold name' (default: False),\n",
      " 'nscutatus_homolog_ensembl_gene': 'Mainland tiger snake gene stable ID' (default: False),\n",
      " 'nscutatus_homolog_ensembl_peptide': 'Mainland tiger snake protein or transcript stable ID' (default: False),\n",
      " 'nscutatus_homolog_goc_score': 'Mainland tiger snake Gene-order conservation score' (default: False),\n",
      " 'nscutatus_homolog_orthology_confidence': 'Mainland tiger snake orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nscutatus_homolog_orthology_type': 'Mainland tiger snake homology type' (default: False),\n",
      " 'nscutatus_homolog_perc_id': '%id. target Mainland tiger snake gene identical to query gene' (default: False),\n",
      " 'nscutatus_homolog_perc_id_r1': '%id. query gene identical to target Mainland tiger snake gene' (default: False),\n",
      " 'nscutatus_homolog_subtype': 'Last common ancestor with Mainland tiger snake' (default: False),\n",
      " 'nscutatus_homolog_wga_coverage': 'Mainland tiger snake Whole-genome alignment coverage' (default: False),\n",
      " 'nvison_homolog_associated_gene_name': 'American mink gene name' (default: False),\n",
      " 'nvison_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'nvison_homolog_chrom_end': 'American mink chromosome/scaffold end (bp)' (default: False),\n",
      " 'nvison_homolog_chrom_start': 'American mink chromosome/scaffold start (bp)' (default: False),\n",
      " 'nvison_homolog_chromosome': 'American mink chromosome/scaffold name' (default: False),\n",
      " 'nvison_homolog_ensembl_gene': 'American mink gene stable ID' (default: False),\n",
      " 'nvison_homolog_ensembl_peptide': 'American mink protein or transcript stable ID' (default: False),\n",
      " 'nvison_homolog_goc_score': 'American mink Gene-order conservation score' (default: False),\n",
      " 'nvison_homolog_orthology_confidence': 'American mink orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'nvison_homolog_orthology_type': 'American mink homology type' (default: False),\n",
      " 'nvison_homolog_perc_id': '%id. target American mink gene identical to query gene' (default: False),\n",
      " 'nvison_homolog_perc_id_r1': '%id. query gene identical to target American mink gene' (default: False),\n",
      " 'nvison_homolog_subtype': 'Last common ancestor with American mink' (default: False),\n",
      " 'nvison_homolog_wga_coverage': 'American mink Whole-genome alignment coverage' (default: False),\n",
      " 'oanatinus_homolog_associated_gene_name': 'Platypus gene name' (default: False),\n",
      " 'oanatinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oanatinus_homolog_chrom_end': 'Platypus chromosome/scaffold end (bp)' (default: False),\n",
      " 'oanatinus_homolog_chrom_start': 'Platypus chromosome/scaffold start (bp)' (default: False),\n",
      " 'oanatinus_homolog_chromosome': 'Platypus chromosome/scaffold name' (default: False),\n",
      " 'oanatinus_homolog_ensembl_gene': 'Platypus gene stable ID' (default: False),\n",
      " 'oanatinus_homolog_ensembl_peptide': 'Platypus protein or transcript stable ID' (default: False),\n",
      " 'oanatinus_homolog_goc_score': 'Platypus Gene-order conservation score' (default: False),\n",
      " 'oanatinus_homolog_orthology_confidence': 'Platypus orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oanatinus_homolog_orthology_type': 'Platypus homology type' (default: False),\n",
      " 'oanatinus_homolog_perc_id': '%id. target Platypus gene identical to query gene' (default: False),\n",
      " 'oanatinus_homolog_perc_id_r1': '%id. query gene identical to target Platypus gene' (default: False),\n",
      " 'oanatinus_homolog_subtype': 'Last common ancestor with Platypus' (default: False),\n",
      " 'oanatinus_homolog_wga_coverage': 'Platypus Whole-genome alignment coverage' (default: False),\n",
      " 'oarambouillet_homolog_associated_gene_name': 'Sheep gene name' (default: False),\n",
      " 'oarambouillet_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oarambouillet_homolog_chrom_end': 'Sheep chromosome/scaffold end (bp)' (default: False),\n",
      " 'oarambouillet_homolog_chrom_start': 'Sheep chromosome/scaffold start (bp)' (default: False),\n",
      " 'oarambouillet_homolog_chromosome': 'Sheep chromosome/scaffold name' (default: False),\n",
      " 'oarambouillet_homolog_ensembl_gene': 'Sheep gene stable ID' (default: False),\n",
      " 'oarambouillet_homolog_ensembl_peptide': 'Sheep protein or transcript stable ID' (default: False),\n",
      " 'oarambouillet_homolog_goc_score': 'Sheep Gene-order conservation score' (default: False),\n",
      " 'oarambouillet_homolog_orthology_confidence': 'Sheep orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oarambouillet_homolog_orthology_type': 'Sheep homology type' (default: False),\n",
      " 'oarambouillet_homolog_perc_id': '%id. target Sheep gene identical to query gene' (default: False),\n",
      " 'oarambouillet_homolog_perc_id_r1': '%id. query gene identical to target Sheep gene' (default: False),\n",
      " 'oarambouillet_homolog_subtype': 'Last common ancestor with Sheep' (default: False),\n",
      " 'oarambouillet_homolog_wga_coverage': 'Sheep Whole-genome alignment coverage' (default: False),\n",
      " 'ocuniculus_homolog_associated_gene_name': 'Rabbit gene name' (default: False),\n",
      " 'ocuniculus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ocuniculus_homolog_chrom_end': 'Rabbit chromosome/scaffold end (bp)' (default: False),\n",
      " 'ocuniculus_homolog_chrom_start': 'Rabbit chromosome/scaffold start (bp)' (default: False),\n",
      " 'ocuniculus_homolog_chromosome': 'Rabbit chromosome/scaffold name' (default: False),\n",
      " 'ocuniculus_homolog_ensembl_gene': 'Rabbit gene stable ID' (default: False),\n",
      " 'ocuniculus_homolog_ensembl_peptide': 'Rabbit protein or transcript stable ID' (default: False),\n",
      " 'ocuniculus_homolog_goc_score': 'Rabbit Gene-order conservation score' (default: False),\n",
      " 'ocuniculus_homolog_orthology_confidence': 'Rabbit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ocuniculus_homolog_orthology_type': 'Rabbit homology type' (default: False),\n",
      " 'ocuniculus_homolog_perc_id': '%id. target Rabbit gene identical to query gene' (default: False),\n",
      " 'ocuniculus_homolog_perc_id_r1': '%id. query gene identical to target Rabbit gene' (default: False),\n",
      " 'ocuniculus_homolog_subtype': 'Last common ancestor with Rabbit' (default: False),\n",
      " 'ocuniculus_homolog_wga_coverage': 'Rabbit Whole-genome alignment coverage' (default: False),\n",
      " 'odegus_homolog_associated_gene_name': 'Degu gene name' (default: False),\n",
      " 'odegus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'odegus_homolog_chrom_end': 'Degu chromosome/scaffold end (bp)' (default: False),\n",
      " 'odegus_homolog_chrom_start': 'Degu chromosome/scaffold start (bp)' (default: False),\n",
      " 'odegus_homolog_chromosome': 'Degu chromosome/scaffold name' (default: False),\n",
      " 'odegus_homolog_ensembl_gene': 'Degu gene stable ID' (default: False),\n",
      " 'odegus_homolog_ensembl_peptide': 'Degu protein or transcript stable ID' (default: False),\n",
      " 'odegus_homolog_goc_score': 'Degu Gene-order conservation score' (default: False),\n",
      " 'odegus_homolog_orthology_confidence': 'Degu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'odegus_homolog_orthology_type': 'Degu homology type' (default: False),\n",
      " 'odegus_homolog_perc_id': '%id. target Degu gene identical to query gene' (default: False),\n",
      " 'odegus_homolog_perc_id_r1': '%id. query gene identical to target Degu gene' (default: False),\n",
      " 'odegus_homolog_subtype': 'Last common ancestor with Degu' (default: False),\n",
      " 'odegus_homolog_wga_coverage': 'Degu Whole-genome alignment coverage' (default: False),\n",
      " 'ogarnettii_homolog_associated_gene_name': 'Bushbaby gene name' (default: False),\n",
      " 'ogarnettii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ogarnettii_homolog_chrom_end': 'Bushbaby chromosome/scaffold end (bp)' (default: False),\n",
      " 'ogarnettii_homolog_chrom_start': 'Bushbaby chromosome/scaffold start (bp)' (default: False),\n",
      " 'ogarnettii_homolog_chromosome': 'Bushbaby chromosome/scaffold name' (default: False),\n",
      " 'ogarnettii_homolog_ensembl_gene': 'Bushbaby gene stable ID' (default: False),\n",
      " 'ogarnettii_homolog_ensembl_peptide': 'Bushbaby protein or transcript stable ID' (default: False),\n",
      " 'ogarnettii_homolog_goc_score': 'Bushbaby Gene-order conservation score' (default: False),\n",
      " 'ogarnettii_homolog_orthology_confidence': 'Bushbaby orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ogarnettii_homolog_orthology_type': 'Bushbaby homology type' (default: False),\n",
      " 'ogarnettii_homolog_perc_id': '%id. target Bushbaby gene identical to query gene' (default: False),\n",
      " 'ogarnettii_homolog_perc_id_r1': '%id. query gene identical to target Bushbaby gene' (default: False),\n",
      " 'ogarnettii_homolog_subtype': 'Last common ancestor with Bushbaby' (default: False),\n",
      " 'ogarnettii_homolog_wga_coverage': 'Bushbaby Whole-genome alignment coverage' (default: False),\n",
      " 'ojavanicus_homolog_associated_gene_name': 'Javanese ricefish gene name' (default: False),\n",
      " 'ojavanicus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ojavanicus_homolog_chrom_end': 'Javanese ricefish chromosome/scaffold end (bp)' (default: False),\n",
      " 'ojavanicus_homolog_chrom_start': 'Javanese ricefish chromosome/scaffold start (bp)' (default: False),\n",
      " 'ojavanicus_homolog_chromosome': 'Javanese ricefish chromosome/scaffold name' (default: False),\n",
      " 'ojavanicus_homolog_ensembl_gene': 'Javanese ricefish gene stable ID' (default: False),\n",
      " 'ojavanicus_homolog_ensembl_peptide': 'Javanese ricefish protein or transcript stable ID' (default: False),\n",
      " 'ojavanicus_homolog_goc_score': 'Javanese ricefish Gene-order conservation score' (default: False),\n",
      " 'ojavanicus_homolog_orthology_confidence': 'Javanese ricefish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ojavanicus_homolog_orthology_type': 'Javanese ricefish homology type' (default: False),\n",
      " 'ojavanicus_homolog_perc_id': '%id. target Javanese ricefish gene identical to query gene' (default: False),\n",
      " 'ojavanicus_homolog_perc_id_r1': '%id. query gene identical to target Javanese ricefish gene' (default: False),\n",
      " 'ojavanicus_homolog_subtype': 'Last common ancestor with Javanese ricefish' (default: False),\n",
      " 'ojavanicus_homolog_wga_coverage': 'Javanese ricefish Whole-genome alignment coverage' (default: False),\n",
      " 'okisutch_homolog_associated_gene_name': 'Coho salmon gene name' (default: False),\n",
      " 'okisutch_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'okisutch_homolog_chrom_end': 'Coho salmon chromosome/scaffold end (bp)' (default: False),\n",
      " 'okisutch_homolog_chrom_start': 'Coho salmon chromosome/scaffold start (bp)' (default: False),\n",
      " 'okisutch_homolog_chromosome': 'Coho salmon chromosome/scaffold name' (default: False),\n",
      " 'okisutch_homolog_ensembl_gene': 'Coho salmon gene stable ID' (default: False),\n",
      " 'okisutch_homolog_ensembl_peptide': 'Coho salmon protein or transcript stable ID' (default: False),\n",
      " 'okisutch_homolog_goc_score': 'Coho salmon Gene-order conservation score' (default: False),\n",
      " 'okisutch_homolog_orthology_confidence': 'Coho salmon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'okisutch_homolog_orthology_type': 'Coho salmon homology type' (default: False),\n",
      " 'okisutch_homolog_perc_id': '%id. target Coho salmon gene identical to query gene' (default: False),\n",
      " 'okisutch_homolog_perc_id_r1': '%id. query gene identical to target Coho salmon gene' (default: False),\n",
      " 'okisutch_homolog_subtype': 'Last common ancestor with Coho salmon' (default: False),\n",
      " 'okisutch_homolog_wga_coverage': 'Coho salmon Whole-genome alignment coverage' (default: False),\n",
      " 'olatipes_homolog_associated_gene_name': 'Japanese medaka HdrR gene name' (default: False),\n",
      " 'olatipes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'olatipes_homolog_chrom_end': 'Japanese medaka HdrR chromosome/scaffold end (bp)' (default: False),\n",
      " 'olatipes_homolog_chrom_start': 'Japanese medaka HdrR chromosome/scaffold start (bp)' (default: False),\n",
      " 'olatipes_homolog_chromosome': 'Japanese medaka HdrR chromosome/scaffold name' (default: False),\n",
      " 'olatipes_homolog_ensembl_gene': 'Japanese medaka HdrR gene stable ID' (default: False),\n",
      " 'olatipes_homolog_ensembl_peptide': 'Japanese medaka HdrR protein or transcript stable ID' (default: False),\n",
      " 'olatipes_homolog_goc_score': 'Japanese medaka HdrR Gene-order conservation score' (default: False),\n",
      " 'olatipes_homolog_orthology_confidence': 'Japanese medaka HdrR orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'olatipes_homolog_orthology_type': 'Japanese medaka HdrR homology type' (default: False),\n",
      " 'olatipes_homolog_perc_id': '%id. target Japanese medaka HdrR gene identical to query gene' (default: False),\n",
      " 'olatipes_homolog_perc_id_r1': '%id. query gene identical to target Japanese medaka HdrR gene' (default: False),\n",
      " 'olatipes_homolog_subtype': 'Last common ancestor with Japanese medaka HdrR' (default: False),\n",
      " 'olatipes_homolog_wga_coverage': 'Japanese medaka HdrR Whole-genome alignment coverage' (default: False),\n",
      " 'omelastigma_homolog_associated_gene_name': 'Indian medaka gene name' (default: False),\n",
      " 'omelastigma_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'omelastigma_homolog_chrom_end': 'Indian medaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'omelastigma_homolog_chrom_start': 'Indian medaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'omelastigma_homolog_chromosome': 'Indian medaka chromosome/scaffold name' (default: False),\n",
      " 'omelastigma_homolog_ensembl_gene': 'Indian medaka gene stable ID' (default: False),\n",
      " 'omelastigma_homolog_ensembl_peptide': 'Indian medaka protein or transcript stable ID' (default: False),\n",
      " 'omelastigma_homolog_goc_score': 'Indian medaka Gene-order conservation score' (default: False),\n",
      " 'omelastigma_homolog_orthology_confidence': 'Indian medaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'omelastigma_homolog_orthology_type': 'Indian medaka homology type' (default: False),\n",
      " 'omelastigma_homolog_perc_id': '%id. target Indian medaka gene identical to query gene' (default: False),\n",
      " 'omelastigma_homolog_perc_id_r1': '%id. query gene identical to target Indian medaka gene' (default: False),\n",
      " 'omelastigma_homolog_subtype': 'Last common ancestor with Indian medaka' (default: False),\n",
      " 'omelastigma_homolog_wga_coverage': 'Indian medaka Whole-genome alignment coverage' (default: False),\n",
      " 'omykiss_homolog_associated_gene_name': 'Rainbow trout gene name' (default: False),\n",
      " 'omykiss_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'omykiss_homolog_chrom_end': 'Rainbow trout chromosome/scaffold end (bp)' (default: False),\n",
      " 'omykiss_homolog_chrom_start': 'Rainbow trout chromosome/scaffold start (bp)' (default: False),\n",
      " 'omykiss_homolog_chromosome': 'Rainbow trout chromosome/scaffold name' (default: False),\n",
      " 'omykiss_homolog_ensembl_gene': 'Rainbow trout gene stable ID' (default: False),\n",
      " 'omykiss_homolog_ensembl_peptide': 'Rainbow trout protein or transcript stable ID' (default: False),\n",
      " 'omykiss_homolog_goc_score': 'Rainbow trout Gene-order conservation score' (default: False),\n",
      " 'omykiss_homolog_orthology_confidence': 'Rainbow trout orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'omykiss_homolog_orthology_type': 'Rainbow trout homology type' (default: False),\n",
      " 'omykiss_homolog_perc_id': '%id. target Rainbow trout gene identical to query gene' (default: False),\n",
      " 'omykiss_homolog_perc_id_r1': '%id. query gene identical to target Rainbow trout gene' (default: False),\n",
      " 'omykiss_homolog_subtype': 'Last common ancestor with Rainbow trout' (default: False),\n",
      " 'oniloticus_homolog_associated_gene_name': 'Nile tilapia gene name' (default: False),\n",
      " 'oniloticus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oniloticus_homolog_chrom_end': 'Nile tilapia chromosome/scaffold end (bp)' (default: False),\n",
      " 'oniloticus_homolog_chrom_start': 'Nile tilapia chromosome/scaffold start (bp)' (default: False),\n",
      " 'oniloticus_homolog_chromosome': 'Nile tilapia chromosome/scaffold name' (default: False),\n",
      " 'oniloticus_homolog_ensembl_gene': 'Nile tilapia gene stable ID' (default: False),\n",
      " 'oniloticus_homolog_ensembl_peptide': 'Nile tilapia protein or transcript stable ID' (default: False),\n",
      " 'oniloticus_homolog_goc_score': 'Nile tilapia Gene-order conservation score' (default: False),\n",
      " 'oniloticus_homolog_orthology_confidence': 'Nile tilapia orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oniloticus_homolog_orthology_type': 'Nile tilapia homology type' (default: False),\n",
      " 'oniloticus_homolog_perc_id': '%id. target Nile tilapia gene identical to query gene' (default: False),\n",
      " 'oniloticus_homolog_perc_id_r1': '%id. query gene identical to target Nile tilapia gene' (default: False),\n",
      " 'oniloticus_homolog_subtype': 'Last common ancestor with Nile tilapia' (default: False),\n",
      " 'oniloticus_homolog_wga_coverage': 'Nile tilapia Whole-genome alignment coverage' (default: False),\n",
      " 'oprinceps_homolog_associated_gene_name': 'Pika gene name' (default: False),\n",
      " 'oprinceps_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'oprinceps_homolog_chrom_end': 'Pika chromosome/scaffold end (bp)' (default: False),\n",
      " 'oprinceps_homolog_chrom_start': 'Pika chromosome/scaffold start (bp)' (default: False),\n",
      " 'oprinceps_homolog_chromosome': 'Pika chromosome/scaffold name' (default: False),\n",
      " 'oprinceps_homolog_ensembl_gene': 'Pika gene stable ID' (default: False),\n",
      " 'oprinceps_homolog_ensembl_peptide': 'Pika protein or transcript stable ID' (default: False),\n",
      " 'oprinceps_homolog_goc_score': 'Pika Gene-order conservation score' (default: False),\n",
      " 'oprinceps_homolog_orthology_confidence': 'Pika orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'oprinceps_homolog_orthology_type': 'Pika homology type' (default: False),\n",
      " 'oprinceps_homolog_perc_id': '%id. target Pika gene identical to query gene' (default: False),\n",
      " 'oprinceps_homolog_perc_id_r1': '%id. query gene identical to target Pika gene' (default: False),\n",
      " 'oprinceps_homolog_subtype': 'Last common ancestor with Pika' (default: False),\n",
      " 'oprinceps_homolog_wga_coverage': 'Pika Whole-genome alignment coverage' (default: False),\n",
      " 'osinensis_homolog_associated_gene_name': 'Chinese medaka gene name' (default: False),\n",
      " 'osinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'osinensis_homolog_chrom_end': 'Chinese medaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'osinensis_homolog_chrom_start': 'Chinese medaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'osinensis_homolog_chromosome': 'Chinese medaka chromosome/scaffold name' (default: False),\n",
      " 'osinensis_homolog_ensembl_gene': 'Chinese medaka gene stable ID' (default: False),\n",
      " 'osinensis_homolog_ensembl_peptide': 'Chinese medaka protein or transcript stable ID' (default: False),\n",
      " 'osinensis_homolog_goc_score': 'Chinese medaka Gene-order conservation score' (default: False),\n",
      " 'osinensis_homolog_orthology_confidence': 'Chinese medaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'osinensis_homolog_orthology_type': 'Chinese medaka homology type' (default: False),\n",
      " 'osinensis_homolog_perc_id': '%id. target Chinese medaka gene identical to query gene' (default: False),\n",
      " 'osinensis_homolog_perc_id_r1': '%id. query gene identical to target Chinese medaka gene' (default: False),\n",
      " 'osinensis_homolog_subtype': 'Last common ancestor with Chinese medaka' (default: False),\n",
      " 'osinensis_homolog_wga_coverage': 'Chinese medaka Whole-genome alignment coverage' (default: False),\n",
      " 'otshawytscha_homolog_associated_gene_name': 'Chinook salmon gene name' (default: False),\n",
      " 'otshawytscha_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'otshawytscha_homolog_chrom_end': 'Chinook salmon chromosome/scaffold end (bp)' (default: False),\n",
      " 'otshawytscha_homolog_chrom_start': 'Chinook salmon chromosome/scaffold start (bp)' (default: False),\n",
      " 'otshawytscha_homolog_chromosome': 'Chinook salmon chromosome/scaffold name' (default: False),\n",
      " 'otshawytscha_homolog_ensembl_gene': 'Chinook salmon gene stable ID' (default: False),\n",
      " 'otshawytscha_homolog_ensembl_peptide': 'Chinook salmon protein or transcript stable ID' (default: False),\n",
      " 'otshawytscha_homolog_goc_score': 'Chinook salmon Gene-order conservation score' (default: False),\n",
      " 'otshawytscha_homolog_orthology_confidence': 'Chinook salmon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'otshawytscha_homolog_orthology_type': 'Chinook salmon homology type' (default: False),\n",
      " 'otshawytscha_homolog_perc_id': '%id. target Chinook salmon gene identical to query gene' (default: False),\n",
      " 'otshawytscha_homolog_perc_id_r1': '%id. query gene identical to target Chinook salmon gene' (default: False),\n",
      " 'otshawytscha_homolog_subtype': 'Last common ancestor with Chinook salmon' (default: False),\n",
      " 'otshawytscha_homolog_wga_coverage': 'Chinook salmon Whole-genome alignment coverage' (default: False),\n",
      " 'p_value': 'P value' (default: False),\n",
      " 'pabelii_homolog_associated_gene_name': 'Sumatran orangutan gene name' (default: False),\n",
      " 'pabelii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pabelii_homolog_chrom_end': 'Sumatran orangutan chromosome/scaffold end (bp)' (default: False),\n",
      " 'pabelii_homolog_chrom_start': 'Sumatran orangutan chromosome/scaffold start (bp)' (default: False),\n",
      " 'pabelii_homolog_chromosome': 'Sumatran orangutan chromosome/scaffold name' (default: False),\n",
      " 'pabelii_homolog_ensembl_gene': 'Sumatran orangutan gene stable ID' (default: False),\n",
      " 'pabelii_homolog_ensembl_peptide': 'Sumatran orangutan protein or transcript stable ID' (default: False),\n",
      " 'pabelii_homolog_goc_score': 'Sumatran orangutan Gene-order conservation score' (default: False),\n",
      " 'pabelii_homolog_orthology_confidence': 'Sumatran orangutan orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pabelii_homolog_orthology_type': 'Sumatran orangutan homology type' (default: False),\n",
      " 'pabelii_homolog_perc_id': '%id. target Sumatran orangutan gene identical to query gene' (default: False),\n",
      " 'pabelii_homolog_perc_id_r1': '%id. query gene identical to target Sumatran orangutan gene' (default: False),\n",
      " 'pabelii_homolog_subtype': 'Last common ancestor with Sumatran orangutan' (default: False),\n",
      " 'pabelii_homolog_wga_coverage': 'Sumatran orangutan Whole-genome alignment coverage' (default: False),\n",
      " 'panubis_homolog_associated_gene_name': 'Olive baboon gene name' (default: False),\n",
      " 'panubis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'panubis_homolog_chrom_end': 'Olive baboon chromosome/scaffold end (bp)' (default: False),\n",
      " 'panubis_homolog_chrom_start': 'Olive baboon chromosome/scaffold start (bp)' (default: False),\n",
      " 'panubis_homolog_chromosome': 'Olive baboon chromosome/scaffold name' (default: False),\n",
      " 'panubis_homolog_ensembl_gene': 'Olive baboon gene stable ID' (default: False),\n",
      " 'panubis_homolog_ensembl_peptide': 'Olive baboon protein or transcript stable ID' (default: False),\n",
      " 'panubis_homolog_goc_score': 'Olive baboon Gene-order conservation score' (default: False),\n",
      " 'panubis_homolog_orthology_confidence': 'Olive baboon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'panubis_homolog_orthology_type': 'Olive baboon homology type' (default: False),\n",
      " 'panubis_homolog_perc_id': '%id. target Olive baboon gene identical to query gene' (default: False),\n",
      " 'panubis_homolog_perc_id_r1': '%id. query gene identical to target Olive baboon gene' (default: False),\n",
      " 'panubis_homolog_subtype': 'Last common ancestor with Olive baboon' (default: False),\n",
      " 'panubis_homolog_wga_coverage': 'Olive baboon Whole-genome alignment coverage' (default: False),\n",
      " 'pcapensis_homolog_associated_gene_name': 'Hyrax gene name' (default: False),\n",
      " 'pcapensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcapensis_homolog_chrom_end': 'Hyrax chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcapensis_homolog_chrom_start': 'Hyrax chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcapensis_homolog_chromosome': 'Hyrax chromosome/scaffold name' (default: False),\n",
      " 'pcapensis_homolog_ensembl_gene': 'Hyrax gene stable ID' (default: False),\n",
      " 'pcapensis_homolog_ensembl_peptide': 'Hyrax protein or transcript stable ID' (default: False),\n",
      " 'pcapensis_homolog_goc_score': 'Hyrax Gene-order conservation score' (default: False),\n",
      " 'pcapensis_homolog_orthology_confidence': 'Hyrax orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcapensis_homolog_orthology_type': 'Hyrax homology type' (default: False),\n",
      " 'pcapensis_homolog_perc_id': '%id. target Hyrax gene identical to query gene' (default: False),\n",
      " 'pcapensis_homolog_perc_id_r1': '%id. query gene identical to target Hyrax gene' (default: False),\n",
      " 'pcapensis_homolog_subtype': 'Last common ancestor with Hyrax' (default: False),\n",
      " 'pcapensis_homolog_wga_coverage': 'Hyrax Whole-genome alignment coverage' (default: False),\n",
      " 'pcatodon_homolog_associated_gene_name': 'Sperm whale gene name' (default: False),\n",
      " 'pcatodon_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcatodon_homolog_chrom_end': 'Sperm whale chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcatodon_homolog_chrom_start': 'Sperm whale chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcatodon_homolog_chromosome': 'Sperm whale chromosome/scaffold name' (default: False),\n",
      " 'pcatodon_homolog_ensembl_gene': 'Sperm whale gene stable ID' (default: False),\n",
      " 'pcatodon_homolog_ensembl_peptide': 'Sperm whale protein or transcript stable ID' (default: False),\n",
      " 'pcatodon_homolog_goc_score': 'Sperm whale Gene-order conservation score' (default: False),\n",
      " 'pcatodon_homolog_orthology_confidence': 'Sperm whale orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcatodon_homolog_orthology_type': 'Sperm whale homology type' (default: False),\n",
      " 'pcatodon_homolog_perc_id': '%id. target Sperm whale gene identical to query gene' (default: False),\n",
      " 'pcatodon_homolog_perc_id_r1': '%id. query gene identical to target Sperm whale gene' (default: False),\n",
      " 'pcatodon_homolog_subtype': 'Last common ancestor with Sperm whale' (default: False),\n",
      " 'pcatodon_homolog_wga_coverage': 'Sperm whale Whole-genome alignment coverage' (default: False),\n",
      " 'pcinereus_homolog_associated_gene_name': 'Koala gene name' (default: False),\n",
      " 'pcinereus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcinereus_homolog_chrom_end': 'Koala chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcinereus_homolog_chrom_start': 'Koala chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcinereus_homolog_chromosome': 'Koala chromosome/scaffold name' (default: False),\n",
      " 'pcinereus_homolog_ensembl_gene': 'Koala gene stable ID' (default: False),\n",
      " 'pcinereus_homolog_ensembl_peptide': 'Koala protein or transcript stable ID' (default: False),\n",
      " 'pcinereus_homolog_goc_score': 'Koala Gene-order conservation score' (default: False),\n",
      " 'pcinereus_homolog_orthology_confidence': 'Koala orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcinereus_homolog_orthology_type': 'Koala homology type' (default: False),\n",
      " 'pcinereus_homolog_perc_id': '%id. target Koala gene identical to query gene' (default: False),\n",
      " 'pcinereus_homolog_perc_id_r1': '%id. query gene identical to target Koala gene' (default: False),\n",
      " 'pcinereus_homolog_subtype': 'Last common ancestor with Koala' (default: False),\n",
      " 'pcinereus_homolog_wga_coverage': 'Koala Whole-genome alignment coverage' (default: False),\n",
      " 'pcoquereli_homolog_associated_gene_name': 'Coquerel's sifaka gene name' (default: False),\n",
      " 'pcoquereli_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pcoquereli_homolog_chrom_end': 'Coquerel's sifaka chromosome/scaffold end (bp)' (default: False),\n",
      " 'pcoquereli_homolog_chrom_start': 'Coquerel's sifaka chromosome/scaffold start (bp)' (default: False),\n",
      " 'pcoquereli_homolog_chromosome': 'Coquerel's sifaka chromosome/scaffold name' (default: False),\n",
      " 'pcoquereli_homolog_ensembl_gene': 'Coquerel's sifaka gene stable ID' (default: False),\n",
      " 'pcoquereli_homolog_ensembl_peptide': 'Coquerel's sifaka protein or transcript stable ID' (default: False),\n",
      " 'pcoquereli_homolog_goc_score': 'Coquerel's sifaka Gene-order conservation score' (default: False),\n",
      " 'pcoquereli_homolog_orthology_confidence': 'Coquerel's sifaka orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pcoquereli_homolog_orthology_type': 'Coquerel's sifaka homology type' (default: False),\n",
      " 'pcoquereli_homolog_perc_id': '%id. target Coquerel's sifaka gene identical to query gene' (default: False),\n",
      " 'pcoquereli_homolog_perc_id_r1': '%id. query gene identical to target Coquerel's sifaka gene' (default: False),\n",
      " 'pcoquereli_homolog_subtype': 'Last common ancestor with Coquerel's sifaka' (default: False),\n",
      " 'pcoquereli_homolog_wga_coverage': 'Coquerel's sifaka Whole-genome alignment coverage' (default: False),\n",
      " 'pdb': 'PDB ID' (default: False),\n",
      " 'peptide': 'Peptide' (default: False),\n",
      " 'peptide_location': 'Protein location (aa)' (default: False),\n",
      " 'peptide_shift': 'Protein allele' (default: False),\n",
      " 'peptide_version': 'Version (protein)' (default: False),\n",
      " 'percentage_gene_gc_content': 'Gene % GC content' (default: False),\n",
      " 'pfam': 'Pfam ID' (default: False),\n",
      " 'pfam_end': 'Pfam end' (default: False),\n",
      " 'pfam_start': 'Pfam start' (default: False),\n",
      " 'pformosa_homolog_associated_gene_name': 'Amazon molly gene name' (default: False),\n",
      " 'pformosa_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pformosa_homolog_chrom_end': 'Amazon molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'pformosa_homolog_chrom_start': 'Amazon molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'pformosa_homolog_chromosome': 'Amazon molly chromosome/scaffold name' (default: False),\n",
      " 'pformosa_homolog_ensembl_gene': 'Amazon molly gene stable ID' (default: False),\n",
      " 'pformosa_homolog_ensembl_peptide': 'Amazon molly protein or transcript stable ID' (default: False),\n",
      " 'pformosa_homolog_goc_score': 'Amazon molly Gene-order conservation score' (default: False),\n",
      " 'pformosa_homolog_orthology_confidence': 'Amazon molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pformosa_homolog_orthology_type': 'Amazon molly homology type' (default: False),\n",
      " 'pformosa_homolog_perc_id': '%id. target Amazon molly gene identical to query gene' (default: False),\n",
      " 'pformosa_homolog_perc_id_r1': '%id. query gene identical to target Amazon molly gene' (default: False),\n",
      " 'pformosa_homolog_subtype': 'Last common ancestor with Amazon molly' (default: False),\n",
      " 'pformosa_homolog_wga_coverage': 'Amazon molly Whole-genome alignment coverage' (default: False),\n",
      " 'pfscan': 'PROSITE profiles ID' (default: False),\n",
      " 'pfscan_end': 'PROSITE profiles end' (default: False),\n",
      " 'pfscan_start': 'PROSITE profiles start' (default: False),\n",
      " 'phalanx_onearray': 'PHALANX OneArray probe' (default: False),\n",
      " 'phase': 'Start phase' (default: False),\n",
      " 'phenotype_description': 'Phenotype description' (default: False),\n",
      " 'pirsf': 'PIRSF ID' (default: False),\n",
      " 'pirsf_end': 'PIRSF end' (default: False),\n",
      " 'pirsf_start': 'PIRSF start' (default: False),\n",
      " 'pkingsleyae_homolog_associated_gene_name': 'Paramormyrops kingsleyae gene name' (default: False),\n",
      " 'pkingsleyae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pkingsleyae_homolog_chrom_end': 'Paramormyrops kingsleyae chromosome/scaffold end (bp)' (default: False),\n",
      " 'pkingsleyae_homolog_chrom_start': 'Paramormyrops kingsleyae chromosome/scaffold start (bp)' (default: False),\n",
      " 'pkingsleyae_homolog_chromosome': 'Paramormyrops kingsleyae chromosome/scaffold name' (default: False),\n",
      " 'pkingsleyae_homolog_ensembl_gene': 'Paramormyrops kingsleyae gene stable ID' (default: False),\n",
      " 'pkingsleyae_homolog_ensembl_peptide': 'Paramormyrops kingsleyae protein or transcript stable ID' (default: False),\n",
      " 'pkingsleyae_homolog_goc_score': 'Paramormyrops kingsleyae Gene-order conservation score' (default: False),\n",
      " 'pkingsleyae_homolog_orthology_confidence': 'Paramormyrops kingsleyae orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pkingsleyae_homolog_orthology_type': 'Paramormyrops kingsleyae homology type' (default: False),\n",
      " 'pkingsleyae_homolog_perc_id': '%id. target Paramormyrops kingsleyae gene identical to query gene' (default: False),\n",
      " 'pkingsleyae_homolog_perc_id_r1': '%id. query gene identical to target Paramormyrops kingsleyae gene' (default: False),\n",
      " 'pkingsleyae_homolog_subtype': 'Last common ancestor with Paramormyrops kingsleyae' (default: False),\n",
      " 'pkingsleyae_homolog_wga_coverage': 'Paramormyrops kingsleyae Whole-genome alignment coverage' (default: False),\n",
      " 'platipinna_homolog_associated_gene_name': 'Sailfin molly gene name' (default: False),\n",
      " 'platipinna_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'platipinna_homolog_chrom_end': 'Sailfin molly chromosome/scaffold end (bp)' (default: False),\n",
      " 'platipinna_homolog_chrom_start': 'Sailfin molly chromosome/scaffold start (bp)' (default: False),\n",
      " 'platipinna_homolog_chromosome': 'Sailfin molly chromosome/scaffold name' (default: False),\n",
      " 'platipinna_homolog_ensembl_gene': 'Sailfin molly gene stable ID' (default: False),\n",
      " 'platipinna_homolog_ensembl_peptide': 'Sailfin molly protein or transcript stable ID' (default: False),\n",
      " 'platipinna_homolog_goc_score': 'Sailfin molly Gene-order conservation score' (default: False),\n",
      " 'platipinna_homolog_orthology_confidence': 'Sailfin molly orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'platipinna_homolog_orthology_type': 'Sailfin molly homology type' (default: False),\n",
      " 'platipinna_homolog_perc_id': '%id. target Sailfin molly gene identical to query gene' (default: False),\n",
      " 'platipinna_homolog_perc_id_r1': '%id. query gene identical to target Sailfin molly gene' (default: False),\n",
      " 'platipinna_homolog_subtype': 'Last common ancestor with Sailfin molly' (default: False),\n",
      " 'platipinna_homolog_wga_coverage': 'Sailfin molly Whole-genome alignment coverage' (default: False),\n",
      " 'pleo_homolog_associated_gene_name': 'Lion gene name' (default: False),\n",
      " 'pleo_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pleo_homolog_chrom_end': 'Lion chromosome/scaffold end (bp)' (default: False),\n",
      " 'pleo_homolog_chrom_start': 'Lion chromosome/scaffold start (bp)' (default: False),\n",
      " 'pleo_homolog_chromosome': 'Lion chromosome/scaffold name' (default: False),\n",
      " 'pleo_homolog_ensembl_gene': 'Lion gene stable ID' (default: False),\n",
      " 'pleo_homolog_ensembl_peptide': 'Lion protein or transcript stable ID' (default: False),\n",
      " 'pleo_homolog_goc_score': 'Lion Gene-order conservation score' (default: False),\n",
      " 'pleo_homolog_orthology_confidence': 'Lion orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pleo_homolog_orthology_type': 'Lion homology type' (default: False),\n",
      " 'pleo_homolog_perc_id': '%id. target Lion gene identical to query gene' (default: False),\n",
      " 'pleo_homolog_perc_id_r1': '%id. query gene identical to target Lion gene' (default: False),\n",
      " 'pleo_homolog_subtype': 'Last common ancestor with Lion' (default: False),\n",
      " 'pleo_homolog_wga_coverage': 'Lion Whole-genome alignment coverage' (default: False),\n",
      " 'pmajor_homolog_associated_gene_name': 'Great Tit gene name' (default: False),\n",
      " 'pmajor_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmajor_homolog_chrom_end': 'Great Tit chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmajor_homolog_chrom_start': 'Great Tit chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmajor_homolog_chromosome': 'Great Tit chromosome/scaffold name' (default: False),\n",
      " 'pmajor_homolog_ensembl_gene': 'Great Tit gene stable ID' (default: False),\n",
      " 'pmajor_homolog_ensembl_peptide': 'Great Tit protein or transcript stable ID' (default: False),\n",
      " 'pmajor_homolog_goc_score': 'Great Tit Gene-order conservation score' (default: False),\n",
      " 'pmajor_homolog_orthology_confidence': 'Great Tit orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmajor_homolog_orthology_type': 'Great Tit homology type' (default: False),\n",
      " 'pmajor_homolog_perc_id': '%id. target Great Tit gene identical to query gene' (default: False),\n",
      " 'pmajor_homolog_perc_id_r1': '%id. query gene identical to target Great Tit gene' (default: False),\n",
      " 'pmajor_homolog_subtype': 'Last common ancestor with Great Tit' (default: False),\n",
      " 'pmajor_homolog_wga_coverage': 'Great Tit Whole-genome alignment coverage' (default: False),\n",
      " 'pmarinus_homolog_associated_gene_name': 'Lamprey gene name' (default: False),\n",
      " 'pmarinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmarinus_homolog_chrom_end': 'Lamprey chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmarinus_homolog_chrom_start': 'Lamprey chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmarinus_homolog_chromosome': 'Lamprey chromosome/scaffold name' (default: False),\n",
      " 'pmarinus_homolog_ensembl_gene': 'Lamprey gene stable ID' (default: False),\n",
      " 'pmarinus_homolog_ensembl_peptide': 'Lamprey protein or transcript stable ID' (default: False),\n",
      " 'pmarinus_homolog_orthology_confidence': 'Lamprey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmarinus_homolog_orthology_type': 'Lamprey homology type' (default: False),\n",
      " 'pmarinus_homolog_perc_id': '%id. target Lamprey gene identical to query gene' (default: False),\n",
      " 'pmarinus_homolog_perc_id_r1': '%id. query gene identical to target Lamprey gene' (default: False),\n",
      " 'pmarinus_homolog_subtype': 'Last common ancestor with Lamprey' (default: False),\n",
      " 'pmarinus_homolog_wga_coverage': 'Lamprey Whole-genome alignment coverage' (default: False),\n",
      " 'pmbairdii_homolog_associated_gene_name': 'Northern American deer mouse gene name' (default: False),\n",
      " 'pmbairdii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmbairdii_homolog_chrom_end': 'Northern American deer mouse chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmbairdii_homolog_chrom_start': 'Northern American deer mouse chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmbairdii_homolog_chromosome': 'Northern American deer mouse chromosome/scaffold name' (default: False),\n",
      " 'pmbairdii_homolog_ensembl_gene': 'Northern American deer mouse gene stable ID' (default: False),\n",
      " 'pmbairdii_homolog_ensembl_peptide': 'Northern American deer mouse protein or transcript stable ID' (default: False),\n",
      " 'pmbairdii_homolog_goc_score': 'Northern American deer mouse Gene-order conservation score' (default: False),\n",
      " 'pmbairdii_homolog_orthology_confidence': 'Northern American deer mouse orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmbairdii_homolog_orthology_type': 'Northern American deer mouse homology type' (default: False),\n",
      " 'pmbairdii_homolog_perc_id': '%id. target Northern American deer mouse gene identical to query gene' (default: False),\n",
      " 'pmbairdii_homolog_perc_id_r1': '%id. query gene identical to target Northern American deer mouse gene' (default: False),\n",
      " 'pmbairdii_homolog_subtype': 'Last common ancestor with Northern American deer mouse' (default: False),\n",
      " 'pmbairdii_homolog_wga_coverage': 'Northern American deer mouse Whole-genome alignment coverage' (default: False),\n",
      " 'pmuralis_homolog_associated_gene_name': 'Common wall lizard gene name' (default: False),\n",
      " 'pmuralis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pmuralis_homolog_chrom_end': 'Common wall lizard chromosome/scaffold end (bp)' (default: False),\n",
      " 'pmuralis_homolog_chrom_start': 'Common wall lizard chromosome/scaffold start (bp)' (default: False),\n",
      " 'pmuralis_homolog_chromosome': 'Common wall lizard chromosome/scaffold name' (default: False),\n",
      " 'pmuralis_homolog_ensembl_gene': 'Common wall lizard gene stable ID' (default: False),\n",
      " 'pmuralis_homolog_ensembl_peptide': 'Common wall lizard protein or transcript stable ID' (default: False),\n",
      " 'pmuralis_homolog_goc_score': 'Common wall lizard Gene-order conservation score' (default: False),\n",
      " 'pmuralis_homolog_orthology_confidence': 'Common wall lizard orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pmuralis_homolog_orthology_type': 'Common wall lizard homology type' (default: False),\n",
      " 'pmuralis_homolog_perc_id': '%id. target Common wall lizard gene identical to query gene' (default: False),\n",
      " 'pmuralis_homolog_perc_id_r1': '%id. query gene identical to target Common wall lizard gene' (default: False),\n",
      " 'pmuralis_homolog_subtype': 'Last common ancestor with Common wall lizard' (default: False),\n",
      " 'pmuralis_homolog_wga_coverage': 'Common wall lizard Whole-genome alignment coverage' (default: False),\n",
      " 'pnattereri_homolog_associated_gene_name': 'Red-bellied piranha gene name' (default: False),\n",
      " 'pnattereri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pnattereri_homolog_chrom_end': 'Red-bellied piranha chromosome/scaffold end (bp)' (default: False),\n",
      " 'pnattereri_homolog_chrom_start': 'Red-bellied piranha chromosome/scaffold start (bp)' (default: False),\n",
      " 'pnattereri_homolog_chromosome': 'Red-bellied piranha chromosome/scaffold name' (default: False),\n",
      " 'pnattereri_homolog_ensembl_gene': 'Red-bellied piranha gene stable ID' (default: False),\n",
      " 'pnattereri_homolog_ensembl_peptide': 'Red-bellied piranha protein or transcript stable ID' (default: False),\n",
      " 'pnattereri_homolog_goc_score': 'Red-bellied piranha Gene-order conservation score' (default: False),\n",
      " 'pnattereri_homolog_orthology_confidence': 'Red-bellied piranha orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pnattereri_homolog_orthology_type': 'Red-bellied piranha homology type' (default: False),\n",
      " 'pnattereri_homolog_perc_id': '%id. target Red-bellied piranha gene identical to query gene' (default: False),\n",
      " 'pnattereri_homolog_perc_id_r1': '%id. query gene identical to target Red-bellied piranha gene' (default: False),\n",
      " 'pnattereri_homolog_subtype': 'Last common ancestor with Red-bellied piranha' (default: False),\n",
      " 'pnattereri_homolog_wga_coverage': 'Red-bellied piranha Whole-genome alignment coverage' (default: False),\n",
      " 'pnyererei_homolog_associated_gene_name': 'Makobe Island cichlid gene name' (default: False),\n",
      " 'pnyererei_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pnyererei_homolog_chrom_end': 'Makobe Island cichlid chromosome/scaffold end (bp)' (default: False),\n",
      " 'pnyererei_homolog_chrom_start': 'Makobe Island cichlid chromosome/scaffold start (bp)' (default: False),\n",
      " 'pnyererei_homolog_chromosome': 'Makobe Island cichlid chromosome/scaffold name' (default: False),\n",
      " 'pnyererei_homolog_ensembl_gene': 'Makobe Island cichlid gene stable ID' (default: False),\n",
      " 'pnyererei_homolog_ensembl_peptide': 'Makobe Island cichlid protein or transcript stable ID' (default: False),\n",
      " 'pnyererei_homolog_goc_score': 'Makobe Island cichlid Gene-order conservation score' (default: False),\n",
      " 'pnyererei_homolog_orthology_confidence': 'Makobe Island cichlid orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pnyererei_homolog_orthology_type': 'Makobe Island cichlid homology type' (default: False),\n",
      " 'pnyererei_homolog_perc_id': '%id. target Makobe Island cichlid gene identical to query gene' (default: False),\n",
      " 'pnyererei_homolog_perc_id_r1': '%id. query gene identical to target Makobe Island cichlid gene' (default: False),\n",
      " 'pnyererei_homolog_subtype': 'Last common ancestor with Makobe Island cichlid' (default: False),\n",
      " 'pnyererei_homolog_wga_coverage': 'Makobe Island cichlid Whole-genome alignment coverage' (default: False),\n",
      " 'polyphen_prediction_2076': 'PolyPhen prediction' (default: False),\n",
      " 'polyphen_score_2076': 'PolyPhen score' (default: False),\n",
      " 'ppaniscus_homolog_associated_gene_name': 'Bonobo gene name' (default: False),\n",
      " 'ppaniscus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ppaniscus_homolog_chrom_end': 'Bonobo chromosome/scaffold end (bp)' (default: False),\n",
      " 'ppaniscus_homolog_chrom_start': 'Bonobo chromosome/scaffold start (bp)' (default: False),\n",
      " 'ppaniscus_homolog_chromosome': 'Bonobo chromosome/scaffold name' (default: False),\n",
      " 'ppaniscus_homolog_ensembl_gene': 'Bonobo gene stable ID' (default: False),\n",
      " 'ppaniscus_homolog_ensembl_peptide': 'Bonobo protein or transcript stable ID' (default: False),\n",
      " 'ppaniscus_homolog_goc_score': 'Bonobo Gene-order conservation score' (default: False),\n",
      " 'ppaniscus_homolog_orthology_confidence': 'Bonobo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ppaniscus_homolog_orthology_type': 'Bonobo homology type' (default: False),\n",
      " 'ppaniscus_homolog_perc_id': '%id. target Bonobo gene identical to query gene' (default: False),\n",
      " 'ppaniscus_homolog_perc_id_r1': '%id. query gene identical to target Bonobo gene' (default: False),\n",
      " 'ppaniscus_homolog_subtype': 'Last common ancestor with Bonobo' (default: False),\n",
      " 'ppaniscus_homolog_wga_coverage': 'Bonobo Whole-genome alignment coverage' (default: False),\n",
      " 'ppardus_homolog_associated_gene_name': 'Leopard gene name' (default: False),\n",
      " 'ppardus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ppardus_homolog_chrom_end': 'Leopard chromosome/scaffold end (bp)' (default: False),\n",
      " 'ppardus_homolog_chrom_start': 'Leopard chromosome/scaffold start (bp)' (default: False),\n",
      " 'ppardus_homolog_chromosome': 'Leopard chromosome/scaffold name' (default: False),\n",
      " 'ppardus_homolog_ensembl_gene': 'Leopard gene stable ID' (default: False),\n",
      " 'ppardus_homolog_ensembl_peptide': 'Leopard protein or transcript stable ID' (default: False),\n",
      " 'ppardus_homolog_goc_score': 'Leopard Gene-order conservation score' (default: False),\n",
      " 'ppardus_homolog_orthology_confidence': 'Leopard orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ppardus_homolog_orthology_type': 'Leopard homology type' (default: False),\n",
      " 'ppardus_homolog_perc_id': '%id. target Leopard gene identical to query gene' (default: False),\n",
      " 'ppardus_homolog_perc_id_r1': '%id. query gene identical to target Leopard gene' (default: False),\n",
      " 'ppardus_homolog_subtype': 'Last common ancestor with Leopard' (default: False),\n",
      " 'ppardus_homolog_wga_coverage': 'Leopard Whole-genome alignment coverage' (default: False),\n",
      " 'preticulata_homolog_associated_gene_name': 'Guppy gene name' (default: False),\n",
      " 'preticulata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'preticulata_homolog_chrom_end': 'Guppy chromosome/scaffold end (bp)' (default: False),\n",
      " 'preticulata_homolog_chrom_start': 'Guppy chromosome/scaffold start (bp)' (default: False),\n",
      " 'preticulata_homolog_chromosome': 'Guppy chromosome/scaffold name' (default: False),\n",
      " 'preticulata_homolog_ensembl_gene': 'Guppy gene stable ID' (default: False),\n",
      " 'preticulata_homolog_ensembl_peptide': 'Guppy protein or transcript stable ID' (default: False),\n",
      " 'preticulata_homolog_goc_score': 'Guppy Gene-order conservation score' (default: False),\n",
      " 'preticulata_homolog_orthology_confidence': 'Guppy orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'preticulata_homolog_orthology_type': 'Guppy homology type' (default: False),\n",
      " 'preticulata_homolog_perc_id': '%id. target Guppy gene identical to query gene' (default: False),\n",
      " 'preticulata_homolog_perc_id_r1': '%id. query gene identical to target Guppy gene' (default: False),\n",
      " 'preticulata_homolog_subtype': 'Last common ancestor with Guppy' (default: False),\n",
      " 'preticulata_homolog_wga_coverage': 'Guppy Whole-genome alignment coverage' (default: False),\n",
      " 'prints': 'Prints ID' (default: False),\n",
      " 'prints_end': 'Prints end' (default: False),\n",
      " 'prints_start': 'Prints start' (default: False),\n",
      " 'protein_id': 'INSDC protein ID' (default: False),\n",
      " 'psimus_homolog_associated_gene_name': 'Greater bamboo lemur gene name' (default: False),\n",
      " 'psimus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psimus_homolog_chrom_end': 'Greater bamboo lemur chromosome/scaffold end (bp)' (default: False),\n",
      " 'psimus_homolog_chrom_start': 'Greater bamboo lemur chromosome/scaffold start (bp)' (default: False),\n",
      " 'psimus_homolog_chromosome': 'Greater bamboo lemur chromosome/scaffold name' (default: False),\n",
      " 'psimus_homolog_ensembl_gene': 'Greater bamboo lemur gene stable ID' (default: False),\n",
      " 'psimus_homolog_ensembl_peptide': 'Greater bamboo lemur protein or transcript stable ID' (default: False),\n",
      " 'psimus_homolog_goc_score': 'Greater bamboo lemur Gene-order conservation score' (default: False),\n",
      " 'psimus_homolog_orthology_confidence': 'Greater bamboo lemur orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psimus_homolog_orthology_type': 'Greater bamboo lemur homology type' (default: False),\n",
      " 'psimus_homolog_perc_id': '%id. target Greater bamboo lemur gene identical to query gene' (default: False),\n",
      " 'psimus_homolog_perc_id_r1': '%id. query gene identical to target Greater bamboo lemur gene' (default: False),\n",
      " 'psimus_homolog_subtype': 'Last common ancestor with Greater bamboo lemur' (default: False),\n",
      " 'psimus_homolog_wga_coverage': 'Greater bamboo lemur Whole-genome alignment coverage' (default: False),\n",
      " 'psinensis_homolog_associated_gene_name': 'Chinese softshell turtle gene name' (default: False),\n",
      " 'psinensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psinensis_homolog_chrom_end': 'Chinese softshell turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'psinensis_homolog_chrom_start': 'Chinese softshell turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'psinensis_homolog_chromosome': 'Chinese softshell turtle chromosome/scaffold name' (default: False),\n",
      " 'psinensis_homolog_ensembl_gene': 'Chinese softshell turtle gene stable ID' (default: False),\n",
      " 'psinensis_homolog_ensembl_peptide': 'Chinese softshell turtle protein or transcript stable ID' (default: False),\n",
      " 'psinensis_homolog_goc_score': 'Chinese softshell turtle Gene-order conservation score' (default: False),\n",
      " 'psinensis_homolog_orthology_confidence': 'Chinese softshell turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psinensis_homolog_orthology_type': 'Chinese softshell turtle homology type' (default: False),\n",
      " 'psinensis_homolog_perc_id': '%id. target Chinese softshell turtle gene identical to query gene' (default: False),\n",
      " 'psinensis_homolog_perc_id_r1': '%id. query gene identical to target Chinese softshell turtle gene' (default: False),\n",
      " 'psinensis_homolog_subtype': 'Last common ancestor with Chinese softshell turtle' (default: False),\n",
      " 'psinensis_homolog_wga_coverage': 'Chinese softshell turtle Whole-genome alignment coverage' (default: False),\n",
      " 'psinus_homolog_associated_gene_name': 'Vaquita gene name' (default: False),\n",
      " 'psinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'psinus_homolog_chrom_end': 'Vaquita chromosome/scaffold end (bp)' (default: False),\n",
      " 'psinus_homolog_chrom_start': 'Vaquita chromosome/scaffold start (bp)' (default: False),\n",
      " 'psinus_homolog_chromosome': 'Vaquita chromosome/scaffold name' (default: False),\n",
      " 'psinus_homolog_ensembl_gene': 'Vaquita gene stable ID' (default: False),\n",
      " 'psinus_homolog_ensembl_peptide': 'Vaquita protein or transcript stable ID' (default: False),\n",
      " 'psinus_homolog_goc_score': 'Vaquita Gene-order conservation score' (default: False),\n",
      " 'psinus_homolog_orthology_confidence': 'Vaquita orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'psinus_homolog_orthology_type': 'Vaquita homology type' (default: False),\n",
      " 'psinus_homolog_perc_id': '%id. target Vaquita gene identical to query gene' (default: False),\n",
      " 'psinus_homolog_perc_id_r1': '%id. query gene identical to target Vaquita gene' (default: False),\n",
      " 'psinus_homolog_subtype': 'Last common ancestor with Vaquita' (default: False),\n",
      " 'psinus_homolog_wga_coverage': 'Vaquita Whole-genome alignment coverage' (default: False),\n",
      " 'ptaltaica_homolog_associated_gene_name': 'Tiger gene name' (default: False),\n",
      " 'ptaltaica_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptaltaica_homolog_chrom_end': 'Tiger chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptaltaica_homolog_chrom_start': 'Tiger chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptaltaica_homolog_chromosome': 'Tiger chromosome/scaffold name' (default: False),\n",
      " 'ptaltaica_homolog_ensembl_gene': 'Tiger gene stable ID' (default: False),\n",
      " 'ptaltaica_homolog_ensembl_peptide': 'Tiger protein or transcript stable ID' (default: False),\n",
      " 'ptaltaica_homolog_goc_score': 'Tiger Gene-order conservation score' (default: False),\n",
      " 'ptaltaica_homolog_orthology_confidence': 'Tiger orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptaltaica_homolog_orthology_type': 'Tiger homology type' (default: False),\n",
      " 'ptaltaica_homolog_perc_id': '%id. target Tiger gene identical to query gene' (default: False),\n",
      " 'ptaltaica_homolog_perc_id_r1': '%id. query gene identical to target Tiger gene' (default: False),\n",
      " 'ptaltaica_homolog_subtype': 'Last common ancestor with Tiger' (default: False),\n",
      " 'ptaltaica_homolog_wga_coverage': 'Tiger Whole-genome alignment coverage' (default: False),\n",
      " 'ptextilis_homolog_associated_gene_name': 'Eastern brown snake gene name' (default: False),\n",
      " 'ptextilis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptextilis_homolog_chrom_end': 'Eastern brown snake chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptextilis_homolog_chrom_start': 'Eastern brown snake chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptextilis_homolog_chromosome': 'Eastern brown snake chromosome/scaffold name' (default: False),\n",
      " 'ptextilis_homolog_ensembl_gene': 'Eastern brown snake gene stable ID' (default: False),\n",
      " 'ptextilis_homolog_ensembl_peptide': 'Eastern brown snake protein or transcript stable ID' (default: False),\n",
      " 'ptextilis_homolog_goc_score': 'Eastern brown snake Gene-order conservation score' (default: False),\n",
      " 'ptextilis_homolog_orthology_confidence': 'Eastern brown snake orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptextilis_homolog_orthology_type': 'Eastern brown snake homology type' (default: False),\n",
      " 'ptextilis_homolog_perc_id': '%id. target Eastern brown snake gene identical to query gene' (default: False),\n",
      " 'ptextilis_homolog_perc_id_r1': '%id. query gene identical to target Eastern brown snake gene' (default: False),\n",
      " 'ptextilis_homolog_subtype': 'Last common ancestor with Eastern brown snake' (default: False),\n",
      " 'ptextilis_homolog_wga_coverage': 'Eastern brown snake Whole-genome alignment coverage' (default: False),\n",
      " 'ptroglodytes_homolog_associated_gene_name': 'Chimpanzee gene name' (default: False),\n",
      " 'ptroglodytes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ptroglodytes_homolog_chrom_end': 'Chimpanzee chromosome/scaffold end (bp)' (default: False),\n",
      " 'ptroglodytes_homolog_chrom_start': 'Chimpanzee chromosome/scaffold start (bp)' (default: False),\n",
      " 'ptroglodytes_homolog_chromosome': 'Chimpanzee chromosome/scaffold name' (default: False),\n",
      " 'ptroglodytes_homolog_ensembl_gene': 'Chimpanzee gene stable ID' (default: False),\n",
      " 'ptroglodytes_homolog_ensembl_peptide': 'Chimpanzee protein or transcript stable ID' (default: False),\n",
      " 'ptroglodytes_homolog_goc_score': 'Chimpanzee Gene-order conservation score' (default: False),\n",
      " 'ptroglodytes_homolog_orthology_confidence': 'Chimpanzee orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ptroglodytes_homolog_orthology_type': 'Chimpanzee homology type' (default: False),\n",
      " 'ptroglodytes_homolog_perc_id': '%id. target Chimpanzee gene identical to query gene' (default: False),\n",
      " 'ptroglodytes_homolog_perc_id_r1': '%id. query gene identical to target Chimpanzee gene' (default: False),\n",
      " 'ptroglodytes_homolog_subtype': 'Last common ancestor with Chimpanzee' (default: False),\n",
      " 'ptroglodytes_homolog_wga_coverage': 'Chimpanzee Whole-genome alignment coverage' (default: False),\n",
      " 'pvampyrus_homolog_associated_gene_name': 'Megabat gene name' (default: False),\n",
      " 'pvampyrus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'pvampyrus_homolog_chrom_end': 'Megabat chromosome/scaffold end (bp)' (default: False),\n",
      " 'pvampyrus_homolog_chrom_start': 'Megabat chromosome/scaffold start (bp)' (default: False),\n",
      " 'pvampyrus_homolog_chromosome': 'Megabat chromosome/scaffold name' (default: False),\n",
      " 'pvampyrus_homolog_ensembl_gene': 'Megabat gene stable ID' (default: False),\n",
      " 'pvampyrus_homolog_ensembl_peptide': 'Megabat protein or transcript stable ID' (default: False),\n",
      " 'pvampyrus_homolog_goc_score': 'Megabat Gene-order conservation score' (default: False),\n",
      " 'pvampyrus_homolog_orthology_confidence': 'Megabat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'pvampyrus_homolog_orthology_type': 'Megabat homology type' (default: False),\n",
      " 'pvampyrus_homolog_perc_id': '%id. target Megabat gene identical to query gene' (default: False),\n",
      " 'pvampyrus_homolog_perc_id_r1': '%id. query gene identical to target Megabat gene' (default: False),\n",
      " 'pvampyrus_homolog_subtype': 'Last common ancestor with Megabat' (default: False),\n",
      " 'pvampyrus_homolog_wga_coverage': 'Megabat Whole-genome alignment coverage' (default: False),\n",
      " 'rank': 'Exon rank in transcript' (default: False),\n",
      " 'rbieti_homolog_associated_gene_name': 'Black snub-nosed monkey gene name' (default: False),\n",
      " 'rbieti_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rbieti_homolog_chrom_end': 'Black snub-nosed monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'rbieti_homolog_chrom_start': 'Black snub-nosed monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'rbieti_homolog_chromosome': 'Black snub-nosed monkey chromosome/scaffold name' (default: False),\n",
      " 'rbieti_homolog_ensembl_gene': 'Black snub-nosed monkey gene stable ID' (default: False),\n",
      " 'rbieti_homolog_ensembl_peptide': 'Black snub-nosed monkey protein or transcript stable ID' (default: False),\n",
      " 'rbieti_homolog_goc_score': 'Black snub-nosed monkey Gene-order conservation score' (default: False),\n",
      " 'rbieti_homolog_orthology_confidence': 'Black snub-nosed monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rbieti_homolog_orthology_type': 'Black snub-nosed monkey homology type' (default: False),\n",
      " 'rbieti_homolog_perc_id': '%id. target Black snub-nosed monkey gene identical to query gene' (default: False),\n",
      " 'rbieti_homolog_perc_id_r1': '%id. query gene identical to target Black snub-nosed monkey gene' (default: False),\n",
      " 'rbieti_homolog_subtype': 'Last common ancestor with Black snub-nosed monkey' (default: False),\n",
      " 'rbieti_homolog_wga_coverage': 'Black snub-nosed monkey Whole-genome alignment coverage' (default: False),\n",
      " 'reactome': 'Reactome ID' (default: False),\n",
      " 'reactome_gene': 'Reactome gene ID' (default: False),\n",
      " 'reactome_transcript': 'Reactome transcript ID' (default: False),\n",
      " 'refseq_mrna': 'RefSeq mRNA ID' (default: False),\n",
      " 'refseq_mrna_predicted': 'RefSeq mRNA predicted ID' (default: False),\n",
      " 'refseq_ncrna': 'RefSeq ncRNA ID' (default: False),\n",
      " 'refseq_ncrna_predicted': 'RefSeq ncRNA predicted ID' (default: False),\n",
      " 'refseq_peptide': 'RefSeq peptide ID' (default: False),\n",
      " 'refseq_peptide_predicted': 'RefSeq peptide predicted ID' (default: False),\n",
      " 'rfam': 'RFAM ID' (default: False),\n",
      " 'rfam_trans_name': 'RFAM transcript name ID' (default: False),\n",
      " 'rferrumequinum_homolog_associated_gene_name': 'Greater horseshoe bat gene name' (default: False),\n",
      " 'rferrumequinum_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rferrumequinum_homolog_chrom_end': 'Greater horseshoe bat chromosome/scaffold end (bp)' (default: False),\n",
      " 'rferrumequinum_homolog_chrom_start': 'Greater horseshoe bat chromosome/scaffold start (bp)' (default: False),\n",
      " 'rferrumequinum_homolog_chromosome': 'Greater horseshoe bat chromosome/scaffold name' (default: False),\n",
      " 'rferrumequinum_homolog_ensembl_gene': 'Greater horseshoe bat gene stable ID' (default: False),\n",
      " 'rferrumequinum_homolog_ensembl_peptide': 'Greater horseshoe bat protein or transcript stable ID' (default: False),\n",
      " 'rferrumequinum_homolog_goc_score': 'Greater horseshoe bat Gene-order conservation score' (default: False),\n",
      " 'rferrumequinum_homolog_orthology_confidence': 'Greater horseshoe bat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rferrumequinum_homolog_orthology_type': 'Greater horseshoe bat homology type' (default: False),\n",
      " 'rferrumequinum_homolog_perc_id': '%id. target Greater horseshoe bat gene identical to query gene' (default: False),\n",
      " 'rferrumequinum_homolog_perc_id_r1': '%id. query gene identical to target Greater horseshoe bat gene' (default: False),\n",
      " 'rferrumequinum_homolog_subtype': 'Last common ancestor with Greater horseshoe bat' (default: False),\n",
      " 'rferrumequinum_homolog_wga_coverage': 'Greater horseshoe bat Whole-genome alignment coverage' (default: False),\n",
      " 'rnacentral': 'RNAcentral ID' (default: False),\n",
      " 'rnorvegicus_homolog_associated_gene_name': 'Rat gene name' (default: False),\n",
      " 'rnorvegicus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rnorvegicus_homolog_chrom_end': 'Rat chromosome/scaffold end (bp)' (default: False),\n",
      " 'rnorvegicus_homolog_chrom_start': 'Rat chromosome/scaffold start (bp)' (default: False),\n",
      " 'rnorvegicus_homolog_chromosome': 'Rat chromosome/scaffold name' (default: False),\n",
      " 'rnorvegicus_homolog_ensembl_gene': 'Rat gene stable ID' (default: False),\n",
      " 'rnorvegicus_homolog_ensembl_peptide': 'Rat protein or transcript stable ID' (default: False),\n",
      " 'rnorvegicus_homolog_goc_score': 'Rat Gene-order conservation score' (default: False),\n",
      " 'rnorvegicus_homolog_orthology_confidence': 'Rat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rnorvegicus_homolog_orthology_type': 'Rat homology type' (default: False),\n",
      " 'rnorvegicus_homolog_perc_id': '%id. target Rat gene identical to query gene' (default: False),\n",
      " 'rnorvegicus_homolog_perc_id_r1': '%id. query gene identical to target Rat gene' (default: False),\n",
      " 'rnorvegicus_homolog_subtype': 'Last common ancestor with Rat' (default: False),\n",
      " 'rnorvegicus_homolog_wga_coverage': 'Rat Whole-genome alignment coverage' (default: False),\n",
      " 'rroxellana_homolog_associated_gene_name': 'Golden snub-nosed monkey gene name' (default: False),\n",
      " 'rroxellana_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'rroxellana_homolog_chrom_end': 'Golden snub-nosed monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'rroxellana_homolog_chrom_start': 'Golden snub-nosed monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'rroxellana_homolog_chromosome': 'Golden snub-nosed monkey chromosome/scaffold name' (default: False),\n",
      " 'rroxellana_homolog_ensembl_gene': 'Golden snub-nosed monkey gene stable ID' (default: False),\n",
      " 'rroxellana_homolog_ensembl_peptide': 'Golden snub-nosed monkey protein or transcript stable ID' (default: False),\n",
      " 'rroxellana_homolog_goc_score': 'Golden snub-nosed monkey Gene-order conservation score' (default: False),\n",
      " 'rroxellana_homolog_orthology_confidence': 'Golden snub-nosed monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'rroxellana_homolog_orthology_type': 'Golden snub-nosed monkey homology type' (default: False),\n",
      " 'rroxellana_homolog_perc_id': '%id. target Golden snub-nosed monkey gene identical to query gene' (default: False),\n",
      " 'rroxellana_homolog_perc_id_r1': '%id. query gene identical to target Golden snub-nosed monkey gene' (default: False),\n",
      " 'rroxellana_homolog_subtype': 'Last common ancestor with Golden snub-nosed monkey' (default: False),\n",
      " 'rroxellana_homolog_wga_coverage': 'Golden snub-nosed monkey Whole-genome alignment coverage' (default: False),\n",
      " 'saraneus_homolog_associated_gene_name': 'Shrew gene name' (default: False),\n",
      " 'saraneus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'saraneus_homolog_chrom_end': 'Shrew chromosome/scaffold end (bp)' (default: False),\n",
      " 'saraneus_homolog_chrom_start': 'Shrew chromosome/scaffold start (bp)' (default: False),\n",
      " 'saraneus_homolog_chromosome': 'Shrew chromosome/scaffold name' (default: False),\n",
      " 'saraneus_homolog_ensembl_gene': 'Shrew gene stable ID' (default: False),\n",
      " 'saraneus_homolog_ensembl_peptide': 'Shrew protein or transcript stable ID' (default: False),\n",
      " 'saraneus_homolog_goc_score': 'Shrew Gene-order conservation score' (default: False),\n",
      " 'saraneus_homolog_orthology_confidence': 'Shrew orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'saraneus_homolog_orthology_type': 'Shrew homology type' (default: False),\n",
      " 'saraneus_homolog_perc_id': '%id. target Shrew gene identical to query gene' (default: False),\n",
      " 'saraneus_homolog_perc_id_r1': '%id. query gene identical to target Shrew gene' (default: False),\n",
      " 'saraneus_homolog_subtype': 'Last common ancestor with Shrew' (default: False),\n",
      " 'saraneus_homolog_wga_coverage': 'Shrew Whole-genome alignment coverage' (default: False),\n",
      " 'saurata_homolog_associated_gene_name': 'Gilthead seabream gene name' (default: False),\n",
      " 'saurata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'saurata_homolog_chrom_end': 'Gilthead seabream chromosome/scaffold end (bp)' (default: False),\n",
      " 'saurata_homolog_chrom_start': 'Gilthead seabream chromosome/scaffold start (bp)' (default: False),\n",
      " 'saurata_homolog_chromosome': 'Gilthead seabream chromosome/scaffold name' (default: False),\n",
      " 'saurata_homolog_ensembl_gene': 'Gilthead seabream gene stable ID' (default: False),\n",
      " 'saurata_homolog_ensembl_peptide': 'Gilthead seabream protein or transcript stable ID' (default: False),\n",
      " 'saurata_homolog_goc_score': 'Gilthead seabream Gene-order conservation score' (default: False),\n",
      " 'saurata_homolog_orthology_confidence': 'Gilthead seabream orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'saurata_homolog_orthology_type': 'Gilthead seabream homology type' (default: False),\n",
      " 'saurata_homolog_perc_id': '%id. target Gilthead seabream gene identical to query gene' (default: False),\n",
      " 'saurata_homolog_perc_id_r1': '%id. query gene identical to target Gilthead seabream gene' (default: False),\n",
      " 'saurata_homolog_subtype': 'Last common ancestor with Gilthead seabream' (default: False),\n",
      " 'saurata_homolog_wga_coverage': 'Gilthead seabream Whole-genome alignment coverage' (default: False),\n",
      " 'sbboliviensis_homolog_associated_gene_name': 'Bolivian squirrel monkey gene name' (default: False),\n",
      " 'sbboliviensis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sbboliviensis_homolog_chrom_end': 'Bolivian squirrel monkey chromosome/scaffold end (bp)' (default: False),\n",
      " 'sbboliviensis_homolog_chrom_start': 'Bolivian squirrel monkey chromosome/scaffold start (bp)' (default: False),\n",
      " 'sbboliviensis_homolog_chromosome': 'Bolivian squirrel monkey chromosome/scaffold name' (default: False),\n",
      " 'sbboliviensis_homolog_ensembl_gene': 'Bolivian squirrel monkey gene stable ID' (default: False),\n",
      " 'sbboliviensis_homolog_ensembl_peptide': 'Bolivian squirrel monkey protein or transcript stable ID' (default: False),\n",
      " 'sbboliviensis_homolog_goc_score': 'Bolivian squirrel monkey Gene-order conservation score' (default: False),\n",
      " 'sbboliviensis_homolog_orthology_confidence': 'Bolivian squirrel monkey orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sbboliviensis_homolog_orthology_type': 'Bolivian squirrel monkey homology type' (default: False),\n",
      " 'sbboliviensis_homolog_perc_id': '%id. target Bolivian squirrel monkey gene identical to query gene' (default: False),\n",
      " 'sbboliviensis_homolog_perc_id_r1': '%id. query gene identical to target Bolivian squirrel monkey gene' (default: False),\n",
      " 'sbboliviensis_homolog_subtype': 'Last common ancestor with Bolivian squirrel monkey' (default: False),\n",
      " 'sbboliviensis_homolog_wga_coverage': 'Bolivian squirrel monkey Whole-genome alignment coverage' (default: False),\n",
      " 'scanaria_homolog_associated_gene_name': 'Common canary gene name' (default: False),\n",
      " 'scanaria_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scanaria_homolog_chrom_end': 'Common canary chromosome/scaffold end (bp)' (default: False),\n",
      " 'scanaria_homolog_chrom_start': 'Common canary chromosome/scaffold start (bp)' (default: False),\n",
      " 'scanaria_homolog_chromosome': 'Common canary chromosome/scaffold name' (default: False),\n",
      " 'scanaria_homolog_ensembl_gene': 'Common canary gene stable ID' (default: False),\n",
      " 'scanaria_homolog_ensembl_peptide': 'Common canary protein or transcript stable ID' (default: False),\n",
      " 'scanaria_homolog_goc_score': 'Common canary Gene-order conservation score' (default: False),\n",
      " 'scanaria_homolog_orthology_confidence': 'Common canary orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scanaria_homolog_orthology_type': 'Common canary homology type' (default: False),\n",
      " 'scanaria_homolog_perc_id': '%id. target Common canary gene identical to query gene' (default: False),\n",
      " 'scanaria_homolog_perc_id_r1': '%id. query gene identical to target Common canary gene' (default: False),\n",
      " 'scanaria_homolog_subtype': 'Last common ancestor with Common canary' (default: False),\n",
      " 'scanaria_homolog_wga_coverage': 'Common canary Whole-genome alignment coverage' (default: False),\n",
      " 'scanprosite': 'PROSITE patterns ID' (default: False),\n",
      " 'scanprosite_end': 'PROSITE patterns end' (default: False),\n",
      " 'scanprosite_start': 'PROSITE patterns start' (default: False),\n",
      " 'scaustralis_homolog_associated_gene_name': 'African ostrich gene name' (default: False),\n",
      " 'scaustralis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scaustralis_homolog_chrom_end': 'African ostrich chromosome/scaffold end (bp)' (default: False),\n",
      " 'scaustralis_homolog_chrom_start': 'African ostrich chromosome/scaffold start (bp)' (default: False),\n",
      " 'scaustralis_homolog_chromosome': 'African ostrich chromosome/scaffold name' (default: False),\n",
      " 'scaustralis_homolog_ensembl_gene': 'African ostrich gene stable ID' (default: False),\n",
      " 'scaustralis_homolog_ensembl_peptide': 'African ostrich protein or transcript stable ID' (default: False),\n",
      " 'scaustralis_homolog_goc_score': 'African ostrich Gene-order conservation score' (default: False),\n",
      " 'scaustralis_homolog_orthology_confidence': 'African ostrich orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scaustralis_homolog_orthology_type': 'African ostrich homology type' (default: False),\n",
      " 'scaustralis_homolog_perc_id': '%id. target African ostrich gene identical to query gene' (default: False),\n",
      " 'scaustralis_homolog_perc_id_r1': '%id. query gene identical to target African ostrich gene' (default: False),\n",
      " 'scaustralis_homolog_subtype': 'Last common ancestor with African ostrich' (default: False),\n",
      " 'scaustralis_homolog_wga_coverage': 'African ostrich Whole-genome alignment coverage' (default: False),\n",
      " 'scerevisiae_homolog_associated_gene_name': 'Saccharomyces cerevisiae gene name' (default: False),\n",
      " 'scerevisiae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'scerevisiae_homolog_chrom_end': 'Saccharomyces cerevisiae chromosome/scaffold end (bp)' (default: False),\n",
      " 'scerevisiae_homolog_chrom_start': 'Saccharomyces cerevisiae chromosome/scaffold start (bp)' (default: False),\n",
      " 'scerevisiae_homolog_chromosome': 'Saccharomyces cerevisiae chromosome/scaffold name' (default: False),\n",
      " 'scerevisiae_homolog_ensembl_gene': 'Saccharomyces cerevisiae gene stable ID' (default: False),\n",
      " 'scerevisiae_homolog_ensembl_peptide': 'Saccharomyces cerevisiae protein or transcript stable ID' (default: False),\n",
      " 'scerevisiae_homolog_orthology_confidence': 'Saccharomyces cerevisiae orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'scerevisiae_homolog_orthology_type': 'Saccharomyces cerevisiae homology type' (default: False),\n",
      " 'scerevisiae_homolog_perc_id': '%id. target Saccharomyces cerevisiae gene identical to query gene' (default: False),\n",
      " 'scerevisiae_homolog_perc_id_r1': '%id. query gene identical to target Saccharomyces cerevisiae gene' (default: False),\n",
      " 'scerevisiae_homolog_subtype': 'Last common ancestor with Saccharomyces cerevisiae' (default: False),\n",
      " 'sdumerili_homolog_associated_gene_name': 'Greater amberjack gene name' (default: False),\n",
      " 'sdumerili_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sdumerili_homolog_chrom_end': 'Greater amberjack chromosome/scaffold end (bp)' (default: False),\n",
      " 'sdumerili_homolog_chrom_start': 'Greater amberjack chromosome/scaffold start (bp)' (default: False),\n",
      " 'sdumerili_homolog_chromosome': 'Greater amberjack chromosome/scaffold name' (default: False),\n",
      " 'sdumerili_homolog_ensembl_gene': 'Greater amberjack gene stable ID' (default: False),\n",
      " 'sdumerili_homolog_ensembl_peptide': 'Greater amberjack protein or transcript stable ID' (default: False),\n",
      " 'sdumerili_homolog_goc_score': 'Greater amberjack Gene-order conservation score' (default: False),\n",
      " 'sdumerili_homolog_orthology_confidence': 'Greater amberjack orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sdumerili_homolog_orthology_type': 'Greater amberjack homology type' (default: False),\n",
      " 'sdumerili_homolog_perc_id': '%id. target Greater amberjack gene identical to query gene' (default: False),\n",
      " 'sdumerili_homolog_perc_id_r1': '%id. query gene identical to target Greater amberjack gene' (default: False),\n",
      " 'sdumerili_homolog_subtype': 'Last common ancestor with Greater amberjack' (default: False),\n",
      " 'sdumerili_homolog_wga_coverage': 'Greater amberjack Whole-genome alignment coverage' (default: False),\n",
      " 'seg': 'Low complexity (Seg)' (default: False),\n",
      " 'seg_end': 'Low complexity (Seg) end' (default: False),\n",
      " 'seg_start': 'Low complexity (Seg) start' (default: False),\n",
      " 'sfld': 'SFLD ID' (default: False),\n",
      " 'sfld_end': 'SFLD end' (default: False),\n",
      " 'sfld_start': 'SFLD start' (default: False),\n",
      " 'sformosus_homolog_associated_gene_name': 'Asian bonytongue gene name' (default: False),\n",
      " 'sformosus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sformosus_homolog_chrom_end': 'Asian bonytongue chromosome/scaffold end (bp)' (default: False),\n",
      " 'sformosus_homolog_chrom_start': 'Asian bonytongue chromosome/scaffold start (bp)' (default: False),\n",
      " 'sformosus_homolog_chromosome': 'Asian bonytongue chromosome/scaffold name' (default: False),\n",
      " 'sformosus_homolog_ensembl_gene': 'Asian bonytongue gene stable ID' (default: False),\n",
      " 'sformosus_homolog_ensembl_peptide': 'Asian bonytongue protein or transcript stable ID' (default: False),\n",
      " 'sformosus_homolog_goc_score': 'Asian bonytongue Gene-order conservation score' (default: False),\n",
      " 'sformosus_homolog_orthology_confidence': 'Asian bonytongue orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sformosus_homolog_orthology_type': 'Asian bonytongue homology type' (default: False),\n",
      " 'sformosus_homolog_perc_id': '%id. target Asian bonytongue gene identical to query gene' (default: False),\n",
      " 'sformosus_homolog_perc_id_r1': '%id. query gene identical to target Asian bonytongue gene' (default: False),\n",
      " 'sformosus_homolog_subtype': 'Last common ancestor with Asian bonytongue' (default: False),\n",
      " 'sformosus_homolog_wga_coverage': 'Asian bonytongue Whole-genome alignment coverage' (default: False),\n",
      " 'sgrahami_homolog_associated_gene_name': 'Golden-line barbel gene name' (default: False),\n",
      " 'sgrahami_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sgrahami_homolog_chrom_end': 'Golden-line barbel chromosome/scaffold end (bp)' (default: False),\n",
      " 'sgrahami_homolog_chrom_start': 'Golden-line barbel chromosome/scaffold start (bp)' (default: False),\n",
      " 'sgrahami_homolog_chromosome': 'Golden-line barbel chromosome/scaffold name' (default: False),\n",
      " 'sgrahami_homolog_ensembl_gene': 'Golden-line barbel gene stable ID' (default: False),\n",
      " 'sgrahami_homolog_ensembl_peptide': 'Golden-line barbel protein or transcript stable ID' (default: False),\n",
      " 'sgrahami_homolog_goc_score': 'Golden-line barbel Gene-order conservation score' (default: False),\n",
      " 'sgrahami_homolog_orthology_confidence': 'Golden-line barbel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sgrahami_homolog_orthology_type': 'Golden-line barbel homology type' (default: False),\n",
      " 'sgrahami_homolog_perc_id': '%id. target Golden-line barbel gene identical to query gene' (default: False),\n",
      " 'sgrahami_homolog_perc_id_r1': '%id. query gene identical to target Golden-line barbel gene' (default: False),\n",
      " 'sgrahami_homolog_subtype': 'Last common ancestor with Golden-line barbel' (default: False),\n",
      " 'sgrahami_homolog_wga_coverage': 'Golden-line barbel Whole-genome alignment coverage' (default: False),\n",
      " 'shabroptila_homolog_associated_gene_name': 'Kakapo gene name' (default: False),\n",
      " 'shabroptila_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'shabroptila_homolog_chrom_end': 'Kakapo chromosome/scaffold end (bp)' (default: False),\n",
      " 'shabroptila_homolog_chrom_start': 'Kakapo chromosome/scaffold start (bp)' (default: False),\n",
      " 'shabroptila_homolog_chromosome': 'Kakapo chromosome/scaffold name' (default: False),\n",
      " 'shabroptila_homolog_ensembl_gene': 'Kakapo gene stable ID' (default: False),\n",
      " 'shabroptila_homolog_ensembl_peptide': 'Kakapo protein or transcript stable ID' (default: False),\n",
      " 'shabroptila_homolog_goc_score': 'Kakapo Gene-order conservation score' (default: False),\n",
      " 'shabroptila_homolog_orthology_confidence': 'Kakapo orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'shabroptila_homolog_orthology_type': 'Kakapo homology type' (default: False),\n",
      " 'shabroptila_homolog_perc_id': '%id. target Kakapo gene identical to query gene' (default: False),\n",
      " 'shabroptila_homolog_perc_id_r1': '%id. query gene identical to target Kakapo gene' (default: False),\n",
      " 'shabroptila_homolog_subtype': 'Last common ancestor with Kakapo' (default: False),\n",
      " 'shabroptila_homolog_wga_coverage': 'Kakapo Whole-genome alignment coverage' (default: False),\n",
      " 'sharrisii_homolog_associated_gene_name': 'Tasmanian devil gene name' (default: False),\n",
      " 'sharrisii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sharrisii_homolog_chrom_end': 'Tasmanian devil chromosome/scaffold end (bp)' (default: False),\n",
      " 'sharrisii_homolog_chrom_start': 'Tasmanian devil chromosome/scaffold start (bp)' (default: False),\n",
      " 'sharrisii_homolog_chromosome': 'Tasmanian devil chromosome/scaffold name' (default: False),\n",
      " 'sharrisii_homolog_ensembl_gene': 'Tasmanian devil gene stable ID' (default: False),\n",
      " 'sharrisii_homolog_ensembl_peptide': 'Tasmanian devil protein or transcript stable ID' (default: False),\n",
      " 'sharrisii_homolog_goc_score': 'Tasmanian devil Gene-order conservation score' (default: False),\n",
      " 'sharrisii_homolog_orthology_confidence': 'Tasmanian devil orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sharrisii_homolog_orthology_type': 'Tasmanian devil homology type' (default: False),\n",
      " 'sharrisii_homolog_perc_id': '%id. target Tasmanian devil gene identical to query gene' (default: False),\n",
      " 'sharrisii_homolog_perc_id_r1': '%id. query gene identical to target Tasmanian devil gene' (default: False),\n",
      " 'sharrisii_homolog_subtype': 'Last common ancestor with Tasmanian devil' (default: False),\n",
      " 'sift_prediction_2076': 'SIFT prediction' (default: False),\n",
      " 'sift_score_2076': 'SIFT score' (default: False),\n",
      " 'sifts_import': 'PDB-ENSP mappings' (default: False),\n",
      " 'sifts_import_end': 'PDB-ENSP mappings end' (default: False),\n",
      " 'sifts_import_start': 'PDB-ENSP mappings start' (default: False),\n",
      " 'signalp': 'Cleavage site (Signalp)' (default: False),\n",
      " 'signalp_end': 'Cleavage site (Signalp) end' (default: False),\n",
      " 'signalp_start': 'Cleavage site (Signalp) start' (default: False),\n",
      " 'sldorsalis_homolog_associated_gene_name': 'Yellowtail amberjack gene name' (default: False),\n",
      " 'sldorsalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sldorsalis_homolog_chrom_end': 'Yellowtail amberjack chromosome/scaffold end (bp)' (default: False),\n",
      " 'sldorsalis_homolog_chrom_start': 'Yellowtail amberjack chromosome/scaffold start (bp)' (default: False),\n",
      " 'sldorsalis_homolog_chromosome': 'Yellowtail amberjack chromosome/scaffold name' (default: False),\n",
      " 'sldorsalis_homolog_ensembl_gene': 'Yellowtail amberjack gene stable ID' (default: False),\n",
      " 'sldorsalis_homolog_ensembl_peptide': 'Yellowtail amberjack protein or transcript stable ID' (default: False),\n",
      " 'sldorsalis_homolog_goc_score': 'Yellowtail amberjack Gene-order conservation score' (default: False),\n",
      " 'sldorsalis_homolog_orthology_confidence': 'Yellowtail amberjack orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sldorsalis_homolog_orthology_type': 'Yellowtail amberjack homology type' (default: False),\n",
      " 'sldorsalis_homolog_perc_id': '%id. target Yellowtail amberjack gene identical to query gene' (default: False),\n",
      " 'sldorsalis_homolog_perc_id_r1': '%id. query gene identical to target Yellowtail amberjack gene' (default: False),\n",
      " 'sldorsalis_homolog_subtype': 'Last common ancestor with Yellowtail amberjack' (default: False),\n",
      " 'sldorsalis_homolog_wga_coverage': 'Yellowtail amberjack Whole-genome alignment coverage' (default: False),\n",
      " 'slucioperca_homolog_associated_gene_name': 'Pike-perch gene name' (default: False),\n",
      " 'slucioperca_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'slucioperca_homolog_chrom_end': 'Pike-perch chromosome/scaffold end (bp)' (default: False),\n",
      " 'slucioperca_homolog_chrom_start': 'Pike-perch chromosome/scaffold start (bp)' (default: False),\n",
      " 'slucioperca_homolog_chromosome': 'Pike-perch chromosome/scaffold name' (default: False),\n",
      " 'slucioperca_homolog_ensembl_gene': 'Pike-perch gene stable ID' (default: False),\n",
      " 'slucioperca_homolog_ensembl_peptide': 'Pike-perch protein or transcript stable ID' (default: False),\n",
      " 'slucioperca_homolog_goc_score': 'Pike-perch Gene-order conservation score' (default: False),\n",
      " 'slucioperca_homolog_orthology_confidence': 'Pike-perch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'slucioperca_homolog_orthology_type': 'Pike-perch homology type' (default: False),\n",
      " 'slucioperca_homolog_perc_id': '%id. target Pike-perch gene identical to query gene' (default: False),\n",
      " 'slucioperca_homolog_perc_id_r1': '%id. query gene identical to target Pike-perch gene' (default: False),\n",
      " 'slucioperca_homolog_subtype': 'Last common ancestor with Pike-perch' (default: False),\n",
      " 'slucioperca_homolog_wga_coverage': 'Pike-perch Whole-genome alignment coverage' (default: False),\n",
      " 'smart': 'SMART ID' (default: False),\n",
      " 'smart_end': 'SMART end' (default: False),\n",
      " 'smart_start': 'SMART start' (default: False),\n",
      " 'smaximus_homolog_associated_gene_name': 'Turbot gene name' (default: False),\n",
      " 'smaximus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'smaximus_homolog_chrom_end': 'Turbot chromosome/scaffold end (bp)' (default: False),\n",
      " 'smaximus_homolog_chrom_start': 'Turbot chromosome/scaffold start (bp)' (default: False),\n",
      " 'smaximus_homolog_chromosome': 'Turbot chromosome/scaffold name' (default: False),\n",
      " 'smaximus_homolog_ensembl_gene': 'Turbot gene stable ID' (default: False),\n",
      " 'smaximus_homolog_ensembl_peptide': 'Turbot protein or transcript stable ID' (default: False),\n",
      " 'smaximus_homolog_goc_score': 'Turbot Gene-order conservation score' (default: False),\n",
      " 'smaximus_homolog_orthology_confidence': 'Turbot orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'smaximus_homolog_orthology_type': 'Turbot homology type' (default: False),\n",
      " 'smaximus_homolog_perc_id': '%id. target Turbot gene identical to query gene' (default: False),\n",
      " 'smaximus_homolog_perc_id_r1': '%id. query gene identical to target Turbot gene' (default: False),\n",
      " 'smaximus_homolog_subtype': 'Last common ancestor with Turbot' (default: False),\n",
      " 'smerianae_homolog_associated_gene_name': 'Argentine black and white tegu gene name' (default: False),\n",
      " 'smerianae_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'smerianae_homolog_chrom_end': 'Argentine black and white tegu chromosome/scaffold end (bp)' (default: False),\n",
      " 'smerianae_homolog_chrom_start': 'Argentine black and white tegu chromosome/scaffold start (bp)' (default: False),\n",
      " 'smerianae_homolog_chromosome': 'Argentine black and white tegu chromosome/scaffold name' (default: False),\n",
      " 'smerianae_homolog_ensembl_gene': 'Argentine black and white tegu gene stable ID' (default: False),\n",
      " 'smerianae_homolog_ensembl_peptide': 'Argentine black and white tegu protein or transcript stable ID' (default: False),\n",
      " 'smerianae_homolog_goc_score': 'Argentine black and white tegu Gene-order conservation score' (default: False),\n",
      " 'smerianae_homolog_orthology_confidence': 'Argentine black and white tegu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'smerianae_homolog_orthology_type': 'Argentine black and white tegu homology type' (default: False),\n",
      " 'smerianae_homolog_perc_id': '%id. target Argentine black and white tegu gene identical to query gene' (default: False),\n",
      " 'smerianae_homolog_perc_id_r1': '%id. query gene identical to target Argentine black and white tegu gene' (default: False),\n",
      " 'smerianae_homolog_subtype': 'Last common ancestor with Argentine black and white tegu' (default: False),\n",
      " 'smerianae_homolog_wga_coverage': 'Argentine black and white tegu Whole-genome alignment coverage' (default: False),\n",
      " 'snp_chromosome_strand': 'Variant chromosome Strand' (default: False),\n",
      " 'somatic_allele': 'Variant alleles' (default: False),\n",
      " 'somatic_cds_end_2076': 'CDS end' (default: False),\n",
      " 'somatic_cds_start_2076': 'CDS start' (default: False),\n",
      " 'somatic_chromosome_end': 'Chromosome/scaffold position end (bp)' (default: False),\n",
      " 'somatic_chromosome_start': 'Chromosome/scaffold position start (bp)' (default: False),\n",
      " 'somatic_mapweight': 'Mapweight' (default: False),\n",
      " 'somatic_peptide_location': 'Protein location (aa)' (default: False),\n",
      " 'somatic_snp_chromosome_strand': 'Variant chromosome/scaffold strand' (default: False),\n",
      " 'somatic_source_description': 'Variant source description' (default: False),\n",
      " 'somatic_source_name': 'Variant source' (default: False),\n",
      " 'somatic_synonymous_status': 'Variant consequence' (default: False),\n",
      " 'somatic_transcript_location': 'Transcript location (bp)' (default: False),\n",
      " 'somatic_validated': 'Variant supporting evidence' (default: False),\n",
      " 'somatic_variation_name': 'Variant name' (default: True),\n",
      " 'source': 'Source (gene)' (default: False),\n",
      " 'source_description': 'Variant source description' (default: False),\n",
      " 'source_name': 'Source name' (default: False),\n",
      " 'spartitus_homolog_associated_gene_name': 'Bicolor damselfish gene name' (default: False),\n",
      " 'spartitus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'spartitus_homolog_chrom_end': 'Bicolor damselfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'spartitus_homolog_chrom_start': 'Bicolor damselfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'spartitus_homolog_chromosome': 'Bicolor damselfish chromosome/scaffold name' (default: False),\n",
      " 'spartitus_homolog_ensembl_gene': 'Bicolor damselfish gene stable ID' (default: False),\n",
      " 'spartitus_homolog_ensembl_peptide': 'Bicolor damselfish protein or transcript stable ID' (default: False),\n",
      " 'spartitus_homolog_goc_score': 'Bicolor damselfish Gene-order conservation score' (default: False),\n",
      " 'spartitus_homolog_orthology_confidence': 'Bicolor damselfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'spartitus_homolog_orthology_type': 'Bicolor damselfish homology type' (default: False),\n",
      " 'spartitus_homolog_perc_id': '%id. target Bicolor damselfish gene identical to query gene' (default: False),\n",
      " 'spartitus_homolog_perc_id_r1': '%id. query gene identical to target Bicolor damselfish gene' (default: False),\n",
      " 'spartitus_homolog_subtype': 'Last common ancestor with Bicolor damselfish' (default: False),\n",
      " 'spartitus_homolog_wga_coverage': 'Bicolor damselfish Whole-genome alignment coverage' (default: False),\n",
      " 'spunctatus_homolog_associated_gene_name': 'Tuatara gene name' (default: False),\n",
      " 'spunctatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'spunctatus_homolog_chrom_end': 'Tuatara chromosome/scaffold end (bp)' (default: False),\n",
      " 'spunctatus_homolog_chrom_start': 'Tuatara chromosome/scaffold start (bp)' (default: False),\n",
      " 'spunctatus_homolog_chromosome': 'Tuatara chromosome/scaffold name' (default: False),\n",
      " 'spunctatus_homolog_ensembl_gene': 'Tuatara gene stable ID' (default: False),\n",
      " 'spunctatus_homolog_ensembl_peptide': 'Tuatara protein or transcript stable ID' (default: False),\n",
      " 'spunctatus_homolog_goc_score': 'Tuatara Gene-order conservation score' (default: False),\n",
      " 'spunctatus_homolog_orthology_confidence': 'Tuatara orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'spunctatus_homolog_orthology_type': 'Tuatara homology type' (default: False),\n",
      " 'spunctatus_homolog_perc_id': '%id. target Tuatara gene identical to query gene' (default: False),\n",
      " 'spunctatus_homolog_perc_id_r1': '%id. query gene identical to target Tuatara gene' (default: False),\n",
      " 'spunctatus_homolog_subtype': 'Last common ancestor with Tuatara' (default: False),\n",
      " 'spunctatus_homolog_wga_coverage': 'Tuatara Whole-genome alignment coverage' (default: False),\n",
      " 'ssalar_homolog_associated_gene_name': 'Atlantic salmon gene name' (default: False),\n",
      " 'ssalar_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ssalar_homolog_chrom_end': 'Atlantic salmon chromosome/scaffold end (bp)' (default: False),\n",
      " 'ssalar_homolog_chrom_start': 'Atlantic salmon chromosome/scaffold start (bp)' (default: False),\n",
      " 'ssalar_homolog_chromosome': 'Atlantic salmon chromosome/scaffold name' (default: False),\n",
      " 'ssalar_homolog_ensembl_gene': 'Atlantic salmon gene stable ID' (default: False),\n",
      " 'ssalar_homolog_ensembl_peptide': 'Atlantic salmon protein or transcript stable ID' (default: False),\n",
      " 'ssalar_homolog_goc_score': 'Atlantic salmon Gene-order conservation score' (default: False),\n",
      " 'ssalar_homolog_orthology_confidence': 'Atlantic salmon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ssalar_homolog_orthology_type': 'Atlantic salmon homology type' (default: False),\n",
      " 'ssalar_homolog_perc_id': '%id. target Atlantic salmon gene identical to query gene' (default: False),\n",
      " 'ssalar_homolog_perc_id_r1': '%id. query gene identical to target Atlantic salmon gene' (default: False),\n",
      " 'ssalar_homolog_subtype': 'Last common ancestor with Atlantic salmon' (default: False),\n",
      " 'sscrofa_homolog_associated_gene_name': 'Pig gene name' (default: False),\n",
      " 'sscrofa_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'sscrofa_homolog_chrom_end': 'Pig chromosome/scaffold end (bp)' (default: False),\n",
      " 'sscrofa_homolog_chrom_start': 'Pig chromosome/scaffold start (bp)' (default: False),\n",
      " 'sscrofa_homolog_chromosome': 'Pig chromosome/scaffold name' (default: False),\n",
      " 'sscrofa_homolog_ensembl_gene': 'Pig gene stable ID' (default: False),\n",
      " 'sscrofa_homolog_ensembl_peptide': 'Pig protein or transcript stable ID' (default: False),\n",
      " 'sscrofa_homolog_goc_score': 'Pig Gene-order conservation score' (default: False),\n",
      " 'sscrofa_homolog_orthology_confidence': 'Pig orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'sscrofa_homolog_orthology_type': 'Pig homology type' (default: False),\n",
      " 'sscrofa_homolog_perc_id': '%id. target Pig gene identical to query gene' (default: False),\n",
      " 'sscrofa_homolog_perc_id_r1': '%id. query gene identical to target Pig gene' (default: False),\n",
      " 'sscrofa_homolog_subtype': 'Last common ancestor with Pig' (default: False),\n",
      " 'sscrofa_homolog_wga_coverage': 'Pig Whole-genome alignment coverage' (default: False),\n",
      " 'start_position': 'Gene start (bp)' (default: False),\n",
      " 'strain_gender': 'Strain gender' (default: False),\n",
      " 'strain_name': 'Strain name' (default: False),\n",
      " 'strand': 'Strand' (default: False),\n",
      " 'strutta_homolog_associated_gene_name': 'Brown trout gene name' (default: False),\n",
      " 'strutta_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'strutta_homolog_chrom_end': 'Brown trout chromosome/scaffold end (bp)' (default: False),\n",
      " 'strutta_homolog_chrom_start': 'Brown trout chromosome/scaffold start (bp)' (default: False),\n",
      " 'strutta_homolog_chromosome': 'Brown trout chromosome/scaffold name' (default: False),\n",
      " 'strutta_homolog_ensembl_gene': 'Brown trout gene stable ID' (default: False),\n",
      " 'strutta_homolog_ensembl_peptide': 'Brown trout protein or transcript stable ID' (default: False),\n",
      " 'strutta_homolog_goc_score': 'Brown trout Gene-order conservation score' (default: False),\n",
      " 'strutta_homolog_orthology_confidence': 'Brown trout orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'strutta_homolog_orthology_type': 'Brown trout homology type' (default: False),\n",
      " 'strutta_homolog_perc_id': '%id. target Brown trout gene identical to query gene' (default: False),\n",
      " 'strutta_homolog_perc_id_r1': '%id. query gene identical to target Brown trout gene' (default: False),\n",
      " 'strutta_homolog_subtype': 'Last common ancestor with Brown trout' (default: False),\n",
      " 'strutta_homolog_wga_coverage': 'Brown trout Whole-genome alignment coverage' (default: False),\n",
      " 'study_external_id': 'Study external reference' (default: False),\n",
      " 'superfamily': 'Superfamily ID' (default: False),\n",
      " 'superfamily_end': 'Superfamily end' (default: False),\n",
      " 'superfamily_start': 'Superfamily start' (default: False),\n",
      " 'svulgaris_homolog_associated_gene_name': 'Eurasian red squirrel gene name' (default: False),\n",
      " 'svulgaris_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'svulgaris_homolog_chrom_end': 'Eurasian red squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'svulgaris_homolog_chrom_start': 'Eurasian red squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'svulgaris_homolog_chromosome': 'Eurasian red squirrel chromosome/scaffold name' (default: False),\n",
      " 'svulgaris_homolog_ensembl_gene': 'Eurasian red squirrel gene stable ID' (default: False),\n",
      " 'svulgaris_homolog_ensembl_peptide': 'Eurasian red squirrel protein or transcript stable ID' (default: False),\n",
      " 'svulgaris_homolog_goc_score': 'Eurasian red squirrel Gene-order conservation score' (default: False),\n",
      " 'svulgaris_homolog_orthology_confidence': 'Eurasian red squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'svulgaris_homolog_orthology_type': 'Eurasian red squirrel homology type' (default: False),\n",
      " 'svulgaris_homolog_perc_id': '%id. target Eurasian red squirrel gene identical to query gene' (default: False),\n",
      " 'svulgaris_homolog_perc_id_r1': '%id. query gene identical to target Eurasian red squirrel gene' (default: False),\n",
      " 'svulgaris_homolog_subtype': 'Last common ancestor with Eurasian red squirrel' (default: False),\n",
      " 'svulgaris_homolog_wga_coverage': 'Eurasian red squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'synonymous_status': 'Variant consequence' (default: False),\n",
      " 'tbelangeri_homolog_associated_gene_name': 'Tree Shrew gene name' (default: False),\n",
      " 'tbelangeri_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tbelangeri_homolog_chrom_end': 'Tree Shrew chromosome/scaffold end (bp)' (default: False),\n",
      " 'tbelangeri_homolog_chrom_start': 'Tree Shrew chromosome/scaffold start (bp)' (default: False),\n",
      " 'tbelangeri_homolog_chromosome': 'Tree Shrew chromosome/scaffold name' (default: False),\n",
      " 'tbelangeri_homolog_ensembl_gene': 'Tree Shrew gene stable ID' (default: False),\n",
      " 'tbelangeri_homolog_ensembl_peptide': 'Tree Shrew protein or transcript stable ID' (default: False),\n",
      " 'tbelangeri_homolog_goc_score': 'Tree Shrew Gene-order conservation score' (default: False),\n",
      " 'tbelangeri_homolog_orthology_confidence': 'Tree Shrew orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tbelangeri_homolog_orthology_type': 'Tree Shrew homology type' (default: False),\n",
      " 'tbelangeri_homolog_perc_id': '%id. target Tree Shrew gene identical to query gene' (default: False),\n",
      " 'tbelangeri_homolog_perc_id_r1': '%id. query gene identical to target Tree Shrew gene' (default: False),\n",
      " 'tbelangeri_homolog_subtype': 'Last common ancestor with Tree Shrew' (default: False),\n",
      " 'tbelangeri_homolog_wga_coverage': 'Tree Shrew Whole-genome alignment coverage' (default: False),\n",
      " 'tctriunguis_homolog_associated_gene_name': 'Three-toed box turtle gene name' (default: False),\n",
      " 'tctriunguis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tctriunguis_homolog_chrom_end': 'Three-toed box turtle chromosome/scaffold end (bp)' (default: False),\n",
      " 'tctriunguis_homolog_chrom_start': 'Three-toed box turtle chromosome/scaffold start (bp)' (default: False),\n",
      " 'tctriunguis_homolog_chromosome': 'Three-toed box turtle chromosome/scaffold name' (default: False),\n",
      " 'tctriunguis_homolog_ensembl_gene': 'Three-toed box turtle gene stable ID' (default: False),\n",
      " 'tctriunguis_homolog_ensembl_peptide': 'Three-toed box turtle protein or transcript stable ID' (default: False),\n",
      " 'tctriunguis_homolog_goc_score': 'Three-toed box turtle Gene-order conservation score' (default: False),\n",
      " 'tctriunguis_homolog_orthology_confidence': 'Three-toed box turtle orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tctriunguis_homolog_orthology_type': 'Three-toed box turtle homology type' (default: False),\n",
      " 'tctriunguis_homolog_perc_id': '%id. target Three-toed box turtle gene identical to query gene' (default: False),\n",
      " 'tctriunguis_homolog_perc_id_r1': '%id. query gene identical to target Three-toed box turtle gene' (default: False),\n",
      " 'tctriunguis_homolog_subtype': 'Last common ancestor with Three-toed box turtle' (default: False),\n",
      " 'tctriunguis_homolog_wga_coverage': 'Three-toed box turtle Whole-genome alignment coverage' (default: False),\n",
      " 'tguttata_homolog_associated_gene_name': 'Zebra finch gene name' (default: False),\n",
      " 'tguttata_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tguttata_homolog_chrom_end': 'Zebra finch chromosome/scaffold end (bp)' (default: False),\n",
      " 'tguttata_homolog_chrom_start': 'Zebra finch chromosome/scaffold start (bp)' (default: False),\n",
      " 'tguttata_homolog_chromosome': 'Zebra finch chromosome/scaffold name' (default: False),\n",
      " 'tguttata_homolog_ensembl_gene': 'Zebra finch gene stable ID' (default: False),\n",
      " 'tguttata_homolog_ensembl_peptide': 'Zebra finch protein or transcript stable ID' (default: False),\n",
      " 'tguttata_homolog_goc_score': 'Zebra finch Gene-order conservation score' (default: False),\n",
      " 'tguttata_homolog_orthology_confidence': 'Zebra finch orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tguttata_homolog_orthology_type': 'Zebra finch homology type' (default: False),\n",
      " 'tguttata_homolog_perc_id': '%id. target Zebra finch gene identical to query gene' (default: False),\n",
      " 'tguttata_homolog_perc_id_r1': '%id. query gene identical to target Zebra finch gene' (default: False),\n",
      " 'tguttata_homolog_subtype': 'Last common ancestor with Zebra finch' (default: False),\n",
      " 'tguttata_homolog_wga_coverage': 'Zebra finch Whole-genome alignment coverage' (default: False),\n",
      " 'tigrfam': 'TIGRFAM ID' (default: False),\n",
      " 'tigrfam_end': 'TIGRFAM end' (default: False),\n",
      " 'tigrfam_start': 'TIGRFAM start' (default: False),\n",
      " 'tmhmm': 'Transmembrane helices' (default: False),\n",
      " 'tmhmm_end': 'Transmembrane helices end' (default: False),\n",
      " 'tmhmm_start': 'Transmembrane helices start' (default: False),\n",
      " 'tnigroviridis_homolog_associated_gene_name': 'Tetraodon gene name' (default: False),\n",
      " 'tnigroviridis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'tnigroviridis_homolog_chrom_end': 'Tetraodon chromosome/scaffold end (bp)' (default: False),\n",
      " 'tnigroviridis_homolog_chrom_start': 'Tetraodon chromosome/scaffold start (bp)' (default: False),\n",
      " 'tnigroviridis_homolog_chromosome': 'Tetraodon chromosome/scaffold name' (default: False),\n",
      " 'tnigroviridis_homolog_ensembl_gene': 'Tetraodon gene stable ID' (default: False),\n",
      " 'tnigroviridis_homolog_ensembl_peptide': 'Tetraodon protein or transcript stable ID' (default: False),\n",
      " 'tnigroviridis_homolog_goc_score': 'Tetraodon Gene-order conservation score' (default: False),\n",
      " 'tnigroviridis_homolog_orthology_confidence': 'Tetraodon orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'tnigroviridis_homolog_orthology_type': 'Tetraodon homology type' (default: False),\n",
      " 'tnigroviridis_homolog_perc_id': '%id. target Tetraodon gene identical to query gene' (default: False),\n",
      " 'tnigroviridis_homolog_perc_id_r1': '%id. query gene identical to target Tetraodon gene' (default: False),\n",
      " 'tnigroviridis_homolog_subtype': 'Last common ancestor with Tetraodon' (default: False),\n",
      " 'tnigroviridis_homolog_wga_coverage': 'Tetraodon Whole-genome alignment coverage' (default: False),\n",
      " 'transcript_appris': 'APPRIS annotation' (default: False),\n",
      " 'transcript_biotype': 'Transcript type' (default: False),\n",
      " 'transcript_count': 'Transcript count' (default: False),\n",
      " 'transcript_end': 'Transcript end (bp)' (default: False),\n",
      " 'transcript_exon_intron': 'Unspliced (Transcript)' (default: False),\n",
      " 'transcript_flank': 'Flank (Transcript)' (default: False),\n",
      " 'transcript_gencode_basic': 'GENCODE basic annotation' (default: False),\n",
      " 'transcript_is_canonical': 'Ensembl Canonical' (default: False),\n",
      " 'transcript_length': 'Transcript length (including UTRs and CDS)' (default: False),\n",
      " 'transcript_location': 'Transcript location (bp)' (default: False),\n",
      " 'transcript_mane_plus_clinical': 'RefSeq match transcript (MANE Plus Clinical)' (default: False),\n",
      " 'transcript_mane_select': 'RefSeq match transcript (MANE Select)' (default: False),\n",
      " 'transcript_source': 'Source (transcript)' (default: False),\n",
      " 'transcript_start': 'Transcript start (bp)' (default: False),\n",
      " 'transcript_tsl': 'Transcript support level (TSL)' (default: False),\n",
      " 'transcript_version': 'Version (transcript)' (default: False),\n",
      " 'transcription_start_site': 'Transcription start site (TSS)' (default: False),\n",
      " 'trubripes_homolog_associated_gene_name': 'Fugu gene name' (default: False),\n",
      " 'trubripes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'trubripes_homolog_chrom_end': 'Fugu chromosome/scaffold end (bp)' (default: False),\n",
      " 'trubripes_homolog_chrom_start': 'Fugu chromosome/scaffold start (bp)' (default: False),\n",
      " 'trubripes_homolog_chromosome': 'Fugu chromosome/scaffold name' (default: False),\n",
      " 'trubripes_homolog_ensembl_gene': 'Fugu gene stable ID' (default: False),\n",
      " 'trubripes_homolog_ensembl_peptide': 'Fugu protein or transcript stable ID' (default: False),\n",
      " 'trubripes_homolog_goc_score': 'Fugu Gene-order conservation score' (default: False),\n",
      " 'trubripes_homolog_orthology_confidence': 'Fugu orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'trubripes_homolog_orthology_type': 'Fugu homology type' (default: False),\n",
      " 'trubripes_homolog_perc_id': '%id. target Fugu gene identical to query gene' (default: False),\n",
      " 'trubripes_homolog_perc_id_r1': '%id. query gene identical to target Fugu gene' (default: False),\n",
      " 'trubripes_homolog_subtype': 'Last common ancestor with Fugu' (default: False),\n",
      " 'trubripes_homolog_wga_coverage': 'Fugu Whole-genome alignment coverage' (default: False),\n",
      " 'ttruncatus_homolog_associated_gene_name': 'Dolphin gene name' (default: False),\n",
      " 'ttruncatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'ttruncatus_homolog_chrom_end': 'Dolphin chromosome/scaffold end (bp)' (default: False),\n",
      " 'ttruncatus_homolog_chrom_start': 'Dolphin chromosome/scaffold start (bp)' (default: False),\n",
      " 'ttruncatus_homolog_chromosome': 'Dolphin chromosome/scaffold name' (default: False),\n",
      " 'ttruncatus_homolog_ensembl_gene': 'Dolphin gene stable ID' (default: False),\n",
      " 'ttruncatus_homolog_ensembl_peptide': 'Dolphin protein or transcript stable ID' (default: False),\n",
      " 'ttruncatus_homolog_goc_score': 'Dolphin Gene-order conservation score' (default: False),\n",
      " 'ttruncatus_homolog_orthology_confidence': 'Dolphin orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'ttruncatus_homolog_orthology_type': 'Dolphin homology type' (default: False),\n",
      " 'ttruncatus_homolog_perc_id': '%id. target Dolphin gene identical to query gene' (default: False),\n",
      " 'ttruncatus_homolog_perc_id_r1': '%id. query gene identical to target Dolphin gene' (default: False),\n",
      " 'ttruncatus_homolog_subtype': 'Last common ancestor with Dolphin' (default: False),\n",
      " 'ttruncatus_homolog_wga_coverage': 'Dolphin Whole-genome alignment coverage' (default: False),\n",
      " 'uamericanus_homolog_associated_gene_name': 'American black bear gene name' (default: False),\n",
      " 'uamericanus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'uamericanus_homolog_chrom_end': 'American black bear chromosome/scaffold end (bp)' (default: False),\n",
      " 'uamericanus_homolog_chrom_start': 'American black bear chromosome/scaffold start (bp)' (default: False),\n",
      " 'uamericanus_homolog_chromosome': 'American black bear chromosome/scaffold name' (default: False),\n",
      " 'uamericanus_homolog_ensembl_gene': 'American black bear gene stable ID' (default: False),\n",
      " 'uamericanus_homolog_ensembl_peptide': 'American black bear protein or transcript stable ID' (default: False),\n",
      " 'uamericanus_homolog_goc_score': 'American black bear Gene-order conservation score' (default: False),\n",
      " 'uamericanus_homolog_orthology_confidence': 'American black bear orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'uamericanus_homolog_orthology_type': 'American black bear homology type' (default: False),\n",
      " 'uamericanus_homolog_perc_id': '%id. target American black bear gene identical to query gene' (default: False),\n",
      " 'uamericanus_homolog_perc_id_r1': '%id. query gene identical to target American black bear gene' (default: False),\n",
      " 'uamericanus_homolog_subtype': 'Last common ancestor with American black bear' (default: False),\n",
      " 'uamericanus_homolog_wga_coverage': 'American black bear Whole-genome alignment coverage' (default: False),\n",
      " 'ucsc': 'UCSC Stable ID' (default: False),\n",
      " 'umaritimus_homolog_associated_gene_name': 'Polar bear gene name' (default: False),\n",
      " 'umaritimus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'umaritimus_homolog_chrom_end': 'Polar bear chromosome/scaffold end (bp)' (default: False),\n",
      " 'umaritimus_homolog_chrom_start': 'Polar bear chromosome/scaffold start (bp)' (default: False),\n",
      " 'umaritimus_homolog_chromosome': 'Polar bear chromosome/scaffold name' (default: False),\n",
      " 'umaritimus_homolog_ensembl_gene': 'Polar bear gene stable ID' (default: False),\n",
      " 'umaritimus_homolog_ensembl_peptide': 'Polar bear protein or transcript stable ID' (default: False),\n",
      " 'umaritimus_homolog_goc_score': 'Polar bear Gene-order conservation score' (default: False),\n",
      " 'umaritimus_homolog_orthology_confidence': 'Polar bear orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'umaritimus_homolog_orthology_type': 'Polar bear homology type' (default: False),\n",
      " 'umaritimus_homolog_perc_id': '%id. target Polar bear gene identical to query gene' (default: False),\n",
      " 'umaritimus_homolog_perc_id_r1': '%id. query gene identical to target Polar bear gene' (default: False),\n",
      " 'umaritimus_homolog_subtype': 'Last common ancestor with Polar bear' (default: False),\n",
      " 'umaritimus_homolog_wga_coverage': 'Polar bear Whole-genome alignment coverage' (default: False),\n",
      " 'uniparc': 'UniParc ID' (default: False),\n",
      " 'uniprot_gn_id': 'UniProtKB Gene Name ID' (default: False),\n",
      " 'uniprot_gn_symbol': 'UniProtKB Gene Name symbol' (default: False),\n",
      " 'uniprot_isoform': 'UniProtKB isoform ID' (default: False),\n",
      " 'uniprotsptrembl': 'UniProtKB/TrEMBL ID' (default: False),\n",
      " 'uniprotswissprot': 'UniProtKB/Swiss-Prot ID' (default: False),\n",
      " 'uparryii_homolog_associated_gene_name': 'Arctic ground squirrel gene name' (default: False),\n",
      " 'uparryii_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'uparryii_homolog_chrom_end': 'Arctic ground squirrel chromosome/scaffold end (bp)' (default: False),\n",
      " 'uparryii_homolog_chrom_start': 'Arctic ground squirrel chromosome/scaffold start (bp)' (default: False),\n",
      " 'uparryii_homolog_chromosome': 'Arctic ground squirrel chromosome/scaffold name' (default: False),\n",
      " 'uparryii_homolog_ensembl_gene': 'Arctic ground squirrel gene stable ID' (default: False),\n",
      " 'uparryii_homolog_ensembl_peptide': 'Arctic ground squirrel protein or transcript stable ID' (default: False),\n",
      " 'uparryii_homolog_goc_score': 'Arctic ground squirrel Gene-order conservation score' (default: False),\n",
      " 'uparryii_homolog_orthology_confidence': 'Arctic ground squirrel orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'uparryii_homolog_orthology_type': 'Arctic ground squirrel homology type' (default: False),\n",
      " 'uparryii_homolog_perc_id': '%id. target Arctic ground squirrel gene identical to query gene' (default: False),\n",
      " 'uparryii_homolog_perc_id_r1': '%id. query gene identical to target Arctic ground squirrel gene' (default: False),\n",
      " 'uparryii_homolog_subtype': 'Last common ancestor with Arctic ground squirrel' (default: False),\n",
      " 'uparryii_homolog_wga_coverage': 'Arctic ground squirrel Whole-genome alignment coverage' (default: False),\n",
      " 'upstream_flank': 'upstream_flank' (default: False),\n",
      " 'validated': 'Variant supporting evidence' (default: False),\n",
      " 'variation_name': 'Variant name' (default: True),\n",
      " 'version': 'Version (gene)' (default: False),\n",
      " 'vpacos_homolog_associated_gene_name': 'Alpaca gene name' (default: False),\n",
      " 'vpacos_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vpacos_homolog_chrom_end': 'Alpaca chromosome/scaffold end (bp)' (default: False),\n",
      " 'vpacos_homolog_chrom_start': 'Alpaca chromosome/scaffold start (bp)' (default: False),\n",
      " 'vpacos_homolog_chromosome': 'Alpaca chromosome/scaffold name' (default: False),\n",
      " 'vpacos_homolog_ensembl_gene': 'Alpaca gene stable ID' (default: False),\n",
      " 'vpacos_homolog_ensembl_peptide': 'Alpaca protein or transcript stable ID' (default: False),\n",
      " 'vpacos_homolog_goc_score': 'Alpaca Gene-order conservation score' (default: False),\n",
      " 'vpacos_homolog_orthology_confidence': 'Alpaca orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vpacos_homolog_orthology_type': 'Alpaca homology type' (default: False),\n",
      " 'vpacos_homolog_perc_id': '%id. target Alpaca gene identical to query gene' (default: False),\n",
      " 'vpacos_homolog_perc_id_r1': '%id. query gene identical to target Alpaca gene' (default: False),\n",
      " 'vpacos_homolog_subtype': 'Last common ancestor with Alpaca' (default: False),\n",
      " 'vpacos_homolog_wga_coverage': 'Alpaca Whole-genome alignment coverage' (default: False),\n",
      " 'vursinus_homolog_associated_gene_name': 'Common wombat gene name' (default: False),\n",
      " 'vursinus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vursinus_homolog_chrom_end': 'Common wombat chromosome/scaffold end (bp)' (default: False),\n",
      " 'vursinus_homolog_chrom_start': 'Common wombat chromosome/scaffold start (bp)' (default: False),\n",
      " 'vursinus_homolog_chromosome': 'Common wombat chromosome/scaffold name' (default: False),\n",
      " 'vursinus_homolog_ensembl_gene': 'Common wombat gene stable ID' (default: False),\n",
      " 'vursinus_homolog_ensembl_peptide': 'Common wombat protein or transcript stable ID' (default: False),\n",
      " 'vursinus_homolog_goc_score': 'Common wombat Gene-order conservation score' (default: False),\n",
      " 'vursinus_homolog_orthology_confidence': 'Common wombat orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vursinus_homolog_orthology_type': 'Common wombat homology type' (default: False),\n",
      " 'vursinus_homolog_perc_id': '%id. target Common wombat gene identical to query gene' (default: False),\n",
      " 'vursinus_homolog_perc_id_r1': '%id. query gene identical to target Common wombat gene' (default: False),\n",
      " 'vursinus_homolog_subtype': 'Last common ancestor with Common wombat' (default: False),\n",
      " 'vursinus_homolog_wga_coverage': 'Common wombat Whole-genome alignment coverage' (default: False),\n",
      " 'vvulpes_homolog_associated_gene_name': 'Red fox gene name' (default: False),\n",
      " 'vvulpes_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'vvulpes_homolog_chrom_end': 'Red fox chromosome/scaffold end (bp)' (default: False),\n",
      " 'vvulpes_homolog_chrom_start': 'Red fox chromosome/scaffold start (bp)' (default: False),\n",
      " 'vvulpes_homolog_chromosome': 'Red fox chromosome/scaffold name' (default: False),\n",
      " 'vvulpes_homolog_ensembl_gene': 'Red fox gene stable ID' (default: False),\n",
      " 'vvulpes_homolog_ensembl_peptide': 'Red fox protein or transcript stable ID' (default: False),\n",
      " 'vvulpes_homolog_goc_score': 'Red fox Gene-order conservation score' (default: False),\n",
      " 'vvulpes_homolog_orthology_confidence': 'Red fox orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'vvulpes_homolog_orthology_type': 'Red fox homology type' (default: False),\n",
      " 'vvulpes_homolog_perc_id': '%id. target Red fox gene identical to query gene' (default: False),\n",
      " 'vvulpes_homolog_perc_id_r1': '%id. query gene identical to target Red fox gene' (default: False),\n",
      " 'vvulpes_homolog_subtype': 'Last common ancestor with Red fox' (default: False),\n",
      " 'vvulpes_homolog_wga_coverage': 'Red fox Whole-genome alignment coverage' (default: False),\n",
      " 'wikigene_description': 'WikiGene description' (default: False),\n",
      " 'wikigene_id': 'WikiGene ID' (default: False),\n",
      " 'wikigene_name': 'WikiGene name' (default: False),\n",
      " 'xmaculatus_homolog_associated_gene_name': 'Platyfish gene name' (default: False),\n",
      " 'xmaculatus_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xmaculatus_homolog_chrom_end': 'Platyfish chromosome/scaffold end (bp)' (default: False),\n",
      " 'xmaculatus_homolog_chrom_start': 'Platyfish chromosome/scaffold start (bp)' (default: False),\n",
      " 'xmaculatus_homolog_chromosome': 'Platyfish chromosome/scaffold name' (default: False),\n",
      " 'xmaculatus_homolog_ensembl_gene': 'Platyfish gene stable ID' (default: False),\n",
      " 'xmaculatus_homolog_ensembl_peptide': 'Platyfish protein or transcript stable ID' (default: False),\n",
      " 'xmaculatus_homolog_goc_score': 'Platyfish Gene-order conservation score' (default: False),\n",
      " 'xmaculatus_homolog_orthology_confidence': 'Platyfish orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xmaculatus_homolog_orthology_type': 'Platyfish homology type' (default: False),\n",
      " 'xmaculatus_homolog_perc_id': '%id. target Platyfish gene identical to query gene' (default: False),\n",
      " 'xmaculatus_homolog_perc_id_r1': '%id. query gene identical to target Platyfish gene' (default: False),\n",
      " 'xmaculatus_homolog_subtype': 'Last common ancestor with Platyfish' (default: False),\n",
      " 'xmaculatus_homolog_wga_coverage': 'Platyfish Whole-genome alignment coverage' (default: False),\n",
      " 'xtropicalis_homolog_associated_gene_name': 'Tropical clawed frog gene name' (default: False),\n",
      " 'xtropicalis_homolog_canonical_transcript_protein': 'Query protein or transcript ID' (default: False),\n",
      " 'xtropicalis_homolog_chrom_end': 'Tropical clawed frog chromosome/scaffold end (bp)' (default: False),\n",
      " 'xtropicalis_homolog_chrom_start': 'Tropical clawed frog chromosome/scaffold start (bp)' (default: False),\n",
      " 'xtropicalis_homolog_chromosome': 'Tropical clawed frog chromosome/scaffold name' (default: False),\n",
      " 'xtropicalis_homolog_ensembl_gene': 'Tropical clawed frog gene stable ID' (default: False),\n",
      " 'xtropicalis_homolog_ensembl_peptide': 'Tropical clawed frog protein or transcript stable ID' (default: False),\n",
      " 'xtropicalis_homolog_goc_score': 'Tropical clawed frog Gene-order conservation score' (default: False),\n",
      " 'xtropicalis_homolog_orthology_confidence': 'Tropical clawed frog orthology confidence [0 low, 1 high]' (default: False),\n",
      " 'xtropicalis_homolog_orthology_type': 'Tropical clawed frog homology type' (default: False),\n",
      " 'xtropicalis_homolog_perc_id': '%id. target Tropical clawed frog gene identical to query gene' (default: False),\n",
      " 'xtropicalis_homolog_perc_id_r1': '%id. query gene identical to target Tropical clawed frog gene' (default: False),\n",
      " 'xtropicalis_homolog_subtype': 'Last common ancestor with Tropical clawed frog' (default: False)}\n"
     ]
    }
   ],
   "source": [
    "#Connect to a Biomart Server\n",
    "server = BiomartServer(\"http://useast.ensembl.org/biomart\")\n",
    "\n",
    "# set verbose to True to get some messages\n",
    "server.verbose = True\n",
    "\n",
    "# use the 'hsapiens_gene_ensembl' dataset\n",
    "hsapiens_gene_ensembl = server.datasets['hsapiens_gene_ensembl']\n",
    "\n",
    "# show all available filters and attributes of the 'hsapiens_gene_ensembl' dataset\n",
    "#hsapiens_gene_ensembl.show_filters()  # uses pprint\n",
    "hsapiens_gene_ensembl.show_attributes()  # uses print"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T12:06:36.189081200Z",
     "start_time": "2023-12-13T12:06:29.687640600Z"
    }
   },
   "id": "a12b2c9cd9988677"
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[BiomartDataset:'hsapiens_gene_ensembl'] Searching using following params:\n",
      "{'attributes': ['ensembl_gene_id',\n",
      "                'hgnc_symbol',\n",
      "                'chromosome_name',\n",
      "                'ensembl_transcript_id',\n",
      "                'ensembl_peptide_id',\n",
      "                'uniprotsptrembl',\n",
      "                'uniprotswissprot']}\n",
      "[BiomartDataset:'hsapiens_gene_ensembl'] Fetching attributes\n",
      "[BiomartDataset] search query:\n",
      "b'<Query virtualSchemaName=\"default\" formatter=\"TSV\" header=\"1\" uniqueRows=\"1\" datasetConfigVersion=\"0.6\" count=\"\"><Dataset name=\"hsapiens_gene_ensembl\" interface=\"default\"><Attribute name=\"ensembl_gene_id\" /><Attribute name=\"hgnc_symbol\" /><Attribute name=\"chromosome_name\" /><Attribute name=\"ensembl_transcript_id\" /><Attribute name=\"ensembl_peptide_id\" /><Attribute name=\"uniprotsptrembl\" /><Attribute name=\"uniprotswissprot\" /></Dataset></Query>'\n"
     ]
    }
   ],
   "source": [
    "# run a search with custom filters and attributes (no header)\n",
    "response = hsapiens_gene_ensembl.search({\n",
    "  #'filters': {\n",
    "  #    'ensembl_gene_id': unique_gene_all,\n",
    "  #},\n",
    "  'attributes': [\n",
    "      'ensembl_gene_id', 'hgnc_symbol', 'chromosome_name', 'ensembl_transcript_id', 'ensembl_peptide_id', 'uniprotsptrembl', 'uniprotswissprot'\n",
    "  ]\n",
    "}, header = 1)\n",
    "\n",
    "# i = -1\n",
    "# #response format is TSV\n",
    "# for line in response.iter_lines():\n",
    "#   i+=1\n",
    "#   line = line.decode('utf-8')\n",
    "#   #print(line.split(\"\\t\"))\n",
    "#   conv_frame.loc[i] = line.split(\"\\t\")\n",
    "\n",
    "if response.ok:\n",
    "    # Convert the response content to a file-like object\n",
    "    content = io.StringIO(response.content.decode('utf-8'))\n",
    "\n",
    "    # Read the content into a DataFrame\n",
    "    conv_frame = pd.read_csv(content, sep='\\t')\n",
    "else:\n",
    "    print(\"Error with the response\")\n",
    "    "
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T13:16:52.474182600Z",
     "start_time": "2023-12-13T13:16:32.359562Z"
    }
   },
   "id": "9b26fc5523d376cb"
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "outputs": [
    {
     "data": {
      "text/plain": "    Gene stable ID HGNC symbol Chromosome/scaffold name Transcript stable ID  \\\n0  ENSG00000198888      MT-ND1                       MT      ENST00000361390   \n1  ENSG00000198763      MT-ND2                       MT      ENST00000361453   \n2  ENSG00000198804      MT-CO1                       MT      ENST00000361624   \n3  ENSG00000198712      MT-CO2                       MT      ENST00000361739   \n4  ENSG00000228253     MT-ATP8                       MT      ENST00000361851   \n\n  Protein stable ID UniProtKB/TrEMBL ID UniProtKB/Swiss-Prot ID  \n0   ENSP00000354687              U5Z754                  P03886  \n1   ENSP00000355046              Q7GXY9                  P03891  \n2   ENSP00000354499              U5YWV7                  P00395  \n3   ENSP00000354876              U5Z487                  P00403  \n4   ENSP00000355265              U5YV54                  P03928  ",
      "text/html": "<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>Gene stable ID</th>\n      <th>HGNC symbol</th>\n      <th>Chromosome/scaffold name</th>\n      <th>Transcript stable ID</th>\n      <th>Protein stable ID</th>\n      <th>UniProtKB/TrEMBL ID</th>\n      <th>UniProtKB/Swiss-Prot ID</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>ENSG00000198888</td>\n      <td>MT-ND1</td>\n      <td>MT</td>\n      <td>ENST00000361390</td>\n      <td>ENSP00000354687</td>\n      <td>U5Z754</td>\n      <td>P03886</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>ENSG00000198763</td>\n      <td>MT-ND2</td>\n      <td>MT</td>\n      <td>ENST00000361453</td>\n      <td>ENSP00000355046</td>\n      <td>Q7GXY9</td>\n      <td>P03891</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>ENSG00000198804</td>\n      <td>MT-CO1</td>\n      <td>MT</td>\n      <td>ENST00000361624</td>\n      <td>ENSP00000354499</td>\n      <td>U5YWV7</td>\n      <td>P00395</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>ENSG00000198712</td>\n      <td>MT-CO2</td>\n      <td>MT</td>\n      <td>ENST00000361739</td>\n      <td>ENSP00000354876</td>\n      <td>U5Z487</td>\n      <td>P00403</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>ENSG00000228253</td>\n      <td>MT-ATP8</td>\n      <td>MT</td>\n      <td>ENST00000361851</td>\n      <td>ENSP00000355265</td>\n      <td>U5YV54</td>\n      <td>P03928</td>\n    </tr>\n  </tbody>\n</table>\n</div>"
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "conv_frame.head()"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T13:16:55.367678500Z",
     "start_time": "2023-12-13T13:16:55.336368600Z"
    }
   },
   "id": "5cb7693939f9d867"
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "outputs": [
    {
     "data": {
      "text/plain": "123120"
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(conv_frame)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T13:17:13.565435500Z",
     "start_time": "2023-12-13T13:17:13.519018Z"
    }
   },
   "id": "44e5e3cdca7120d9"
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "outputs": [
    {
     "data": {
      "text/plain": "4095"
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(conv_frame[conv_frame['Gene stable ID'].isin(unique_gene_all)])"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2023-12-13T13:18:18.458669200Z",
     "start_time": "2023-12-13T13:18:18.386034100Z"
    }
   },
   "id": "8817350f6fd9c5cc"
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
